---
title: "Integrate_GeneModels_Obiroi"
output: 
 html_document:
  code_folding: hide
date: "2024-11-04"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE,message = FALSE)
require(rtracklayer)
require(GenomicFeatures)
require(tidyverse)
require(tibble)
require(magrittr)
require(DT)
```

# Assess confounding gene models from LongRead against established gene models.

## Download and import data from previous gene models and databases 

### Import data from previous Gene Models

```{r cars}
BRCDK_GTF <- import("Originals/Updated_2024/Obiroi_BRCDK__5_4.gtf.gz")
RefSeq_GTF <- import("Originals/Refseq/Obiroi_RefSeq__5_4_renamed.gtf.gz")
GenBank_GTF <- import("Originals/Genbank/Obiroi_GenBank__5_4_renamed.gtf.gz")

```

### Download additional files from UCSC genome hubs and rename chromosomes.

```{r,eval=FALSE}
dir.create("Originals/UCSC_Augustus/",showWarnings = FALSE,recursive = TRUE)
dir.create("Originals/UCSC_EnsGene/",showWarnings = FALSE,recursive = TRUE)
dir.create("Originals/UCSC_RefSeq/",showWarnings = FALSE,recursive = TRUE)
dir.create("Originals/UCSC_XenoRefGene/",showWarnings = FALSE,recursive = TRUE)
dir.create("Originals/UCSC_ChromMap/",showWarnings = FALSE,recursive = TRUE)

download.file("https://hgdownload.soe.ucsc.edu/hubs/GCF/003/672/135/GCF_003672135.1/genes/GCF_003672135.1_Obir_v5.4.augustus.gtf.gz",
       "Originals/UCSC_Augustus/GCF_003672135.1_Obir_v5.4.augustus.gtf.gz")
download.file("https://hgdownload.soe.ucsc.edu/hubs/GCF/003/672/135/GCF_003672135.1/genes/GCF_003672135.1_Obir_v5.4.ensGene.2021_12.gtf.gz",
       "Originals/UCSC_EnsGene/GCF_003672135.1_Obir_v5.4.ensGene.2021_12.gtf.gz")
download.file("https://hgdownload.soe.ucsc.edu/hubs/GCF/003/672/135/GCF_003672135.1/genes/GCF_003672135.1_Obir_v5.4.ncbiRefSeq.gtf.gz",
       "Originals/UCSC_RefSeq/GCF_003672135.1_Obir_v5.4.ncbiRefSeq.gtf.gz")
download.file("https://hgdownload.soe.ucsc.edu/hubs/GCF/003/672/135/GCF_003672135.1/genes/GCF_003672135.1_Obir_v5.4.xenoRefGene.gtf.gz",
       "Originals/UCSC_XenoRefGene/GCF_003672135.1_Obir_v5.4.xenoRefGene.gtf.gz")
download.file("https://hgdownload.soe.ucsc.edu/hubs/GCF/003/672/135/GCF_003672135.1/GCF_003672135.1.chromAlias.txt",
       "Originals/UCSC_ChromMap/GCF_003672135.1.chromAlias.txt")


UCSC_Refseq_GTF <- import("Originals/UCSC_RefSeq/GCF_003672135.1_Obir_v5.4.ncbiRefSeq.gtf.gz")
UCSC_Augustus_GTF <- import("Originals/UCSC_Augustus/GCF_003672135.1_Obir_v5.4.augustus.gtf.gz")
UCSC_EnsGene_GTF <- import("Originals/UCSC_EnsGene/GCF_003672135.1_Obir_v5.4.ensGene.2021_12.gtf.gz")
UCSC_xenoRefGene_GTF <- import("Originals/UCSC_XenoRefGene/GCF_003672135.1_Obir_v5.4.xenoRefGene.gtf.gz")

chr_map <- read.delim("Originals/UCSC_ChromMap/GCF_003672135.1.chromAlias.txt")

UCSC_Refseq_GTF_2 <- renameSeqlevels(UCSC_Refseq_GTF,
               chr_map %>% pull(assembly,name = "X..refseq"))
UCSC_EnsGene_GTF_2 <- renameSeqlevels(UCSC_EnsGene_GTF,
               chr_map %>% pull(assembly,name = "X..refseq"))
UCSC_Augustus_GTF_2 <- renameSeqlevels(UCSC_Augustus_GTF,
               chr_map %>% pull(assembly,name = "X..refseq"))
UCSC_xenoRefGene_GTF_2 <- renameSeqlevels(UCSC_xenoRefGene_GTF,
               chr_map %>% pull(assembly,name = "X..refseq"))
export(UCSC_Refseq_GTF_2,"Originals/UCSC_RefSeq/GCF_003672135.1_Obir_v5.4.ncbiRefSeq_renamed.gtf")
export(UCSC_EnsGene_GTF_2,"Originals/UCSC_EnsGene/GCF_003672135.1_Obir_v5.4.ensGene.2021_12_renamed.gtf")
export(UCSC_Augustus_GTF_2,"Originals/UCSC_Augustus/GCF_003672135.1_Obir_v5.4.augustus_renamed.gtf")
export(UCSC_xenoRefGene_GTF_2,"Originals/UCSC_XenoRefGene/GCF_003672135.1_Obir_v5.4.xenoRefGenerenamed.gtf")
R.utils::gzip("Originals/UCSC_RefSeq/GCF_003672135.1_Obir_v5.4.ncbiRefSeq_renamed.gtf")
R.utils::gzip("Originals/UCSC_EnsGene/GCF_003672135.1_Obir_v5.4.ensGene.2021_12_renamed.gtf")
R.utils::gzip("Originals/UCSC_Augustus/GCF_003672135.1_Obir_v5.4.augustus_renamed.gtf")
R.utils::gzip("Originals/UCSC_XenoRefGene/GCF_003672135.1_Obir_v5.4.xenoRefGenerenamed.gtf")
```


## Import data from Long and short read data

### From Short Read data

```{r pressure}
ShortStringtie_GTF <- import("Illumina/StringTie_NoGTF//stringtie/stringtie/merged_stringtie.gtf")
ShortStringtieMaleAndFemale_GTF <- import("Illumina/StringTie_MaleAndFemale/stringtie/stringtie/merged_stringtie.gtf")
```

### From Long Read data

```{r}
LongStringtie_GTF <- import("PacBio/Stringtie/Stringtie_Sample1.gtf")
```

### From mixed Short and Long Read data

```{r}
LongAndShortStringtie_GTF <- import("PabioAndIllumina/Stringtie/Stringtie_Mixedmode_LongAndShort.gtf")
```

## Assess confounding gene models

### LongRead, ShortRead, BRCDK, RefSeq, GenBank to Txdb objects

```{r}
BRCDK_txdb <- makeTxDbFromGRanges(BRCDK_GTF)

LongStringtie_GTF <- LongStringtie_GTF[strand(LongStringtie_GTF) != "*"]
ShortStringtie_GTF <- ShortStringtie_GTF[strand(ShortStringtie_GTF) != "*"]

LongStringtie_txdb <- makeTxDbFromGRanges(LongStringtie_GTF)
ShortStringtie_txdb <- makeTxDbFromGRanges(ShortStringtie_GTF)
GenBank_txdb <- makeTxDbFromGRanges(GenBank_GTF)
RefSeq_txdb <- makeTxDbFromGRanges(RefSeq_GTF)
```

### Create exon by gene models

```{r}
BRCDK_exonsByGene <- exonsBy(BRCDK_txdb,by="gene")
LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb,by="gene")
ShortStringtie_exonsByGene <- exonsBy(ShortStringtie_txdb,by="gene")
GenBank_exonsByGene <- exonsBy(GenBank_txdb,by="gene")

```

## Assess confounding gene models 

### Identify potentially confounding gene models

```{r}
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]

QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[SubMultiHits]

noFiltering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
      Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
 set_colnames("Counfounding_Gene_Number") %>% datatable()
```


#### Assessing manually

```{r,eval=FALSE,echo=FALSE}
 # Evaluate multi_BRCDK
overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% "LOC105274949",]
"Ir317.3" - Keep stringtie Long Read
"LOC105274442" - Keep stringtie Long Read
"LOC105274450" - Keep stringtie Long Read
"LOC105274452" - Keep stringtie Long Read 
"LOC105274459" - Keep stringtie Long Read
"LOC105274478" - Weird one Overlaps multi both ways
"LOC105274484" - Keep stringtie Long Read - May lose a variant - Downstream single exon gene
"LOC105274520" - Keep stringtie Long Read - May lose a variant - Downstream single exon gene
"LOC105274588" - Keep stringtie Long Read
"LOC105274623" - Keep stringtie Long read - May lose long variant - Downstream single exon gene
"LOC105274666" - Weird one Overlaps multi both ways
"LOC105274693" - Keep stringtie Long Read
"LOC105274707" - Keep stringtie Long Read - May lose long variant - Downstream single exon gene
"LOC105274722" - Keep stringtie Long Read 
"LOC105274725" - Weird one - Made of single read exons
"LOC105274756" - Keep stringtie Long Read - May lose long variant - Downstream single exon gene
"LOC105274778" - Keep stringtie Long Read - May lose long variant - Downstream single exon gene
"LOC105274931" - Keep stringtie Long Read - May lose long variant - Downstream single exon gene
"LOC105274939" - Weird one Overlaps multi both ways
"LOC105274940" - Weird one Overlaps multi both ways and weird Downstream single/double exon gene
"LOC105274949" - Keep stringtie Long Read
"LOC105274965" - Keep stringtie Long Read - May lose long variant - Downstream single exon gene
"LOC105274982" - Weird one Overlaps multi both ways and weird Downstream single/double exon gene
"LOC105274987" - Keep stringtie Long Read


```


```{r,eval=FALSE,echo=FALSE}
tempQF <- overlappingGenesDF[overlappingGenesDF$subjectHits == 9347,]

BRCDK_exonsByGene[tempQF$queryHits]
LongStringtie_exonsByGene[9347]
```


```{r,eval=FALSE,echo=FALSE}
tempQF <- overlappingGenesDF[overlappingGenesDF$queryHits == 7751,]


LongStringtie_exonsByGene[tempQF$subjectHits]
BRCDK_exonsByGene[7751]
```

# Clean up GTF

## Assess confounded by single exon genes in Long Read data

### Filter single exon longread models and reassess confounding gene models

```{r}
singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]

QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]
singleExon_Filtering <- multi_BRCDK


data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
      Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
 set_colnames("Counfounding_Gene_Number") %>% datatable()

```

## Assess gene models against GenBank established gene models

### Identity where Long read and Refseq have confounded gene models but Long read agrees with GenBank

```{r}
overlappingGenesDF_MultiHits <- overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% multi_BRCDK,]
unique_BRCDK_MultiHits <- overlappingGenesDF_MultiHits %>% pull(queryHitsNames) %>% unique

# GenBank_exonsByGene_noSingle <- GenBank_exonsByGene[lengths(GenBank_exonsByGene) > 1]
filterVec <- vector("logical",length = length(unique_BRCDK_MultiHits))
for(i in 1:length(unique_BRCDK_MultiHits)){
 message(i)
 StringTie_Genes_Temp <- overlappingGenesDF_MultiHits[overlappingGenesDF_MultiHits$queryHitsNames %in% 
                 unique_BRCDK_MultiHits[i],
                "subjectHitsNames"]
 OverlapTableTemp <- as.data.frame(findOverlaps(LongStringtie_exonsByGene[StringTie_Genes_Temp],GenBank_exonsByGene))
 if(any(duplicated(OverlapTableTemp$queryHits))){
  filterVec[i] <- FALSE
 }else if(any(duplicated(OverlapTableTemp$subjectHits))){
  filterVec[i] <- FALSE
 }else{
  filterVec[i] <- TRUE
 }
}
names(filterVec) <- unique_BRCDK_MultiHits
```

### Apply Genbank filter and assess confounding gene models

```{r}
singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

GenbankFiltering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
      Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
 set_colnames("Counfounding_Gene_Number") %>% datatable()
```

#### Assess manually

```{r eval=FALSE,echo=FALSE}

"LOC105274442" - Genes within genes in Refseq
"LOC105274459" - Floating UTRs
"LOC105274478" - multioverlapping
"LOC105274588" - RefSeq gene overlaps
"LOC105274940" - Floating UTRs
"LOC105274987" - Overlap in Pacbio UTR to 5'UTR of Refseq
"LOC105275001" - Genes within genes in Refseq
"LOC105275005" - Refseq isoform doesnt exist in Pacbio or strinngtie
"LOC105275287" - Stringtie short read matches Refseq
"LOC105275722" - Floating UTRs perhaps
"LOC105275770" - Stringtie short read matches Refseq
"LOC105275835"
"LOC105275877" - Stringtie short read matches Pacbio
"LOC105275890" - Stringtie short read matches Pacbio
"LOC105275915" - Stringtie short read matches Pacbio
"LOC105275967"
"LOC105275979" - Looks like run on UTR in ShortRead
"LOC105275990" - Stringtie short read matches Refseq
"LOC105276245"
"LOC105276271"
"LOC105276288"
"LOC105276354"
"LOC105276377"
"LOC105276416"
```


## Assess and fix gene models confounded within Refseq Annotation.

### Find and fix transcripts in genes which overlap other genes within the Refseq annotation.

```{r,eval=TRUE}
BRCDK_exonsByTranscript <- exonsBy(BRCDK_txdb,by="tx",use.names=TRUE)
BRCDK_GTF_GeneToTranscript <- BRCDK_GTF %>% mcols %>% 
 as.data.frame() %>% 
 dplyr::select(gene_id,transcript_id) %>% 
 distinct()

selfOverlaps <- findOverlaps(BRCDK_exonsByTranscript,BRCDK_exonsByTranscript)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(BRCDK_exonsByTranscript)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(BRCDK_exonsByTranscript)[temp_selfOverlaps$subjectHits]

temp_selfOverlaps$queryHitsGeneNames <- BRCDK_GTF_GeneToTranscript$gene_id[match(temp_selfOverlaps$queryHitsNames,
                       BRCDK_GTF_GeneToTranscript$transcript_id)]

temp_selfOverlaps$subjectHitsGeneNames <- BRCDK_GTF_GeneToTranscript$gene_id[match(temp_selfOverlaps$subjectHitsNames,
                       BRCDK_GTF_GeneToTranscript$transcript_id)]

temp_selfOverlaps$subjectHitsGeneNames[is.na(temp_selfOverlaps$subjectHitsGeneNames)] <- temp_selfOverlaps$subjectHitsNames[is.na(temp_selfOverlaps$subjectHitsGeneNames)]

temp_selfOverlaps$queryHitsGeneNames[is.na(temp_selfOverlaps$queryHitsGeneNames)] <- temp_selfOverlaps$queryHitsNames[is.na(temp_selfOverlaps$queryHitsGeneNames)]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHitsGeneNames != temp_selfOverlaps$subjectHitsGeneNames,]
nrow(temp_selfOverlaps)

allSelfOverlapGenes <- temp_selfOverlaps[,3] %>% unique

datatable(
 data.frame(Genes_Overlapping_Other_Genes_Within_Refseq_Models=temp_selfOverlaps[,5] %>% unique %>% length())
)
if(!file.exists("RObjects/BRCDK_NoGeneModelOverlap.gtf")){



while(nrow(temp_selfOverlaps) > 0){
message("Overlapping transcripts - ",nrow(temp_selfOverlaps)) 
 loop <- 1
# XM_026973633.1 1372
updatedGene <- GRangesList()
for(i in 1:nrow(temp_selfOverlaps)){
 geneA <- temp_selfOverlaps[i,3]
 geneB <- temp_selfOverlaps[i,4]
 exons_geneA <- BRCDK_exonsByTranscript[geneA]
 exons_geneB <- BRCDK_exonsByTranscript[geneB]
 if(all(strand(exons_geneA) == "+")){
  if(start(exons_geneA[[1]][1]) < start(exons_geneB[[1]][1])){
   proceedingGene <- exons_geneA
   followingGene <- exons_geneB
  }else{
   proceedingGene <- exons_geneB
   followingGene <- exons_geneA
  }
  rangeToRestrict <- c(start(proceedingGene[[1]][1]),start(followingGene[[1]][1])-20)
  updatedGene <- c(updatedGene,restrict(proceedingGene[1],start = rangeToRestrict[1],end=rangeToRestrict[2]))
   # print("+")
 }else if(all(strand(exons_geneA) == "-")){
  if(end(exons_geneA[[1]][length(exons_geneA[[1]])]) > end(exons_geneB[[1]][length(exons_geneB[[1]])])){
   proceedingGene <- exons_geneA
   followingGene <- exons_geneB
  }else{
   proceedingGene <- exons_geneB
   followingGene <- exons_geneA
  }
  rangeToRestrict <- c(end(followingGene[[1]][1])+20,end(proceedingGene[[1]][1]))
  updatedGene <- c(updatedGene,restrict(proceedingGene[1],start = rangeToRestrict[1],end=rangeToRestrict[2]))
   # print("-")
 }
 if(i %% 100 == 0)message(loop,":",i)
}

BRCDK_exonsByTranscript[names(updatedGene)] <- updatedGene


selfOverlaps <- findOverlaps(BRCDK_exonsByTranscript,BRCDK_exonsByTranscript)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(BRCDK_exonsByTranscript)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(BRCDK_exonsByTranscript)[temp_selfOverlaps$subjectHits]

temp_selfOverlaps$queryHitsGeneNames <- BRCDK_GTF_GeneToTranscript$gene_id[match(temp_selfOverlaps$queryHitsNames,
                       BRCDK_GTF_GeneToTranscript$transcript_id)]

temp_selfOverlaps$subjectHitsGeneNames <- BRCDK_GTF_GeneToTranscript$gene_id[match(temp_selfOverlaps$subjectHitsNames,
                       BRCDK_GTF_GeneToTranscript$transcript_id)]

temp_selfOverlaps$subjectHitsGeneNames[is.na(temp_selfOverlaps$subjectHitsGeneNames)] <- temp_selfOverlaps$subjectHitsNames[is.na(temp_selfOverlaps$subjectHitsGeneNames)]

temp_selfOverlaps$queryHitsGeneNames[is.na(temp_selfOverlaps$queryHitsGeneNames)] <- temp_selfOverlaps$queryHitsNames[is.na(temp_selfOverlaps$queryHitsGeneNames)]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHitsGeneNames != temp_selfOverlaps$subjectHitsGeneNames,]
loop <- loop+1

}
temp <- unlist(BRCDK_exonsByTranscript)
df_temp <- data.frame(source="StringTie_curated",
             type="exon",
             score=NA,phase=NA,
             gene_id=BRCDK_GTF_GeneToTranscript$gene_id[
              match(names(temp),
                       BRCDK_GTF_GeneToTranscript$transcript_id)
              ],
             transcript_id=names(temp))

df_temp$gene_id[is.na(df_temp$gene_id)] <- df_temp$transcript_id[is.na(df_temp$gene_id)]
mcols(temp) <- df_temp
class(temp$phase) <- "integer"
rtracklayer::export(temp[temp$transcript_id %in% allSelfOverlapGenes,],con="BRCDK_NoGeneModelOverlap.gtf") 
rtracklayer::export(temp,con="RObjects/BRCDK_NoGeneModelOverlapFull.gtf") 
saveRDS(temp,"RObjects/BRCDK_NoGeneModelOverlapFull.RDS")
}

# BRCDK_GTF <- import("BRCDK_NoGeneModelOverlapFull.gtf")
BRCDK_GTF <- readRDS("RObjects/BRCDK_NoGeneModelOverlapFull.RDS")
BRCDK_txdb <- makeTxDbFromGRanges(BRCDK_GTF)
BRCDK_exonsByGene <- exonsBy(BRCDK_txdb,by = "gene")
selfOverlaps <- findOverlaps(BRCDK_exonsByGene,BRCDK_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(BRCDK_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(BRCDK_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]
#nrow(temp_selfOverlaps)
```

### Apply RefSeq self-overlap filter and then assess confounding gene models between LongRead and RefSeq.


```{r}

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

overlappingRefSeqFiltering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
      Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
  set_colnames("Counfounding_Gene_Number") %>% datatable()
```

#### Manually assess

```{r,eval=FALSE,echo=FALSE}

"LOC105274442" - Genes within genes in Refseq
"LOC105274459" - Floating UTRs
"LOC105274478" - multioverlapping
"LOC105274588" - RefSeq gene overlaps
"LOC105274940" - Floating UTRs
"LOC105274987" - Overlap in Pacbio UTR to 5'UTR of Refseq
"LOC105275001" - Genes within genes in Refseq
"LOC105275005" - Refseq isoform doesnt exist in Pacbio or strinngtie
"LOC105275287" - Stringtie short read matches Refseq
"LOC105275722" - Floating UTRs perhaps
"LOC105275770" - Stringtie short read matches Refseq
"LOC105275835"
"LOC105275877"
"LOC105275890"
"LOC105275915"
"LOC105275967"
"LOC105275979"
"LOC105275990"
"LOC105276245"
"LOC105276271"
"LOC105276288"
```


## Use agreement between long read and short read models to evaluate gene models vs RefSeq 

### Identity where Long read and Refseq have confounded gene models but Long read agrees with Short Read.


```{r}
overlappingGenesDF_MultiHits <- overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% multi_BRCDK,]
unique_BRCDK_MultiHits <- overlappingGenesDF_MultiHits %>% pull(queryHitsNames) %>% unique

ShortStringtie_exonsByGene_noSingle <- ShortStringtie_exonsByGene[lengths(ShortStringtie_exonsByGene) > 1]

filterVec2 <- vector("logical",length = length(unique_BRCDK_MultiHits))
for(i in 1:length(unique_BRCDK_MultiHits)){
 message(i)
 StringTie_Genes_Temp <- overlappingGenesDF_MultiHits[overlappingGenesDF_MultiHits$queryHitsNames %in% 
                 unique_BRCDK_MultiHits[i],
                "subjectHitsNames"]
 OverlapTableTemp <- as.data.frame(findOverlaps(LongStringtie_exonsByGene[StringTie_Genes_Temp],ShortStringtie_exonsByGene))
 if(any(duplicated(OverlapTableTemp$queryHits))){
  filterVec2[i] <- FALSE
 }else if(any(duplicated(OverlapTableTemp$subjectHits))){
  filterVec2[i] <- FALSE
 }else{
  filterVec2[i] <- TRUE
 }
}
names(filterVec2) <- unique_BRCDK_MultiHits
```

### Filter out gene models where short read and long read agree but are different from RefSeq.

```{r}

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

ShortReadAndLongReadMatchFiltering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
      Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>%
 set_colnames("Counfounding_Gene_Number") %>% datatable()
```

#### Manually assess

```{r,eval=FALSE,echo=FALSE}
"LOC105274459" - Floating 2 exon UTR and overlapping 3'UTRs (lots of Ts (As as on opposite strand) downstream) from Refseq with downstream 5 UTR from Pacbio. -- 7 Ts in a row (7/10)
"LOC105274478" - Floating multi exon gene (lots of As downstream) -- 6As in a row (9/10)
"LOC105274940" - Floating multi exon gene (lots of As downstream) -- 4As in a row (7/10)
"LOC105275287" - Floating multi exon gene (lots of A/Gs downstream) -- 2As in a row (3/10)
"LOC105275770" - Floating multi exon gene (lots of Ts downstream) -- 5As in a row (7/10)
"LOC105275877" - 
"LOC105275915"
"LOC105275979"
"LOC105276245"
"LOC105276271"
"LOC105276288"
"LOC105276416"
"LOC105276519"
"LOC105276799"
"LOC105276868"
"LOC105276940"
"LOC105276941"
"LOC105277123"
"LOC105277125"
"LOC105277149"
"LOC105277202"
"LOC105277234"
"LOC105277394"
"LOC105277425"
"LOC105277488"
"LOC105277606"
"LOC105277680"
"LOC105277768"
"LOC105277793"
"LOC105277941"
"LOC105278064"
"LOC105278066"
"LOC105278244"
"LOC105278262"
"LOC105278327"
"LOC105278329"
"LOC105278470"
"LOC105278474"
"LOC105278569"
"LOC105278631"
"LOC105278639"
"LOC105278891"
"LOC105278994"
"LOC105279031"
"LOC105279274"
"LOC105279301"
"LOC105279462"
"LOC105279464"
"LOC105279644"
"LOC105279661"
"LOC105279687"
"LOC105279752"
"LOC105279816"
"LOC105280056"
"LOC105280265"
"LOC105280521"
"LOC105280692"
"LOC105280761"
"LOC105281207"
"LOC105281330"
"LOC105281402"
"LOC105281448"
"LOC105281587"
"LOC105281609"
"LOC105281789"
"LOC105281903"
"LOC105282012"
"LOC105282112"
"LOC105282237"
"LOC105282359"
"LOC105282423"
"LOC105282895"
"LOC105282908"
"LOC105283079"
"LOC105283145"
"LOC105283180"
"LOC105283212"
"LOC105283214"
"LOC105283491"
"LOC105283503"
"LOC105283615"
"LOC105283711"
"LOC105283806"
"LOC105283824"
"LOC105283841"
"LOC105284073"
"LOC105284102"
"LOC105284155"
"LOC105284264"
"LOC105284265"
"LOC105284340"
"LOC105284377"
"LOC105284454"
"LOC105284484"
"LOC105284524"
"LOC105284529"
"LOC105284540"
"LOC105284788"
"LOC105284864"
"LOC105284876"
"LOC105284886"
"LOC105284909"
"LOC105284944"
"LOC105285222"
"LOC105285464"
"LOC105285560"
"LOC105285619"
"LOC105285919"
"LOC105286207"
"LOC105286340"
"LOC105286518"
"LOC105286657"
"LOC105286692"
"LOC105286732"
"LOC105286777"
"LOC105286974"
"LOC105287099"
"LOC105287672"
"LOC105287773"
"LOC105287787"
"LOC105287969"
"LOC113562368"
"LOC113563199"
"LOC113563593"

```


## Identify and attempt to rescue potential internal-primed transcripts within Long Read data

### Capture information on polyA repeat and polyA proportion downstream from LongRead transcripts.

```{r}
require(Biostrings)
require(Rsamtools)
allstringtie_Transcripts <- transcripts(LongStringtie_txdb)

genome <- FaFile("Genome/GenBank.fa")
genomeLevel <- seqlengths(genome)
genomeRanges <- GRanges(names(genomeLevel),IRanges(1,genomeLevel))
genomeFreq <- alphabetFrequency(readDNAStringSet("Genome/GenBank.fa"))

# allBRCDK_Transcripts <- allBRCDK_Transcripts[seqnames(allBRCDK_Transcripts) %in% names(genomeLevel)]
# allBRCDK_Transcripts[]
PacBioStringtie_20tails <- flank(allstringtie_Transcripts,start=FALSE,width = 10)
PacBioStringtie_20tails <- PacBioStringtie_20tails[findOverlaps(PacBioStringtie_20tails,genomeRanges,type = "within")@from %>% unique]
PacBioStringtie_20tailSeqs <- getSeq(genome,PacBioStringtie_20tails)
names(PacBioStringtie_20tailSeqs) <- PacBioStringtie_20tails$tx_name
PacBioStringtie_20AF <- alphabetFrequency(PacBioStringtie_20tailSeqs)
rownames(PacBioStringtie_20AF) <-PacBioStringtie_20tails$tx_name

polyA <- ifelse((vcountPattern("AAAAAA",PacBioStringtie_20tailSeqs) > 0),6,
        ifelse((vcountPattern("AAAAA",PacBioStringtie_20tailSeqs) > 0),5,
            ifelse((vcountPattern("AAAA",PacBioStringtie_20tailSeqs) > 0),4,0)))

PacBioStringtie_20AF <- PacBioStringtie_20AF %>% as.data.frame %>% mutate(polyA=polyA)

PBani_GTF <- import("~/Downloads/Anindita_stringtie/Stringtie_Sample1.gtf")
PBani_GTF <- PBani_GTF[strand(PBani_GTF) != "*"]

PBani_txdb <- makeTxDbFromGRanges(PBani_GTF)
PBani_Transcripts <- transcripts(PBani_txdb)
PBani_20tails <- flank(PBani_Transcripts,start=FALSE,width = 10)
PBani_20tails <-PBani_20tails[findOverlaps(PBani_20tails,genomeRanges,type = "within")@from %>% unique]
PBani_20tailSeqs <- getSeq(genome,PBani_20tails)
names(PBani_20tailSeqs) <-PBani_20tails$tx_name
PBani_20AF <- alphabetFrequency(PBani_20tailSeqs)
rownames(PBani_20AF) <-PBani_20tails$tx_name

polyA <- ifelse((vcountPattern("AAAAAA",PBani_20tailSeqs) > 0),6,
        ifelse((vcountPattern("AAAAA",PBani_20tailSeqs) > 0),5,
            ifelse((vcountPattern("AAAA",PBani_20tailSeqs) > 0),4,0)))

PBani_20AF <- PBani_20AF %>% as.data.frame %>% mutate(polyA=polyA)

require(Biostrings)
require(Rsamtools)
BRCDK_Transcripts <- transcripts(BRCDK_txdb)

BRCDK_20tails <- flank(BRCDK_Transcripts,start=FALSE,width = 10)
BRCDK_20tails <- BRCDK_20tails[findOverlaps(BRCDK_20tails,genomeRanges,type = "within")@from %>% unique]
BRCDK_20tailSeqs <- getSeq(genome,BRCDK_20tails)
names(BRCDK_20tailSeqs) <- BRCDK_20tails$tx_name
BRCDK_20AF <- alphabetFrequency(BRCDK_20tailSeqs)
rownames(BRCDK_20AF) <-BRCDK_20tails$tx_name


polyA <- ifelse((vcountPattern("AAAAAA",BRCDK_20tailSeqs) > 0),6,
        ifelse((vcountPattern("AAAAA",BRCDK_20tailSeqs) > 0),5,
            ifelse((vcountPattern("AAAA",BRCDK_20tailSeqs) > 0),4,0)))

BRCDK_20AF <- BRCDK_20AF %>% as.data.frame %>% mutate(polyA=polyA)



require(Biostrings)
require(Rsamtools)
RefSeq_Transcripts <- transcripts(RefSeq_txdb)

RefSeq_20tails <- flank(RefSeq_Transcripts,start=FALSE,width = 10)
RefSeq_20tails <- RefSeq_20tails[findOverlaps(RefSeq_20tails,genomeRanges,type = "within")@from %>% unique]
RefSeq_20tailSeqs <- getSeq(genome,RefSeq_20tails)
names(RefSeq_20tailSeqs) <- RefSeq_20tails$tx_name
RefSeq_20AF <- alphabetFrequency(RefSeq_20tailSeqs)
rownames(RefSeq_20AF) <-RefSeq_20tails$tx_name

polyA <- ifelse((vcountPattern("AAAAAA",RefSeq_20tailSeqs) > 0),6,
        ifelse((vcountPattern("AAAAA",RefSeq_20tailSeqs) > 0),5,
            ifelse((vcountPattern("AAAA",RefSeq_20tailSeqs) > 0),4,0)))

RefSeq_20AF <- RefSeq_20AF %>% as.data.frame %>% mutate(polyA=polyA)





require(Biostrings)
require(Rsamtools)
GenBank_Transcripts <- transcripts(GenBank_txdb)

GenBank_20tails <- flank(GenBank_Transcripts,start=FALSE,width = 10)
GenBank_20tails <- GenBank_20tails[findOverlaps(GenBank_20tails,genomeRanges,type = "within")@from %>% unique]
GenBank_20tailSeqs <- getSeq(genome,GenBank_20tails)
names(GenBank_20tailSeqs) <- GenBank_20tails$tx_name
GenBank_20AF <- alphabetFrequency(GenBank_20tailSeqs)
rownames(GenBank_20AF) <-GenBank_20tails$tx_name

polyA <- ifelse((vcountPattern("AAAAAA",GenBank_20tailSeqs) > 0),6,
        ifelse((vcountPattern("AAAAA",GenBank_20tailSeqs) > 0),5,
            ifelse((vcountPattern("AAAA",GenBank_20tailSeqs) > 0),4,0)))

GenBank_20AF <- GenBank_20AF %>% as.data.frame %>% mutate(polyA=polyA)


PBLong <- PacBioStringtie_20AF %>% 
 as.data.frame() %>% 
 select(A,G,C,T,polyA) %>% 
 pivot_longer(cols = !matches("poly")) %>% 
 mutate(Models="PacBio")
GBLong <- GenBank_20AF %>% 
 as.data.frame() %>% 
 select(A,G,C,T,polyA) %>% 
 pivot_longer(cols = !matches("poly")) %>% 
 mutate(Models="GenBank")
RefSeqLong <- RefSeq_20AF %>% 
 as.data.frame() %>% 
 select(A,G,C,T,polyA) %>% 
 pivot_longer(cols = !matches("poly")) %>% 
 mutate(Models="RefSeq")
PBaniLong <- PBani_20AF %>% 
 as.data.frame() %>% 
 select(A,G,C,T,polyA) %>% 
 pivot_longer(cols = !matches("poly")) %>% 
 mutate(Models="Anindita")

toPlot <- rbind(PBLong,PBaniLong,GBLong,RefSeqLong)

ggplot(toPlot,aes(x=value,fill=Models))+
 geom_histogram()+
 facet_grid(Models~name)


ggplot(toPlot %>% filter(name=="A"),
    aes(x=value,fill=Models))+
 geom_histogram()+
 facet_grid(Models~polyA)


potentialPolyFakes <- PacBioStringtie_20AF[(PacBioStringtie_20AF$polyA ==4 & PacBioStringtie_20AF$A >= 7) | (PacBioStringtie_20AF$polyA > 4),]

potentialPolyFakeGenes <- rownames(potentialPolyFakes) %>% unique %>% gsub("\\.\\d+$","",.)
```

### Assess potential impact of polyA transcripts on confounded models

```{r}

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]

overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% potentialPolyFakeGenes
,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

preInternalPrimeFiltering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
      Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
  set_colnames("Counfounding_Gene_Number") %>% datatable()
```

#### Manually assess

```{r,eval=FALSE,echo=FALSE}
"LOC105274459" 
"LOC105275287"
"LOC105275877"
"LOC105276245"
"LOC105276271"
"LOC105276416"
"LOC105276799"
"LOC105276941"
"LOC105277123"
"LOC105277125"
"LOC105277202"
"LOC105277394"
"LOC105277425"
"LOC105277606"
"LOC105277793"
"LOC105277941"
"LOC105278064"
"LOC105278066"
"LOC105278244"
"LOC105278262"
"LOC105278327"
"LOC105278994"
"LOC105279031"
"LOC105279462"
"LOC105280056"
"LOC105280761"
"LOC105281207"
"LOC105281402"
"LOC105281448"
"LOC105281903"
"LOC105282112"
"LOC105282359"
"LOC105283145"
"LOC105283180"
"LOC105283212"
"LOC105283491"
"LOC105283711"
"LOC105283824"
"LOC105283841"
"LOC105284102"
"LOC105284264"
"LOC105284340"
"LOC105284377"
"LOC105284524"
"LOC105284540"
"LOC105284788"
"LOC105284876"
"LOC105284886"
"LOC105284909"
"LOC105284944"
"LOC105286207"
"LOC105286518"
"LOC105286657"
"LOC105286732"
"LOC105286777"
"LOC105287099"
"LOC105287787"
"LOC105287969"
"LOC113562368"
```


### Try and rescue the confounded and internally primed long read gene models


```{r}
singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]


polAToClean <- overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% multi_BRCDK,"subjectHitsNames"]


require(Biostrings)
require(Rsamtools)
allstringtie_Transcripts <- transcripts(LongStringtie_txdb)

genome <- FaFile("Genome/GenBank.fa")
genomeLevel <- seqlengths(genome)
genomeRanges <- GRanges(names(genomeLevel),IRanges(1,genomeLevel))
genomeFreq <- alphabetFrequency(readDNAStringSet("Genome/GenBank.fa"))

# allBRCDK_Transcripts <- allBRCDK_Transcripts[seqnames(allBRCDK_Transcripts) %in% names(genomeLevel)]
# allBRCDK_Transcripts[]
PacBioStringtie_20tails <- flank(allstringtie_Transcripts,start=FALSE,width = 10)
PacBioStringtie_20tails <- PacBioStringtie_20tails[findOverlaps(PacBioStringtie_20tails,genomeRanges,type = "within")@from %>% unique]
PacBioStringtie_20tailSeqs <- getSeq(genome,PacBioStringtie_20tails)
names(PacBioStringtie_20tailSeqs) <- PacBioStringtie_20tails$tx_name
PacBioStringtie_20AF <- alphabetFrequency(PacBioStringtie_20tailSeqs)
rownames(PacBioStringtie_20AF) <-PacBioStringtie_20tails$tx_name

polyA <- ifelse((vcountPattern("AAAAAA",PacBioStringtie_20tailSeqs) > 0),6,
        ifelse((vcountPattern("AAAAA",PacBioStringtie_20tailSeqs) > 0),5,
            ifelse((vcountPattern("AAAA",PacBioStringtie_20tailSeqs) > 0),4,0)))

PacBioStringtie_20AF <- PacBioStringtie_20AF %>% as.data.frame %>% mutate(polyA=polyA)

potentialPolyFakes <- PacBioStringtie_20AF[(PacBioStringtie_20AF$polyA ==4 & PacBioStringtie_20AF$A >= 7) | (PacBioStringtie_20AF$polyA > 4),]

potentialPolyFakeTranscripts <- rownames(potentialPolyFakes) %>% unique 


LongStringtie_exonsByTranscript <- exonsBy(LongStringtie_txdb,by="tx",use.names=TRUE)

ShortStringtie_exonsByTranscript <- exonsBy(ShortStringtie_txdb,by="tx",use.names=TRUE)
stringtie_TranscriptsWithTXandGene <- transcripts(LongStringtie_txdb,columns=c("tx_id", "tx_name","gene_id"))

stringtie_TranscriptsWithTXandGene_Neg <- stringtie_TranscriptsWithTXandGene[strand(stringtie_TranscriptsWithTXandGene) == "-"]
stringtie_TranscriptsWithTXandGene_Neg$length <- width(stringtie_TranscriptsWithTXandGene_Neg)
# stringtie_TranscriptsWithTXandGene_Neg_Ends <- resize(stringtie_TranscriptsWithTXandGene_Neg,fix="end",width=1)
stringtie_TranscriptsWithTXandGene_NegFilt <- stringtie_TranscriptsWithTXandGene_Neg %>% 
 as.data.frame() %>% 
 arrange(seqnames,start,desc(length)) %>% 
 distinct(gene_id,.keep_all = TRUE) %>% GRanges()

stringtie_TranscriptsWithTXandGene_NegFilt_PolyA <- stringtie_TranscriptsWithTXandGene_NegFilt[stringtie_TranscriptsWithTXandGene_NegFilt$tx_name %in% potentialPolyFakeTranscripts]

stringtie_TranscriptsWithTXandGene_NegFilt_PolyA <- stringtie_TranscriptsWithTXandGene_NegFilt_PolyA[unlist(stringtie_TranscriptsWithTXandGene_NegFilt_PolyA$gene_id) %in% polAToClean,]

follow_LongStringtie_exonsByTranscript <- precede(stringtie_TranscriptsWithTXandGene_NegFilt_PolyA,allstringtie_Transcripts)

# "LOC105274459" - Floating 2 exon UTR and overlapping 3'UTRs (lots of Ts (As as on opposite strand) downstream) from Refseq with downstream 5 UTR from Pacbio. -- 7 Ts in a row (7/10) - STRG.10781.1
# "LOC105274478" - Floating multi exon gene (lots of As downstream) -- 6As in a row (9/10)
# "LOC105274940" - Floating multi exon gene (lots of As downstream) -- 4As in a row (7/10)
# "LOC105275770" - Floating multi exon gene (lots of Ts downstream) -- 5As in a row (7/10) - STRG.4147.1

# STRG.10781 - nice rescue
# STRG.11314 - Nice rescue
polAmodelList <- GRangesList()
reason <- list()
renameFollowersInList <- list()



stringtie_TranscriptsWithTXandGene_NegFilt_PolyA %<>% sort(decreasing = FALSE)



for(i in 1:length(stringtie_TranscriptsWithTXandGene_NegFilt_PolyA)){
 message("Primed gene ",i)
 toi <- stringtie_TranscriptsWithTXandGene_NegFilt_PolyA$tx_name[i]
 followingTranscript <- allstringtie_Transcripts[follow_LongStringtie_exonsByTranscript[stringtie_TranscriptsWithTXandGene_NegFilt_PolyA$tx_name %in% toi]]
 polyTranscript <- allstringtie_Transcripts[allstringtie_Transcripts$tx_name %in% toi]
 
 overlapRange <- GRanges(seqnames(polyTranscript),
             IRanges(start=end(followingTranscript),end=start(polyTranscript)),strand = strand(polyTranscript)
             )
 potentialTranscript <- c(polyTranscript,followingTranscript)
 
 followingModel <- LongStringtie_exonsByTranscript[[followingTranscript$tx_name]]
 polyAModel <- LongStringtie_exonsByTranscript[[polyTranscript$tx_name]]
 
 if(any(duplicated(findOverlaps(potentialTranscript,BRCDK_exonsByGene)@to))){
 
 
 if(any(duplicated(findOverlaps(potentialTranscript,ShortStringtie_exonsByGene)@to))){
 
 
 AltModels <- ShortStringtie_exonsByTranscript[ShortStringtie_exonsByTranscript %over% overlapRange]
 
 if(length(AltModels) > 0){
 
 polyA_lastExon <- polyAModel[length(polyAModel)]
 following_firstExon <- followingModel[1]
  
 commonExonsOverlap <- intersect(findOverlaps(AltModels,c(polyA_lastExon))@from,
 findOverlaps(AltModels,c(following_firstExon))@from)
 if(length(commonExonsOverlap) > 0){
  AltModels <- AltModels[commonExonsOverlap]
 }
 AltModel <- AltModels[[which.max(sum(width(AltModels)))]]
 overlapAlt <- findOverlaps(overlapRange,AltModel,type = "within")
 if(length(overlapAlt) == 1){
  tempFollowerList <- list(followingTranscript$tx_name)
  names(tempFollowerList) <- polyTranscript$tx_name
  renameFollowersInList <- c(renameFollowersInList,tempFollowerList)
  polyAModel <- GenomicRanges::reduce(c(polyAModel,overlapRange,followingModel))
  temp <- list("Taking model from Short Read - creating a merged exon")
  names(temp) <- toi
  reason <- c(reason,temp)
  message("Making a merge by joining by extending exons")
 }else{
  message("Making a merge by adding new junction")
  
  temp <- list("Taking model from Short Read - filling in model with junctions")
  names(temp) <- toi
  reason <- c(reason,temp)
  
  overlapAlt2 <- findOverlaps(AltModel,overlapRange,type = "within")
  missingExons <- NULL
  if(length(overlapAlt2)> 0){
   message("Help")
   missingExons <- AltModel[overlapAlt2@from]
  }
  
  polyA_lastExon <- polyAModel[length(polyAModel)]
  following_firstExon <- followingModel[1]
  AltModel_overlap_polyA_lastExon <- AltModel[AltModel %over% polyA_lastExon]
  AltModel_overlap_following_firstExon <- AltModel[AltModel %over% following_firstExon]
  if(length(AltModel_overlap_polyA_lastExon) > 0){
   start(polyAModel[length(polyAModel)]) <- start(AltModel_overlap_polyA_lastExon)
  }else{
   polyAModel <- polyAModel[-length(polyAModel)]
  }
  end(followingModel[1]) <- end(AltModel_overlap_following_firstExon)
  

  tempFollowerList <- list(followingTranscript$tx_name)
  names(tempFollowerList) <- polyTranscript$tx_name
  renameFollowersInList <- c(renameFollowersInList,tempFollowerList)
  polyAModel <- c(polyAModel,missingExons,followingModel)
 }
 }else{
  temp <- list("No overlapping exons, connectioing by junction")
  tempFollowerList <- list(followingTranscript$tx_name)
  names(tempFollowerList) <- polyTranscript$tx_name
  renameFollowersInList <- c(renameFollowersInList,tempFollowerList)
  polyAModel <- c(polyAModel,followingModel)
  names(temp) <- toi
  reason <- c(reason,temp)
 } 
 }else{
  temp <- list("Following didnt overlap same Shortread Stringtie gene as polyA gene")
  names(temp) <- toi
  reason <- c(reason,temp)
  
 }
 }else{
  temp <- list("Following didnt overlap same RefSeq gene as polyA gene")
  names(temp) <- toi
  reason <- c(reason,temp)
 } 
 mcols(polyAModel) <- NULL
 mcols(polyAModel) <- data.frame(source="StringTie_curated",type="exon",score=NA,phase=NA,gene_id=gsub("\\.\\d+$","",toi),transcript_id=toi)
 tempGR <- GRangesList(polyAModel)
 names(tempGR) <- toi
 polAmodelList <- c(polAmodelList,tempGR)
}

temp <- unlist(polAmodelList)
rtracklayer::export(temp,con="test_neg1.gtf") 

tempOverlap <- findOverlaps(polAmodelList,polAmodelList) %>% as.data.frame()
tempOverlap <- apply(tempOverlap,1,sort) %>% t %>% as.data.frame %>% distinct
tempOverlap <- tempOverlap[tempOverlap[,1] != tempOverlap[,2],]

polAmodelList2 <- polAmodelList
for(i in 1:nrow(tempOverlap)){
 temp <- GenomicRanges::reduce(c(polAmodelList[[tempOverlap[i,2]]],polAmodelList[[tempOverlap[i,1]]]))
 mcols(temp) <- NULL
 mcols(temp) <- data.frame(source="StringTie_curated",type="exon",score=NA,phase=NA,
              gene_id=gsub("\\.\\d+$","",names(polAmodelList[tempOverlap[i,2]])),
              transcript_id=names(polAmodelList[tempOverlap[i,2]]))
 polAmodelList2[[tempOverlap[i,2]]] <- temp
}

polAmodelList2 <- polAmodelList2[-tempOverlap[,1]]

temp <- unlist(polAmodelList2)
rtracklayer::export(temp,con="test_neg2.gtf") 


LongStringtie_GTF2 <- import("PacBio/Stringtie/Stringtie_Sample1.gtf")
LongStringtie_GTF2 <- LongStringtie_GTF2[strand(LongStringtie_GTF2) != "*"]
LongStringtie_GTF2 <- LongStringtie_GTF2[LongStringtie_GTF2$type == "exon"]


for(i in 1:length(polAmodelList2)){
 x <- 1
 message(i)
 toi <- names(polAmodelList2)[i]
 goi <- gsub("\\.\\d+$","",toi)
 removedByMerge_toi <- renameFollowersInList[[toi]]
  while(x == 1){ 
  if(any(names(renameFollowersInList) %in% removedByMerge_toi[length(removedByMerge_toi)])){
   removedByMerge_toi <- c(removedByMerge_toi,unlist(renameFollowersInList[removedByMerge_toi[length(removedByMerge_toi)]]))
  }else{
   x <- 0
  }
 }
 removedByMerge_goi <- gsub("\\.\\d+$","",removedByMerge_toi)
 LongStringtie_GTF2 <- LongStringtie_GTF2[!LongStringtie_GTF2$transcript_id %in% toi,]
 LongStringtie_GTF2 <- LongStringtie_GTF2[!LongStringtie_GTF2$transcript_id %in% removedByMerge_toi,]
 LongStringtie_GTF2$gene_id[LongStringtie_GTF2$gene_id %in% removedByMerge_goi] <- goi
 LongStringtie_GTF2 <- c(LongStringtie_GTF2,polAmodelList2[[i]])
}

export(LongStringtie_GTF2,"test_neg_full.gtf")

## Positive genes

stringtie_TranscriptsWithTXandGene_Pos <- stringtie_TranscriptsWithTXandGene[strand(stringtie_TranscriptsWithTXandGene) == "+"]
stringtie_TranscriptsWithTXandGene_Pos$length <- width(stringtie_TranscriptsWithTXandGene_Pos)


stringtie_TranscriptsWithTXandGene_PosFilt <- stringtie_TranscriptsWithTXandGene_Pos %>% 
 as.data.frame() %>% 
 arrange(seqnames,start,desc(length)) %>% 
 distinct(gene_id,.keep_all = TRUE) %>% GRanges()

stringtie_TranscriptsWithTXandGene_PosFilt_PolyA <- stringtie_TranscriptsWithTXandGene_PosFilt[stringtie_TranscriptsWithTXandGene_PosFilt$tx_name %in% potentialPolyFakeTranscripts]

stringtie_TranscriptsWithTXandGene_PosFilt_PolyA <- stringtie_TranscriptsWithTXandGene_PosFilt_PolyA[unlist(stringtie_TranscriptsWithTXandGene_PosFilt_PolyA$gene_id) %in% polAToClean,]




stringtie_TranscriptsWithTXandGene_PosFilt_PolyA %<>% sort(decreasing = TRUE)

follow_LongStringtie_exonsByTranscript <- precede(stringtie_TranscriptsWithTXandGene_PosFilt_PolyA,allstringtie_Transcripts)


polAmodelList <- GRangesList()
reason <- list()
renameFollowersInList <- list()



for(i in 1:length(stringtie_TranscriptsWithTXandGene_PosFilt_PolyA)){
 message("Primed gene ",i)
 toi <- stringtie_TranscriptsWithTXandGene_PosFilt_PolyA$tx_name[i]
 followingTranscript <- allstringtie_Transcripts[follow_LongStringtie_exonsByTranscript[stringtie_TranscriptsWithTXandGene_PosFilt_PolyA$tx_name %in% toi]]
 polyTranscript <- allstringtie_Transcripts[allstringtie_Transcripts$tx_name %in% toi]
 
 overlapRange <- GRanges(seqnames(polyTranscript),
             IRanges(start=end(polyTranscript),end=start(followingTranscript)),strand = strand(polyTranscript)
             )
 potentialTranscript <- c(polyTranscript,followingTranscript)
 
 followingModel <- LongStringtie_exonsByTranscript[[followingTranscript$tx_name]]
 polyAModel <- LongStringtie_exonsByTranscript[[polyTranscript$tx_name]]
 
 if(any(duplicated(findOverlaps(potentialTranscript,BRCDK_exonsByGene)@to))){
 
 
 if(any(duplicated(findOverlaps(potentialTranscript,ShortStringtie_exonsByGene)@to))){
 
 
 AltModels <- ShortStringtie_exonsByTranscript[ShortStringtie_exonsByTranscript %over% overlapRange]
 
 if(length(AltModels) > 0){
 
 polyA_lastExon <- polyAModel[length(polyAModel)]
 following_firstExon <- followingModel[1]
  
 commonExonsOverlap <- intersect(findOverlaps(AltModels,c(polyA_lastExon))@from,
 findOverlaps(AltModels,c(following_firstExon))@from)
 if(length(commonExonsOverlap) > 0){
  AltModels <- AltModels[commonExonsOverlap]
 }
 AltModel <- AltModels[[which.max(sum(width(AltModels)))]]
 overlapAlt <- findOverlaps(overlapRange,AltModel,type = "within")
 if(length(overlapAlt) == 1){
  tempFollowerList <- list(followingTranscript$tx_name)
  names(tempFollowerList) <- polyTranscript$tx_name
  renameFollowersInList <- c(renameFollowersInList,tempFollowerList)
  polyAModel <- GenomicRanges::reduce(c(polyAModel,overlapRange,followingModel))
  temp <- list("Taking model from Short Read - creating a merged exon")
  names(temp) <- toi
  reason <- c(reason,temp)
  message("Making a merge by joining by extending exons")
 }else{
  message("Making a merge by adding new junction")
  
  temp <- list("Taking model from Short Read - filling in model with junctions")
  names(temp) <- toi
  reason <- c(reason,temp)
  
  overlapAlt2 <- findOverlaps(AltModel,overlapRange,type = "within")
  missingExons <- NULL
  if(length(overlapAlt2)> 0){
   message("Help")
   missingExons <- AltModel[overlapAlt2@from]
  }
  
  polyA_lastExon <- polyAModel[length(polyAModel)]
  following_firstExon <- followingModel[1]
  AltModel_overlap_polyA_lastExon <- AltModel[AltModel %over% polyA_lastExon]
  AltModel_overlap_following_firstExon <- AltModel[AltModel %over% following_firstExon]
  if(length(AltModel_overlap_polyA_lastExon) > 0){
   end(polyAModel[length(polyAModel)]) <- end(AltModel_overlap_polyA_lastExon)
  }else{
   polyAModel <- polyAModel[-length(polyAModel)]
  }
  start(followingModel[1]) <- start(AltModel_overlap_following_firstExon)
  
  # AltModel_followEnd <- end(AltModel[precede(polyTranscript,AltModel)])
  # AltModel_proceedStart <- start(AltModel[follow(followingTranscript,AltModel)])
  # start(polyAModel[length(polyAModel)]) <- AltModel_proceedStart
  # end(followingModel[1]) <- AltModel_followEnd
  tempFollowerList <- list(followingTranscript$tx_name)
  names(tempFollowerList) <- polyTranscript$tx_name
  renameFollowersInList <- c(renameFollowersInList,tempFollowerList)
  polyAModel <- c(polyAModel,missingExons,followingModel)
 }
 }else{
  temp <- list("No overlapping exons, connectioing by junction")
  tempFollowerList <- list(followingTranscript$tx_name)
  names(tempFollowerList) <- polyTranscript$tx_name
  renameFollowersInList <- c(renameFollowersInList,tempFollowerList)
  polyAModel <- c(polyAModel,followingModel)
  names(temp) <- toi
  reason <- c(reason,temp)
 } 
 }else{
  temp <- list("Following didnt overlap same Shortread Stringtie gene as polyA gene")
  names(temp) <- toi
  reason <- c(reason,temp)
  
 }
 }else{
  temp <- list("Following didnt overlap same RefSeq gene as polyA gene")
  names(temp) <- toi
  reason <- c(reason,temp)
 } 
 mcols(polyAModel) <- NULL
 mcols(polyAModel) <- data.frame(source="StringTie_curated",type="exon",score=NA,phase=NA,gene_id=gsub("\\.\\d+$","",toi),transcript_id=toi)
 tempGR <- GRangesList(polyAModel)
 names(tempGR) <- toi
 polAmodelList <- c(polAmodelList,tempGR)
}

temp <- unlist(polAmodelList)
rtracklayer::export(temp,con="test_pos1.gtf") 

tempOverlap <- findOverlaps(polAmodelList,polAmodelList) %>% as.data.frame()
tempOverlap <- apply(tempOverlap,1,sort) %>% t %>% as.data.frame %>% distinct
tempOverlap <- tempOverlap[tempOverlap[,1] != tempOverlap[,2],]

polAmodelList2 <- polAmodelList
for(i in 1:nrow(tempOverlap)){
 temp <- GenomicRanges::reduce(c(polAmodelList[[tempOverlap[i,2]]],polAmodelList[[tempOverlap[i,1]]]))
 mcols(temp) <- NULL
 mcols(temp) <- data.frame(source="StringTie_curated",type="exon",score=NA,phase=NA,
              gene_id=gsub("\\.\\d+$","",names(polAmodelList[tempOverlap[i,2]])),
              transcript_id=names(polAmodelList[tempOverlap[i,2]]))
 polAmodelList2[[tempOverlap[i,2]]] <- temp
}

polAmodelList2 <- polAmodelList2[-tempOverlap[,1]]

temp <- unlist(polAmodelList2)
rtracklayer::export(temp,con="test_pos2.gtf") 


LongStringtie_GTF3 <- import("PacBio/Stringtie/Stringtie_Sample1.gtf")
LongStringtie_GTF3 <- LongStringtie_GTF3[strand(LongStringtie_GTF3) != "*"]
LongStringtie_GTF3 <- LongStringtie_GTF3[LongStringtie_GTF3$type == "exon"]


for(i in 1:length(polAmodelList2)){
 x <- 1
 message(i)
 toi <- names(polAmodelList2)[i]
 goi <- gsub("\\.\\d+$","",toi)
 removedByMerge_toi <- renameFollowersInList[[toi]]
  while(x == 1){ 
  if(any(names(renameFollowersInList) %in% removedByMerge_toi[length(removedByMerge_toi)])){
   removedByMerge_toi <- c(removedByMerge_toi,unlist(renameFollowersInList[removedByMerge_toi[length(removedByMerge_toi)]]))
  }else{
   x <- 0
  }
 }
 removedByMerge_goi <- gsub("\\.\\d+$","",removedByMerge_toi)
 LongStringtie_GTF3 <- LongStringtie_GTF3[!LongStringtie_GTF3$transcript_id %in% toi,]
 LongStringtie_GTF3 <- LongStringtie_GTF3[!LongStringtie_GTF3$transcript_id %in% removedByMerge_toi,]
 LongStringtie_GTF3$gene_id[LongStringtie_GTF3$gene_id %in% removedByMerge_goi] <- goi
 LongStringtie_GTF3 <- c(LongStringtie_GTF3,polAmodelList2[[i]])
}


# STRG.11313 -- didnt work -- should be named STRG.11315
export(LongStringtie_GTF3,"test_pos_full.gtf")



for(i in 1:length(polAmodelList2)){
 x <- 1
 message(i)
 toi <- names(polAmodelList2)[i]
 goi <- gsub("\\.\\d+$","",toi)
 removedByMerge_toi <- renameFollowersInList[[toi]]
  while(x == 1){ 
  if(any(names(renameFollowersInList) %in% removedByMerge_toi[length(removedByMerge_toi)])){
   removedByMerge_toi <- c(removedByMerge_toi,unlist(renameFollowersInList[removedByMerge_toi[length(removedByMerge_toi)]]))
  }else{
   x <- 0
  }
 }
 removedByMerge_goi <- gsub("\\.\\d+$","",removedByMerge_toi)
 LongStringtie_GTF2 <- LongStringtie_GTF2[!LongStringtie_GTF2$transcript_id %in% toi,]
 LongStringtie_GTF2 <- LongStringtie_GTF2[!LongStringtie_GTF2$transcript_id %in% removedByMerge_toi,]
 LongStringtie_GTF2$gene_id[LongStringtie_GTF2$gene_id %in% removedByMerge_goi] <- goi
 LongStringtie_GTF2 <- c(LongStringtie_GTF2,polAmodelList2[[i]])
}

# STRG.7191
# STRG.11313 -- didnt work -- should be named STRG.11315
export(LongStringtie_GTF2,"test_polyA_fixed.gtf")


```


### Assess confounded gene models after polyA internal priming rescue
```{r}
LongStringtie_txdb2 <- makeTxDbFromGRanges(LongStringtie_GTF2)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb2,by="gene")

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]

# overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% potentialPolyFakeGenes
# ,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

postPolyAFiltering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
       Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
  set_colnames("Counfounding_Gene_Number") %>% datatable()
```

#### Manually assess

```{r,eval=FALSE,echo=FALSE}
"LOC105274459" - Refseq UTR overlaps 5' of stringtie
"LOC105274940" - Floating UTR (Refseq thinks two, short read thinks 1 -doesnt pass polyA filter for overlaps)
"LOC105275287" - Short read and refseq match - Dont accept my model
"LOC105275877" - Short read has gap, long read has gap
"LOC105275915" - Short read has gap, long read has gap
"LOC105276245" - Short read has what looks like run on when long read has gap
"LOC105276271" - Short read has gap, long read has gap
"LOC105276416" - Short read and refseq match - Dont accept my model -- all exons represented in short read
"LOC105276519" - Short read has what looks like run on when long read has gap -- but more complex example
"LOC105276799" - Short read and refseq match - Dont accept my model -- all exons represented in short read
"LOC105276940"
"LOC105276941"
"LOC105277123"
"LOC105277125"
"LOC105277202"
"LOC105277234"
"LOC105277394"
"LOC105277406"
"LOC105277606"
"LOC105277680"
"LOC105277793"
"LOC105277941"
"LOC105277974"
"LOC105278064"
"LOC105278066"
"LOC105278262"
"LOC105278327"
"LOC105278329"
"LOC105278470"
"LOC105278569"
"LOC105278631"
"LOC105278732"
"LOC105278891"
"LOC105278994"
"LOC105279031"
"LOC105279292"
"LOC105279462"
"LOC105279542"
"LOC105280056"
"LOC105280265"
"LOC105280271"
"LOC105280668"
"LOC105280761"
"LOC105280850"
"LOC105281370"
"LOC105281448"
"LOC105281730"
"LOC105281903"
"LOC105282012"
"LOC105282112"
"LOC105282359"
"LOC105282423"
"LOC105283145"
"LOC105283180"
"LOC105283212"
"LOC105283214"
"LOC105283491"
"LOC105283615"
"LOC105283711"
"LOC105283806"
"LOC105283824"
"LOC105283841"
"LOC105283935"
"LOC105284155"
"LOC105284264"
"LOC105284265"
"LOC105284340"
"LOC105284377"
"LOC105284484"
"LOC105284524"
"LOC105284529"
"LOC105284540"
"LOC105284687"
"LOC105284786"
"LOC105284864"
"LOC105284876"
"LOC105284886"
"LOC105284909"
"LOC105285560"
"LOC105285619"
"LOC105285912"
"LOC105286340"
"LOC105286632"
"LOC105286657"
"LOC105286732"
"LOC105286777"
"LOC105287099"
"LOC105287672"
"LOC105287773"
"LOC105287787"
"LOC105287969"
"LOC105287977"
"LOC105288145"
"LOC113561858"
"LOC113562368"
"LOC113563199"
"LOC113563593"
```


## Use ShortRead models to confirm LongRead models which split RefSeq genes.

### Assess whether gap between LongRead genes overlapping single RefSeq gene exist in ShortRead gene models

```{r}
overlappingGenesDF_MultiHits <- overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% multi_BRCDK,]
unique_BRCDK_MultiHits <- overlappingGenesDF_MultiHits %>% pull(queryHitsNames) %>% unique

# GenBank_exonsByGene_noSingle <- GenBank_exonsByGene[lengths(GenBank_exonsByGene) > 1]
filterVec_3 <- vector("logical",length = length(unique_BRCDK_MultiHits))
for(i in 1:length(unique_BRCDK_MultiHits)){
 message(i)
 StringTie_Genes_Temp <- overlappingGenesDF_MultiHits[overlappingGenesDF_MultiHits$queryHitsNames %in% 
                 unique_BRCDK_MultiHits[i],
                "subjectHitsNames"]
 OverlapTableTemp_1 <- as.data.frame(findOverlaps(LongStringtie_exonsByGene[StringTie_Genes_Temp],ShortStringtie_exonsByGene))
 if(!any(duplicated(OverlapTableTemp_1$subjectHits)) & nrow(OverlapTableTemp_1) == 2){
  filterVec_3[i] <- TRUE
 }else{
  filterVec_3[i] <- FALSE
 }
}
names(filterVec_3) <- unique_BRCDK_MultiHits
```

### Filter out genes and assess confounded gene models after rescue

```{r}
LongStringtie_txdb2 <- makeTxDbFromGRanges(LongStringtie_GTF2)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb2,by="gene")

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_3[filterVec_3])
,]
# overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% potentialPolyFakeGenes
# ,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]
multi_BRCDKOld <- multi_BRCDK


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

ShortReadAndLongReadGapFiltering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
       Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
  set_colnames("Counfounding_Gene_Number") %>% datatable()
```

#### Manually assess

```{r,eval=FALSE,echo=FALSE}
"LOC105274459" - Refseq UTR overlaps 5' of stringtie
"LOC105274940" - Floating UTR (Refseq thinks two, short read thinks 1 -doesnt pass polyA filter for overlaps)
"LOC105274987" - Refseq 3' UTR overlaps 3' of stringtie
"LOC105275005" - Short read has what looks like run on when long read has gap
"LOC105275287" - Short read and refseq match - Dont accept my model
"LOC105275722" - Short read has what looks like run on when long read has gap .. tthis looks like it may just miss poly tail
"LOC105275990" - Short read and refseq match - Dont accept my model
"LOC105276245" - Short read has what looks like run on when long read has gap
"LOC105276354" - Long read overlaps another long read gene, long read and short read match
"LOC105276377" - Short read and refseq match - Dont accept my model -- all exons represented in short read
"LOC105276416" - Short read and refseq match - Dont accept my model -- all exons represented in short read
"LOC105276519" - Short read has what looks like run on when long read has gap -- but more complex example
"LOC105276799" - Short read and refseq match - Dont accept my model -- all exons represented in short read
"LOC105276868"
"LOC105276940"
"LOC105276941"
"LOC105277123"
"LOC105277234"
"LOC105277394"
"LOC105277425"
"LOC105277680"
"LOC105277793"
"LOC105278262"
"LOC105278327"
"LOC105278470"
"LOC105278474"
"LOC105278631"
"LOC105278802"
"LOC105278994"
"LOC105279031"
"LOC105279301"
"LOC105279462"
"LOC105279542"
"LOC105280056"
"LOC105280265"
"LOC105280271"
"LOC105280421"
"LOC105280692"
"LOC105280850"
"LOC105281207"
"LOC105281330"
"LOC105281370"
"LOC105281402"
"LOC105281448"
"LOC105281587"
"LOC105281730"
"LOC105281789"
"LOC105281903"
"LOC105282012"
"LOC105282112"
"LOC105282237"
"LOC105282423"
"LOC105283145"
"LOC105283180"
"LOC105283214"
"LOC105283491"
"LOC105283615"
"LOC105283711"
"LOC105283806"
"LOC105283824"
"LOC105283841"
"LOC105283935"
"LOC105284155"
"LOC105284264"
"LOC105284265"
"LOC105284340"
"LOC105284377"
"LOC105284484"
"LOC105284524"
"LOC105284529"
"LOC105284540"
"LOC105284788"
"LOC105284864"
"LOC105284876"
"LOC105284886"
"LOC105284909"
"LOC105285619"
"LOC105285912"
"LOC105286207"
"LOC105286340"
"LOC105286518"
"LOC105286632"
"LOC105286657"
"LOC105286692"
"LOC105286777"
"LOC105286974"
"LOC105287099"
"LOC105287188"
"LOC105287672"
"LOC105287773"
"LOC105287969"
"LOC105287977"
"LOC105288058"
"LOC105288145"
"LOC113561858"
"LOC113563199"
```

## Clean up LongRead models where genes overlap

### Identify and clean up transcripts i overlapping gene models within LongRead data.

```{r,eval=TRUE}

LongStringtie_exonsByTranscript <- exonsBy(LongStringtie_txdb2,by="tx",use.names=TRUE)
LongStringtie_GTF_GeneToTranscript <- LongStringtie_GTF2 %>% mcols %>% 
 as.data.frame() %>% 
 dplyr::select(gene_id,transcript_id) %>% 
 distinct()

selfOverlaps <- findOverlaps(LongStringtie_exonsByTranscript,LongStringtie_exonsByTranscript)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongStringtie_exonsByTranscript)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongStringtie_exonsByTranscript)[temp_selfOverlaps$subjectHits]

temp_selfOverlaps$queryHitsGeneNames <- LongStringtie_GTF_GeneToTranscript$gene_id[match(temp_selfOverlaps$queryHitsNames,
                       LongStringtie_GTF_GeneToTranscript$transcript_id)]

temp_selfOverlaps$subjectHitsGeneNames <- LongStringtie_GTF_GeneToTranscript$gene_id[match(temp_selfOverlaps$subjectHitsNames,
                       LongStringtie_GTF_GeneToTranscript$transcript_id)]

temp_selfOverlaps$subjectHitsGeneNames[is.na(temp_selfOverlaps$subjectHitsGeneNames)] <- temp_selfOverlaps$subjectHitsNames[is.na(temp_selfOverlaps$subjectHitsGeneNames)]

temp_selfOverlaps$queryHitsGeneNames[is.na(temp_selfOverlaps$queryHitsGeneNames)] <- temp_selfOverlaps$queryHitsNames[is.na(temp_selfOverlaps$queryHitsGeneNames)]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHitsGeneNames != temp_selfOverlaps$subjectHitsGeneNames,]
nrow(temp_selfOverlaps)

allSelfOverlapGenes <- temp_selfOverlaps[,3] %>% unique

if(!file.exists("RObjects/LongStringtie_NoGeneModelOverlap.gtf")){



while(nrow(temp_selfOverlaps) > 0){
message("Overlapping transcripts - ",nrow(temp_selfOverlaps)) 
 loop <- 1
# XM_026973633.1 1372
updatedGene <- GRangesList()
for(i in 1:nrow(temp_selfOverlaps)){
 geneA <- temp_selfOverlaps[i,3]
 geneB <- temp_selfOverlaps[i,4]
 exons_geneA <- LongStringtie_exonsByTranscript[geneA]
 exons_geneB <- LongStringtie_exonsByTranscript[geneB]
 if(all(strand(exons_geneA) == "+")){
  if(start(exons_geneA[[1]][1]) < start(exons_geneB[[1]][1])){
   proceedingGene <- exons_geneA
   followingGene <- exons_geneB
  }else{
   proceedingGene <- exons_geneB
   followingGene <- exons_geneA
  }
  rangeToRestrict <- c(start(proceedingGene[[1]][1]),start(followingGene[[1]][1])-20)
  updatedGene <- c(updatedGene,restrict(proceedingGene[1],start = rangeToRestrict[1],end=rangeToRestrict[2]))
   # print("+")
 }else if(all(strand(exons_geneA) == "-")){
  if(end(exons_geneA[[1]][length(exons_geneA[[1]])]) > end(exons_geneB[[1]][length(exons_geneB[[1]])])){
   proceedingGene <- exons_geneA
   followingGene <- exons_geneB
  }else{
   proceedingGene <- exons_geneB
   followingGene <- exons_geneA
  }
  rangeToRestrict <- c(end(followingGene[[1]][1])+20,end(proceedingGene[[1]][1]))
  updatedGene <- c(updatedGene,restrict(proceedingGene[1],start = rangeToRestrict[1],end=rangeToRestrict[2]))
   # print("-")
 }
 if(i %% 100 == 0)message(loop,":",i)
}

LongStringtie_exonsByTranscript[names(updatedGene)] <- updatedGene
# 
# temp <- unlist(LongStringtie_exonsByTranscript)
# df_temp <- data.frame(source="StringTie_curated",
#              type="exon",
#              score=NA,phase=NA,
#              gene_id=LongStringtie_GTF_GeneToTranscript$gene_id[
#               match(names(temp),
#                        LongStringtie_GTF_GeneToTranscript$transcript_id)
#               ],
#              transcript_id=names(temp))
# 
# df_temp$gene_id[is.na(df_temp$gene_id)] <- df_temp$transcript_id[is.na(df_temp$gene_id)]
# mcols(temp) <- df_temp
# class(temp$phase) <- "integer"
# LongStringtie_txdb <- makeTxDbFromGRanges(temp)
# LongStringtie_exonsByTranscript <- exonsBy(LongStringtie_txdb,by="tx",use.names=TRUE)
# LongStringtie_GTF_GeneToTranscript <- LongStringtie_GTF %>% mcols %>% 
#  as.data.frame() %>% 
#  dplyr::filter(type=="transcript") %>% 
#  dplyr::select(gene_id,transcript_id) %>% 
#  distinct()

selfOverlaps <- findOverlaps(LongStringtie_exonsByTranscript,LongStringtie_exonsByTranscript)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongStringtie_exonsByTranscript)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongStringtie_exonsByTranscript)[temp_selfOverlaps$subjectHits]

temp_selfOverlaps$queryHitsGeneNames <- LongStringtie_GTF_GeneToTranscript$gene_id[match(temp_selfOverlaps$queryHitsNames,
                       LongStringtie_GTF_GeneToTranscript$transcript_id)]

temp_selfOverlaps$subjectHitsGeneNames <- LongStringtie_GTF_GeneToTranscript$gene_id[match(temp_selfOverlaps$subjectHitsNames,
                       LongStringtie_GTF_GeneToTranscript$transcript_id)]

temp_selfOverlaps$subjectHitsGeneNames[is.na(temp_selfOverlaps$subjectHitsGeneNames)] <- temp_selfOverlaps$subjectHitsNames[is.na(temp_selfOverlaps$subjectHitsGeneNames)]

temp_selfOverlaps$queryHitsGeneNames[is.na(temp_selfOverlaps$queryHitsGeneNames)] <- temp_selfOverlaps$queryHitsNames[is.na(temp_selfOverlaps$queryHitsGeneNames)]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHitsGeneNames != temp_selfOverlaps$subjectHitsGeneNames,]
loop <- loop+1

}
temp <- unlist(LongStringtie_exonsByTranscript)
df_temp <- data.frame(source="StringTie_curated",
             type="exon",
             score=NA,phase=NA,
             gene_id=LongStringtie_GTF_GeneToTranscript$gene_id[
              match(names(temp),
                       LongStringtie_GTF_GeneToTranscript$transcript_id)
              ],
             transcript_id=names(temp))

df_temp$gene_id[is.na(df_temp$gene_id)] <- df_temp$transcript_id[is.na(df_temp$gene_id)]
mcols(temp) <- df_temp
class(temp$phase) <- "integer"
rtracklayer::export(temp[temp$transcript_id %in% allSelfOverlapGenes,],con="LongStringtie_NoGeneModelOverlap.gtf") 
rtracklayer::export(temp,con="LongStringtie_NoGeneModelOverlapFull.gtf") 
saveRDS(temp,"LongStringtie_NoGeneModelOverlapFull.RDS")
}

# LongStringtie_GTF <- import("LongStringtie_NoGeneModelOverlapFull.gtf")
LongStringtie_GTF2 <- readRDS("RObjects/LongStringtie_NoGeneModelOverlapFull.RDS")
LongStringtie_txdb <- makeTxDbFromGRanges(LongStringtie_GTF2)
LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb,by = "gene")
selfOverlaps <- findOverlaps(LongStringtie_exonsByGene,LongStringtie_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongStringtie_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongStringtie_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]
nrow(temp_selfOverlaps)
```

### Assess confounded gene models after fixing overlapping LongRead gene models

```{r}
LongStringtie_txdb2 <- makeTxDbFromGRanges(LongStringtie_GTF2)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb2,by="gene")

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_3[filterVec_3])
,]
# overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% potentialPolyFakeGenes
# ,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

SelfOverlapStringtie_Filtering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
       Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
  set_colnames("Counfounding_Gene_Number") %>% datatable()

```


## Assess whether LongRead models splitting RefSeq genes are caused by ShortRead run on across gene models.

### Evaluae whether gap exist in LongRead gene models is within a single exon in ShortRead gene models

```{r}
overlappingGenesDF_MultiHits <- overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% multi_BRCDK,]
unique_BRCDK_MultiHits <- overlappingGenesDF_MultiHits %>% pull(queryHitsNames) %>% unique

# GenBank_exonsByGene_noSingle <- GenBank_exonsByGene[lengths(GenBank_exonsByGene) > 1]
filterVec_4 <- vector("logical",length = length(unique_BRCDK_MultiHits))
for(i in 1:length(unique_BRCDK_MultiHits)){
 message(i)
 StringTie_Genes_Temp <- overlappingGenesDF_MultiHits[overlappingGenesDF_MultiHits$queryHitsNames %in% 
                 unique_BRCDK_MultiHits[i],
                "subjectHitsNames"]

 if(length(StringTie_Genes_Temp) == 2){
  
  exons_geneA <- LongStringtie_exonsByGene[StringTie_Genes_Temp[1]]
  exons_geneB <- LongStringtie_exonsByGene[StringTie_Genes_Temp[2]]
  gene_geneA <- GRanges(unique(as.character(seqnames(exons_geneA[[1]]))),
             IRanges(min(start(exons_geneA)),max(end(exons_geneA))))
  gene_geneB <- GRanges(unique(as.character(seqnames(exons_geneB[[1]]))),
             IRanges(min(start(exons_geneB)),max(end(exons_geneB)))) 
  withinCheck <- findOverlaps(gene_geneB,gene_geneA,type = "within")
  withinCheck2 <- findOverlaps(gene_geneA,gene_geneB,type = "within")
  if(length(withinCheck) == 0 & length(withinCheck2) == 0 ){
   # findOverlaps(exons_geneA,exons_geneB)
   
   if(all(strand(exons_geneA) == "+")){
    if(start(exons_geneA[[1]][1]) < start(exons_geneB[[1]][1])){
     proceedingGene <- exons_geneA
     followingGene <- exons_geneB
    }else{
     proceedingGene <- exons_geneB
     followingGene <- exons_geneA
    }
    rangeToEval <- GRanges(unique(as.character(seqnames(followingGene[[1]]))),
                IRanges(start=end(proceedingGene[[1]][length(proceedingGene[[1]])]),end=start(followingGene[[1]][1])),strand="+")
    # print("+")
   }else if(all(strand(exons_geneA) == "-")){
    if(end(exons_geneA[[1]][length(exons_geneA[[1]])]) > end(exons_geneB[[1]][length(exons_geneB[[1]])])){
     proceedingGene <- exons_geneA
     followingGene <- exons_geneB
    }else{
     proceedingGene <- exons_geneB
     followingGene <- exons_geneA
    }
    rangeToEval <- GRanges(unique(as.character(seqnames(followingGene[[1]]))),
                IRanges(start=end(followingGene[[1]][length(followingGene[[1]])]),
                    end=start(proceedingGene[[1]][1])),
                strand="-")
    # print("-")
   }
   tempWithinExon <- findOverlaps(rangeToEval,BRCDK_exonsByGene[[unique_BRCDK_MultiHits[i]]],type="within")
   if(length(tempWithinExon) > 0){
    filterVec_4[i] <- TRUE
   }
  }
 }

}
names(filterVec_4) <- unique_BRCDK_MultiHits
```


### Assess effect on confounding gene models.

```{r}
LongStringtie_txdb2 <- makeTxDbFromGRanges(LongStringtie_GTF2)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb2,by="gene")

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_3[filterVec_3])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_4[filterVec_4])
,]
# overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% potentialPolyFakeGenes
# ,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

GapAllInExon_Filtering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
       Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
  set_colnames("Counfounding_Gene_Number") %>% datatable()

```



#### Manually assess
```{r, eval=FALSE,echo=FALSE}
"LOC105274459"
"LOC105275005"
"LOC105275287"
"LOC105275990"
"LOC105276245"
"LOC105276377"
"LOC105276416"
"LOC105276519"
"LOC105276799"
"LOC105276868"
"LOC105276940"
"LOC105277123"
"LOC105277234"
"LOC105277394"
"LOC105277680"
"LOC105278262"
"LOC105278470"
"LOC105278474"
"LOC105278631"
"LOC105278802"
"LOC105278994"
"LOC105279031"
"LOC105279301"
"LOC105279462"
"LOC105280056"
"LOC105280265"
"LOC105280271"
"LOC105280421"
"LOC105280692"
"LOC105280850"
"LOC105281330"
"LOC105281370"
"LOC105281402"
"LOC105281448"
"LOC105281587"
"LOC105281730"
"LOC105281789"
"LOC105281903"
"LOC105282237"
"LOC105282423"
"LOC105283145"
"LOC105283180"
"LOC105283491"
"LOC105283615"
"LOC105283711"
"LOC105283841"
"LOC105283935"
"LOC105284155"
"LOC105284264"
"LOC105284377"
"LOC105284484"
"LOC105284524"
"LOC105284529"
"LOC105284540"
"LOC105284788"
"LOC105284864"
"LOC105284876"
"LOC105284886"
"LOC105285619"
"LOC105285912"
"LOC105286207"
"LOC105286340"
"LOC105286518"
"LOC105286632"
"LOC105286657"
"LOC105286692"
"LOC105286974"
"LOC105287099"
"LOC105287672"
"LOC105287773"
"LOC105287969"
"LOC105287977"
"LOC105288145"
"LOC113561858"
"LOC113563199"
```

## Does ShortRead and RefSeq agree where gaps in Long read exist

### Assess whether gaps in LongRead data contain junctions/exons where ShortRead and RefSeq agree.

```{r}
overlappingGenesDF_MultiHits <- overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% multi_BRCDK,]
unique_BRCDK_MultiHits <- overlappingGenesDF_MultiHits %>% pull(queryHitsNames) %>% unique

# GenBank_exonsByGene_noSingle <- GenBank_exonsByGene[lengths(GenBank_exonsByGene) > 1]
filterVec_5 <- vector("logical",length = length(unique_BRCDK_MultiHits))
for(i in 1:length(unique_BRCDK_MultiHits)){
 message(i)
 StringTie_Genes_Temp <- overlappingGenesDF_MultiHits[overlappingGenesDF_MultiHits$queryHitsNames %in% 
                 unique_BRCDK_MultiHits[i],
                "subjectHitsNames"]

 if(length(StringTie_Genes_Temp) == 2){
  
  exons_geneA <- LongStringtie_exonsByGene[StringTie_Genes_Temp[1]]
  exons_geneB <- LongStringtie_exonsByGene[StringTie_Genes_Temp[2]]
  gene_geneA <- GRanges(unique(as.character(seqnames(exons_geneA[[1]]))),
             IRanges(min(start(exons_geneA)),max(end(exons_geneA))))
  gene_geneB <- GRanges(unique(as.character(seqnames(exons_geneB[[1]]))),
             IRanges(min(start(exons_geneB)),max(end(exons_geneB)))) 
  withinCheck <- findOverlaps(gene_geneB,gene_geneA,type = "within")
  withinCheck2 <- findOverlaps(gene_geneA,gene_geneB,type = "within")
  if(length(withinCheck) == 0 & length(withinCheck2) == 0 ){
   # findOverlaps(exons_geneA,exons_geneB)
   
   if(all(strand(exons_geneA) == "+")){
    if(start(exons_geneA[[1]][1]) < start(exons_geneB[[1]][1])){
     proceedingGene <- exons_geneA
     followingGene <- exons_geneB
    }else{
     proceedingGene <- exons_geneB
     followingGene <- exons_geneA
    }
    rangeToEval <- GRanges(unique(as.character(seqnames(followingGene[[1]]))),
                IRanges(start=end(proceedingGene[[1]][length(proceedingGene[[1]])]),end=start(followingGene[[1]][1])),strand="+")
    # print("+")
   }else if(all(strand(exons_geneA) == "-")){
    if(end(exons_geneA[[1]][length(exons_geneA[[1]])]) > end(exons_geneB[[1]][length(exons_geneB[[1]])])){
     proceedingGene <- exons_geneA
     followingGene <- exons_geneB
    }else{
     proceedingGene <- exons_geneB
     followingGene <- exons_geneA
    }
    rangeToEval <- GRanges(unique(as.character(seqnames(followingGene[[1]]))),
                IRanges(start=end(followingGene[[1]][length(followingGene[[1]])]),
                    end=start(proceedingGene[[1]][1])),
                strand="-")
    # print("-")
   }
   AltExonsHits <- findOverlaps(rangeToEval,BRCDK_exonsByGene[[unique_BRCDK_MultiHits[i]]])
   ShortReadExonsHits <- findOverlaps(rangeToEval,unlist(ShortStringtie_exonsByGene))
   AltExons <- unique(BRCDK_exonsByGene[[unique_BRCDK_MultiHits[i]]][AltExonsHits@to])
   ShortReadExons <- unique(unlist(ShortStringtie_exonsByGene)[ShortReadExonsHits@to])
   equalOverlap <- findOverlaps(AltExons,ShortReadExons,type = "equal")
   if(length(equalOverlap@from) == length(AltExons))
    filterVec_5[i] <- TRUE
   }
  }
 }

names(filterVec_5) <- unique_BRCDK_MultiHits
```



### Assess confounded gene models after assessing ShortRead and RefSeq agreement in LongRead gaps.

```{r}
LongStringtie_txdb2 <- makeTxDbFromGRanges(LongStringtie_GTF2)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb2,by="gene")

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_3[filterVec_3])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_4[filterVec_4])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_5[filterVec_5])
,]
# overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% potentialPolyFakeGenes
# ,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

GapMatchesRefseqAndShortRead_Filtering <- multi_BRCDK

```

#### Manually assess

```{r,eval=FALSE,echo=FALSE}
"LOC105274459" - Accept both - stringtie 5' prime overlaps 3' Refseq
"LOC105275990" - Not sure why this wasnt filtered.. shortread stringtie and Refseq match
"LOC105276245" - Accept Stringtie long 
"LOC105277680" - Reject stringtie -accept Refseq
"LOC105278262" - Reject stringtie -accept Refseq
"LOC105278631" - Reject stringtie -accept Refseq
"LOC105278994" - Accept Stringtie long 
"LOC105279031" - Accept Stringtie long 
"LOC105280271" - Accept both - Remove STRG.118.1 and merge
"LOC105280692" - Reject stringtie -accept Refseq
"LOC105281370" - Reject stringtie -accept Refseq
"LOC105281402" - Accept Stringtie long  
"LOC105281448" - Reject stringtie -accept Refseq
"LOC105281730" - Reject stringtie -accept Refseq
"LOC105282423" - Accept Stringtie long
"LOC105283711" - Accept Stringtie long - after merge STRG.10347 to STRG.10344.4
"LOC105283935" - Accept Stringtie long 
"LOC105284377" - Reject stringtie -accept Refseq
"LOC105284524" - Reject stringtie -accept Refseq
"LOC105284788" - Accept Stringtie long
"LOC105286207" - Accept Stringtie long
"LOC105286340" - Reject stringtie -accept Refseq Not sure why this wasnt filtered.. shortread stringtie and Refseq match
"LOC105286632" - Accept Stringtie long
"LOC105286657" - Reject stringtie -accept Refseq 
"LOC105287969" - Reject stringtie -accept ShortRead 
"LOC105287977" - Accept Stringtie long
"LOC105288145" - Accept Stringtie long
"LOC113561858" - Trim STRG.4488.2 and STRG.4488.1 - Move STRG.4488.2 to new gene. then Accept Stringtie long
"LOC113563199" - Accept Stringtie long
```

## Fix last few manually before rejecting gene models

### Manual fix for last few gene models by assessment of Long and Short Read coverage across tissues and existing models.

```{r,eval=TRUE}
BRCDK_exonsByTranscript <- exonsBy(BRCDK_txdb,by="tx",use.names=TRUE)
LongStringtie_exonsByTranscript <- exonsBy(LongStringtie_txdb2,by="tx",use.names=TRUE)


BRCDK_GTF_GeneToTranscript <- BRCDK_GTF %>% mcols %>% 
 as.data.frame() %>% 
 dplyr::select(gene_id,transcript_id) %>% 
 distinct()

transcriptsOfI <- BRCDK_GTF_GeneToTranscript[BRCDK_GTF_GeneToTranscript[,1] %in% "LOC105274459",2] %>% unique %>% .[!. %in% ""]

conflicting_end <- end(LongStringtie_exonsByTranscript[["STRG.10779.2"]][1])

for(i in 1:length(transcriptsOfI)){
 BRCDK_exonsByTranscript[[transcriptsOfI[i]]][length(BRCDK_exonsByTranscript[[transcriptsOfI[i]]])] <- restrict(BRCDK_exonsByTranscript[[transcriptsOfI[i]]][length(BRCDK_exonsByTranscript[[transcriptsOfI[i]]])],
      start = conflicting_end+20,
      end=end(BRCDK_exonsByTranscript[[transcriptsOfI[i]]][length(BRCDK_exonsByTranscript[[transcriptsOfI[i]]])]))
}

temp <- unlist(BRCDK_exonsByTranscript)
df_temp <- data.frame(source="StringTie_curated",
             type="exon",
             score=NA,phase=NA,
             gene_id=BRCDK_GTF_GeneToTranscript$gene_id[
              match(names(temp),
                       BRCDK_GTF_GeneToTranscript$transcript_id)
              ],
             transcript_id=names(temp))

df_temp$gene_id[is.na(df_temp$gene_id)] <- df_temp$transcript_id[is.na(df_temp$gene_id)]
mcols(temp) <- df_temp
class(temp$phase) <- "integer"
rtracklayer::export(temp[temp$transcript_id %in% transcriptsOfI,],con="BRCDK_LOC105274459.gtf") 

# BRCDK_GTF <- import("BRCDK_NoGeneModelOverlapFull.gtf")
BRCDK_GTF <- temp
BRCDK_txdb <- makeTxDbFromGRanges(BRCDK_GTF)
BRCDK_exonsByGene <- exonsBy(BRCDK_txdb,by = "gene")
selfOverlaps <- findOverlaps(BRCDK_exonsByGene,BRCDK_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(BRCDK_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(BRCDK_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]
nrow(temp_selfOverlaps)
```

```{r}
LongStringtie_GTF2 <- LongStringtie_GTF2[!LongStringtie_GTF2$transcript_id == "STRG.118.1",]

LongStringtie_GTF2[LongStringtie_GTF2$transcript_id == "STRG.10347.1"]$gene_id <- "STRG.10344"
LongStringtie_GTF2[LongStringtie_GTF2$transcript_id == "STRG.10347.1"]$transcript_id <- "STRG.10344.1"
LongStringtie_GTF2 <- LongStringtie_GTF2[-which(LongStringtie_GTF2$transcript_id == "STRG.4488.2")[1]]

```


### Assess confounded gene models after rescue
```{r}
LongStringtie_txdb2 <- makeTxDbFromGRanges(LongStringtie_GTF2)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb2,by="gene")

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_3[filterVec_3])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_4[filterVec_4])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_5[filterVec_5])
,]
# overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% potentialPolyFakeGenes
# ,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

AfterManualTouchUp_Filtering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
       Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
   set_colnames("Counfounding_Gene_Number") %>% datatable()
```



#### Assess manually

```{r,eval=FALSE,echo=FALSE}
"LOC105275990" - Reject stringtie -accept Refseq
"LOC105276245" - Accept Stringtie long 
"LOC105277680" - Reject stringtie -accept Refseq
"LOC105278262" - Reject stringtie -accept Refseq
"LOC105278631" - Reject stringtie -accept Refseq
"LOC105278994" - Accept Stringtie long 
"LOC105279031" - Accept Stringtie long 
"LOC105280692" - Reject stringtie -accept Refseq
"LOC105281370" - Reject stringtie -accept Refseq
"LOC105281402" - Accept Stringtie long  
"LOC105281448" - Reject stringtie -accept Refseq
"LOC105281730" - Reject stringtie -accept Refseq
"LOC105282423" - Accept Stringtie long
"LOC105283711" - Accept Stringtie long - after merge STRG.10347 to STRG.10344.4
"LOC105283935" - Accept Stringtie long 
"LOC105284377" - Reject stringtie -accept Refseq
"LOC105284524" - Reject stringtie -accept Refseq
"LOC105284788" - Accept Stringtie long
"LOC105286207" - Accept Stringtie long
"LOC105286340" - Reject stringtie -accept Refseq Not sure why this wasnt filtered.. shortread stringtie and Refseq match
"LOC105286632" - Accept Stringtie long
"LOC105286657" - Reject stringtie -accept Refseq 
"LOC105287969" - Reject stringtie -accept ShortRead 
"LOC105287977" - Accept Stringtie long
"LOC105288145" - Accept Stringtie long
"LOC113561858" - Trim STRG.4488.2 and STRG.4488.1 - Move STRG.4488.2 to new gene. then Accept Stringtie long
"LOC113563199" - Accept Stringtie long


```

### Manually accept gene models based on 

```{r}
accept_refSeq_manual <- c("LOC105275990","LOC105277680","LOC105278262","LOC105278631",
"LOC105280692","LOC105281370","LOC105281448","LOC105281730","LOC105284377",
"LOC105284524","LOC105286340","LOC105286657")

accept_LongStringtie_manual <- c("LOC105276245","LOC105278994","LOC105279031","LOC105281402","LOC105282423",
 "LOC105283711","LOC105283935","LOC105284788","LOC105286207","LOC105286632","LOC105287977","LOC105288145",
 "LOC113561858","LOC113563199")

accept_ShortStringtie_manual <- c("LOC105287969")
```


## Assess confounded gene models after rescue
```{r}
LongStringtie_txdb2 <- makeTxDbFromGRanges(LongStringtie_GTF2)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb2,by="gene")

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% singleExonGene,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec[filterVec])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec2[filterVec2])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_3[filterVec_3])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_4[filterVec_4])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% names(filterVec_5[filterVec_5])
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% accept_refSeq_manual
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% accept_LongStringtie_manual
,]
overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$queryHitsNames %in% accept_ShortStringtie_manual
,]
# overlappingGenesDF <- overlappingGenesDF[!overlappingGenesDF$subjectHitsNames %in% potentialPolyFakeGenes
# ,]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

AfterManualAssignUp_Filtering <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
       Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
   set_colnames("Counfounding_Gene_Number") %>% datatable()
```

# Summarise step-wise filtering and rescue steps

```{r}

data.frame(NoFilter=length(noFiltering),
After_Single_Exon_filter=length(singleExon_Filtering),
After_GenBank_Agreement_filter=length(GenbankFiltering),
After_RefSeq_SelfOverlap_CleanUp=length(overlappingRefSeqFiltering) ,
After_ShortRead_Agreement_filter=length(ShortReadAndLongReadMatchFiltering), 
# Afterlength(preInternalPrimeFiltering) 
After_Internal_Priming_CleanUp=length(postPolyAFiltering) ,
After_ShortRead_And_LongRead_Agree_On_Gaps_filter=length(ShortReadAndLongReadGapFiltering) ,
After_LongRead_SelfOverlap_CleanUp=length(SelfOverlapStringtie_Filtering) ,
After_LongRead_Gap_ContainedIn_ShortRead_Exon_filter=length(GapAllInExon_Filtering),
After_ShortRead_And_RefSeq_agree_In_LongRead_Gap_filter=length(GapMatchesRefseqAndShortRead_Filtering),
After_Human_Curation_Fixing=length(AfterManualTouchUp_Filtering),
After_Human_Curation_Accepting=length(AfterManualAssignUp_Filtering)) %>% t %>% 
 datatable

```

# Create final gene models based on filters and fixes.

## Merge RefSeq and LongRead gene models.

### Fix GTFs by established filters and fixes for merging Refseq and LongRead gene models.

```{r}
# BRCDK_GTF
# LongStringtie_GTF2
LongStringtie_txdb2 <- makeTxDbFromGRanges(LongStringtie_GTF2)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb2,by="gene")

singleExonGene <- names(LongStringtie_exonsByGene)[lengths(LongStringtie_exonsByGene) == 1]

ShortStringtie_Cleaned <- ShortStringtie_GTF[ShortStringtie_GTF$gene_id %in% c("MSTRG.8281","MSTRG.8282","MSTRG.8283")]
ShortStringtie_txdb2 <- makeTxDbFromGRanges(ShortStringtie_Cleaned)

ShortStringtie_exonsByGene <- exonsBy(ShortStringtie_txdb2,by="gene")
overlappingGenes <- findOverlaps(ShortStringtie_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(ShortStringtie_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
nrow(overlappingGenesDF)

overlappingGenes <- findOverlaps(ShortStringtie_exonsByGene,BRCDK_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(ShortStringtie_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$subjectHits]
nrow(overlappingGenesDF)
OverlapIncludedShortReadGenes <- overlappingGenesDF$subjectHitsNames

overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]




gapFilter <- overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% names(filterVec_5[filterVec_5]),]$subjectHitsNames
manFilter <- overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% accept_refSeq_manual,]$subjectHitsNames
shortReadFilter <- overlappingGenesDF[overlappingGenesDF$queryHitsNames %in% accept_ShortStringtie_manual,]$subjectHitsNames


LongStringtie_Cleaned <- LongStringtie_GTF2[!LongStringtie_GTF2$gene_id %in% singleExonGene,]
LongStringtie_Cleaned <- LongStringtie_Cleaned[!LongStringtie_Cleaned$gene_id %in% gapFilter,]
LongStringtie_Cleaned <- LongStringtie_Cleaned[!LongStringtie_Cleaned$gene_id %in% manFilter,]
LongStringtie_Cleaned <- LongStringtie_Cleaned[!LongStringtie_Cleaned$gene_id %in% shortReadFilter,]

BRCDK_Cleaned <- BRCDK_GTF[!BRCDK_GTF$gene_id %in% names(filterVec[filterVec]),]
BRCDK_Cleaned <- BRCDK_Cleaned[!BRCDK_Cleaned$gene_id %in% names(filterVec2[filterVec2]),]
BRCDK_Cleaned <- BRCDK_Cleaned[!BRCDK_Cleaned$gene_id %in% names(filterVec_3[filterVec_3]),]
BRCDK_Cleaned <- BRCDK_Cleaned[!BRCDK_Cleaned$gene_id %in% names(filterVec_4[filterVec_4]),]
BRCDK_Cleaned <- BRCDK_Cleaned[!BRCDK_Cleaned$gene_id %in% names(filterVec_4[filterVec_4]),]
BRCDK_Cleaned <- BRCDK_Cleaned[!BRCDK_Cleaned$gene_id %in% accept_LongStringtie_manual,]
BRCDK_Cleaned <- BRCDK_Cleaned[!BRCDK_Cleaned$gene_id %in% accept_ShortStringtie_manual,]
BRCDK_Cleaned <- BRCDK_Cleaned[!BRCDK_Cleaned$gene_id %in% OverlapIncludedShortReadGenes,]




LongStringtie_txdb_Cleaned <- makeTxDbFromGRanges(LongStringtie_Cleaned)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb_Cleaned,by="gene")

BRCDK_txdb_Cleaned <- makeTxDbFromGRanges(BRCDK_Cleaned)

BRCDK_exonsByGene <- exonsBy(BRCDK_txdb_Cleaned,by="gene")



# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

Cleaned_filter <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
       Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
   set_colnames("Counfounding_Gene_Number") %>% datatable()
```

Must remove or update names for genes which were many refsq to one stringtie
In fact could that better of all...maybe before merging.
STRG.2373 this is strange one..check rules on when to merge genes for polyA tailed genes...

### Accept LongRead gene models which fuse multiple RefSeq genes.

```{r}
toCleanMultiBRCDK <- overlappingGenesDF[overlappingGenesDF$subjectHitsNames %in% multi_longStringtie,]$queryHitsNames %>% unique
```

```{r}
BRCDK_Cleaned <- BRCDK_Cleaned[!BRCDK_Cleaned$gene_id %in% toCleanMultiBRCDK,]




LongStringtie_txdb_Cleaned <- makeTxDbFromGRanges(LongStringtie_Cleaned)

LongStringtie_exonsByGene <- exonsBy(LongStringtie_txdb_Cleaned,by="gene")

BRCDK_txdb_Cleaned <- makeTxDbFromGRanges(BRCDK_Cleaned)

BRCDK_exonsByGene <- exonsBy(BRCDK_txdb_Cleaned,by="gene")



# BRCDK_exonsByGene[names(updatedGene)] <- updatedGene
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]


QueryHitFreqFrame <- overlappingGenesDF$queryHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Query_Hit","Freq"))

QueryMultOverlapTotal <- QueryHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

QueryMultiHits <- QueryHitFreqFrame %>%
 filter(Freq > 1) %>%
 pull(Query_Hit)

multi_BRCDK <- names(BRCDK_exonsByGene)[as.numeric(unfactor(QueryMultiHits))]


SubjectHitFreqFrame <- overlappingGenesDF$subjectHits %>% 
 table %>% 
 as.data.frame() %>% 
 set_colnames(c("Subject_Hit","Freq"))

SubjectMultOverlapTotal <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 nrow

SubMultiHits <- SubjectHitFreqFrame %>% 
 filter(Freq > 1) %>% 
 pull(Subject_Hit) 

multi_longStringtie <- names(LongStringtie_exonsByGene)[as.numeric(unfactor(SubMultiHits))]

Cleaned_filter <- multi_BRCDK

data.frame(Multi_RefSeq_to_Single_LongRead=length(multi_longStringtie),
       Multi_LongRead_to_Single_RefSeq=length(multi_BRCDK)) %>% t %>% 
   set_colnames("Counfounding_Gene_Number") %>% datatable()

```

### Check merged gene models have no conflicts.

```{r}
overlappingGenes <- findOverlaps(BRCDK_exonsByGene,LongStringtie_exonsByGene)
overlappingGenesDF <- as.data.frame(overlappingGenes)
overlappingGenesDF$queryHitsNames <- names(BRCDK_exonsByGene)[overlappingGenesDF$queryHits]
overlappingGenesDF$subjectHitsNames <- names(LongStringtie_exonsByGene)[overlappingGenesDF$subjectHits]
duplicated(overlappingGenesDF$queryHitsNames) %>% any
duplicated(overlappingGenesDF$subjectHitsNames) %>% any

LongToBRCDK_geneId <- overlappingGenesDF$subjectHitsNames[
              match(BRCDK_Cleaned$gene_id,overlappingGenesDF$queryHitsNames)
              ]
BRCDK_Cleaned$old_gene_id <- BRCDK_Cleaned$gene_id
BRCDK_Cleaned$gene_id <- LongToBRCDK_geneId
BRCDK_Cleaned$gene_id[is.na(BRCDK_Cleaned$gene_id)] <- BRCDK_Cleaned$old_gene_id[is.na(BRCDK_Cleaned$gene_id)]
# BRCDK_Cleaned$gene_id %>% unique %>% length
LongandRefseq_GTF <- c(LongStringtie_Cleaned,BRCDK_Cleaned)

rtracklayer::export(LongandRefseq_GTF,con = "LongandRefseq_GTF.gtf")

LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseq_GTF)
LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by = "gene")
selfOverlaps <- findOverlaps(LongandRefseq_exonsByGene,LongandRefseq_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]

```

## Merge ShortRead and LongRead gene models.

What is going on with LOC105285984;LOC105285985 and MSTRG.2668 RU.2397.1

### Add ShortRead trascripts which are contained within single LongRead gene models.

```{r}

LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseq_GTF)

LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by="gene")

ShortStringtie_exonsByTranscript <- exonsBy(ShortStringtie_txdb,by="tx",use.names=TRUE)
ShortStringtie_GTF_GeneToTranscript <- ShortStringtie_GTF %>% mcols %>% 
 as.data.frame() %>% 
 dplyr::select(gene_id,transcript_id) %>% 
 distinct()

someOverlaps <- findOverlaps(LongandRefseq_exonsByGene,ShortStringtie_exonsByTranscript)
temp_someOverlaps <- as.data.frame(someOverlaps)

temp_someOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_someOverlaps$queryHits]
temp_someOverlaps$subjectHitsNames <- names(ShortStringtie_exonsByTranscript)[temp_someOverlaps$subjectHits]


temp_someOverlaps$subjectHitsGeneNames <- ShortStringtie_GTF_GeneToTranscript$gene_id[match(temp_someOverlaps$subjectHitsNames,
                       ShortStringtie_GTF_GeneToTranscript$transcript_id)]

shortOverlappingTrans <- temp_someOverlaps %>% 
 mutate(dup=duplicated(subjectHitsNames)) %>% filter(dup) %>% pull(subjectHitsNames)

# shortOverlappingTrans2 <- temp_someOverlaps %>% 
#  mutate(dup=duplicated(queryHitsNames)) %>% filter(dup) %>% pull(queryHitsNames)
# 
# [temp_someOverlaps$queryHitsNames %in% shortOverlappingTrans2,]


ShortStringtie_Cleaned <- ShortStringtie_GTF[!ShortStringtie_GTF$transcript_id %in% shortOverlappingTrans]
ShortStringtie_txdb2 <- makeTxDbFromGRanges(ShortStringtie_Cleaned)


ShortStringtie_exonsByTranscript <- exonsBy(ShortStringtie_txdb2,by="tx",use.names=TRUE)
ShortStringtie_GTF_GeneToTranscript <- ShortStringtie_Cleaned %>% mcols %>% 
 as.data.frame() %>% 
 dplyr::select(gene_id,transcript_id) %>% 
 distinct()

someOverlaps <- findOverlaps(LongandRefseq_exonsByGene,ShortStringtie_exonsByTranscript)
temp_someOverlaps <- as.data.frame(someOverlaps)

temp_someOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_someOverlaps$queryHits]
temp_someOverlaps$subjectHitsNames <- names(ShortStringtie_exonsByTranscript)[temp_someOverlaps$subjectHits]


temp_someOverlaps$subjectHitsGeneNames <- ShortStringtie_GTF_GeneToTranscript$gene_id[match(temp_someOverlaps$subjectHitsNames,
                       ShortStringtie_GTF_GeneToTranscript$transcript_id)]

LongToShort_geneId <- temp_someOverlaps$queryHitsNames[
              match(ShortStringtie_Cleaned$transcript_id,temp_someOverlaps$subjectHitsNames)
              ]

ShortStringtie_Cleaned$old_gene_id <- ShortStringtie_Cleaned$gene_id
ShortStringtie_Cleaned$gene_id <- LongToShort_geneId
ShortStringtie_Cleaned$gene_id[is.na(ShortStringtie_Cleaned$gene_id)] <- ShortStringtie_Cleaned$old_gene_id[is.na(ShortStringtie_Cleaned$gene_id)]
# BRCDK_Cleaned$gene_id %>% unique %>% length
LongandRefseqandShort_GTF <- c(LongandRefseq_GTF,ShortStringtie_Cleaned)



LongandRefseqandShort_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_GTF)
LongandRefseqandShort_exonsByGene <- exonsBy(LongandRefseqandShort_txdb,by = "gene")
LongandRefseqandShort_transcriptsByGene <- exonsBy(LongandRefseqandShort_txdb,by = "tx",use.names=TRUE)
selfOverlaps <- findOverlaps(LongandRefseqandShort_exonsByGene,LongandRefseqandShort_transcriptsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongandRefseqandShort_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongandRefseqandShort_transcriptsByGene)[temp_selfOverlaps$subjectHits]


shortOverlappingTrans2 <- temp_selfOverlaps %>% 
 mutate(dup=duplicated(subjectHitsNames)) %>% filter(dup) %>% pull(subjectHitsNames)

LongandRefseqandShort_Cleaned_GTF <- LongandRefseqandShort_GTF[!LongandRefseqandShort_GTF$transcript_id %in% shortOverlappingTrans2]
rtracklayer::export(LongandRefseqandShort_Cleaned_GTF,con = "LongandRefseqandShort_Cleaned_GTF.gtf")


LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned_GTF)
LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by = "gene")
selfOverlaps <- findOverlaps(LongandRefseq_exonsByGene,LongandRefseq_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]
# nrow(temp_selfOverlaps)


```

### Clean up to make sure no ShortRead transcripts overlap multiple LongRead genes.

```{r}

LongandRefseq_transcriptsByGene <- exonsBy(LongandRefseq_txdb,by = "tx",use.names=TRUE)
selfOverlaps <- findOverlaps(LongandRefseq_transcriptsByGene,LongandRefseq_transcriptsByGene,type = "equal")
temp_selfOverlaps <- as.data.frame(selfOverlaps)
temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]
temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_transcriptsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_transcriptsByGene)[temp_selfOverlaps$subjectHits]
toFilter <- temp_selfOverlaps %>% select(queryHitsNames,subjectHitsNames) %>% apply(1,sort) %>% t %>% as.data.frame %>% distinct

repeatTranscripts <- unique(toFilter$V2)

LongandRefseqandShort_Cleaned2_GTF <- LongandRefseqandShort_Cleaned_GTF[!LongandRefseqandShort_Cleaned_GTF$transcript_id %in% 
                                     repeatTranscripts]

```


```{r}
LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_GTF %>% as.data.frame(row.names = NULL)

unique_genes <- unique(LongandRefseqandShort_Cleaned2_DF$gene_id)
new_gene_ids <- paste0("RU.", seq_along(unique_genes))
gene_map <- setNames(new_gene_ids, unique_genes)

LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_DF %>%
 mutate(ru_gene_id = gene_map[gene_id])


LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_DF %>%
 group_by(ru_gene_id) %>%
 mutate(ru_trans_id = paste0(ru_gene_id, ".", dense_rank(transcript_id))) %>%
 ungroup()

LongandRefseqandShort_Cleaned2_DF %<>% 
 mutate(old_gene_id=ifelse(is.na(old_gene_id),gene_id,old_gene_id))


LongandRefseqandShort_Cleaned2_DF %<>% 
 mutate(gene_id=ru_gene_id,transcript_id=ru_trans_id) %>% 
 filter(type == "exon") %>% 
 select(-ru_gene_id,-ru_trans_id)

LongandRefseqandShort_Cleaned2_GTF <- GRanges(LongandRefseqandShort_Cleaned2_DF)

export(LongandRefseqandShort_Cleaned2_GTF,con = "LongandRefseqandShort_Cleaned3_GTF.gtf")
```

# Annotate final gene models for overlaps with RefSeq, Genbank and Ensembl gene/transcript models

```{r}
UCSC_EnsGene_GTF <- import("Originals/UCSC_EnsGene/GCF_003672135.1_Obir_v5.4.ensGene.2021_12.gtf.gz")
chr_map <- read.delim("Originals/UCSC_Map/GCF_003672135.1.chromAlias.txt")
UCSC_EnsGene_GTF_2 <- renameSeqlevels(UCSC_EnsGene_GTF,
               chr_map %>% pull(assembly,name = "X..refseq"))

```


```{r}
Combined_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
EnsGene_txdb <- makeTxDbFromGRanges(UCSC_EnsGene_GTF_2)
GenBank_txdb <- makeTxDbFromGRanges(GenBank_GTF)
RefSeq_txdb <- makeTxDbFromGRanges(RefSeq_GTF)
```


```{r}
Combined_exonsByGene <- exonsBy(Combined_txdb,by="gene")
EnsGene_exonsByGene <- exonsBy(EnsGene_txdb,by="gene")
RefSeq_exonsByGene <- exonsBy(RefSeq_txdb,by="gene")
GenBank_exonsByGene <- exonsBy(GenBank_txdb,by="gene")
```

```{r}
EnsOverlaps <- findOverlaps(Combined_exonsByGene,EnsGene_exonsByGene)
temp_EnsOverlaps <- as.data.frame(EnsOverlaps)

temp_EnsOverlaps$queryHitsNames <- names(Combined_exonsByGene)[temp_EnsOverlaps$queryHits]
temp_EnsOverlaps$subjectHitsNames <- names(EnsGene_exonsByGene)[temp_EnsOverlaps$subjectHits]
temp_EnsOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Ens_gene_id=paste(subjectHitsNames,collapse = ";"))

RefseqOverlaps <- findOverlaps(Combined_exonsByGene,RefSeq_exonsByGene)
temp_RefseqOverlaps <- as.data.frame(RefseqOverlaps)

temp_RefseqOverlaps$queryHitsNames <- names(Combined_exonsByGene)[temp_RefseqOverlaps$queryHits]
temp_RefseqOverlaps$subjectHitsNames <- names(RefSeq_exonsByGene)[temp_RefseqOverlaps$subjectHits]
temp_RefseqOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Refseq_gene_id=paste(subjectHitsNames,collapse = ";"))

GenbankOverlaps <- findOverlaps(Combined_exonsByGene,GenBank_exonsByGene)
temp_GenbankOverlaps <- as.data.frame(GenbankOverlaps)

temp_GenbankOverlaps$queryHitsNames <- names(Combined_exonsByGene)[temp_GenbankOverlaps$queryHits]
temp_GenbankOverlaps$subjectHitsNames <- names(GenBank_exonsByGene)[temp_GenbankOverlaps$subjectHits]
temp_GenbankOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Genbank_gene_id=paste(subjectHitsNames,collapse = ";"))

LongandRefseqandShort_Cleaned2_DF <- as.data.frame(LongandRefseqandShort_Cleaned2_GTF)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_RefseqOverlaps,
                      by.x="gene_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_EnsOverlaps,
                      by.x="gene_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_GenbankOverlaps,
                      by.x="gene_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_GTF <- GRanges(LongandRefseqandShort_Cleaned2_DF)
Combined_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)

```



```{r}
Combined_exonsBytranscript <- exonsBy(Combined_txdb,by="tx",use.names=TRUE)
EnsGene_exonsBytranscript <- exonsBy(EnsGene_txdb,by="tx",use.names=TRUE)
RefSeq_exonsBytranscript <- exonsBy(RefSeq_txdb,by="tx",use.names=TRUE)
GenBank_exonsBytranscript <- exonsBy(GenBank_txdb,by="tx",use.names=TRUE)
```

```{r}
EnsOverlaps <- findOverlaps(Combined_exonsBytranscript,EnsGene_exonsBytranscript)
temp_EnsOverlaps <- as.data.frame(EnsOverlaps)

temp_EnsOverlaps$queryHitsNames <- names(Combined_exonsBytranscript)[temp_EnsOverlaps$queryHits]
temp_EnsOverlaps$subjectHitsNames <- names(EnsGene_exonsBytranscript)[temp_EnsOverlaps$subjectHits]
temp_EnsOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Ens_transcript_id=paste(subjectHitsNames,collapse = ";"))

RefseqOverlaps <- findOverlaps(Combined_exonsBytranscript,RefSeq_exonsBytranscript)
temp_RefseqOverlaps <- as.data.frame(RefseqOverlaps)

temp_RefseqOverlaps$queryHitsNames <- names(Combined_exonsBytranscript)[temp_RefseqOverlaps$queryHits]
temp_RefseqOverlaps$subjectHitsNames <- names(RefSeq_exonsBytranscript)[temp_RefseqOverlaps$subjectHits]
temp_RefseqOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Refseq_transcript_id=paste(subjectHitsNames,collapse = ";"))

GenbankOverlaps <- findOverlaps(Combined_exonsBytranscript,GenBank_exonsBytranscript)
temp_GenbankOverlaps <- as.data.frame(GenbankOverlaps)

temp_GenbankOverlaps$queryHitsNames <- names(Combined_exonsBytranscript)[temp_GenbankOverlaps$queryHits]
temp_GenbankOverlaps$subjectHitsNames <- names(GenBank_exonsBytranscript)[temp_GenbankOverlaps$subjectHits]
temp_GenbankOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Genbank_transcript_id=paste(subjectHitsNames,collapse = ";"))

LongandRefseqandShort_Cleaned2_DF <- as.data.frame(LongandRefseqandShort_Cleaned2_GTF)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_RefseqOverlaps,
                      by.x="transcript_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_EnsOverlaps,
                      by.x="transcript_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_GenbankOverlaps,
                      by.x="transcript_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_GTF <- GRanges(LongandRefseqandShort_Cleaned2_DF)

export(LongandRefseqandShort_Cleaned2_GTF,con = "Obiroi_ru_combined.gtf")


```

# QC final gene models with gffcompare, squanti3 and Busco

### Create output for BUSCO assessment

```{r}
genomeSeq <- readDNAStringSet(filepath = "Genome/GenBank.fa")
LongandRefseqandShort_Cleaned2_GTF <- LongandRefseqandShort_Cleaned2_GTF[seqnames(LongandRefseqandShort_Cleaned2_GTF) %in% names(genomeSeq),]

GenBank_GTF[GenBank_GTF$transcript_id == ""]$transcript_id <- NA
RefSeq_GTF[RefSeq_GTF$transcript_id == ""]$transcript_id <- NA
export(GenBank_GTF,con = "Genbank.gtf")
export(RefSeq_GTF,con = "Refseq.gtf")

GenBank_GTF <- GenBank_GTF[seqnames(GenBank_GTF) %in% names(genomeSeq),]
RefSeq_GTF <- RefSeq_GTF[seqnames(RefSeq_GTF) %in% names(genomeSeq),]


export(LongandRefseqandShort_Cleaned2_GTF,con = "Obiroi_ru_combined.gtf")


Combined_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
RefSeq_txdb <- makeTxDbFromGRanges(RefSeq_GTF)
GenBank_txdb <- makeTxDbFromGRanges(GenBank_GTF)

Combined_exonsBytranscript <- exonsBy(Combined_txdb,by="tx",use.names=TRUE)
Genbank_exonsBytranscript <- exonsBy(GenBank_txdb,by="tx",use.names=TRUE)
RefSeq_exonsBytranscript <- exonsBy(RefSeq_txdb,by="tx",use.names=TRUE)

Combined_transSeq <- GenomicFeatures::extractTranscriptSeqs(FaFile("Genome/GenBank.fa"),Combined_exonsBytranscript)
Genbank_transSeq <- GenomicFeatures::extractTranscriptSeqs(FaFile("Genome/GenBank.fa"),Genbank_exonsBytranscript)
Refseq_transSeq <- GenomicFeatures::extractTranscriptSeqs(FaFile("Genome/GenBank.fa"),RefSeq_exonsBytranscript)

writeXStringSet(Combined_transSeq,"Combined_transSeq.fa")
writeXStringSet(Genbank_transSeq,"Genbank_transSeq.fa")
writeXStringSet(Refseq_transSeq,"Refseq_transSeq.fa")
```

### Create output for squanti QC assessment and correction

```{r}
Combined_transcript <- transcripts(Combined_txdb,columns=c("tx_name","gene_id"))
Combined_genes <- genes(Combined_txdb,columns=c("gene_id"))

Genbank_transcript <- exonsBy(GenBank_txdb,by="tx",use.names=TRUE)
RefSeq_transcript <- exonsBy(RefSeq_txdb,by="tx",use.names=TRUE)


Combined_transcript <- as.data.frame(Combined_transcript) %>% mutate(transcript_id=tx_name,type="transcript") %>% select(-tx_name) %>% GRanges()
Combined_genes <- as.data.frame(Combined_genes) %>% mutate(type="gene") %>% GRanges()

Combined_GFT <- c(Combined_transcript,Combined_genes,LongandRefseqandShort_Cleaned2_GTF) %>% sort
export(Combined_GFT,con = "Obiroi_ru_combined_full.gtf")

```


### Import squanti corrected gene models and reannotate

```{r}
SQ_corrected <- import("Squanti/squanti3_combined_vs_refseq/combine_vs_refseq.gtf_corrected.gtf")
SQCombined_txdb <- makeTxDbFromGRanges(SQ_corrected)
```


```{r}
SQCombined_exonsByGene <- exonsBy(SQCombined_txdb,by="gene")
EnsGene_exonsByGene <- exonsBy(EnsGene_txdb,by="gene")
RefSeq_exonsByGene <- exonsBy(RefSeq_txdb,by="gene")
GenBank_exonsByGene <- exonsBy(GenBank_txdb,by="gene")
```

```{r}

EnsOverlaps <- findOverlaps(SQCombined_exonsByGene,EnsGene_exonsByGene)
temp_EnsOverlaps <- as.data.frame(EnsOverlaps)

temp_EnsOverlaps$queryHitsNames <- names(SQCombined_exonsByGene)[temp_EnsOverlaps$queryHits]
temp_EnsOverlaps$subjectHitsNames <- names(EnsGene_exonsByGene)[temp_EnsOverlaps$subjectHits]
temp_EnsOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Ens_gene_id=paste(subjectHitsNames,collapse = ";"))

RefseqOverlaps <- findOverlaps(SQCombined_exonsByGene,RefSeq_exonsByGene)
temp_RefseqOverlaps <- as.data.frame(RefseqOverlaps)

temp_RefseqOverlaps$queryHitsNames <- names(SQCombined_exonsByGene)[temp_RefseqOverlaps$queryHits]
temp_RefseqOverlaps$subjectHitsNames <- names(RefSeq_exonsByGene)[temp_RefseqOverlaps$subjectHits]
temp_RefseqOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Refseq_gene_id=paste(subjectHitsNames,collapse = ";"))

GenbankOverlaps <- findOverlaps(SQCombined_exonsByGene,GenBank_exonsByGene)
temp_GenbankOverlaps <- as.data.frame(GenbankOverlaps)

temp_GenbankOverlaps$queryHitsNames <- names(SQCombined_exonsByGene)[temp_GenbankOverlaps$queryHits]
temp_GenbankOverlaps$subjectHitsNames <- names(GenBank_exonsByGene)[temp_GenbankOverlaps$subjectHits]
temp_GenbankOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Genbank_gene_id=paste(subjectHitsNames,collapse = ";"))

LongandRefseqandShort_Cleaned2_DF <- as.data.frame(SQ_corrected)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_RefseqOverlaps,
                      by.x="gene_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_EnsOverlaps,
                      by.x="gene_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_GenbankOverlaps,
                      by.x="gene_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_GTF <- GRanges(LongandRefseqandShort_Cleaned2_DF)
SQCombined_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)

```



```{r}
SQCombined_exonsBytranscript <- exonsBy(SQCombined_txdb,by="tx",use.names=TRUE)
EnsGene_exonsBytranscript <- exonsBy(EnsGene_txdb,by="tx",use.names=TRUE)
RefSeq_exonsBytranscript <- exonsBy(RefSeq_txdb,by="tx",use.names=TRUE)
GenBank_exonsBytranscript <- exonsBy(GenBank_txdb,by="tx",use.names=TRUE)
```

```{r}

EnsOverlaps <- findOverlaps(SQCombined_exonsBytranscript,EnsGene_exonsBytranscript)
temp_EnsOverlaps <- as.data.frame(EnsOverlaps)

temp_EnsOverlaps$queryHitsNames <- names(SQCombined_exonsBytranscript)[temp_EnsOverlaps$queryHits]
temp_EnsOverlaps$subjectHitsNames <- names(EnsGene_exonsBytranscript)[temp_EnsOverlaps$subjectHits]
temp_EnsOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Ens_transcript_id=paste(subjectHitsNames,collapse = ";"))

RefseqOverlaps <- findOverlaps(SQCombined_exonsBytranscript,RefSeq_exonsBytranscript)
temp_RefseqOverlaps <- as.data.frame(RefseqOverlaps)

temp_RefseqOverlaps$queryHitsNames <- names(SQCombined_exonsBytranscript)[temp_RefseqOverlaps$queryHits]
temp_RefseqOverlaps$subjectHitsNames <- names(RefSeq_exonsBytranscript)[temp_RefseqOverlaps$subjectHits]
temp_RefseqOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Refseq_transcript_id=paste(subjectHitsNames,collapse = ";"))

GenbankOverlaps <- findOverlaps(SQCombined_exonsBytranscript,GenBank_exonsBytranscript)
temp_GenbankOverlaps <- as.data.frame(GenbankOverlaps)

temp_GenbankOverlaps$queryHitsNames <- names(SQCombined_exonsBytranscript)[temp_GenbankOverlaps$queryHits]
temp_GenbankOverlaps$subjectHitsNames <- names(GenBank_exonsBytranscript)[temp_GenbankOverlaps$subjectHits]
temp_GenbankOverlaps %<>% 
 select(queryHitsNames,subjectHitsNames) %>% 
 group_by(queryHitsNames) %>% 
 summarise(Genbank_transcript_id=paste(subjectHitsNames,collapse = ";"))

LongandRefseqandShort_Cleaned2_DF <- as.data.frame(LongandRefseqandShort_Cleaned2_GTF)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_RefseqOverlaps,
                      by.x="transcript_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_EnsOverlaps,
                      by.x="transcript_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,temp_GenbankOverlaps,
                      by.x="transcript_id",by.y="queryHitsNames",
                      all.x=TRUE,all.y=FALSE)

LongandRefseqandShort_Cleaned2_GTF <- GRanges(LongandRefseqandShort_Cleaned2_DF)
```


### Create output for gffcompare QC

```{r}
# levels(LongandRefseqandShort_Cleaned2_GTF$source) <- c("PacBio_Stringtie","Illumina_Stringtie","Refseq")
tempSource <- vector("character",length=length(LongandRefseqandShort_Cleaned2_GTF))

tempSource[grep("^STRG",LongandRefseqandShort_Cleaned2_GTF$old_gene_id)] <- "PacBio_Stringtie"
tempSource[grep("^MSTRG",LongandRefseqandShort_Cleaned2_GTF$old_gene_id)] <- "Illumina_Stringtie"
tempSource[-grep("^MSTRG|^STRG",LongandRefseqandShort_Cleaned2_GTF$old_gene_id)] <- "RefSeq"
LongandRefseqandShort_Cleaned2_GTF$source <- tempSource
export(LongandRefseqandShort_Cleaned2_GTF,con = "Obiroi_ru_SQCombined.gtf")


```


### Install gffcompare
```{r,eval=FALSE}
Herper::install_CondaTools("gffcompare",
              "gffcompare",
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")
```

### Run gffcompare
```{r,eval=FALSE}
dir.create("gffcompare",showWarnings = FALSE,recursive = TRUE)
Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Refseq.gtf -o gffcompare/Obiroi_ru_SQCombined_refseq Obiroi_ru_SQCombined.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")

Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Genbank.gtf -o gffcompare/Obiroi_ru_SQCombined_genbank Obiroi_ru_SQCombined.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")

```

# Parse QC and make XLSx report.

```{r}
parse_cuffCompare <- function(stats_file){
lines <- readLines(stats_file)
prec_sens <- lines[grepl("level",lines)] %>% 
 str_trim() %>% 
 gsub("Intron chain","Intron_chain",.) %>% 
 gsub(" level:","_level",.) %>% 
 gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V3,-V5) %>% set_colnames(c("level","Sensitivity","Precision"))

Query_mrna <- lines[grepl("Query mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4,-V5) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))
 
Reference_mrna <- lines[grepl("Reference mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=4,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))

matching <- lines[grepl("Matching",lines)] %>% str_trim() %>%
   gsub(".*:","",.) %>% str_trim() %>% as.numeric %>% as.data.frame() %>% t %>% 
 set_colnames(c("Matching_intron_chains","Matching_transcripts","Matching_loci"))

myRownames <- lines[grepl("Missed|Novel",lines)] %>% 
 str_trim() %>% 
 gsub(":.*","",.) %>% gsub(" ","_",.)
Novel_missing <- lines[grepl("Missed|Novel",lines)] %>% str_trim() %>% 
 gsub(".*:","",.) %>% str_trim() %>% 
 gsub("\t.*","",.) %>% str_trim() %>% str_split("/") %>% unlist %>% matrix(ncol=2,byrow = TRUE) %>% 
 as.data.frame() %>% set_colnames(c("n_of","Total")) %>% set_rownames(myRownames)

toReturn <- list(prec_sens,Query_mrna,Reference_mrna,matching,Novel_missing)
names(toReturn) <- c("prec_sens","Query_mrna","Reference_mrna","matching","Novel_missing")
toReturn
}

refseq_gffcompare <- parse_cuffCompare("gffcompare/Obiroi_ru_SQCombined_refseq.stats")
genbank_gffcompare <- parse_cuffCompare("gffcompare/Obiroi_ru_SQCombined_genbank.stats")

gffcompare_sensitivity <- merge(refseq_gffcompare$prec_sens[,1:2],
                    genbank_gffcompare$prec_sens[,1:2],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Sensitivity",sep="_"))) 

gffcompare_precision <- merge(refseq_gffcompare$prec_sens[,c(1,3)],
                    genbank_gffcompare$prec_sens[,c(1,3)],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Precision",sep="_"))) 

gffcompare_ns<- cbind(t(refseq_gffcompare$Query_mrna),
                    t(refseq_gffcompare$Reference_mrna),t(genbank_gffcompare$Reference_mrna)) %>% 
 set_colnames(c("RU","RefSeq","GenBank")) %>% as.data.frame() %>% .[c(2,1,3),] %>% rownames_to_column()

gffcompare_matching<- cbind(t(refseq_gffcompare$matching),
                    t(genbank_gffcompare$matching)) %>% 
 set_colnames(c("RefSeq","GenBank")) %>% as.data.frame() %>% .[c(3,2,1),] %>% rownames_to_column()

gffcompare_NovelAndMissing <- cbind(refseq_gffcompare$Novel_missing %>% 
                   set_colnames(paste0(colnames(.),"_against_RefSeq")),
                  genbank_gffcompare$Novel_missing %>%
                   set_colnames(paste0(colnames(.),"_against_Genbank"))) %>% 
 as.data.frame() %>% rownames_to_column()

gffcompare_stats <- list(gffcompare_Summary_numbers=gffcompare_ns,
             gffcompare_NovelAndMissing=gffcompare_NovelAndMissing,
             gffcompare_matching=gffcompare_matching,
             gffcompare_sensitivity=gffcompare_precision,
             gffcompare_precision=gffcompare_precision)
             
             
 

```

```{r}
parse_squantiQC <- function(QCfile){
 temp <- read.delim(paste0(QCfile,"_classification.txt"))
 broadQC <- temp %>% group_by(structural_category) %>% summarise(Total=n())
 subQC <- temp %>% group_by(structural_category,subcategory) %>% summarise(Total=n()) %>% as.data.frame()
 coding <- temp %>% group_by(coding) %>% summarise(Total=n())
 novel_coding <- temp %>% filter(associated_transcript == "novel") %>% group_by(coding) %>% summarise(Total=n())
 transLevel <- list(broadQC,subQC,coding,novel_coding)
 
 temp <- read.delim(paste0(QCfile,"_junctions.txt"))
 subQC <- temp %>% group_by(junction_category,canonical) %>% summarise(Total=n()) %>% as.data.frame()
 subQC2 <- temp %>% group_by(junction_category,canonical,RTS_junction) %>% summarise(Total=n()) %>% as.data.frame()

 juncLevel <- list(subQC,subQC2) 
 list(transLevel=transLevel,juncLevel=juncLevel)
}
refseq_squanti <- parse_squantiQC("Squanti/squanti3_combined_vs_refseq/combine_vs_refseq.gtf")
genbank_squanti <- parse_squantiQC("Squanti/squanti3_combined_vs_genbank/combine_vs_genbank.gtf")

squanti_transcript_categories <- merge(refseq_squanti$transLevel[[1]],
                    genbank_squanti$transLevel[[1]],by=1,all=TRUE) %>% 
 set_colnames(c("structural_category",
         "RU_vs_RefSeq",
         "RU_vs_GenBank")) #%>% mutate(across(1:3,function(x){x[is.na(x)] <- 0}))
squanti_transcript_categories[is.na(squanti_transcript_categories)] <- 0

squanti_subtranscript_categories <- merge(refseq_squanti$transLevel[[2]],
                    genbank_squanti$transLevel[[2]],by=c(1,2),all=TRUE) %>% 
 set_colnames(c("structural_category","sub_category",
         "RU_vs_RefSeq",
         "RU_vs_GenBank")) #%>% mutate(across(1:3,function(x){x[is.na(x)] <- 0}))
squanti_subtranscript_categories[is.na(squanti_subtranscript_categories)] <- 0

squanti_coding_categories <- merge(refseq_squanti$transLevel[[3]],
                    genbank_squanti$transLevel[[3]],by=c(1),all=TRUE) %>% 
 set_colnames(c("Coding",
         "RU_vs_RefSeq",
         "RU_vs_GenBank")) %>% mutate(Class="Total") %>% 
 select(Coding,Class,everything()) 
squanti_novelcoding_categories <- merge(refseq_squanti$transLevel[[4]],
                    genbank_squanti$transLevel[[4]],by=c(1),all=TRUE) %>% 
 set_colnames(c("Coding",
         "RU_vs_RefSeq",
         "RU_vs_GenBank")) %>% mutate(Class="Novel") %>% 
 select(Coding,Class,everything()) 

squanti_coding_categories <- rbind(squanti_coding_categories,
                  squanti_novelcoding_categories)

squanti_junction_categories <- merge(refseq_squanti$juncLevel[[1]],
                    genbank_squanti$juncLevel[[1]],by=c(1,2),all=TRUE) %>% 
 set_colnames(c("Junction_category","Canonical",
         "RU_vs_RefSeq",
         "RU_vs_GenBank")) 

squanti_stats <- list(squanti_transcript_categories=squanti_transcript_categories,
           squanti_subtranscript_categories=squanti_subtranscript_categories,
           squanti_coding_categories=squanti_coding_categories,
           squanti_junction_categories=squanti_junction_categories)
           
           
```

```{r}
parse_busco <- function(busfile){
 temp <- jsonlite::read_json(busfile)
 result <- data.frame(Total_BUSCO_Markers=temp$lineage_dataset$number_of_buscos %>% as.numeric(),
       Percent_BUSCO_Markers_Complete=temp$results$Complete %>% as.numeric(),
       Percent_BUSCO_Markers_Fragmented=temp$results$Fragmented %>% as.numeric(),
       Percent_BUSCO_Markers_Missing=temp$results$Missing %>% as.numeric()
       )
 result[1,] <- result[1,,drop=FALSE]
}

# genbank_squanti <- parse_busco("busco/Genbank_busco/short_summary.specific.arachnida_odb10.Genbank_busco.json")
# refseq_squanti <- parse_busco("busco/Refseq_busco/short_summary.specific.arachnida_odb10.Refseq_busco.json")
# ru_squanti <- parse_busco("busco/combined_busco/short_summary.specific.arachnida_odb10.busco_corr.json")

genbank_squanti <- parse_busco("busco/Genbank_busco_insect//short_summary.specific.insecta_odb10.Genbank_busco_insect.json")
refseq_squanti <- parse_busco("busco/Refseq_busco_insect//short_summary.specific.insecta_odb10.Refseq_busco_insect.json")
ru_squanti <- parse_busco("busco/busco_corr_insect//short_summary.specific.insecta_odb10.busco_corr_insect.json")

busco_results <- cbind(t(ru_squanti),t(refseq_squanti),t(genbank_squanti)) %>% set_colnames(c("RU","RefSeq","GenBank")) %>% as.data.frame() %>% rownames_to_column()
busco_stats <- list(busco_results=busco_results)
```

```{r}
c(gffcompare_stats,squanti_stats,busco_stats) %>% rio::export(file="GeneModel_stats_RUvsRefseqAndGenbank.xlsx")
```

```{r}
devtools::session_info()
```

## Add Busco information back to GTF.

```{r}
busco_Combined <- read.delim("busco/busco_corr_insect/run_insecta_odb10//full_table.tsv",comment.char = "#",header=FALSE) %>% 
 set_colnames(c("Busco_id","Status","Sequence","Score","Length","OrthoDB_url","Description"))
busco_Genbank <- read.delim("busco/Genbank_busco_insect/run_insecta_odb10///full_table.tsv",comment.char = "#",header=FALSE) %>% 
 set_colnames(c("Busco_id","Status","Sequence","Score","Length","OrthoDB_url","Description"))
busco_RefSeq <- read.delim("busco/Refseq_busco_insect/run_insecta_odb10///full_table.tsv",comment.char = "#",header=FALSE) %>% 
 set_colnames(c("Busco_id","Status","Sequence","Score","Length","OrthoDB_url","Description"))

All_busco <- busco_Combined %>% pull(Busco_id) %>% unique()

Busco_Found_Combined <- busco_Combined %>% filter(!Status %in% "Missing") %>% pull(Busco_id) %>% unique()
Busco_Found_RefSeq <- busco_RefSeq %>% filter(!Status %in% "Missing") %>% pull(Busco_id) %>% unique()
Busco_Found_Genbank <- busco_Genbank %>% filter(!Status %in% "Missing") %>% pull(Busco_id) %>% unique()

busco_table <- data.frame(row.names=All_busco)
busco_table$RU <- rownames(busco_table) %in% Busco_Found_Combined
busco_table$RefSeq <- rownames(busco_table) %in% Busco_Found_RefSeq
busco_table$GenBank <- rownames(busco_table) %in% Busco_Found_Genbank

plot(eulerr::euler(busco_table))

RefSeqNotRU <- busco_table[busco_table$RefSeq & !busco_table$RU,] %>% rownames_to_column() %>% pull(rowname)
RUNotRefSeq <- busco_table[!busco_table$RefSeq & busco_table$RU,] %>% rownames_to_column() %>% pull(rowname)

RefSeqNotRU_RefseqIDs <- busco_RefSeq[busco_RefSeq$Busco_id %in% RefSeqNotRU,]$Sequence %>% gsub(":.*","",.) %>% unique
RUNotRefSeq_RUIDs <- busco_Combined[busco_Combined$Busco_id %in% RUNotRefSeq,]$Sequence %>% gsub(":.*","",.) %>% unique


```

## Add in Busco gene info

```{r}
busco_Combined$RUID <- busco_Combined$Sequence %>% gsub(":.*","",.) 
busco_Combined %<>% mutate(BuscoInfo=paste("Busco_id=",Busco_id,";OrthoDB_url=",OrthoDB_url,";Description",Description,sep=""))
LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_GTF %>% as.data.frame 

busco_Combined_toMerge <- busco_Combined %>% 
 select(RUID,BuscoInfo) %>% 
 group_by(RUID) %>% 
 summarise(BuscoInformation=paste(BuscoInfo,sep=":",collapse=":")) %>% 
 ungroup() %>% 
 select(RUID,BuscoInformation)

LongandRefseqandShort_Cleaned2_DF <- merge(LongandRefseqandShort_Cleaned2_DF,busco_Combined_toMerge,
   by.x="transcript_id",by.y="RUID",all.x=TRUE,all.y=FALSE)
LongandRefseqandShort_Cleaned2_GTF <- LongandRefseqandShort_Cleaned2_DF %>% GRanges()
```


## Add in source for transcripts info

```{r}

Combined_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
Combined_exonsByTX <- exonsBy(Combined_txdb,by="tx",use.names=TRUE)


BRCDK_GTF <- readRDS("RObjects/BRCDK_NoGeneModelOverlapFull.RDS")
BRCDK_txdb <- makeTxDbFromGRanges(BRCDK_GTF)
BRCDK_exonsByTX <- exonsBy(BRCDK_txdb,by="tx",use.names=TRUE)


Long_txdb <- makeTxDbFromGRanges(LongStringtie_GTF2)
Long_exonsByTX <- exonsBy(Long_txdb,by="tx",use.names=TRUE)

Short_GTF <- import("Illumina/StringTie_NoGTF/stringtie/stringtie/merged_stringtie.gtf")
Short_GTF <- Short_GTF[strand(Short_GTF) != "*"]
Short_txdb <- makeTxDbFromGRanges(Short_GTF)
Short_exonsByTX <- exonsBy(Short_txdb,by="tx",use.names=TRUE)

tempa <- findOverlaps(Combined_exonsByTX,Combined_exonsByTX,type = "equal")

temp1 <- findOverlaps(Combined_exonsByTX,BRCDK_exonsByTX,type = "equal")
temp2 <- findOverlaps(Combined_exonsByTX,Long_exonsByTX,type = "equal")
temp3 <- findOverlaps(Combined_exonsByTX,Short_exonsByTX,type = "equal")

RefSeq_source <- names(Combined_exonsByTX)[temp1@from] %>% unique
PacBio_source <- names(Combined_exonsByTX)[temp2@from] %>% unique
Illumina_source <- names(Combined_exonsByTX)[temp3@from] %>% unique

source_Refseq <- LongandRefseqandShort_Cleaned2_GTF$source
source_PacBio <- LongandRefseqandShort_Cleaned2_GTF$source
source_Illumina <- LongandRefseqandShort_Cleaned2_GTF$source

source_Refseq[LongandRefseqandShort_Cleaned2_GTF$transcript_id %in% RefSeq_source] <- "RefSeq"
source_PacBio[LongandRefseqandShort_Cleaned2_GTF$transcript_id %in% PacBio_source] <- "PacBio_StringTie2"
source_Illumina[LongandRefseqandShort_Cleaned2_GTF$transcript_id %in% Illumina_source] <- "Illumina_StringTie2"
source <- paste(source_Refseq,source_PacBio,source_Illumina,sep=",") %>% gsub(",,",",",.) %>% gsub("^,|,$","",.)
source[source==""] <- "PacBio_StringTie2"
LongandRefseqandShort_Cleaned2_GTF$source <- source
```

## Add in Squanti filtering info

```{r}
tempF <- read.delim("Squanti/squanti3_filter_combined_vs_refseq//combine_vs_refseq.gtf_RulesFilter_result_classification.txt")
tempF %<>% as.data.frame %>% select(isoform,
                  structural_category,
                  subcategory,
                  RTS_stage,
                  all_canonical,
                  FSM_class,
                  coding,
                  predicted_NMD,
                  perc_A_downstream_TTS,
                  filter_result)
LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_GTF %>% as.data.frame()
LongandRefseqandShort_Cleaned2_GTF <- merge(LongandRefseqandShort_Cleaned2_DF,
   tempF,by.x="transcript_id",by.y="isoform",all.x=TRUE,all.y=FALSE) %>% GRanges()

export(LongandRefseqandShort_Cleaned2_GTF,con = "LongandRefseqandShort_Cleaned2_GTF.gtf")

LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_GTF %>% as.data.frame()
LongandRefseqandShort_Cleaned2_DF %>% 
 select(seqnames,start,end,width,strand,transcript_id,gene_id,source,type,score,phase) %>% GRanges() %>% 
 export(con = "LongandRefseqandShort_Cleaned2_GTF_simple.gtf")
```

```{r,eval=FALSE}
Herper::with_CondaEnv("transdecoder",{
 
 fasta <- "/rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_Reference_noGTF/workflow_data/extdata/GenBank.fa.gz"           
 output_dir <- "/rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/cleanedGTF2/"
 gtf <- "/rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/cleanedGTF/LongandRefseqandShort_Cleaned2_GTF_simple.gtf"
 transdecode_File <- "Transdecode_LongandRefseqandShort_Cleaned2_GTF.gtf"
 transcode_output <- file.path(output_dir,"transdecode")
 unzippedFasta <- file.path(output_dir,gsub("\\.gz","",basename(fasta)))
 transfasta <- file.path(output_dir,"transcripts.fasta")
 gffTemp <- file.path(output_dir,gsub("\\.gtf",".gff",basename(gtf)))
 R.utils::gunzip(fasta,unzippedFasta,remove=FALSE)
 
 cmd <- "gtf_genome_to_cdna_fasta.pl"
 args <- paste(gtf,unzippedFasta,">",transfasta)
 system(paste(cmd,args))

 cmd <- "gtf_to_alignment_gff3.pl"
 args <- paste(gtf,">",gffTemp)
 system(paste(cmd,args))

 system(paste("TransDecoder.LongOrfs -m 75 --output_dir",transcode_output,"-t",
        transfasta))

 system(paste("TransDecoder.Predict --single_best_only --output_dir",transcode_output,"-t",
        transfasta))
 system(paste("cdna_alignment_orf_to_genome_orf.pl",
        paste0(file.path(transcode_output,basename(transfasta)),".transdecoder.gff3"),
        gffTemp,transfasta,">",paste0(file.path(transcode_output,basename(transfasta)),".transdecoder.genome.gff3")))

 system(paste("gffread -E -T",file.path(transcode_output,
                     paste0(
                      basename(transfasta),
                      ".transdecoder.genome.gff3")),
        ">",
        file.path(output_dir,
             transdecode_File)),
 )
 R.utils::gzip(file.path(output_dir,
             transdecode_File),
        paste0(
         file.path(output_dir,
              transdecode_File),
         ".gz"),
        remove=FALSE)
 },
 pathToMiniConda = "/ru-auth/local/home/brc_pipeline/brc_conda/")



```

## Add rRNA annotations from Refseq

```{r}
RefSeq_GTF <- import("Originals/Refseq/Obiroi_RefSeq__5_4_renamed.gtf.gz")
RefSeq_txdb <- makeTxDbFromGRanges(RefSeq_GTF)
RefSeq_tx <- exonsBy(RefSeq_txdb,by = "tx",use.names=TRUE)
RefSeq_RNAs <- RefSeq_GTF[RefSeq_GTF$transcript_biotype %in%"rRNA"]$transcript_id %>% unique
RefSeq_RNA_tx <- RefSeq_tx[RefSeq_RNAs]

Comb_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
Comb_tx <- exonsBy(Comb_txdb,by = "tx",use.names=TRUE)

hits_rRNA <- findOverlaps(Comb_tx,RefSeq_RNA_tx,type = "equal")
rRNAgenes <- names(Comb_tx)[hits_rRNA@from]
rRNAfound <- names(RefSeq_RNA_tx)[hits_rRNA@to]

hits_rRNA2 <- findOverlaps(Comb_tx,RefSeq_RNA_tx)
rRNAfound2 <- names(RefSeq_RNA_tx)[hits_rRNA2@to] %>% unique
OverlappingRNA <- rRNAfound2[!rRNAfound2 %in% rRNAfound]

rRNA_to_add <- RefSeq_RNAs[!RefSeq_RNAs %in% rRNAfound2]


rRNAs_to_Label <- rRNAgenes
rRNA_to_add_GR <- RefSeq_GTF[RefSeq_GTF$transcript_id %in% rRNA_to_add]
mcols(rRNA_to_add_GR) <- mcols(rRNA_to_add_GR) %>% as.data.frame %>% select(source,type,score,phase)
```

## Add tRNA annotations from Refseq

```{r}
RefSeq_GTF <- import("Originals/Refseq/Obiroi_RefSeq__5_4_renamed.gtf.gz")
RefSeq_txdb <- makeTxDbFromGRanges(RefSeq_GTF)
RefSeq_tx <- exonsBy(RefSeq_txdb,by = "tx",use.names=TRUE)
RefSeq_tRNAs <- RefSeq_GTF[RefSeq_GTF$transcript_biotype %in%"tRNA"]$transcript_id %>% unique
RefSeq_tRNA_tx <- RefSeq_tx[RefSeq_tRNAs]

Comb_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
Comb_tx <- exonsBy(Comb_txdb,by = "tx",use.names=TRUE)

hits_tRNA <- findOverlaps(Comb_tx,RefSeq_tRNA_tx,type = "equal")
tRNAgenes <- names(Comb_tx)[hits_tRNA@from]
tRNAfound <- names(RefSeq_tRNA_tx)[hits_tRNA@to]

hits_tRNA2 <- findOverlaps(Comb_tx,RefSeq_tRNA_tx)
tRNAfound2 <- names(RefSeq_tRNA_tx)[hits_tRNA2@to] %>% unique
OverlappingtRNA <- tRNAfound2[!tRNAfound2 %in% tRNAfound]

tRNA_to_add <- RefSeq_tRNAs[!RefSeq_tRNAs %in% tRNAfound2]


tRNAs_to_Label <- tRNAgenes
tRNA_to_add_GR <- RefSeq_GTF[RefSeq_GTF$transcript_id %in% tRNA_to_add]
mcols(tRNA_to_add_GR) <- mcols(tRNA_to_add_GR) %>% as.data.frame %>% select(source,type,score,phase)

```

## Add snoRNA annotations from Refseq

```{r}
RefSeq_GTF <- import("Originals/Refseq/Obiroi_RefSeq__5_4_renamed.gtf.gz")
RefSeq_txdb <- makeTxDbFromGRanges(RefSeq_GTF)
RefSeq_tx <- exonsBy(RefSeq_txdb,by = "tx",use.names=TRUE)
RefSeq_snoRNAs <- RefSeq_GTF[RefSeq_GTF$transcript_biotype %in%"snoRNA"]$transcript_id %>% unique
RefSeq_snoRNA_tx <- RefSeq_tx[RefSeq_snoRNAs]

Comb_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
Comb_tx <- exonsBy(Comb_txdb,by = "tx",use.names=TRUE)

hits_snoRNA <- findOverlaps(Comb_tx,RefSeq_snoRNA_tx,type = "equal")
snoRNAgenes <- names(Comb_tx)[hits_snoRNA@from]
snoRNAfound <- names(RefSeq_snoRNA_tx)[hits_snoRNA@to]

hits_snoRNA2 <- findOverlaps(Comb_tx,RefSeq_snoRNA_tx)
snoRNAfound2 <- names(RefSeq_snoRNA_tx)[hits_snoRNA2@to] %>% unique
OverlappingsnoRNA <- snoRNAfound2[!snoRNAfound2 %in% snoRNAfound]

snoRNA_to_add <- RefSeq_snoRNAs[!RefSeq_snoRNAs %in% snoRNAfound2]


snoRNAs_to_Label <- snoRNAgenes
snoRNA_to_add_GR <- RefSeq_GTF[RefSeq_GTF$transcript_id %in% snoRNA_to_add]
mcols(snoRNA_to_add_GR) <- mcols(snoRNA_to_add_GR) %>% as.data.frame %>% select(source,type,score,phase)

```

## Add snRNA annotations from Refseq

```{r}
RefSeq_GTF <- import("Originals/Refseq/Obiroi_RefSeq__5_4_renamed.gtf.gz")
RefSeq_txdb <- makeTxDbFromGRanges(RefSeq_GTF)
RefSeq_tx <- exonsBy(RefSeq_txdb,by = "tx",use.names=TRUE)
RefSeq_snRNAs <- RefSeq_GTF[RefSeq_GTF$transcript_biotype %in%"snRNA"]$transcript_id %>% unique
RefSeq_snRNA_tx <- RefSeq_tx[RefSeq_snRNAs]

Comb_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
Comb_tx <- exonsBy(Comb_txdb,by = "tx",use.names=TRUE)

hits_snRNA <- findOverlaps(Comb_tx,RefSeq_snRNA_tx,type = "equal")
snRNAgenes <- names(Comb_tx)[hits_snRNA@from]
snRNAfound <- names(RefSeq_snRNA_tx)[hits_snRNA@to]

hits_snRNA2 <- findOverlaps(Comb_tx,RefSeq_snRNA_tx)
snRNAfound2 <- names(RefSeq_snRNA_tx)[hits_snRNA2@to] %>% unique
OverlappingsnRNA <- snRNAfound2[!snRNAfound2 %in% snRNAfound]

snRNA_to_add <- RefSeq_snRNAs[!RefSeq_snRNAs %in% snRNAfound2]


snRNAs_to_Label <- snRNAgenes
snRNA_to_add_GR <- RefSeq_GTF[RefSeq_GTF$transcript_id %in% snRNA_to_add]
mcols(snRNA_to_add_GR) <- mcols(snRNA_to_add_GR) %>% as.data.frame %>% select(source,type,score,phase)

```

## Add piRNA annotations from pilfer

```{r}
piRNAbed <- import.bed("piRNAs/piRNAs_pilfer.bed")
piRNAbed <- renameSeqlevels(piRNAbed,
               chr_map %>% pull(assembly,name = "X..refseq"))
export(piRNAbed,"piRNAs/piRNAs_pilfer_renamed.bed")

piRNAbed <- GenomicRanges::reduce(piRNAbed)
export(piRNAbed,"piRNAs/piRNAs_pilfer_renamed_reduced.bed")


Comb_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
Comb_tx <- exonsBy(Comb_txdb,by = "tx",use.names=TRUE)

hits_piRNA <- findOverlaps(Comb_tx,piRNAbed)

```

```{r}
LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_GTF %>% 
 as.data.frame()
LongandRefseqandShort_Cleaned2_DF$transcript_type <- NA
LongandRefseqandShort_Cleaned2_DF[LongandRefseqandShort_Cleaned2_DF$type == "transcript",]$transcript_type <- "mRNA"

LongandRefseqandShort_Cleaned2_DF[LongandRefseqandShort_Cleaned2_DF$type == "transcript" & LongandRefseqandShort_Cleaned2_DF$transcript_id %in% rRNAs_to_Label,]$transcript_type <- "rRNA"

LongandRefseqandShort_Cleaned2_DF[LongandRefseqandShort_Cleaned2_DF$type == "transcript" & LongandRefseqandShort_Cleaned2_DF$transcript_id %in% tRNAs_to_Label,]$transcript_type <- "tRNA"

LongandRefseqandShort_Cleaned2_DF[LongandRefseqandShort_Cleaned2_DF$type == "transcript" & LongandRefseqandShort_Cleaned2_DF$transcript_id %in% snoRNAs_to_Label,]$transcript_type <- "snoRNA"

LongandRefseqandShort_Cleaned2_DF[LongandRefseqandShort_Cleaned2_DF$type == "transcript" & LongandRefseqandShort_Cleaned2_DF$transcript_id %in% snRNAs_to_Label,]$transcript_type <- "snRNA"

LongandRefseqandShort_Cleaned2_GTF <- LongandRefseqandShort_Cleaned2_DF %>% GRanges()

rRNA_to_add_df <- rRNA_to_add_GR %>% as.data.frame() %>% mutate(source="RefSeq",Refseq_gene_id=NA,Ens_gene_id=NA,Genbank_gene_id=NA,Refseq_transcript_id=NA,
                       Ens_transcript_id=NA,Genbank_transcript_id=NA,BuscoInformation=NA,
                       structural_category=NA,subcategory=NA,
                    RTS_stage=NA,all_canonical=NA,FSM_class=NA,coding=NA,
                    predicted_NMD=NA,perc_A_downstream_TTS=NA,
                       filter_result="Isoform",transcript_type="rRNA") 

maxID <- LongandRefseqandShort_Cleaned2_GTF$gene_id %>% gsub("RU\\.","",.) %>% as.numeric() %>% max
rRNA_to_add_df$gene_id <- NA
rRNA_to_add_df$transcript_id <- NA
rRNA_to_add_df[rRNA_to_add_df$type == "transcript",]$gene_id <- paste0("RU.",seq(maxID+1,maxID+nrow(rRNA_to_add_df[rRNA_to_add_df$type %in% "transcript",])))
rRNA_to_add_df[rRNA_to_add_df$type == "exon",]$gene_id <- paste0("RU.",seq(maxID+1,maxID+nrow(rRNA_to_add_df[rRNA_to_add_df$type %in% "exon",])))
rRNA_to_add_df[rRNA_to_add_df$type == "transcript",]$transcript_id <- paste0(rRNA_to_add_df[rRNA_to_add_df$type == "transcript",]$gene_id,".1")
rRNA_to_add_df[rRNA_to_add_df$type == "exon",]$transcript_id <- paste0(rRNA_to_add_df[rRNA_to_add_df$type == "exon",]$gene_id,".1")
rRNA_to_add_GR <- rRNA_to_add_df %>% GRanges()
LongandRefseqandShort_Cleaned2_GTF <- c(LongandRefseqandShort_Cleaned2_GTF,rRNA_to_add_GR) %>% sort
```

```{r}
LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by = "gene")
selfOverlaps <- findOverlaps(LongandRefseq_exonsByGene,LongandRefseq_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]

```


```{r}
transTemp <- import("transdecoder/Transdecode_75bpORF_LongandRefseqandShort_Cleaned2_GTF.gtf")
transTemp$transcript_id <- transTemp$transcript_id %>% 
 gsub("\\.p.*","",.)
transTemp$gene_id <- transTemp$gene_id %>% 
 gsub("\\^.*","",.)
transTempStrandSummary <- transTemp %>% as.data.frame() %>% 
 select(transcript_id,strand) %>% 
 group_by(transcript_id) %>% summarise(Strand=unique(strand))

LRStradSummary <- LongandRefseqandShort_Cleaned2_GTF %>% as.data.frame %>% filter(!is.na(transcript_id)) %>% 
 select(transcript_id,strand) %>% 
 group_by(transcript_id) %>% summarise(Strand=unique(strand))

compareStrand <- merge(transTempStrandSummary,LRStradSummary,by=1,all=FALSE)
noCodingOnCorrectSrand <- compareStrand[compareStrand$Strand.x != compareStrand$Strand.y,"transcript_id"]

transTemp <- transTemp[!transTemp$transcript_id %in% noCodingOnCorrectSrand]
transTemp_CDS <- transTemp[transTemp$type == "CDS"]
LongandRefseqandShort_Cleaned3_GTF <- c(LongandRefseqandShort_Cleaned2_GTF,transTemp_CDS) %>% sort


allTranscripts <- LongandRefseqandShort_Cleaned3_GTF[LongandRefseqandShort_Cleaned3_GTF$transcript_type %in% "mRNA",]$transcript_id %>% unique
codingTranscripts <- LongandRefseqandShort_Cleaned3_GTF[LongandRefseqandShort_Cleaned3_GTF$type == "CDS" & LongandRefseqandShort_Cleaned3_GTF$type == "CDS",]$transcript_id %>% unique
lncRNAs <- allTranscripts[!allTranscripts %in% codingTranscripts] %>% unique
LongandRefseqandShort_Cleaned3_GTF[LongandRefseqandShort_Cleaned3_GTF$transcript_id %in% lncRNAs & LongandRefseqandShort_Cleaned3_GTF$type == "transcript",]$transcript_type <- "lncRNA"
# makeTxDbFromGRanges(LongandRefseqandShort_Cleaned3_GTF)
```


```{r,eval=FALSE}
LongandRefseqandShort_Cleaned3_GTF[LongandRefseqandShort_Cleaned3_GTF$coding %in% "non_coding" & LongandRefseqandShort_Cleaned3_GTF$type == "transcript"]$transcript_type %>% table %>% as.data.frame %>% DT::datatable()

LongandRefseqandShort_Cleaned3_GTF[LongandRefseqandShort_Cleaned3_GTF$coding %in% "coding" & LongandRefseqandShort_Cleaned3_GTF$type == "transcript"]$transcript_type %>% table %>% as.data.frame %>% DT::datatable()

LongandRefseqandShort_Cleaned3_GTF[LongandRefseqandShort_Cleaned3_GTF$transcript_type %in% "lncRNA" & LongandRefseqandShort_Cleaned3_GTF$type == "transcript"]$coding %>% table %>% as.data.frame %>% DT::datatable()

LongandRefseqandShort_Cleaned3_GTF[LongandRefseqandShort_Cleaned3_GTF$transcript_type %in% "mRNA" & LongandRefseqandShort_Cleaned3_GTF$type == "transcript"]$coding %>% table %>% as.data.frame %>% DT::datatable()




```

```{r}
codingInSquanti <- LongandRefseqandShort_Cleaned3_GTF[LongandRefseqandShort_Cleaned3_GTF$transcript_type %in% "lncRNA" & LongandRefseqandShort_Cleaned3_GTF$type == "transcript" & LongandRefseqandShort_Cleaned3_GTF$coding == "coding"]$transcript_id
squanti_cds <- import("Squanti/squanti3_combined_vs_refseq/combine_vs_refseq.gtf_corrected.gtf.cds.gff")
squanti_justcds <- squanti_cds[squanti_cds$transcript_id %in% codingInSquanti & squanti_cds$type == "CDS",]

squanti_justcds$gene_id <- gsub("\\.\\d+$","",squanti_justcds$transcript_id)

LongandRefseqandShort_Cleaned4_GTF <- c(LongandRefseqandShort_Cleaned3_GTF,squanti_justcds) %>% sort

LongandRefseqandShort_Cleaned4_GTF[LongandRefseqandShort_Cleaned4_GTF$transcript_id %in% codingInSquanti & LongandRefseqandShort_Cleaned4_GTF$type == "transcript"]$transcript_type <- "mRNA"


```

```{r,eval=FALSE}
LongandRefseqandShort_Cleaned4_GTF[LongandRefseqandShort_Cleaned4_GTF$transcript_type %in% "lncRNA" & LongandRefseqandShort_Cleaned4_GTF$type == "transcript"]$coding %>% table %>% as.data.frame %>% DT::datatable()
```

```{r}
LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned4_GTF)
LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by = "gene")
selfOverlaps <- findOverlaps(LongandRefseq_exonsByGene,LongandRefseq_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]

```

```{r}
LongandRefseqandShort_Cleaned4_DF <- LongandRefseqandShort_Cleaned4_GTF %>% as.data.frame()
LongandRefseqandShort_Cleaned4_DF_exon <- LongandRefseqandShort_Cleaned4_DF[LongandRefseqandShort_Cleaned4_DF$type == "exon",]
LongandRefseqandShort_Cleaned4_DF_cds <- LongandRefseqandShort_Cleaned4_DF[LongandRefseqandShort_Cleaned4_DF$type == "CDS",]
LongandRefseqandShort_Cleaned4_DF_exon <- LongandRefseqandShort_Cleaned4_DF_exon %>% select(seqnames,start,end,width,strand,transcript_id,gene_id,type,score,phase)
LongandRefseqandShort_Cleaned4_DF_cds <- LongandRefseqandShort_Cleaned4_DF_cds %>% select(seqnames,start,end,width,strand,transcript_id,gene_id,source,type,score,phase)
#LongandRefseqandShort_Cleaned4_DF_cds$source %>% is.na %>% table
#LongandRefseqandShort_Cleaned4_DF_cds$source %>% table
LongandRefseqandShort_Cleaned4_DF_cds$source  <- ifelse(LongandRefseqandShort_Cleaned4_DF_cds$source == "PacBio","GeneMark","transdecoder")

LongandRefseqandShort_Cleaned4_DF_transcript <- LongandRefseqandShort_Cleaned4_DF[LongandRefseqandShort_Cleaned4_DF$type == "transcript",]
# LongandRefseqandShort_Cleaned4_DF_transcript$transcript_id %>% is.na %>% table
# LongandRefseqandShort_Cleaned4_DF_transcript$gene_id %>% is.na %>% table
# LongandRefseqandShort_Cleaned4_DF_transcript$source  %>% is.na %>% table
# LongandRefseqandShort_Cleaned4_DF_transcript$source  %>% table

source <- LongandRefseqandShort_Cleaned4_DF_transcript$source %>% as.character()
source[grepl("RefSeq",LongandRefseqandShort_Cleaned4_DF_transcript$source) & is.na(LongandRefseqandShort_Cleaned4_DF_transcript$Refseq_gene_id)] <- gsub("RefSeq","GenBank",LongandRefseqandShort_Cleaned4_DF_transcript[grepl("RefSeq",LongandRefseqandShort_Cleaned4_DF_transcript$source) & is.na(LongandRefseqandShort_Cleaned4_DF_transcript$Refseq_gene_id),]$source)
 LongandRefseqandShort_Cleaned4_DF_transcript$source <- factor(source)

LongandRefseqandShort_Cleaned4_DF_transcript[LongandRefseqandShort_Cleaned4_DF_transcript$seqnames == "MT",]$transcript_type <- "mRNA"

LongandRefseqandShort_Cleaned4_GTF <- c(LongandRefseqandShort_Cleaned4_DF_transcript %>% GRanges(),
LongandRefseqandShort_Cleaned4_DF_cds %>% GRanges(),
LongandRefseqandShort_Cleaned4_DF_exon %>% GRanges()) %>% sort

```

```{r}
LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned4_GTF)
LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by = "gene")
selfOverlaps <- findOverlaps(LongandRefseq_exonsByGene,LongandRefseq_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]

```

```{r}
names(LongandRefseqandShort_Cleaned4_GTF) <- NULL
# LongandRefseqandShort_Cleaned4_GTF

chr_map <- read.delim("Originals/UCSC_Map/GCF_003672135.1.chromAlias.txt")

LongandRefseqandShort_Cleaned4_GTF_1 <- LongandRefseqandShort_Cleaned4_GTF[!grepl("NW_|QO",seqnames(LongandRefseqandShort_Cleaned4_GTF))]

LongandRefseqandShort_Cleaned4_GTF_2 <- LongandRefseqandShort_Cleaned4_GTF[grepl("NW_",seqnames(LongandRefseqandShort_Cleaned4_GTF))]
LongandRefseqandShort_Cleaned4_GTF_2 <- renameSeqlevels(LongandRefseqandShort_Cleaned4_GTF_2,
               chr_map %>% pull(assembly,name = "X..refseq"))

LongandRefseqandShort_Cleaned4_GTF_3 <- LongandRefseqandShort_Cleaned4_GTF[grepl("QO",seqnames(LongandRefseqandShort_Cleaned4_GTF))]

LongandRefseqandShort_Cleaned4_GTF_3 <- renameSeqlevels(LongandRefseqandShort_Cleaned4_GTF_3,
               chr_map %>% pull(assembly,name = "genbank"))

LongandRefseqandShort_Cleaned4_GTF_4 <- c(LongandRefseqandShort_Cleaned4_GTF_1,LongandRefseqandShort_Cleaned4_GTF_2,LongandRefseqandShort_Cleaned4_GTF_3)


LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned4_GTF_4)
LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by = "gene")
selfOverlaps <- findOverlaps(LongandRefseq_exonsByGene,LongandRefseq_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]

LongandRefseqandShort_Cleaned4_GTF_4 <- LongandRefseqandShort_Cleaned4_GTF_4[!LongandRefseqandShort_Cleaned4_GTF_4$gene_id %in% c("RU.15009"),]

LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned4_GTF_4)
LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by = "gene")
selfOverlaps <- findOverlaps(LongandRefseq_exonsByGene,LongandRefseq_exonsByGene)
temp_selfOverlaps <- as.data.frame(selfOverlaps)

temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$queryHits]
temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$subjectHits]


temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,]

LongandRefseqandShort_Cleaned4_GTF <- LongandRefseqandShort_Cleaned4_GTF_4
# export(LongandRefseqandShort_Cleaned4_GTF,con = "LongandRefseqandShort_Cleaned4_GTF.gtf")
```


<!-- ### Add M_and_F ShortRead trascripts which are contained within single LongRead gene models. -->

<!-- ```{r} -->

<!-- LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned4_GTF) -->

<!-- LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by="gene") -->

<!-- ShortStringtie_MandF_gtf <- import("Illumina/StringTie_MaleAndFemale/stringtie/stringtie/merged_stringtie.gtf") -->
<!-- ShortStringtie_MandF_gtf <- ShortStringtie_MandF_gtf[strand(ShortStringtie_MandF_gtf) != "*"] -->

<!-- ShortStringtie_MandF_txdb <- makeTxDbFromGRanges(ShortStringtie_MandF_gtf) -->
<!-- ShortStringtie_MandF_exonsByTranscript <- exonsBy(ShortStringtie_MandF_txdb,by="tx",use.names=TRUE) -->
<!-- ShortStringtie_MandF_GTF_GeneToTranscript <- ShortStringtie_MandF_gtf %>% mcols %>%  -->
<!--  as.data.frame() %>%  -->
<!--  dplyr::select(gene_id,transcript_id) %>%  -->
<!--  distinct() -->

<!-- someOverlaps <- findOverlaps(LongandRefseq_exonsByGene,ShortStringtie_MandF_exonsByTranscript) -->
<!-- temp_someOverlaps <- as.data.frame(someOverlaps) -->

<!-- temp_someOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_someOverlaps$queryHits] -->
<!-- temp_someOverlaps$subjectHitsNames <- names(ShortStringtie_MandF_exonsByTranscript)[temp_someOverlaps$subjectHits] -->


<!-- temp_someOverlaps$subjectHitsGeneNames <- ShortStringtie_MandF_GTF_GeneToTranscript$gene_id[match(temp_someOverlaps$subjectHitsNames, -->
<!--                        ShortStringtie_MandF_GTF_GeneToTranscript$transcript_id)] -->

<!-- shortOverlappingTrans <- temp_someOverlaps %>%  -->
<!--  mutate(dup=duplicated(subjectHitsNames)) %>% filter(dup) %>% pull(subjectHitsNames) -->

<!-- # shortOverlappingTrans2 <- temp_someOverlaps %>%  -->
<!-- #  mutate(dup=duplicated(queryHitsNames)) %>% filter(dup) %>% pull(queryHitsNames) -->
<!-- #  -->
<!-- # [temp_someOverlaps$queryHitsNames %in% shortOverlappingTrans2,] -->


<!-- ShortStringtie_MandF_Cleaned <- ShortStringtie_MandF_gtf[!ShortStringtie_MandF_gtf$transcript_id %in% shortOverlappingTrans] -->
<!-- ShortStringtie_MandF_txdb2 <- makeTxDbFromGRanges(ShortStringtie_MandF_Cleaned) -->


<!-- ShortStringtie_MandF_exonsByTranscript <- exonsBy(ShortStringtie_MandF_txdb2,by="tx",use.names=TRUE) -->
<!-- ShortStringtie_MandF_GTF_GeneToTranscript <- ShortStringtie_MandF_Cleaned %>% mcols %>%  -->
<!--  as.data.frame() %>%  -->
<!--  dplyr::select(gene_id,transcript_id) %>%  -->
<!--  distinct() -->

<!-- someOverlaps <- findOverlaps(LongandRefseq_exonsByGene,ShortStringtie_MandF_exonsByTranscript) -->
<!-- temp_someOverlaps <- as.data.frame(someOverlaps) -->

<!-- temp_someOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_someOverlaps$queryHits] -->
<!-- temp_someOverlaps$subjectHitsNames <- names(ShortStringtie_MandF_exonsByTranscript)[temp_someOverlaps$subjectHits] -->


<!-- temp_someOverlaps$subjectHitsGeneNames <- ShortStringtie_MandF_GTF_GeneToTranscript$gene_id[match(temp_someOverlaps$subjectHitsNames, -->
<!--                        ShortStringtie_MandF_GTF_GeneToTranscript$transcript_id)] -->

<!-- LongToShort_geneId <- temp_someOverlaps$queryHitsNames[ -->
<!--               match(ShortStringtie_MandF_Cleaned$transcript_id,temp_someOverlaps$subjectHitsNames) -->
<!--               ] -->

<!-- ShortStringtie_MandF_Cleaned$old_gene_id <- ShortStringtie_MandF_Cleaned$gene_id -->
<!-- ShortStringtie_MandF_Cleaned$gene_id <- LongToShort_geneId -->
<!-- ShortStringtie_MandF_Cleaned$gene_id[is.na(ShortStringtie_MandF_Cleaned$gene_id)] <- ShortStringtie_MandF_Cleaned$old_gene_id[is.na(ShortStringtie_MandF_Cleaned$gene_id)] -->
<!-- # BRCDK_Cleaned$gene_id %>% unique %>% length -->
<!-- LongandRefseqandShort_GTF <- c(LongandRefseqandShort_Cleaned4_GTF,ShortStringtie_MandF_Cleaned) -->



<!-- LongandRefseqandShort_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_GTF) -->
<!-- LongandRefseqandShort_exonsByGene <- exonsBy(LongandRefseqandShort_txdb,by = "gene") -->
<!-- LongandRefseqandShort_transcriptsByGene <- exonsBy(LongandRefseqandShort_txdb,by = "tx",use.names=TRUE) -->
<!-- selfOverlaps <- findOverlaps(LongandRefseqandShort_exonsByGene,LongandRefseqandShort_transcriptsByGene) -->
<!-- temp_selfOverlaps <- as.data.frame(selfOverlaps) -->

<!-- temp_selfOverlaps$queryHitsNames <- names(LongandRefseqandShort_exonsByGene)[temp_selfOverlaps$queryHits] -->
<!-- temp_selfOverlaps$subjectHitsNames <- names(LongandRefseqandShort_transcriptsByGene)[temp_selfOverlaps$subjectHits] -->


<!-- shortOverlappingTrans2 <- temp_selfOverlaps %>%  -->
<!--  mutate(dup=duplicated(subjectHitsNames)) %>% filter(dup) %>% pull(subjectHitsNames) -->

<!-- LongandRefseqandShort_Cleaned_GTF <- LongandRefseqandShort_GTF[!LongandRefseqandShort_GTF$transcript_id %in% shortOverlappingTrans2] -->
<!-- # rtracklayer::export(LongandRefseqandShort_Cleaned_GTF,con = "LongandRefseqandShort_Cleaned_GTF.gtf") -->


<!-- LongandRefseq_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned_GTF) -->
<!-- LongandRefseq_exonsByGene <- exonsBy(LongandRefseq_txdb,by = "gene") -->
<!-- selfOverlaps <- findOverlaps(LongandRefseq_exonsByGene,LongandRefseq_exonsByGene) -->
<!-- temp_selfOverlaps <- as.data.frame(selfOverlaps) -->

<!-- temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$queryHits] -->
<!-- temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_exonsByGene)[temp_selfOverlaps$subjectHits] -->


<!-- temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,] -->
<!-- # nrow(temp_selfOverlaps) -->


<!-- ``` -->

<!-- ### Clean up to make sure no ShortRead transcripts overlap multiple LongRead genes. -->

<!-- ```{r} -->

<!-- LongandRefseq_transcriptsByGene <- exonsBy(LongandRefseq_txdb,by = "tx",use.names=TRUE) -->
<!-- selfOverlaps <- findOverlaps(LongandRefseq_transcriptsByGene,LongandRefseq_transcriptsByGene,type = "equal") -->
<!-- temp_selfOverlaps <- as.data.frame(selfOverlaps) -->
<!-- temp_selfOverlaps <- temp_selfOverlaps[temp_selfOverlaps$queryHits != temp_selfOverlaps$subjectHits,] -->
<!-- temp_selfOverlaps$queryHitsNames <- names(LongandRefseq_transcriptsByGene)[temp_selfOverlaps$queryHits] -->
<!-- temp_selfOverlaps$subjectHitsNames <- names(LongandRefseq_transcriptsByGene)[temp_selfOverlaps$subjectHits] -->
<!-- toFilter <- temp_selfOverlaps %>% select(queryHitsNames,subjectHitsNames) %>% apply(1,sort) %>% t %>% as.data.frame %>% distinct -->

<!-- repeatTranscripts <- unique(toFilter$V2) -->

<!-- LongandRefseqandShort_Cleaned2_GTF <- LongandRefseqandShort_Cleaned_GTF[!LongandRefseqandShort_Cleaned_GTF$transcript_id %in%  -->
<!--                                      repeatTranscripts] -->

<!-- ``` -->


<!-- ```{r} -->
<!-- LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_GTF %>% as.data.frame(row.names = NULL) -->

<!-- unique_genes <- unique(LongandRefseqandShort_Cleaned2_DF$gene_id) -->
<!-- new_gene_ids <- paste0("RU.", seq_along(unique_genes)) -->
<!-- gene_map <- setNames(new_gene_ids, unique_genes) -->

<!-- LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_DF %>% -->
<!--  mutate(ru_gene_id = gene_map[gene_id]) -->


<!-- LongandRefseqandShort_Cleaned2_DF <- LongandRefseqandShort_Cleaned2_DF %>% -->
<!--  group_by(ru_gene_id) %>% -->
<!--  mutate(ru_trans_id = paste0(ru_gene_id, ".", dense_rank(transcript_id))) %>% -->
<!--  ungroup() -->

<!-- LongandRefseqandShort_Cleaned2_DF %<>%  -->
<!--  mutate(old_gene_id=ifelse(is.na(old_gene_id),gene_id,old_gene_id)) -->


<!-- LongandRefseqandShort_Cleaned2_DF %<>%  -->
<!--  mutate(gene_id=ru_gene_id,transcript_id=ru_trans_id) %>%  -->
<!--  filter(type == "exon") %>%  -->
<!--  select(-ru_gene_id,-ru_trans_id) -->

<!-- LongandRefseqandShort_Cleaned2_GTF <- GRanges(LongandRefseqandShort_Cleaned2_DF) -->

<!-- export(LongandRefseqandShort_Cleaned2_GTF,con = "LongandRefseqandShort_Cleaned3_GTF.gtf") -->
<!-- ``` -->

```{r}
filesToParse <- dir("mirdeep/",pattern="*.csv",full.names=TRUE)
temp <- list()
for(i in 1:length(filesToParse)){
 tempa <- read.delim(filesToParse[i],skip = 26,sep="\t",header = TRUE)
 tempB <- tempa$precursor.coordinate %>% GRanges()
 tempB$miRNA_ID <- paste0(ifelse(tempa$example.miRBase.miRNA.with.the.same.seed == "-","Novel",tempa$example.miRBase.miRNA.with.the.same.seed),
              ":",tempa$consensus.mature.sequence,":",tempa$consensus.precursor.sequence)
 names(tempB) <- tempB$miRNA_ID
 temp[[i]] <- tempB
}
temp <- GRangesList(temp) %>% unlist 
temp2 <- GenomicRanges::reduce(temp,with.revmap=T)
mirnaname <- list()
for(i in 1:length(temp2$revmap)){
 mirnaname[[i]] <- paste(unique(temp$miRNA_ID[temp2$revmap[[i]]]),collapse=";")
}
temp2$miRNA_ID <- unlist(mirnaname)
miRNA_to_add_GR <- temp2 %>% as.data.frame() %>% select(-revmap) %>% GRanges
miRNA_to_add_GR <- renameSeqlevels(miRNA_to_add_GR,
               chr_map %>% pull(assembly,name = "X..refseq"))

miRNA_to_add_df <- miRNA_to_add_GR %>% as.data.frame() %>% mutate(source="mirdeep2",Refseq_gene_id=NA,Ens_gene_id=NA,Genbank_gene_id=NA,Refseq_transcript_id=NA,
                       Ens_transcript_id=NA,Genbank_transcript_id=NA,BuscoInformation=NA,
                       structural_category=NA,subcategory=NA,
                    RTS_stage=NA,all_canonical=NA,FSM_class=NA,coding=NA,
                    predicted_NMD=NA,perc_A_downstream_TTS=NA,
                       filter_result="Isoform",transcript_type="miRNA") 

maxID <- LongandRefseqandShort_Cleaned4_GTF$gene_id %>% gsub("RU\\.","",.) %>% as.numeric() %>% max
miRNA_to_add_df$gene_id <- NA
miRNA_to_add_df$transcript_id <- NA
miRNA_to_add_df$gene_id <- paste0("RU.",seq(maxID+1,maxID+nrow(miRNA_to_add_df)))
miRNA_to_add_df$transcript_id <- paste0(miRNA_to_add_df$gene_id,".1")
miRNA_to_add_df_trans <- miRNA_to_add_df
miRNA_to_add_df_exon <- miRNA_to_add_df
miRNA_to_add_df_exon$type <- "exon"
miRNA_to_add_df_trans$type <- "transcript"
# miRNA_to_add_df_exon$transcript_id <- NA
miRNA_to_add_df_exon$filter_result <- NA
miRNA_to_add_GR <- rbind(miRNA_to_add_df_exon,miRNA_to_add_df_trans) %>% GRanges()
LongandRefseqandShort_Cleaned5_GTF <- c(LongandRefseqandShort_Cleaned4_GTF,miRNA_to_add_GR) %>% sort

# findOverlaps(LongandRefseq_exonsByGene,miRNA_to_add_GR)
```

```{r}
piRNA_to_add_df <- piRNAbed %>% as.data.frame() %>% mutate(source="Pilfer",Refseq_gene_id=NA,Ens_gene_id=NA,Genbank_gene_id=NA,Refseq_transcript_id=NA,
                       Ens_transcript_id=NA,Genbank_transcript_id=NA,BuscoInformation=NA,
                       structural_category=NA,subcategory=NA,
                    RTS_stage=NA,all_canonical=NA,FSM_class=NA,coding=NA,
                    predicted_NMD=NA,perc_A_downstream_TTS=NA,
                       filter_result="Isoform",transcript_type="piRNA") 

maxID <- LongandRefseqandShort_Cleaned5_GTF$gene_id %>% gsub("RU\\.","",.) %>% as.numeric() %>% max
piRNA_to_add_df$gene_id <- NA
piRNA_to_add_df$transcript_id <- NA
piRNA_to_add_df$gene_id <- paste0("RU.",seq(maxID+1,maxID+nrow(piRNA_to_add_df)))
piRNA_to_add_df$transcript_id <- paste0(piRNA_to_add_df$gene_id,".1")
piRNA_to_add_df_trans <- piRNA_to_add_df
piRNA_to_add_df_exon <- piRNA_to_add_df
piRNA_to_add_df_exon$type <- "exon"
piRNA_to_add_df_trans$type <- "transcript"
# piRNA_to_add_df_exon$transcript_id <- NA
piRNA_to_add_df_exon$filter_result <- NA
piRNA_to_add_GR <- rbind(piRNA_to_add_df_exon,piRNA_to_add_df_trans) %>% GRanges()
LongandRefseqandShort_Cleaned6_GTF <- c(LongandRefseqandShort_Cleaned5_GTF,piRNA_to_add_GR) %>% sort

```




```{r}
Combined_all <- LongandRefseqandShort_Cleaned6_GTF
piRNAs <- Combined_all[Combined_all$type == "transcript" & Combined_all$transcript_type == "piRNA"]$transcript_id
sq_filter <- Combined_all[Combined_all$type == "transcript" & 
                            Combined_all$transcript_type %in% c("mRNA","lncRNA") & 
                            Combined_all$source == "PacBio_StringTie2" &
                            is.na(Combined_all$BuscoInformation)]$transcript_id

sqSum <- Combined_all %>% 
  as.data.frame %>% 
  select(structural_category,subcategory,RTS_stage,all_canonical,FSM_class,coding,predicted_NMD,perc_A_downstream_TTS,filter_result) %>% 
  apply(1,paste,sep=";",collapse=";")

Combined_all2 <- Combined_all %>% as.data.frame %>% 
  select(-structural_category,-subcategory,-RTS_stage,-all_canonical,-FSM_class,-coding,-predicted_NMD,-perc_A_downstream_TTS,-filter_result)
Combined_all2$SQinfo <- sqSum
Combined_all2 %<>% GRanges()

Combined_all_no_pirna <- Combined_all2[!Combined_all2$transcript_id %in% piRNAs]
Combined_all_no_pirna_filtered <- Combined_all2[!Combined_all2$transcript_id %in% piRNAs & 
                                         !Combined_all2$transcript_id %in% sq_filter]

export(Combined_all2,
       con = "Combined.gtf")
export(Combined_all_no_pirna,
       con = "Combined_noPiRNA.gtf")
export(Combined_all_no_pirna_filtered,
       con = "Combined_noPiRNA_filtered.gtf")
```


# QC final gene models with gffcompare, squanti3 and Busco

### Create output for BUSCO assessment

```{r}
genomeSeq <- readDNAStringSet(filepath = "Genome/GenBank.fa")
Combined_all_no_pirna2 <- Combined_all_no_pirna[seqnames(Combined_all_no_pirna) %in% names(genomeSeq),]
Combined_txdb <- makeTxDbFromGRanges(Combined_all_no_pirna2)
Combined_exonsBytranscript <- exonsBy(Combined_txdb,by="tx",use.names=TRUE)
Combined_transSeq <- GenomicFeatures::extractTranscriptSeqs(FaFile("Genome/GenBank.fa"),Combined_exonsBytranscript)
writeXStringSet(Combined_transSeq,"Combined_all_noPiRNA_transSeq.fa")

genomeSeq <- readDNAStringSet(filepath = "Genome/GenBank.fa")
Combined_all_no_pirna_filtered22 <- Combined_all_no_pirna_filtered[seqnames(Combined_all_no_pirna_filtered) %in% names(genomeSeq),]
Combined_txdb <- makeTxDbFromGRanges(Combined_all_no_pirna_filtered22)
Combined_exonsBytranscript <- exonsBy(Combined_txdb,by="tx",use.names=TRUE)
Combined_transSeq <- GenomicFeatures::extractTranscriptSeqs(FaFile("Genome/GenBank.fa"),Combined_exonsBytranscript)
writeXStringSet(Combined_transSeq,"Combined_all_noPiRNA_filtered_transSeq.fa")

```

### Create output for squanti QC assessment and correction

```{r}
Combined_all_no_pirna_filteredtxdb <- makeTxDbFromGRanges(Combined_all_no_pirna_filtered)
Combined_genes <- genes(Combined_all_no_pirna_filteredtxdb,columns=c("gene_id"))
Combined_genes <- as.data.frame(Combined_genes) %>% mutate(type="gene") %>% GRanges()

Combined_GFT <- c(Combined_genes,Combined_all_no_pirna_filtered) %>% sort
names(Combined_GFT) <- NULL
export(Combined_GFT,con = "Combined_all_no_pirna_filtered_forSQ.gtf")

Combined_all_no_pirnatxdb <- makeTxDbFromGRanges(Combined_all_no_pirna)
Combined_genes <- genes(Combined_all_no_pirnatxdb,columns=c("gene_id"))
Combined_genes <- as.data.frame(Combined_genes) %>% mutate(type="gene") %>% GRanges()

Combined_GFT <- c(Combined_genes,Combined_all_no_pirna) %>% sort
names(Combined_GFT) <- NULL
export(Combined_GFT,con = "Combined_all_no_pirna_forSQ.gtf")

```



### Run gffcompare
```{r,eval=FALSE}
dir.create("gffcompare_2025",showWarnings = FALSE,recursive = TRUE)
Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Refseq.gtf -o gffcompare_2025/Combined_all_no_pirna_filtered_forSQ_refseq Combined_all_no_pirna_filtered_forSQ.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")

Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Refseq.gtf -o gffcompare_2025/Combined_all_no_pirna_forSQ_refseq Combined_all_no_pirna_forSQ.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")

Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Genbank.gtf -o gffcompare_2025/Combined_all_no_pirna_filtered_forSQ_genbank Combined_all_no_pirna_filtered_forSQ.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")

Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Genbank.gtf -o gffcompare_2025/Combined_all_no_pirna_forSQ_genbank Combined_all_no_pirna_forSQ.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")

```


# Parse QC and make XLSx report for unfiltered

```{r}
parse_cuffCompare <- function(stats_file){
lines <- readLines(stats_file)
prec_sens <- lines[grepl("level",lines)] %>% 
 str_trim() %>% 
 gsub("Intron chain","Intron_chain",.) %>% 
 gsub(" level:","_level",.) %>% 
 gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V3,-V5) %>% set_colnames(c("level","Sensitivity","Precision"))

Query_mrna <- lines[grepl("Query mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4,-V5) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))
 
Reference_mrna <- lines[grepl("Reference mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=4,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))

matching <- lines[grepl("Matching",lines)] %>% str_trim() %>%
   gsub(".*:","",.) %>% str_trim() %>% as.numeric %>% as.data.frame() %>% t %>% 
 set_colnames(c("Matching_intron_chains","Matching_transcripts","Matching_loci"))

myRownames <- lines[grepl("Missed|Novel",lines)] %>% 
 str_trim() %>% 
 gsub(":.*","",.) %>% gsub(" ","_",.)
Novel_missing <- lines[grepl("Missed|Novel",lines)] %>% str_trim() %>% 
 gsub(".*:","",.) %>% str_trim() %>% 
 gsub("\t.*","",.) %>% str_trim() %>% str_split("/") %>% unlist %>% matrix(ncol=2,byrow = TRUE) %>% 
 as.data.frame() %>% set_colnames(c("n_of","Total")) %>% set_rownames(myRownames)

toReturn <- list(prec_sens,Query_mrna,Reference_mrna,matching,Novel_missing)
names(toReturn) <- c("prec_sens","Query_mrna","Reference_mrna","matching","Novel_missing")
toReturn
}

refseq_gffcompare <- parse_cuffCompare("gffcompare_2025/Combined_all_no_pirna_forSQ_refseq.stats")
genbank_gffcompare <- parse_cuffCompare("gffcompare_2025/Combined_all_no_pirna_forSQ_genbank.stats")

gffcompare_sensitivity <- merge(refseq_gffcompare$prec_sens[,1:2],
                    genbank_gffcompare$prec_sens[,1:2],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Sensitivity",sep="_"))) 

gffcompare_precision <- merge(refseq_gffcompare$prec_sens[,c(1,3)],
                    genbank_gffcompare$prec_sens[,c(1,3)],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Precision",sep="_"))) 

gffcompare_ns<- cbind(t(refseq_gffcompare$Query_mrna),
                    t(refseq_gffcompare$Reference_mrna),t(genbank_gffcompare$Reference_mrna)) %>% 
 set_colnames(c("RU","RefSeq","GenBank")) %>% as.data.frame() %>% .[c(2,1,3),] %>% rownames_to_column()

gffcompare_matching<- cbind(t(refseq_gffcompare$matching),
                    t(genbank_gffcompare$matching)) %>% 
 set_colnames(c("RefSeq","GenBank")) %>% as.data.frame() %>% .[c(3,2,1),] %>% rownames_to_column()

gffcompare_NovelAndMissing <- cbind(refseq_gffcompare$Novel_missing %>% 
                   set_colnames(paste0(colnames(.),"_against_RefSeq")),
                  genbank_gffcompare$Novel_missing %>%
                   set_colnames(paste0(colnames(.),"_against_Genbank"))) %>% 
 as.data.frame() %>% rownames_to_column()

gffcompare_stats_unfiltered <- list(gffcompare_Summary_numbers=gffcompare_ns,
             gffcompare_NovelAndMissing=gffcompare_NovelAndMissing,
             gffcompare_matching=gffcompare_matching,
             gffcompare_sensitivity=gffcompare_sensitivity,
             gffcompare_precision=gffcompare_precision)
             
             
 

```

# Parse QC and make XLSx report for filtered

```{r}
parse_cuffCompare <- function(stats_file){
lines <- readLines(stats_file)
prec_sens <- lines[grepl("level",lines)] %>% 
 str_trim() %>% 
 gsub("Intron chain","Intron_chain",.) %>% 
 gsub(" level:","_level",.) %>% 
 gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V3,-V5) %>% set_colnames(c("level","Sensitivity","Precision"))

Query_mrna <- lines[grepl("Query mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4,-V5) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))
 
Reference_mrna <- lines[grepl("Reference mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=4,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))

matching <- lines[grepl("Matching",lines)] %>% str_trim() %>%
   gsub(".*:","",.) %>% str_trim() %>% as.numeric %>% as.data.frame() %>% t %>% 
 set_colnames(c("Matching_intron_chains","Matching_transcripts","Matching_loci"))

myRownames <- lines[grepl("Missed|Novel",lines)] %>% 
 str_trim() %>% 
 gsub(":.*","",.) %>% gsub(" ","_",.)
Novel_missing <- lines[grepl("Missed|Novel",lines)] %>% str_trim() %>% 
 gsub(".*:","",.) %>% str_trim() %>% 
 gsub("\t.*","",.) %>% str_trim() %>% str_split("/") %>% unlist %>% matrix(ncol=2,byrow = TRUE) %>% 
 as.data.frame() %>% set_colnames(c("n_of","Total")) %>% set_rownames(myRownames)

toReturn <- list(prec_sens,Query_mrna,Reference_mrna,matching,Novel_missing)
names(toReturn) <- c("prec_sens","Query_mrna","Reference_mrna","matching","Novel_missing")
toReturn
}

refseq_gffcompare <- parse_cuffCompare("gffcompare_2025/Combined_all_no_pirna_filtered_forSQ_refseq.stats")
genbank_gffcompare <- parse_cuffCompare("gffcompare_2025/Combined_all_no_pirna_filtered_forSQ_genbank.stats")

gffcompare_sensitivity <- merge(refseq_gffcompare$prec_sens[,1:2],
                    genbank_gffcompare$prec_sens[,1:2],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Sensitivity",sep="_"))) 

gffcompare_precision <- merge(refseq_gffcompare$prec_sens[,c(1,3)],
                    genbank_gffcompare$prec_sens[,c(1,3)],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Precision",sep="_"))) 

gffcompare_ns<- cbind(t(refseq_gffcompare$Query_mrna),
                    t(refseq_gffcompare$Reference_mrna),t(genbank_gffcompare$Reference_mrna)) %>% 
 set_colnames(c("RU","RefSeq","GenBank")) %>% as.data.frame() %>% .[c(2,1,3),] %>% rownames_to_column()

gffcompare_matching<- cbind(t(refseq_gffcompare$matching),
                    t(genbank_gffcompare$matching)) %>% 
 set_colnames(c("RefSeq","GenBank")) %>% as.data.frame() %>% .[c(3,2,1),] %>% rownames_to_column()

gffcompare_NovelAndMissing <- cbind(refseq_gffcompare$Novel_missing %>% 
                   set_colnames(paste0(colnames(.),"_against_RefSeq")),
                  genbank_gffcompare$Novel_missing %>%
                   set_colnames(paste0(colnames(.),"_against_Genbank"))) %>% 
 as.data.frame() %>% rownames_to_column()

gffcompare_stats_filtered <- list(gffcompare_Summary_numbers=gffcompare_ns,
             gffcompare_NovelAndMissing=gffcompare_NovelAndMissing,
             gffcompare_matching=gffcompare_matching,
             gffcompare_sensitivity=gffcompare_sensitivity,
             gffcompare_precision=gffcompare_precision)
             
             
 

```

```{r}
parse_squantiQC <- function(QCfile){
 temp <- read.delim(paste0(QCfile,"_classification.txt"))
 broadQC <- temp %>% group_by(structural_category) %>% summarise(Total=n())
 subQC <- temp %>% group_by(structural_category,subcategory) %>% summarise(Total=n()) %>% as.data.frame()
 coding <- temp %>% group_by(coding) %>% summarise(Total=n())
 novel_coding <- temp %>% filter(associated_transcript == "novel") %>% group_by(coding) %>% summarise(Total=n())
 transLevel <- list(broadQC,subQC,coding,novel_coding)
 
 temp <- read.delim(paste0(QCfile,"_junctions.txt"))
 subQC <- temp %>% group_by(junction_category,canonical) %>% summarise(Total=n()) %>% as.data.frame()
 subQC2 <- temp %>% group_by(junction_category,canonical,RTS_junction) %>% summarise(Total=n()) %>% as.data.frame()

 juncLevel <- list(subQC,subQC2) 
 list(transLevel=transLevel,juncLevel=juncLevel)
}
refseq_squanti <- parse_squantiQC("Squanti_2025/sq_combined_all_no_pirna_vs_refseq/combined_all_no_pirna_vs_refseq.gtf")
genbank_squanti <- parse_squantiQC("Squanti_2025/sq_combined_all_no_pirna_vs_genbank//combined_all_no_pirna_vs_genbank.gtf")

squanti_transcript_categories <- merge(refseq_squanti$transLevel[[1]],
                    genbank_squanti$transLevel[[1]],by=1,all=TRUE) %>% 
 set_colnames(c("structural_category",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) #%>% mutate(across(1:3,function(x){x[is.na(x)] <- 0}))
squanti_transcript_categories[is.na(squanti_transcript_categories)] <- 0

squanti_subtranscript_categories <- merge(refseq_squanti$transLevel[[2]],
                    genbank_squanti$transLevel[[2]],by=c(1,2),all=TRUE) %>% 
 set_colnames(c("structural_category","sub_category",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) #%>% mutate(across(1:3,function(x){x[is.na(x)] <- 0}))
squanti_subtranscript_categories[is.na(squanti_subtranscript_categories)] <- 0

squanti_coding_categories <- merge(refseq_squanti$transLevel[[3]],
                    genbank_squanti$transLevel[[3]],by=c(1),all=TRUE) %>% 
 set_colnames(c("Coding",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) %>% mutate(Class="Total") %>% 
 select(Coding,Class,everything()) 
squanti_novelcoding_categories <- merge(refseq_squanti$transLevel[[4]],
                    genbank_squanti$transLevel[[4]],by=c(1),all=TRUE) %>% 
 set_colnames(c("Coding",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) %>% mutate(Class="Novel") %>% 
 select(Coding,Class,everything()) 

squanti_coding_categories <- rbind(squanti_coding_categories,
                  squanti_novelcoding_categories)

squanti_junction_categories <- merge(refseq_squanti$juncLevel[[1]],
                    genbank_squanti$juncLevel[[1]],by=c(1,2),all=TRUE) %>% 
 set_colnames(c("Junction_category","Canonical",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) 

squanti_stats_unfiltered <- list(squanti_transcript_categories=squanti_transcript_categories,
           squanti_subtranscript_categories=squanti_subtranscript_categories,
           squanti_coding_categories=squanti_coding_categories,
           squanti_junction_categories=squanti_junction_categories)
           
           
```

```{r}
parse_squantiQC <- function(QCfile){
 temp <- read.delim(paste0(QCfile,"_classification.txt"))
 broadQC <- temp %>% group_by(structural_category) %>% summarise(Total=n())
 subQC <- temp %>% group_by(structural_category,subcategory) %>% summarise(Total=n()) %>% as.data.frame()
 coding <- temp %>% group_by(coding) %>% summarise(Total=n())
 novel_coding <- temp %>% filter(associated_transcript == "novel") %>% group_by(coding) %>% summarise(Total=n())
 transLevel <- list(broadQC,subQC,coding,novel_coding)
 
 temp <- read.delim(paste0(QCfile,"_junctions.txt"))
 subQC <- temp %>% group_by(junction_category,canonical) %>% summarise(Total=n()) %>% as.data.frame()
 subQC2 <- temp %>% group_by(junction_category,canonical,RTS_junction) %>% summarise(Total=n()) %>% as.data.frame()

 juncLevel <- list(subQC,subQC2) 
 list(transLevel=transLevel,juncLevel=juncLevel)
}
refseq_squanti <- parse_squantiQC("Squanti_2025/sq_combined_all_no_pirna_vs_refseq_filtered//combined_all_no_pirna_filtered_vs_refseq.gtf")
genbank_squanti <- parse_squantiQC("Squanti_2025/sq_combined_all_no_pirna_vs_genbank_filtered///combined_all_no_pirna_filtered_vs_genbank.gtf")

squanti_transcript_categories <- merge(refseq_squanti$transLevel[[1]],
                    genbank_squanti$transLevel[[1]],by=1,all=TRUE) %>% 
 set_colnames(c("structural_category",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) #%>% mutate(across(1:3,function(x){x[is.na(x)] <- 0}))
squanti_transcript_categories[is.na(squanti_transcript_categories)] <- 0

squanti_subtranscript_categories <- merge(refseq_squanti$transLevel[[2]],
                    genbank_squanti$transLevel[[2]],by=c(1,2),all=TRUE) %>% 
 set_colnames(c("structural_category","sub_category",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) #%>% mutate(across(1:3,function(x){x[is.na(x)] <- 0}))
squanti_subtranscript_categories[is.na(squanti_subtranscript_categories)] <- 0

squanti_coding_categories <- merge(refseq_squanti$transLevel[[3]],
                    genbank_squanti$transLevel[[3]],by=c(1),all=TRUE) %>% 
 set_colnames(c("Coding",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) %>% mutate(Class="Total") %>% 
 select(Coding,Class,everything()) 
squanti_novelcoding_categories <- merge(refseq_squanti$transLevel[[4]],
                    genbank_squanti$transLevel[[4]],by=c(1),all=TRUE) %>% 
 set_colnames(c("Coding",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) %>% mutate(Class="Novel") %>% 
 select(Coding,Class,everything()) 

squanti_coding_categories <- rbind(squanti_coding_categories,
                  squanti_novelcoding_categories)

squanti_junction_categories <- merge(refseq_squanti$juncLevel[[1]],
                    genbank_squanti$juncLevel[[1]],by=c(1,2),all=TRUE) %>% 
 set_colnames(c("Junction_category","Canonical",
         "RU_vs_RefSeq",
         "RU_vs_Genbank")) 

squanti_stats_filtered <- list(squanti_transcript_categories=squanti_transcript_categories,
           squanti_subtranscript_categories=squanti_subtranscript_categories,
           squanti_coding_categories=squanti_coding_categories,
           squanti_junction_categories=squanti_junction_categories)
           
           
```

```{r}
parse_busco <- function(busfile){
 temp <- jsonlite::read_json(busfile)
 result <- data.frame(Total_BUSCO_Markers=temp$lineage_dataset$number_of_buscos %>% as.numeric(),
       Percent_BUSCO_Markers_Complete=temp$results$Complete %>% as.numeric(),
       Percent_BUSCO_Markers_Fragmented=temp$results$Fragmented %>% as.numeric(),
       Percent_BUSCO_Markers_Missing=temp$results$Missing %>% as.numeric()
       )
 result[1,] <- result[1,,drop=FALSE]
}

# genbank_squanti <- parse_busco("busco/Genbank_busco/short_summary.specific.arachnida_odb10.Genbank_busco.json")
# refseq_squanti <- parse_busco("busco/Refseq_busco/short_summary.specific.arachnida_odb10.Refseq_busco.json")
# ru_squanti <- parse_busco("busco/combined_busco/short_summary.specific.arachnida_odb10.busco_corr.json")

genbank_squanti <- parse_busco("busco/Genbank_busco_insect//short_summary.specific.insecta_odb10.Genbank_busco_insect.json")
refseq_squanti <- parse_busco("busco/Refseq_busco_insect//short_summary.specific.insecta_odb10.Refseq_busco_insect.json")
ru_unfiltered_squanti <- parse_busco("Busco_2025/busco_combined_all_no_pirna/short_summary.specific.insecta_odb10.busco_combined_all_no_pirna.json")
ru_filtered_squanti <- parse_busco("Busco_2025/busco_combined_all_no_pirna_filtered//short_summary.specific.insecta_odb10.busco_combined_all_no_pirna_filtered.json")

busco_results <- cbind(t(ru_unfiltered_squanti),t(ru_filtered_squanti),t(refseq_squanti),t(genbank_squanti)) %>% set_colnames(c("RU","RU_filtered","RefSeq","GenBank")) %>% as.data.frame() %>% rownames_to_column()
busco_stats <- list(busco_results=busco_results)
busco_stats_unfiltered <- list(busco_results=busco_results[,-3])
busco_stats_filtered <- list(busco_results=busco_results[,-2] %>% set_colnames(c("rowname","RU","RefSeq","GenBank")))
```

```{r}

Combined_all_no_pirna <- Combined_all2[!Combined_all2$transcript_id %in% piRNAs]
Combined_all_no_pirna_filtered <- Combined_all2[!Combined_all2$transcript_id %in% piRNAs & 
                                         !Combined_all2$transcript_id %in% sq_filter]

Combined_all <- Combined_all2
Combined_all_filtered <- Combined_all2[!Combined_all2$transcript_id %in% sq_filter]


unfiltered_transcripts <- Combined_all[Combined_all$type == "transcript"]$transcript_type %>% table %>% as.data.frame() %>% set_colnames(c("Transcript_type","N_Of_transcripts"))
filtered_transcripts <- Combined_all_filtered[Combined_all_filtered$type == "transcript"]$transcript_type %>% table %>% as.data.frame() %>% set_colnames(c("Transcript_type","N_Of_transcripts"))

unfiltered_gene_count <- list()
for(i in 1:length(unfiltered_transcripts$Transcript_type)){
  unfiltered_gene_count[[i]] <- Combined_all[Combined_all$type == "transcript" & Combined_all$transcript_type == unfiltered_transcripts$Transcript_type[i]]$gene_id %>% unique %>% length
}
names(unfiltered_gene_count) <- unfiltered_transcripts$Transcript_type
unfiltered_gene_count <- unfiltered_gene_count %>% unlist %>% as.data.frame %>% rownames_to_column %>% set_colnames(c("Gene_type","N_Of_genes"))

filtered_gene_count <- list()
for(i in 1:length(filtered_transcripts$Transcript_type)){
  filtered_gene_count[[i]] <- Combined_all_filtered[Combined_all_filtered$type == "transcript" & Combined_all_filtered$transcript_type == filtered_transcripts$Transcript_type[i]]$gene_id %>% unique %>% length
}
names(filtered_gene_count) <- filtered_transcripts$Transcript_type
filtered_gene_count <- filtered_gene_count %>% unlist %>% as.data.frame %>% rownames_to_column %>% set_colnames(c("Gene_type","N_Of_genes"))

filtered_transcript_types <- list(transcript_types=filtered_transcripts)
unfiltered_transcript_types <- list(transcript_types=unfiltered_transcripts)
```



```{r,eval=FALSE}
Combined_all_filtered[!Combined_all_filtered$transcript_type %in% "piRNA"]$gene_id %>% unique() %>% length()
Combined_all[!Combined_all$transcript_type %in% "piRNA"]$gene_id %>% unique() %>% length()
filtered_gene_count %>% filter(Gene_type != "piRNA") %>% summarise(sum(N_Of_genes))
unfiltered_gene_count %>% filter(Gene_type != "piRNA") %>% summarise(sum(N_Of_genes))

filtered_gene_count %>% filter(Gene_type != "piRNA") %>% summarise(sum(N_Of_genes))
unfiltered_gene_count %>% filter(Gene_type != "piRNA") %>% summarise(sum(N_Of_genes))

filtered_transcripts %>% filter(Transcript_type != "piRNA") %>% summarise(sum(N_Of_transcripts))
unfiltered_transcripts %>% filter(Transcript_type != "piRNA") %>% summarise(sum(N_Of_transcripts))



tempTest <- makeTxDbFromGRanges(Combined_all[!Combined_all$transcript_type %in% "piRNA"])
tempTest_Genes <- genes(tempTest)
tempTest_ras <- transcripts(tempTest)
(gsub("\\.\\d+$","",tempTest_ras$tx_name)) %>% unique %>% length
tempTest_ras <- transcripts(tempTest,columns=c("tx_name","gene_id"))
(gsub("\\.\\d+$","",tempTest_ras$tx_name) == as.character(tempTest_ras$gene_id)) %>% all

Combined_all %>% as.data.frame %>% filter(type=="transcript") %>% 
  filter(!transcript_type %in% "piRNA") %>% 
  group_by(transcript_type) %>% 
  summarise(N_of_Gene=length(unique(gene_id))) %>% ungroup  %>% 
  summarise(TotalGene=sum(N_of_Gene))

Combined_all %>% as.data.frame %>% filter(type=="transcript") %>% 
  filter(!transcript_type %in% "piRNA") %>% summarise(TotalGene=length(unique(gene_id)))


temme <- Combined_all %>% as.data.frame %>% filter(type=="transcript") %>% 
  filter(!transcript_type %in% "piRNA")

(temme$gene_id == gsub("\\.\\d+$","",temme$transcript_id)) %>% all

Combined_all %>% as.data.frame %>% filter(type=="transcript") %>% 
  filter(!transcript_type %in% "piRNA") %>% select(transcript_type,gene_id) %>% distinct() %>% nrow

Combined_all %>% as.data.frame %>% filter(type=="transcript") %>% 
  filter(!transcript_type %in% "piRNA") %>% select(transcript_type,gene_id) %>% distinct(gene_id)  %>% nrow

jsjs <- Combined_all %>% as.data.frame %>% filter(type=="transcript") %>% 
  filter(!transcript_type %in% "piRNA") %>% select(transcript_type,gene_id) %>% distinct() %>% pull() 

Combined_all %>% as.data.frame %>% filter(type=="transcript") %>% 
  filter(transcript_type %in% "tRNA")

sllss <- jsjs[duplicated(jsjs)]
Combined_all[Combined_all$gene_id %in% sllss]$transcript_type %>% table

Combined_all[Combined_all$gene_id %in% sllss & Combined_all$transcript_type %in% "tRNA",]
Combined_all[Combined_all$gene_id %in% sllss & Combined_all$transcript_type %in% "rRNA",]
Combined_all[Combined_all$gene_id %in% sllss & Combined_all$transcript_type %in% "snRNA",]


Combined_all[Combined_all$gene_id %in% "RU.9522",]
RU.9522

temp <- read.delim("gffcompare_2025/Combined_all_no_pirna_forSQ_refseq.tracking",header = FALSE)
tempA <- temp$V5 %>% str_split("\\|") %>% unlist %>% matrix(ncol=7,byrow=TRUE)
tempB <- tempA[,1] %>% str_split(":") %>% unlist %>% matrix(ncol=2,byrow=TRUE)
temp <- cbind(temp[,1:4],tempB,tempA[,-1]) %>% as.data.frame()
temp$V2 %>% unique %>% length
temp[,6] %>% unique %>% length

tempA <- temp %>% select(2,6) %>% distinct %>% 
  group_by(V2) %>% 
  summarise(n())
tempA[tempA$`n()` > 1,]
temp[temp$V2 %in% "XLOC_000142",]
temp[temp$V2 %in% "XLOC_000176",]
temp[temp$V2 %in% "XLOC_000324",]


tempC <- temp %>% select(2,6) %>% distinct %>% set_colnames(c("V2","V3")) %>% 
  group_by(V3) %>% 
  summarise(n())

```

```{r}
Combined_all2DF <- Combined_all2 %>% as.data.frame()
maxID <- gsub("RU\\.","",Combined_all2DF$gene_id) %>% as.numeric() %>% max
Combined_all2DF[Combined_all2DF$transcript_id %in% "RU.4951.2",]$gene_id <- paste0("RU.",maxID+1)
newTransID <-  paste0(Combined_all2DF[Combined_all2DF$transcript_id %in% "RU.4951.2",]$gene_id,".1")
Combined_all2DF[Combined_all2DF$transcript_id %in% "RU.4951.2",]$transcript_id <- newTransID

sq_filter <- c(unique(newTransID),sq_filter)

allmRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "mRNA",]$gene_id
allrRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "rRNA",]$gene_id
allsnRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "snRNA",]$gene_id
allsnoRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "snoRNA",]$gene_id
alltRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "tRNA",]$gene_id
alllncRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "lncRNA",]$gene_id

rRNAandmRNAGenes <- allrRNA_Genes[allrRNA_Genes %in% c(allmRNA_Genes,alllncRNA_Genes)]
snRNAandmRNAGenes <- allsnRNA_Genes[allsnRNA_Genes %in% c(allmRNA_Genes,alllncRNA_Genes)]
snoRNAandmRNAGenes <- allsnoRNA_Genes[allsnoRNA_Genes %in% c(allmRNA_Genes,alllncRNA_Genes)]
tRNAandmRNAGenes <- alltRNA_Genes[alltRNA_Genes %in% c(allmRNA_Genes,alllncRNA_Genes)]
lncRNAandmRNAGenes <- alllncRNA_Genes[alllncRNA_Genes %in% allmRNA_Genes]
lncRNAandNotmRNAGenes <- alllncRNA_Genes[!alllncRNA_Genes %in% allmRNA_Genes]

# toTstFilters <- c("RU.4951.2",rRNAandmRNAGenes,snRNAandmRNAGenes,snoRNAandmRNAGenes,tRNAandmRNAGenes,lncRNAandmRNAGenes,lncRNAandNotmRNAGenes)
# 
# toTstFilters[toTstFilters %in% sq_filter]

maxID <- gsub("RU\\.","",Combined_all2DF$gene_id) %>% as.numeric() %>% max

newIDs <- paste0("RU.",seq(maxID+1,maxID+length(rRNAandmRNAGenes)))
tempO <- cbind(Combined_all2DF[Combined_all2DF$gene_id %in% rRNAandmRNAGenes & 
                  Combined_all2DF$transcript_type %in% "rRNA",]$transcript_id,newIDs) %>% 
  as.data.frame()
tempO <- cbind(tempO,paste0(tempO$newIDs,".",1))
# Combined_all2DF[match(tempO$V1,Combined_all2DF$transcript_id,nomatch = 0),]
# tempO$`paste0(tempO$newIDs, ".", 1)`[match(Combined_all2DF$transcript_id,tempO$V1,nomatch = 0)]
for(i in 1:nrow(tempO)){
  indexOfI <- Combined_all2DF$transcript_id %in% tempO[i,1]
  Combined_all2DF[indexOfI,]$transcript_id <- tempO[i,3]
  Combined_all2DF[indexOfI,]$gene_id <- tempO[i,2]  
}


maxID <- gsub("RU\\.","",Combined_all2DF$gene_id) %>% as.numeric() %>% max



newIDs <- paste0("RU.",seq(maxID+1,maxID+length(snRNAandmRNAGenes)))
tempO <- cbind(Combined_all2DF[Combined_all2DF$gene_id %in% snRNAandmRNAGenes & 
                  Combined_all2DF$transcript_type %in% "snRNA",]$transcript_id,newIDs) %>% 
  as.data.frame()
tempO <- cbind(tempO,paste0(tempO$newIDs,".",1))
for(i in 1:nrow(tempO)){
  indexOfI <- Combined_all2DF$transcript_id %in% tempO[i,1]
  Combined_all2DF[indexOfI,]$transcript_id <- tempO[i,3]
  Combined_all2DF[indexOfI,]$gene_id <- tempO[i,2]  
}



maxID <- gsub("RU\\.","",Combined_all2DF$gene_id) %>% as.numeric() %>% max

newIDs <- paste0("RU.",seq(maxID+1,maxID+length(snoRNAandmRNAGenes)))
tempO <- cbind(Combined_all2DF[Combined_all2DF$gene_id %in% snoRNAandmRNAGenes & 
                  Combined_all2DF$transcript_type %in% "snoRNA",]$transcript_id,newIDs) %>% 
  as.data.frame()
tempO <- cbind(tempO,paste0(tempO$newIDs,".",1))
for(i in 1:nrow(tempO)){
  indexOfI <- Combined_all2DF$transcript_id %in% tempO[i,1]
  Combined_all2DF[indexOfI,]$transcript_id <- tempO[i,3]
  Combined_all2DF[indexOfI,]$gene_id <- tempO[i,2]  
}


maxID <- gsub("RU\\.","",Combined_all2DF$gene_id) %>% as.numeric() %>% max

newIDs <- paste0("RU.",seq(maxID+1,maxID+length(tRNAandmRNAGenes)))
tempO <- cbind(Combined_all2DF[Combined_all2DF$gene_id %in% tRNAandmRNAGenes & 
                  Combined_all2DF$transcript_type %in% "tRNA",]$transcript_id,newIDs) %>% 
  as.data.frame()
tempO <- cbind(tempO,paste0(tempO$newIDs,".",1))
for(i in 1:nrow(tempO)){
  indexOfI <- Combined_all2DF$transcript_id %in% tempO[i,1]
  Combined_all2DF[indexOfI,]$transcript_id <- tempO[i,3]
  Combined_all2DF[indexOfI,]$gene_id <- tempO[i,2]  
}


# allmRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "mRNA",]$gene_id
# allrRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "rRNA",]$gene_id
# allsnRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "snRNA",]$gene_id
# allsnoRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "snoRNA",]$gene_id
# alltRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "tRNA",]$gene_id
# alllncRNA_Genes <- Combined_all2DF[Combined_all$transcript_type %in% "lncRNA",]$gene_id
# 
# rRNAandmRNAGenes <- allrRNA_Genes[allrRNA_Genes %in% c(allmRNA_Genes,alllncRNA_Genes)]
# snRNAandmRNAGenes <- allsnRNA_Genes[allsnRNA_Genes %in% c(allmRNA_Genes,alllncRNA_Genes)]
# snoRNAandmRNAGenes <- allsnoRNA_Genes[allsnoRNA_Genes %in% c(allmRNA_Genes,alllncRNA_Genes)]
# tRNAandmRNAGenes <- alltRNA_Genes[alltRNA_Genes %in% c(allmRNA_Genes,alllncRNA_Genes)]
# lncRNAandmRNAGenes <- alllncRNA_Genes[alllncRNA_Genes %in% allmRNA_Genes]
# lncRNAandNotmRNAGenes <- alllncRNA_Genes[!alllncRNA_Genes %in% allmRNA_Genes]

Combined_all3 <- Combined_all2DF %>% GRanges()
Combined_all3$gene_type <- Combined_all3$transcript_type
allCodingAndNonCoding <- Combined_all3 %>% as.data.frame() %>% select(gene_id,gene_type) %>% 
  filter(gene_type %in% c("lncRNA","mRNA")) %>% distinct %>% group_by(gene_id) %>% 
  summarise(n())

proteincoding <- allCodingAndNonCoding %>% filter(`n()`>1) %>% pull(gene_id)
Combined_all3[Combined_all3$gene_type %in% "mRNA"]$gene_type <- "protein_coding"
Combined_all3[Combined_all3$gene_id %in% proteincoding]$gene_type <- "protein_coding"

Combined_all3 %>% as.data.frame() %>% select(gene_id,gene_type) %>% 
  filter(gene_type %in% c("lncRNA","protein_coding")) %>% distinct %>% group_by(gene_id) %>% 
  summarise(n()) %>% filter(`n()`>1) 
```

```{r}

Combined_all_no_pirna <- Combined_all3[!Combined_all3$transcript_id %in% piRNAs]
Combined_all_no_pirna_filtered <- Combined_all3[!Combined_all3$transcript_id %in% piRNAs & 
                                         !Combined_all3$transcript_id %in% sq_filter]

Combined_all <- Combined_all3
Combined_all_filtered <- Combined_all3[!Combined_all3$transcript_id %in% sq_filter]


unfiltered_transcripts <- Combined_all[Combined_all$type == "transcript"]$transcript_type %>% table %>% as.data.frame() %>% set_colnames(c("Transcript_type","N_Of_transcripts"))
filtered_transcripts <- Combined_all_filtered[Combined_all_filtered$type == "transcript"]$transcript_type %>% table %>% as.data.frame() %>% set_colnames(c("Transcript_type","N_Of_transcripts"))

gene_type <- Combined_all$gene_type %>% na.omit %>% unique
unfiltered_gene_count <- list()
for(i in 1:length(gene_type)){
  unfiltered_gene_count[[i]] <- Combined_all[Combined_all$type == "transcript" & Combined_all$gene_type == gene_type[i]]$gene_id %>% unique %>% length
}
names(unfiltered_gene_count) <- gene_type
unfiltered_gene_count <- unfiltered_gene_count %>% unlist %>% as.data.frame %>% rownames_to_column %>% set_colnames(c("Gene_type","N_Of_genes"))

filtered_gene_count <- list()
for(i in 1:length(gene_type)){
  filtered_gene_count[[i]] <- Combined_all_filtered[Combined_all_filtered$type == "transcript" & Combined_all_filtered$gene_type == gene_type[i]]$gene_id %>% unique %>% length
}
names(filtered_gene_count) <- gene_type
filtered_gene_count <- filtered_gene_count %>% unlist %>% as.data.frame %>% rownames_to_column %>% set_colnames(c("Gene_type","N_Of_genes"))

filtered_transcript_types <- list(transcript_types=filtered_transcripts)
unfiltered_transcript_types <- list(transcript_types=unfiltered_transcripts)
filtered_gene_types <- list(gene_types=filtered_gene_count)
unfiltered_gene_types <- list(gene_types=unfiltered_gene_count)
```

```{r,eval=FALSE}
Combined_all_filtered[!Combined_all_filtered$transcript_type %in% "piRNA"]$gene_id %>% unique() %>% length()
Combined_all[!Combined_all$transcript_type %in% "piRNA"]$gene_id %>% unique() %>% length()
filtered_gene_count %>% filter(Gene_type != "piRNA") %>% summarise(sum(N_Of_genes))
unfiltered_gene_count %>% filter(Gene_type != "piRNA") %>% summarise(sum(N_Of_genes))

filtered_gene_count %>% filter(Gene_type != "piRNA") %>% summarise(sum(N_Of_genes))
unfiltered_gene_count %>% filter(Gene_type != "piRNA") %>% summarise(sum(N_Of_genes))

filtered_transcripts %>% filter(Transcript_type != "piRNA") %>% summarise(sum(N_Of_transcripts))
unfiltered_transcripts %>% filter(Transcript_type != "piRNA") %>% summarise(sum(N_Of_transcripts))

```

```{r}
c(filtered_gene_types,filtered_transcript_types,gffcompare_stats_filtered,squanti_stats_filtered,busco_stats_filtered) %>% rio::export(file="2025_GeneModel_stats_RUvsRefseqAndGenbank_filtered.xlsx")
c(unfiltered_gene_types,unfiltered_transcript_types,gffcompare_stats_unfiltered,squanti_stats_unfiltered,busco_stats_unfiltered) %>% rio::export(file="2025_GeneModel_stats_RUvsRefseqAndGenbank_full.xlsx")
```


```{r}


piRNAs <- Combined_all[Combined_all$type == "transcript" & Combined_all$transcript_type == "piRNA"]$transcript_id
toCount <- Combined_all[Combined_all$type == "transcript" & Combined_all$gene_type %in% c("lncRNA","protein_coding")]$gene_id


Combined_all_no_pirna <- Combined_all[!Combined_all$transcript_id %in% piRNAs]
Combined_all_no_pirna_filtered <- Combined_all[!Combined_all$transcript_id %in% piRNAs & 
                                         !Combined_all$transcript_id %in% sq_filter]

Combined_all_toCount <- Combined_all[Combined_all$gene_id %in% toCount]

Combined_all_toCount_filtered <- Combined_all[Combined_all$gene_id %in% toCount & 
                                         !Combined_all$transcript_id %in% sq_filter]

# Combined_all_toCount_filtered$gene_id %>% unique %>% length
# Combined_all_toCount$gene_id %>% unique %>% length
# filtered_gene_count %>% filter(Gene_type %in% c("lncRNA","protein_coding")) %>% summarise(sum(N_Of_genes))
# unfiltered_gene_count %>% filter(Gene_type %in% c("lncRNA","protein_coding")) %>% summarise(sum(N_Of_genes))

export(Combined_all2,
       con = "Combined_2025.gtf")
export(Combined_all_no_pirna,
       con = "Combined_noPiRNA_2025.gtf")
export(Combined_all_no_pirna_filtered,
       con = "Combined_noPiRNA_filtered_2025.gtf")
export(Combined_all_toCount,
       con = "Combined_forDifferentialExpression_2025.gtf")
export(Combined_all_toCount_filtered,
       con = "Combined_forDifferentialExpression_filtered_2025.gtf")
```

```{r}
# GenBank_GTF <- GenBank_GTF[seqnames(GenBank_GTF) %in% names(genomeSeq),]
# RefSeq_GTF <- RefSeq_GTF[seqnames(RefSeq_GTF) %in% names(genomeSeq),]

Ref_transTable <- RefSeq_GTF %>% as.data.frame %>% filter(type=="transcript") %>% pull(transcript_biotype) %>% table
Genbank_transTable <- GenBank_GTF %>% as.data.frame %>% filter(type=="transcript") %>% pull(transcript_biotype) %>% table
Combined_transTable <- Combined_all %>% as.data.frame %>% filter(type=="transcript") %>% pull(transcript_type) %>% table
names(Ref_transTable)[2] <- "lncRNA"

Combined_transTable <- Combined_transTable %>% as.data.frame %>% set_colnames(c("Var1","Freq"))
Ref_transTable <- Ref_transTable %>% as.data.frame
Genbank_transTable <- Genbank_transTable %>% as.data.frame  %>% set_colnames(c("Var1","Freq"))


gene_type <- Combined_all$gene_type %>% na.omit %>% unique
unfiltered_gene_count <- list()
for(i in 1:length(gene_type)){
  unfiltered_gene_count[[i]] <- Combined_all[Combined_all$type == "transcript" & Combined_all$gene_type == gene_type[i]]$gene_id %>% unique %>% length
}
names(unfiltered_gene_count) <- gene_type
Combined_geneTable <- unfiltered_gene_count %>% unlist %>% as.data.frame %>% rownames_to_column %>% set_colnames(c("Gene_type","N_Of_genes"))

gene_type <- RefSeq_GTF$gene_biotype %>% na.omit %>% unique
unfiltered_gene_count <- list()
for(i in 1:length(gene_type)){
  unfiltered_gene_count[[i]] <- RefSeq_GTF[RefSeq_GTF$type == "gene" & RefSeq_GTF$gene_biotype == gene_type[i]]$gene_id %>% unique %>% length
}
names(unfiltered_gene_count) <- gene_type
RefSeq_geneTable <- unfiltered_gene_count %>% unlist %>% as.data.frame %>% rownames_to_column %>% set_colnames(c("Gene_type","N_Of_genes"))

gene_type <- GenBank_GTF$gene_biotype %>% na.omit %>% unique
unfiltered_gene_count <- list()
for(i in 1:length(gene_type)){
  unfiltered_gene_count[[i]] <- GenBank_GTF[GenBank_GTF$type == "gene" & GenBank_GTF$gene_biotype == gene_type[i]]$gene_id %>% unique %>% length
}
names(unfiltered_gene_count) <- gene_type
GenBank_geneTable <- unfiltered_gene_count %>% unlist %>% as.data.frame %>% rownames_to_column %>% set_colnames(c("Gene_type","N_Of_genes"))



total_transtable <- merge(Combined_transTable,Ref_transTable,by=1,all=TRUE) %>% 
  merge(Genbank_transTable,by=1,all=TRUE) %>% 
  set_colnames(c("Transcript_type","RU","RefSeq","GenBank"))

total_genetable <- merge(Combined_geneTable,RefSeq_geneTable,by=1,all=TRUE) %>% 
  merge(GenBank_geneTable,by=1,all=TRUE) %>% 
  set_colnames(c("Gene_type","RU","RefSeq","GenBank"))

total_genetable[is.na(total_genetable)] <- 0
total_transtable[is.na(total_transtable)] <- 0

list(Gene_Biotypes=total_genetable,
     Transcript_Biotypes=total_transtable) %>% rio::export(file="2025_BioTypes_stats_RUvsRefseqAndGenbank_full.xlsx")

full_biotypes <- list(Gene_Biotypes=total_genetable,
     Transcript_Biotypes=total_transtable)

Combined_all_filtered <- Combined_all[!Combined_all$transcript_id %in% sq_filter]

Combined_transTable <- Combined_all_filtered %>% as.data.frame %>% filter(type=="transcript") %>% pull(transcript_type) %>% table

Combined_transTable <- Combined_transTable %>% as.data.frame %>% set_colnames(c("Var1","Freq"))


gene_type <- Combined_all_filtered$gene_type %>% na.omit %>% unique
unfiltered_gene_count <- list()
for(i in 1:length(gene_type)){
  unfiltered_gene_count[[i]] <- Combined_all_filtered[Combined_all_filtered$type == "transcript" & Combined_all_filtered$gene_type == gene_type[i]]$gene_id %>% unique %>% length
}
names(unfiltered_gene_count) <- gene_type
Combined_geneTable <- unfiltered_gene_count %>% unlist %>% as.data.frame %>% rownames_to_column %>% set_colnames(c("Gene_type","N_Of_genes"))



total_transtable <- merge(Combined_transTable,Ref_transTable,by=1,all=TRUE) %>% 
  merge(Genbank_transTable,by=1,all=TRUE) %>% 
  set_colnames(c("Transcript_type","RU","RefSeq","GenBank"))

total_genetable <- merge(Combined_geneTable,RefSeq_geneTable,by=1,all=TRUE) %>% 
  merge(GenBank_geneTable,by=1,all=TRUE) %>% 
  set_colnames(c("Gene_type","RU","RefSeq","GenBank"))

total_genetable[is.na(total_genetable)] <- 0
total_transtable[is.na(total_transtable)] <- 0

list(Gene_Biotypes=total_genetable,
     Transcript_Biotypes=total_transtable) %>% rio::export(file="2025_BioTypes_stats_RUvsRefseqAndGenbank_filtered.xlsx")

filtered_biotypes <- list(Gene_Biotypes=total_genetable,
     Transcript_Biotypes=total_transtable) 



```

```{r}
c(filtered_biotypes,filtered_gene_types,filtered_transcript_types,gffcompare_stats_filtered,squanti_stats_filtered,busco_stats_filtered) %>% rio::export(file="2025_GeneModelandBioTypes_stats_RUvsRefseqAndGenbank_filtered.xlsx")
c(full_biotypes,unfiltered_gene_types,unfiltered_transcript_types,gffcompare_stats_unfiltered,squanti_stats_unfiltered,busco_stats_unfiltered) %>% rio::export(file="2025_GeneModelandBioTypes_stats_RUvsRefseqAndGenbank_full.xlsx")
```


```{r}
Combined_all %>% 
  as.data.frame %>% 
  filter(type %in% "transcript") %>% 
  filter(transcript_type %in% c("lncRNA","mRNA")) %>% summarise(hasRefSeq=sum(!is.na(Refseq_gene_id)))
  


Combined_all_with_pirna <- Combined_all
Combined_all_with_pirna_filtered <- Combined_all[!Combined_all$transcript_id %in% sq_filter]

names(Combined_all_with_pirna) <- NULL
export(Combined_all_with_pirna,con = "Combined_all_with_pirna.gtf")

names(Combined_all_with_pirna_filtered) <- NULL
export(Combined_all_with_pirna_filtered,con = "Combined_all_with_pirna_filtered.gtf")
```



### Run gffcompare
```{r,eval=FALSE}
dir.create("gffcompare_2025",showWarnings = FALSE,recursive = TRUE)
Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Refseq.gtf -o gffcompare_2025/Combined_all_with_pirna_refseq Combined_all_with_pirna.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")

Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Refseq.gtf -o gffcompare_2025/Combined_all_with_pirna_filtered_refseq Combined_all_with_pirna_filtered.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")

Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Genbank.gtf -o gffcompare_2025/Combined_all_with_pirna_genbank Combined_all_with_pirna.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")

Herper::with_CondaEnv("gffcompare",
              system("gffcompare -V -r Genbank.gtf -o gffcompare_2025/Combined_all_with_pirna_filtered_genbank Combined_all_with_pirna_filtered.gtf"),
              pathToMiniConda = "/Users/thomascarroll/Downloads/Kip_Delivery_GeneModels/condaenvs/")



```


```{r}
parse_cuffCompare <- function(stats_file){
lines <- readLines(stats_file)
prec_sens <- lines[grepl("level",lines)] %>% 
 str_trim() %>% 
 gsub("Intron chain","Intron_chain",.) %>% 
 gsub(" level:","_level",.) %>% 
 gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V3,-V5) %>% set_colnames(c("level","Sensitivity","Precision"))

Query_mrna <- lines[grepl("Query mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4,-V5) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))
 
Reference_mrna <- lines[grepl("Reference mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=4,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))

matching <- lines[grepl("Matching",lines)] %>% str_trim() %>%
   gsub(".*:","",.) %>% str_trim() %>% as.numeric %>% as.data.frame() %>% t %>% 
 set_colnames(c("Matching_intron_chains","Matching_transcripts","Matching_loci"))

myRownames <- lines[grepl("Missed|Novel",lines)] %>% 
 str_trim() %>% 
 gsub(":.*","",.) %>% gsub(" ","_",.)
Novel_missing <- lines[grepl("Missed|Novel",lines)] %>% str_trim() %>% 
 gsub(".*:","",.) %>% str_trim() %>% 
 gsub("\t.*","",.) %>% str_trim() %>% str_split("/") %>% unlist %>% matrix(ncol=2,byrow = TRUE) %>% 
 as.data.frame() %>% set_colnames(c("n_of","Total")) %>% set_rownames(myRownames)

toReturn <- list(prec_sens,Query_mrna,Reference_mrna,matching,Novel_missing)
names(toReturn) <- c("prec_sens","Query_mrna","Reference_mrna","matching","Novel_missing")
toReturn
}

refseq_gffcompare <- parse_cuffCompare("gffcompare_2025/Combined_all_with_pirna_filtered_refseq.stats")
genbank_gffcompare <- parse_cuffCompare("gffcompare_2025/Combined_all_with_pirna_filtered_genbank.stats")

gffcompare_sensitivity <- merge(refseq_gffcompare$prec_sens[,1:2],
                    genbank_gffcompare$prec_sens[,1:2],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Sensitivity",sep="_"))) 

gffcompare_precision <- merge(refseq_gffcompare$prec_sens[,c(1,3)],
                    genbank_gffcompare$prec_sens[,c(1,3)],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Precision",sep="_"))) 

gffcompare_ns<- cbind(t(refseq_gffcompare$Query_mrna),
                    t(refseq_gffcompare$Reference_mrna),t(genbank_gffcompare$Reference_mrna)) %>% 
 set_colnames(c("RU","RefSeq","GenBank")) %>% as.data.frame() %>% .[c(2,1,3),] %>% rownames_to_column()

gffcompare_matching<- cbind(t(refseq_gffcompare$matching),
                    t(genbank_gffcompare$matching)) %>% 
 set_colnames(c("RefSeq","GenBank")) %>% as.data.frame() %>% .[c(3,2,1),] %>% rownames_to_column()

gffcompare_NovelAndMissing <- cbind(refseq_gffcompare$Novel_missing %>% 
                   set_colnames(paste0(colnames(.),"_against_RefSeq")),
                  genbank_gffcompare$Novel_missing %>%
                   set_colnames(paste0(colnames(.),"_against_Genbank"))) %>% 
 as.data.frame() %>% rownames_to_column()

gffcompare_wihpirna_stats_filtered <- list(gffcompare_Summary_numbers=gffcompare_ns,
             gffcompare_NovelAndMissing=gffcompare_NovelAndMissing,
             gffcompare_matching=gffcompare_matching,
             gffcompare_sensitivity=gffcompare_sensitivity,
             gffcompare_precision=gffcompare_precision)
             
             
 

```


```{r}
parse_cuffCompare <- function(stats_file){
lines <- readLines(stats_file)
prec_sens <- lines[grepl("level",lines)] %>% 
 str_trim() %>% 
 gsub("Intron chain","Intron_chain",.) %>% 
 gsub(" level:","_level",.) %>% 
 gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V3,-V5) %>% set_colnames(c("level","Sensitivity","Precision"))

Query_mrna <- lines[grepl("Query mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=5,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4,-V5) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))
 
Reference_mrna <- lines[grepl("Reference mRNAs :",lines)] %>% 
  gsub(".*: +|in|loci|\\(","",.) %>% 
  gsub(" +","\t",.) %>% str_split("\t") %>% unlist %>% matrix(ncol=4,byrow = TRUE) %>% 
 as.data.frame() %>% 
 select(-V4) %>% 
 set_colnames(c("trascripts","genes","multi-exon_transcripts"))

matching <- lines[grepl("Matching",lines)] %>% str_trim() %>%
   gsub(".*:","",.) %>% str_trim() %>% as.numeric %>% as.data.frame() %>% t %>% 
 set_colnames(c("Matching_intron_chains","Matching_transcripts","Matching_loci"))

myRownames <- lines[grepl("Missed|Novel",lines)] %>% 
 str_trim() %>% 
 gsub(":.*","",.) %>% gsub(" ","_",.)
Novel_missing <- lines[grepl("Missed|Novel",lines)] %>% str_trim() %>% 
 gsub(".*:","",.) %>% str_trim() %>% 
 gsub("\t.*","",.) %>% str_trim() %>% str_split("/") %>% unlist %>% matrix(ncol=2,byrow = TRUE) %>% 
 as.data.frame() %>% set_colnames(c("n_of","Total")) %>% set_rownames(myRownames)

toReturn <- list(prec_sens,Query_mrna,Reference_mrna,matching,Novel_missing)
names(toReturn) <- c("prec_sens","Query_mrna","Reference_mrna","matching","Novel_missing")
toReturn
}

refseq_gffcompare <- parse_cuffCompare("gffcompare_2025/Combined_all_with_pirna_refseq.stats")
genbank_gffcompare <- parse_cuffCompare("gffcompare_2025/Combined_all_with_pirna_genbank.stats")

gffcompare_sensitivity <- merge(refseq_gffcompare$prec_sens[,1:2],
                    genbank_gffcompare$prec_sens[,1:2],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Sensitivity",sep="_"))) 

gffcompare_precision <- merge(refseq_gffcompare$prec_sens[,c(1,3)],
                    genbank_gffcompare$prec_sens[,c(1,3)],by=1,all=TRUE) %>% 
 set_colnames(c("level",
         paste(c("RefSeq",
         "GenBank"),"Precision",sep="_"))) 

gffcompare_ns<- cbind(t(refseq_gffcompare$Query_mrna),
                    t(refseq_gffcompare$Reference_mrna),t(genbank_gffcompare$Reference_mrna)) %>% 
 set_colnames(c("RU","RefSeq","GenBank")) %>% as.data.frame() %>% .[c(2,1,3),] %>% rownames_to_column()

gffcompare_matching<- cbind(t(refseq_gffcompare$matching),
                    t(genbank_gffcompare$matching)) %>% 
 set_colnames(c("RefSeq","GenBank")) %>% as.data.frame() %>% .[c(3,2,1),] %>% rownames_to_column()

gffcompare_NovelAndMissing <- cbind(refseq_gffcompare$Novel_missing %>% 
                   set_colnames(paste0(colnames(.),"_against_RefSeq")),
                  genbank_gffcompare$Novel_missing %>%
                   set_colnames(paste0(colnames(.),"_against_Genbank"))) %>% 
 as.data.frame() %>% rownames_to_column()

gffcompare_wihpirna_stats <- list(gffcompare_Summary_numbers=gffcompare_ns,
             gffcompare_NovelAndMissing=gffcompare_NovelAndMissing,
             gffcompare_matching=gffcompare_matching,
             gffcompare_sensitivity=gffcompare_sensitivity,
             gffcompare_precision=gffcompare_precision)
             
             
 

```

```{r}
c(filtered_biotypes,filtered_gene_types,filtered_transcript_types,gffcompare_wihpirna_stats_filtered,squanti_stats_filtered,busco_stats_filtered) %>% rio::export(file="2025_GeneModelandBioTypes_stats_RUvsRefseqAndGenbank_withpiRNAs_filtered.xlsx")
c(full_biotypes,unfiltered_gene_types,unfiltered_transcript_types,gffcompare_wihpirna_stats,squanti_stats_unfiltered,busco_stats_unfiltered) %>% rio::export(file="2025_GeneModelandBioTypes_stats_RUvsRefseqAndGenbank_withpiRNAs_full.xlsx")
```

```{r,eval=FALSE}
allIDs <- Combined_all_with_pirna %>% as.data.frame %>% filter(type=="transcript") %>% 
  pull(transcript_id) %>% unique
allIDs2 <- Combined_all_with_pirna %>% as.data.frame %>% 
  pull(transcript_id) %>% unique

temp <- allIDs2[!allIDs2 %in% allIDs]
Combined_all_with_pirna[Combined_all_with_pirna$gene_id %in% "RU.8978"]

temmp <- makeTxDbFromGRanges(RefSeq_GTF)
myTrans <- transcripts(temmp)
myExonsTrans <- exonsBy(temmp,by = "tx",use.names=TRUE)
length(myTrans)
length(unique(myTrans))
RefSeq_GTF$transcript_id %>% unique %>% length()
full_biotypes$Transcript_Biotypes$RefSeq %>% sum

RefSeq_GTF %>% as.data.frame %>%  pull(transcript_id) %>% unique %>% length
RefSeq_GTF %>% as.data.frame %>% filter(type=="transcript") %>% pull(transcript_id) %>% unique %>% length
RefSeq_GTF %>% as.data.frame %>% filter(type=="transcript") %>% pull(transcript_biotype) %>% table %>% sum
RefSeq_GTF %>% as.data.frame %>% filter(type=="exon") %>% pull(transcript_id) %>% unique %>% length
RefSeq_GTF %>% as.data.frame %>% filter(type=="gene") %>% pull(transcript_id) %>% unique %>% length

teew <- RefSeq_GTF %>% as.data.frame %>%  pull(transcript_id) %>% unique
teew2 <- RefSeq_GTF %>% as.data.frame %>% filter(type=="transcript") %>% pull(transcript_id) %>% unique 
teew[!teew %in% teew2]

myTrans[myTrans$tx_name %in% teew2]
RefSeq_GTF[RefSeq_GTF$transcript_id %in% "",]
RefSeq_GTF[RefSeq_GTF$gene_id %in% "LOC113563558",]

RefSeq_GTF[RefSeq_GTF %over% GRanges("Chr1:7590224-7602851")]

RefSeq_GTF[RefSeq_GTF$type %in% "transcript",]

temp <- read.delim("gffcompare_2025/Combined_all_with_pirna_refseq.tracking",header = FALSE)
tempA <- temp$V5 %>% str_split("\\|") %>% unlist %>% matrix(ncol=7,byrow=TRUE)
tempB <- tempA[,1] %>% str_split(":") %>% unlist %>% matrix(ncol=2,byrow=TRUE)
temp <- cbind(temp[,1:4],tempB,tempA[,-1]) %>% as.data.frame()
temp$V2 %>% unique %>% length
temp[,6] %>% unique %>% length

temp[!temp$V3 %in% "-",] %>% as.data.frame %>% .$V3 %>% str_split("\\|") %>% unlist %>% matrix(ncol=2,byrow=TRUE) %>% .[,2] %>% unique

tempA <- temp %>% select(2,6) %>% distinct %>% 
  group_by(V2) %>% 
  summarise(n())
tempA[tempA$`n()` > 1,]
temp[temp$V2 %in% "XLOC_000142",]
temp[temp$V2 %in% "XLOC_000176",]
temp[temp$V2 %in% "XLOC_000324",]


tempC <- temp %>% select(2,6) %>% distinct %>% set_colnames(c("V2","V3")) %>% 
  group_by(V3) %>% 
  summarise(n())


temp <- read.delim("Combined_all_with_pirna_refseq.Combined_all_with_pirna.gtf.tmap",header = TRUE)
temp[,2] %>% unique %>% length


RefSeq_GTF <- import("Refseq.gtf")
RefSeq_txdb <- makeTxDbFromGRanges(RefSeq_GTF)

RefSeq_exByTx <- exonsBy(RefSeq_txdb,by = "tx",use.names=TRUE)

sssa <- findOverlaps(RefSeq_exByTx,RefSeq_exByTx,type = "equal")

sssa <- sssa %>% as.data.frame
sssa[sssa$queryHits != sssa$subjectHits,]


RefSeq_exByTx[740]
RefSeq_exByTx[741]
RefSeq_exByTx[17714]
RefSeq_exByTx[17715]






```

```{r}
devtools::session_info()
```

```{r,eval=FALSE,echo=FALSE}
# Anno_Combined_txdb <- makeTxDbFromGRanges(LongandRefseqandShort_Cleaned2_GTF)
# Trans_Anno_Combined_txdb <- makeTxDbFromGRanges(Transdecode_LongandRefseqandShort_Cleaned2_GTF)
# Trans_Anno_Combined_trans <- transcripts(Trans_Anno_Combined_txdb)
# Anno_Combined_trans <- transcripts(Anno_Combined_txdb)
# 
# Trans_Anno_Combined_trans$tx_name %>% unique %>% length
# Trans_Anno_Combined_trans$tx_name %>% gsub("\\.p.*","",.) %>% unique %>% length()
# Anno_Combined_trans$tx_name %>% unique %>% length()

XM_011334874.3 - Merge in non conflicting refsq gene models? Overlap at 5'end of extended Stringtie
XM_011344031.2 - Could this be a polyA to join but missed
XM_011332670.3 - Overlaps in 3' so exlcuded
XM_020031084.2 - Merge in non conflicting refsq gene models?..it has fucked up gene name /gene model in refseq
XM_011335248.3 - Merge in non conflicting refsq gene models?
XM_026974150.1 - two in refseq vs stringtie..has a missing exoon in stringtie

RU.2286.1 What.why has this been joined..
RU.1083.1 Refseq missing the gene entirely
RU.9585.1 Overlaps above bu on othee sttrand
RU.1768. missing exon combinarion
RU.3199.2 missing long exon
RU.3344 as above and opposite strand
RU.3643.1 - more complete gene..longer in RU
RU.5535 - mpre xonns variants
RU.7022.7 - more vairants and longer exons
RU.7421.1 - Longer 5'uTR
#https://www.nature.com/articles/s41592-024-02298-3
```

## Get counts over all gene models

```{r,eval=FALSE}
require(bambu)
dir.create("RObjects")
bambuCounts <- bambu("PacBio/PBTools/mergedBams/Sample1_pbmm2.bam",
           annotations="PacBio/Stringtie/Stringtie_Sample1.gtf",
           genome = "Genome/GenBank.fa",
           discovery=FALSE,
           quant=TRUE)
saveRDS(bambuCounts,"RObjects/Stringtie_bambuCounts.RData")
Obiroi_BRCDK_TXDB <- prepareAnnotations("Originals/Updated_2024/Obiroi_BRCDK__5_4.gtf.gz")
bambuCounts2 <- bambu("PacBio/PBTools/mergedBams/Sample1_pbmm2.bam",
           annotations=Obiroi_BRCDK_TXDB,
           genome = "Genome/GenBank.fa",
           discovery=FALSE,
           quant=TRUE)
saveRDS(bambuCounts2,"RObjects/Obiroi_BRCDK_bambuCounts.RData")

Obiroi_GenBank_TXDB <- prepareAnnotations("Originals/Genbank/Obiroi_GenBank__5_4_renamed.gtf.gz")
bambuCounts3 <- bambu("PacBio/PBTools/mergedBams/Sample1_pbmm2.bam",
           annotations=Obiroi_GenBank_TXDB,
           genome = "Genome/GenBank.fa",
           discovery=FALSE,
           quant=TRUE)
saveRDS(bambuCounts3,"RObjects/Obiroi_GenBank_bambuCounts.RData")

Obiroi_RefSeq_TXDB <- prepareAnnotations("Originals/Refseq/Obiroi_RefSeq__5_4_renamed.gtf.gz")
bambuCounts4 <- bambu("PacBio/PBTools/mergedBams/Sample1_pbmm2.bam",
           annotations=Obiroi_RefSeq_TXDB,
           genome = "Genome/GenBank.fa",
           discovery=FALSE,
           quant=TRUE)
saveRDS(bambuCounts4,"RObjects/Obiroi_RefSeq_bambuCounts.RData")

Obiroi_xenoRef_TXDB <- prepareAnnotations("Originals/USCS_XenoRefGene/GCF_003672135.1_Obir_v5.4.xenoRefGenerenamed.gtf.gz")
bambuCounts5 <- bambu("PacBio/PBTools/mergedBams/Sample1_pbmm2.bam",
           annotations=Obiroi_xenoRef_TXDB,
           genome = "Genome/GenBank.fa",
           discovery=FALSE,
           quant=TRUE)
saveRDS(bambuCounts5,"RObjects/Obiroi_xenoRef_bambuCounts.RData")
```

```{r,eval=FALSE}
require(Herper)
dir.create("stringtie_transcript_longread_counts",showWarnings = FALSE)
dir.create("stringtie_transcript_longread_counts/Stringtie_Sample1_counts")
Herper::with_CondaEnv(new = "stringtie",
           system(paste("stringtie","PacBio/PBTools/mergedBams/Sample1_pbmm2.bam","-G","PacBio/Stringtie/Stringtie_Sample1.gtf",
                  "-eB","-o","stringtie_transcript_longread_counts/Stringtie_Sample1_counts/"),
               wait=TRUE
               )
           )

temp <- rtracklayer::import("Originals/Updated_2024/Obiroi_BRCDK__5_4.gtf.gz")
temp[temp$transcript_id == "",]$transcript_id <- NA
tempGTF <- paste0(tempfile(),".gtf")
export(temp,tempGTF)
Herper::with_CondaEnv(new = "stringtie",
           system(paste("stringtie","PacBio/PBTools/mergedBams/Sample1_pbmm2.bam","-G",tempGTF,
                  "-eB","-o","stringtie_transcript_longread_counts/Obiroi_BRCDK_counts"),
               wait=TRUE
               )
           )


temp <- rtracklayer::import("Originals/Genbank/Obiroi_GenBank__5_4_renamed.gtf.gz")
temp[temp$transcript_id == "",]$transcript_id <- NA
tempGTF <- paste0(tempfile(),".gtf")
export(temp,tempGTF)
Herper::with_CondaEnv(new = "stringtie",
           system(paste("stringtie","PacBio/PBTools/mergedBams/Sample1_pbmm2.bam","-G",tempGTF,
                  "-eB","-o","stringtie_transcript_longread_counts/Obiroi_GenBank_counts"),
               wait=TRUE
               )
           )

temp <- rtracklayer::import("Originals/Refseq/Obiroi_RefSeq__5_4_renamed.gtf.gz")
temp[temp$transcript_id == "",]$transcript_id <- NA
tempGTF <- paste0(tempfile(),".gtf")
export(temp,tempGTF)
Herper::with_CondaEnv(new = "stringtie",
           system(paste("stringtie","PacBio/PBTools/mergedBams/Sample1_pbmm2.bam","-G",tempGTF,
                  "-eB","-o","stringtie_transcript_longread_counts/Obiroi_Refseq_counts"),
               wait=TRUE
               )
           )

temp <- rtracklayer::import("Originals/USCS_XenoRefGene/GCF_003672135.1_Obir_v5.4.xenoRefGenerenamed.gtf.gz")
# temp[temp$transcript_id == "",]$transcript_id <- NA
tempGTF <- paste0(tempfile(),".gtf")
export(temp,tempGTF)
Herper::with_CondaEnv(new = "stringtie",
           system(paste("stringtie","PacBio/PBTools/mergedBams/Sample1_pbmm2.bam","-G",tempGTF,
                  "-eB","-o","stringtie_transcript_longread_counts/Obiroi_xenoRef_counts"),
               wait=TRUE
               )
           )

```
