
---
output:
  html_document
---

```{css,echo=FALSE}
  .title {
    margin-top: 0.5em !important;
    margin-bottom: 0.5em;
  }

  .section h1,  .section h2,  .section h3,  .section h4,  .section h5 {
    padding-top: 0.5em !important; /*adding so much padding because theme adds a margin-top: -65px by default */
      margin-bottom: 0.2em;
  }

  @media (min-width: 992px) {
    .col-md-9 {
      width: 75%;
      padding-left: 5em !important;
    }
  }

  #TOC::before {
  content: "";
  display: block;
  height: 200px;
  margin: 2em 20px 40px 20px;
  background-image: url("https://avatars.githubusercontent.com/u/24759038?s=400&u=7ae0265948793e7698efd079e4f3391ca147ed6b&v=4");
  background-size: contain;
  background-position: center center;
  background-repeat: no-repeat;
}

```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
```
## Params


## Setting output, environments, libraries and temporary directories.

First the directories for workflow output, conda and conda environments used by Herper, R libaries from Renv and for temporary files must be created.

<details>

<style>
 div.JSasR {
      display: block;
      padding: 9.5px;
      margin: 0 0 10px;
      font-size: 13px;
      line-height: 1.42857143;
      word-break: break-all;
      word-wrap: break-word;
      color: #333333;
        background-color: #f5f5f5;
        border: 1px solid #cccccc;
      border-radius: 4px;
  }
</style>
<div class = "JSasR">
```{js,echo=FALSE,eval=TRUE}
var myStringArray = "output_dir";
var myStringArray2 = "output_dir";
var arrayLength = myStringArray.length;
for (let i = 0; i < arrayLength; i++) {
  var setwdStart2 = myStringArray[i];
  var setwdStart = ' <- "';
  var setwdStart3 = myStringArray2[i];
  var setwdEnd = '"';
  var loc = window.location.pathname;
  var dir = loc.substring(0, loc.lastIndexOf('/'));
  var dir2 = dir.substring(0, dir.lastIndexOf('/'));
  var path = setwdStart2.concat(setwdStart).concat(dir2).concat("/").concat(setwdStart3).concat(setwdEnd);
  document.write(path);
  document.write("<br>");
}
```
</div>

```{r}
dir.create(output_dir, showWarnings = FALSE, recursive = TRUE)
```
</details>
## Create whitelisted fasta file

This section descibes how we create our reference fasta file

<details>
```{r , include=TRUE}
Obiroi_BRCDK_5_4_whitelist <- {
  options(timeout = 600 * 60)
  download_extdata_dir <- file.path(output_dir, "extdata")
  dir.create(download_extdata_dir, showWarnings = FALSE, recursive = TRUE)
  download.file("https://rubioinformatics.s3.amazonaws.com/Reference/Obiroi_BRCDK__5_4_whitelist.txt", file.path(download_extdata_dir, "Obiroi_BRCDK__5_4_whitelist.txt"))
  Obiroi_BRCDK_5_4_whitelist_MD5 <- tools::md5sum(file.path(download_extdata_dir, "Obiroi_BRCDK__5_4_whitelist.txt"))
  if (Obiroi_BRCDK_5_4_whitelist_MD5 == "bd85d59677a4f60e9b7341ff1d1da2f0") {
    write.table(Obiroi_BRCDK_5_4_whitelist_MD5, file.path(download_extdata_dir, "Obiroi_BRCDK_5_4_whitelist_MD5.txt"))
  } else {
    unlink(file.path(download_extdata_dir, "Obiroi_BRCDK__5_4_whitelist.txt"))
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_extdata_dir, "Obiroi_BRCDK__5_4_whitelist.txt")
}
Obiroi_BRCDK_5_4_fasta <- {
  options(timeout = 600 * 60)
  download_extdata_dir <- file.path(output_dir, "extdata")
  dir.create(download_extdata_dir, showWarnings = FALSE, recursive = TRUE)
  download.file("https://rubioinformatics.s3.amazonaws.com/Reference/GenBank.fa.gz", file.path(download_extdata_dir, "GenBank.fa.gz"))
  Obiroi_BRCDK_5_4_fasta_MD5 <- tools::md5sum(file.path(download_extdata_dir, "GenBank.fa.gz"))
  if (Obiroi_BRCDK_5_4_fasta_MD5 == "b9a982a884fb6345e305147bbd876096") {
    write.table(Obiroi_BRCDK_5_4_fasta_MD5, file.path(download_extdata_dir, "Obiroi_BRCDK_5_4_fasta_MD5.txt"))
  } else {
    unlink(file.path(download_extdata_dir, "GenBank.fa.gz"))
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_extdata_dir, "GenBank.fa.gz")
}
BRCDK54_Bzipped <- {
  unzipped <- gunzip(Obiroi_BRCDK_5_4_fasta, destname = tempfile(), remove = FALSE)
  bgzip(unzipped, dest = file.path(output_dir, "BRCDK54_Bzipped.fa.bgz"), overwrite = TRUE)
  file.path(output_dir, "BRCDK54_Bzipped.fa.bgz")
}
BRCDK54_BzippedIndex <- {
  indexFa(BRCDK54_Bzipped)
  "BRCDK54_Bzipped.fa.bgz.fai"
}
BRCDK54_BzippedWhite <- {
  FaFile <- FaFile(file = BRCDK54_Bzipped, index = BRCDK54_BzippedIndex)
  WhiteListChrs <- read.table(Obiroi_BRCDK_5_4_whitelist, header = TRUE, row.names = 1) %>% rownames()
  dnastringset <- scanFa(FaFile, as = c("DNAStringSet"))
  fastawhite_File <- tempfile()
  writeXStringSet(dnastringset[WhiteListChrs], file = fastawhite_File)
  bgzip(fastawhite_File, dest = file.path(output_dir, "BRCDK54_BzippedWhite.fa.bgz"))
  file.path(output_dir, "BRCDK54_BzippedWhite.fa.bgz")
}
```
```
trying URL 'https://rubioinformatics.s3.amazonaws.com/Reference/Obiroi_BRCDK__5_4_whitelist.txt'
Content type 'text/plain' length 3955 bytes
==================================================
downloaded 3955 bytes


trying URL 'https://rubioinformatics.s3.amazonaws.com/Reference/GenBank.fa.gz'
Content type 'application/x-gzip' length 67490813 bytes (64.4 MB)
==================================================
downloaded 64.4 MB





```
</details>
## Create Hisat2 index for reference genome (Rhisat2)

This section descibes how we build the the index files for Rhisat2 alignment.

<details>
```{r , include=TRUE}
BRCDK54_hisat2index <- {
  unzipped_fasta <- paste0(tempfile(), ".fa")
  gunzip(BRCDK54_BzippedWhite, destname = unzipped_fasta, ext = "bgz", FUN = gzfile, remove = FALSE)
  hisat2_build(references = unzipped_fasta, outdir = file.path(output_dir, "BRCDK54_hisat2index"), force = TRUE, prefix = "index")
  file.path(output_dir, "BRCDK54_hisat2index", "index.1.ht2")
}
```
```

```
</details>
## Process

## Create whitelisted fasta file

This section descibes how we create our reference fasta file

<details>
```{r , include=TRUE}
BRCDK54_BzippedWhiteIndex <- {
  indexFa(BRCDK54_BzippedWhite)
  file.path(output_dir, "BRCDK54_BzippedWhite.fa.bgz.fai")
}
```
```

```
</details>
## Build sequence dictionary files for Picard

Here we create sequence dictionary files required for Picard.

<details>
```{r , include=TRUE}
general_conda <- {
  "/ru-auth/local/home/brc_pipeline/brc_conda/"
}
general_env <- {
  "general_rnaseq"
}
BRCDK54_dictionary <- {
  Herper::with_CondaEnv(general_env,
    {
      tempFasta <- paste0(tempfile(), ".fa")
      gunzip(BRCDK54_BzippedWhite, destname = tempFasta, ext = "bgz", FUN = gzfile, remove = FALSE)
      command <- paste0("picard", " ", "CreateSequenceDictionary", " ", "R=", tempFasta, " ", "O=", file.path(output_dir, "BRCDK54_dictionary.dict"))
      ruTargets::system(command)
      file.path(output_dir, "BRCDK54_dictionary.dict")
    },
    pathToMiniConda = general_conda
  )
}
```
```




```
</details>
## FastQ retrieval and integrity checking of fastQ files

This section describes the retrieval of FastQ and testing of their
integrity of files by checking the md5 sum of downloaded files versus known md5.

<details>
```{r , include=TRUE}
FourthInstarLarvae_I1_Pair1 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//4thInstarLarvae_S3_R1_001.fastq.gz", file.path(download_process_dir, "4thInstarLarvae_S3_R1_001.fastq.gz"), overwrite = TRUE)
  FourthInstarLarvae_I1_Pair1MD5 <- tools::md5sum(file.path(download_process_dir, "4thInstarLarvae_S3_R1_001.fastq.gz"))
  if (FourthInstarLarvae_I1_Pair1MD5 == "fe6792f5f4adf702a2617ee87c1e9d67") {
    write.table("fe6792f5f4adf702a2617ee87c1e9d67", file.path(download_process_dir, "FourthInstarLarvae_I1_Pair1MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "4thInstarLarvae_S3_R1_001.fastq.gz")
}
Pupae_I1_Pair2 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//Pupae_S5_R2_001.fastq.gz", file.path(download_process_dir, "Pupae_S5_R2_001.fastq.gz"), overwrite = TRUE)
  Pupae_I1_Pair2MD5 <- tools::md5sum(file.path(download_process_dir, "Pupae_S5_R2_001.fastq.gz"))
  if (Pupae_I1_Pair2MD5 == "238d7b1944d3ae0386bb93bbd564ab35") {
    write.table("238d7b1944d3ae0386bb93bbd564ab35", file.path(download_process_dir, "Pupae_I1_Pair2MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "Pupae_S5_R2_001.fastq.gz")
}
Eggs_I1_Pair1 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//Eggs_S1_R1_001.fastq.gz", file.path(download_process_dir, "Eggs_S1_R1_001.fastq.gz"), overwrite = TRUE)
  Eggs_I1_Pair1MD5 <- tools::md5sum(file.path(download_process_dir, "Eggs_S1_R1_001.fastq.gz"))
  if (Eggs_I1_Pair1MD5 == "249f5ea4c813049bea4f9c077bd287f6") {
    write.table("249f5ea4c813049bea4f9c077bd287f6", file.path(download_process_dir, "Eggs_I1_Pair1MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "Eggs_S1_R1_001.fastq.gz")
}
Eggs_I1_Pair2 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//Eggs_S1_R2_001.fastq.gz", file.path(download_process_dir, "Eggs_S1_R2_001.fastq.gz"), overwrite = TRUE)
  Eggs_I1_Pair2MD5 <- tools::md5sum(file.path(download_process_dir, "Eggs_S1_R2_001.fastq.gz"))
  if (Eggs_I1_Pair2MD5 == "cfc77b2d48e37d19fc6255ef42c6374d") {
    write.table("cfc77b2d48e37d19fc6255ef42c6374d", file.path(download_process_dir, "Eggs_I1_Pair2MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "Eggs_S1_R2_001.fastq.gz")
}
Adults_I1_Pair1 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//Adults_S6_R1_001.fastq.gz", file.path(download_process_dir, "Adults_S6_R1_001.fastq.gz"), overwrite = TRUE)
  Adults_I1_Pair1MD5 <- tools::md5sum(file.path(download_process_dir, "Adults_S6_R1_001.fastq.gz"))
  if (Adults_I1_Pair1MD5 == "3a59e54cc98fdf1a9d5c6a533b8dbb31") {
    write.table("3a59e54cc98fdf1a9d5c6a533b8dbb31", file.path(download_process_dir, "Adults_I1_Pair1MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "Adults_S6_R1_001.fastq.gz")
}
Adults_I1_Pair2 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//Adults_S6_R2_001.fastq.gz", file.path(download_process_dir, "Adults_S6_R2_001.fastq.gz"), overwrite = TRUE)
  Adults_I1_Pair2MD5 <- tools::md5sum(file.path(download_process_dir, "Adults_S6_R2_001.fastq.gz"))
  if (Adults_I1_Pair2MD5 == "ef9ab3b9b369214958529efd5071d3d3") {
    write.table("ef9ab3b9b369214958529efd5071d3d3", file.path(download_process_dir, "Adults_I1_Pair2MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "Adults_S6_R2_001.fastq.gz")
}
Prepupae_I1_Pair1 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//Prepupae_S4_R1_001.fastq.gz", file.path(download_process_dir, "Prepupae_S4_R1_001.fastq.gz"), overwrite = TRUE)
  Prepupae_I1_Pair1MD5 <- tools::md5sum(file.path(download_process_dir, "Prepupae_S4_R1_001.fastq.gz"))
  if (Prepupae_I1_Pair1MD5 == "949fac340af43be172c2dcacab7982af") {
    write.table("949fac340af43be172c2dcacab7982af", file.path(download_process_dir, "Prepupae_I1_Pair1MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "Prepupae_S4_R1_001.fastq.gz")
}
YoungLarvae_I1_Pair1 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//YoungLarvae_S2_R1_001.fastq.gz", file.path(download_process_dir, "YoungLarvae_S2_R1_001.fastq.gz"), overwrite = TRUE)
  YoungLarvae_I1_Pair1MD5 <- tools::md5sum(file.path(download_process_dir, "YoungLarvae_S2_R1_001.fastq.gz"))
  if (YoungLarvae_I1_Pair1MD5 == "e998a3daf02c5979673c7cd2880cc177") {
    write.table("e998a3daf02c5979673c7cd2880cc177", file.path(download_process_dir, "YoungLarvae_I1_Pair1MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "YoungLarvae_S2_R1_001.fastq.gz")
}
Prepupae_I1_Pair2 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//Prepupae_S4_R2_001.fastq.gz", file.path(download_process_dir, "Prepupae_S4_R2_001.fastq.gz"), overwrite = TRUE)
  Prepupae_I1_Pair2MD5 <- tools::md5sum(file.path(download_process_dir, "Prepupae_S4_R2_001.fastq.gz"))
  if (Prepupae_I1_Pair2MD5 == "6b14d952a94719d73322b3633ea5052c") {
    write.table("6b14d952a94719d73322b3633ea5052c", file.path(download_process_dir, "Prepupae_I1_Pair2MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "Prepupae_S4_R2_001.fastq.gz")
}
Pupae_I1_Pair1 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//Pupae_S5_R1_001.fastq.gz", file.path(download_process_dir, "Pupae_S5_R1_001.fastq.gz"), overwrite = TRUE)
  Pupae_I1_Pair1MD5 <- tools::md5sum(file.path(download_process_dir, "Pupae_S5_R1_001.fastq.gz"))
  if (Pupae_I1_Pair1MD5 == "b33434fa6df2b2bb9127f69ea956d7b7") {
    write.table("b33434fa6df2b2bb9127f69ea956d7b7", file.path(download_process_dir, "Pupae_I1_Pair1MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "Pupae_S5_R1_001.fastq.gz")
}
YoungLarvae_I1_Pair2 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//YoungLarvae_S2_R2_001.fastq.gz", file.path(download_process_dir, "YoungLarvae_S2_R2_001.fastq.gz"), overwrite = TRUE)
  YoungLarvae_I1_Pair2MD5 <- tools::md5sum(file.path(download_process_dir, "YoungLarvae_S2_R2_001.fastq.gz"))
  if (YoungLarvae_I1_Pair2MD5 == "104ad601816e2da13415f3f419d29b3e") {
    write.table("104ad601816e2da13415f3f419d29b3e", file.path(download_process_dir, "YoungLarvae_I1_Pair2MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "YoungLarvae_S2_R2_001.fastq.gz")
}
FourthInstarLarvae_I1_Pair2 <- {
  download_process_dir <- file.path(output_dir, "FQ")
  dir.create(download_process_dir, showWarnings = FALSE, recursive = TRUE)
  file.copy("/rugpfs/fs0/brc/scratch/krozen/BRC_Kip/rRNA_depletion/220421_A00815_0362_BH2WVHDRX2/Kip_04122022//4thInstarLarvae_S3_R2_001.fastq.gz", file.path(download_process_dir, "4thInstarLarvae_S3_R2_001.fastq.gz"), overwrite = TRUE)
  FourthInstarLarvae_I1_Pair2MD5 <- tools::md5sum(file.path(download_process_dir, "4thInstarLarvae_S3_R2_001.fastq.gz"))
  if (FourthInstarLarvae_I1_Pair2MD5 == "8fe388e40511ee57f46cd59116f9090d") {
    write.table("8fe388e40511ee57f46cd59116f9090d", file.path(download_process_dir, "FourthInstarLarvae_I1_Pair2MD5.txt"))
  } else {
    stop("Calculated MD5 didn't match expected")
  }
  file.path(download_process_dir, "4thInstarLarvae_S3_R2_001.fastq.gz")
}
```
```












```
</details>
## FastQ Concatenation

This section descibes the concantenation of FastQ files from same sample and read pair into
a single FastQ file.

To do this we will use the Rfastp package's catFastQ function.

<details>
```{r , include=TRUE}
FourthInstarLarvae_Pair1FQ <- {
  FourthInstarLarvae_I1_Pair1
}
Pupae_Pair2FQ <- {
  Pupae_I1_Pair2
}
Eggs_Pair1FQ <- {
  Eggs_I1_Pair1
}
Eggs_Pair2FQ <- {
  Eggs_I1_Pair2
}
Adults_Pair1FQ <- {
  Adults_I1_Pair1
}
Adults_Pair2FQ <- {
  Adults_I1_Pair2
}
Prepupae_Pair1FQ <- {
  Prepupae_I1_Pair1
}
YoungLarvae_Pair1FQ <- {
  YoungLarvae_I1_Pair1
}
Prepupae_Pair2FQ <- {
  Prepupae_I1_Pair2
}
Pupae_Pair1FQ <- {
  Pupae_I1_Pair1
}
YoungLarvae_Pair2FQ <- {
  YoungLarvae_I1_Pair2
}
FourthInstarLarvae_Pair2FQ <- {
  FourthInstarLarvae_I1_Pair2
}
```
```












```
</details>
## FastQ Quality Control

This section descibes how quality control is performed on FastQ files prior to alignment.

FastQ quality is assessed here using the rfastp packages rfastp function.

To do this we will use the Rfastp package's catFastQ function.

<details>
```{r , include=TRUE}
minReadLength <- {
  15
}
qualityFilterPercent <- {
  40
}
maxNfilter <- {
  5
}
qualityFilterPhred <- {
  15
}
Eggs_QC <- {
  fqqc_path <- file.path(output_dir, "FQ_QC")
  dir.create(fqqc_path, showWarnings = FALSE, recursive = TRUE)
  Rfastp::rfastp(read1 = path.expand(Eggs_Pair1FQ), read2 = path.expand(Eggs_Pair2FQ), outputFastq = path.expand(file.path(fqqc_path, "Eggs_QC")), qualityFilterPhred = qualityFilterPhred, maxNfilter = maxNfilter, minReadLength = minReadLength, qualityFilterPercent = qualityFilterPercent)
  file.path(fqqc_path, paste0("Eggs_QC", ".json"))
}
Adults_QC <- {
  fqqc_path <- file.path(output_dir, "FQ_QC")
  dir.create(fqqc_path, showWarnings = FALSE, recursive = TRUE)
  Rfastp::rfastp(read1 = path.expand(Adults_Pair1FQ), read2 = path.expand(Adults_Pair2FQ), outputFastq = path.expand(file.path(fqqc_path, "Adults_QC")), qualityFilterPhred = qualityFilterPhred, maxNfilter = maxNfilter, minReadLength = minReadLength, qualityFilterPercent = qualityFilterPercent)
  file.path(fqqc_path, paste0("Adults_QC", ".json"))
}
Prepupae_QC <- {
  fqqc_path <- file.path(output_dir, "FQ_QC")
  dir.create(fqqc_path, showWarnings = FALSE, recursive = TRUE)
  Rfastp::rfastp(read1 = path.expand(Prepupae_Pair1FQ), read2 = path.expand(Prepupae_Pair2FQ), outputFastq = path.expand(file.path(fqqc_path, "Prepupae_QC")), qualityFilterPhred = qualityFilterPhred, maxNfilter = maxNfilter, minReadLength = minReadLength, qualityFilterPercent = qualityFilterPercent)
  file.path(fqqc_path, paste0("Prepupae_QC", ".json"))
}
Pupae_QC <- {
  fqqc_path <- file.path(output_dir, "FQ_QC")
  dir.create(fqqc_path, showWarnings = FALSE, recursive = TRUE)
  Rfastp::rfastp(read1 = path.expand(Pupae_Pair1FQ), read2 = path.expand(Pupae_Pair2FQ), outputFastq = path.expand(file.path(fqqc_path, "Pupae_QC")), qualityFilterPhred = qualityFilterPhred, maxNfilter = maxNfilter, minReadLength = minReadLength, qualityFilterPercent = qualityFilterPercent)
  file.path(fqqc_path, paste0("Pupae_QC", ".json"))
}
YoungLarvae_QC <- {
  fqqc_path <- file.path(output_dir, "FQ_QC")
  dir.create(fqqc_path, showWarnings = FALSE, recursive = TRUE)
  Rfastp::rfastp(read1 = path.expand(YoungLarvae_Pair1FQ), read2 = path.expand(YoungLarvae_Pair2FQ), outputFastq = path.expand(file.path(fqqc_path, "YoungLarvae_QC")), qualityFilterPhred = qualityFilterPhred, maxNfilter = maxNfilter, minReadLength = minReadLength, qualityFilterPercent = qualityFilterPercent)
  file.path(fqqc_path, paste0("YoungLarvae_QC", ".json"))
}
FourthInstarLarvae_QC <- {
  fqqc_path <- file.path(output_dir, "FQ_QC")
  dir.create(fqqc_path, showWarnings = FALSE, recursive = TRUE)
  Rfastp::rfastp(read1 = path.expand(FourthInstarLarvae_Pair1FQ), read2 = path.expand(FourthInstarLarvae_Pair2FQ), outputFastq = path.expand(file.path(fqqc_path, "FourthInstarLarvae_QC")), qualityFilterPhred = qualityFilterPhred, maxNfilter = maxNfilter, minReadLength = minReadLength, qualityFilterPercent = qualityFilterPercent)
  file.path(fqqc_path, paste0("FourthInstarLarvae_QC", ".json"))
}
```
```




Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[13:58:10] start to load data
[13:58:18] loaded 1M read pairs
[13:58:30] loaded 2M read pairs
[13:58:42] loaded 3M read pairs
[13:58:55] loaded 4M read pairs
[13:59:8] loaded 5M read pairs
[13:59:21] loaded 6M read pairs
[13:59:34] loaded 7M read pairs
[13:59:47] loaded 8M read pairs
[14:0:0] loaded 9M read pairs
[14:0:13] loaded 10M read pairs
[14:0:26] loaded 11M read pairs
[14:0:39] loaded 12M read pairs
[14:0:52] loaded 13M read pairs
[14:1:5] loaded 14M read pairs
[14:1:17] loaded 15M read pairs
[14:1:29] loaded 16M read pairs
[14:1:41] loaded 17M read pairs
[14:1:52] loaded 18M read pairs
[14:2:3] loaded 19M read pairs
[14:2:13] loaded 20M read pairs
[14:2:24] loaded 21M read pairs
[14:2:35] loaded 22M read pairs
[14:2:46] loaded 23M read pairs
[14:2:58] loaded 24M read pairs
[14:3:10] loaded 25M read pairs
[14:3:21] loaded 26M read pairs
[14:3:31] loaded 27M read pairs
[14:3:42] loaded 28M read pairs
[14:3:53] loaded 29M read pairs
[14:4:3] loaded 30M read pairs
[14:4:14] loaded 31M read pairs
[14:4:25] loaded 32M read pairs
[14:4:36] loaded 33M read pairs
[14:4:46] loaded 34M read pairs
[14:4:57] loaded 35M read pairs
[14:5:7] loaded 36M read pairs
[14:5:18] loaded 37M read pairs
[14:5:29] loaded 38M read pairs
[14:5:39] loaded 39M read pairs
[14:5:50] loaded 40M read pairs
[14:6:1] loaded 41M read pairs
[14:6:11] loaded 42M read pairs
[14:6:22] loaded 43M read pairs
[14:6:33] loaded 44M read pairs
[14:6:44] loaded 45M read pairs
[14:6:54] loaded 46M read pairs
[14:7:5] loaded 47M read pairs
[14:7:16] loaded 48M read pairs
[14:7:27] loaded 49M read pairs
[14:7:38] loaded 50M read pairs
[14:7:49] loaded 51M read pairs
[14:7:59] loaded 52M read pairs
[14:8:10] loaded 53M read pairs
[14:8:21] loaded 54M read pairs
[14:8:31] loaded 55M read pairs
[14:8:42] loaded 56M read pairs
[14:8:53] loaded 57M read pairs
[14:9:4] loaded 58M read pairs
[14:9:14] loaded 59M read pairs
[14:9:25] loaded 60M read pairs
[14:9:36] loaded 61M read pairs
[14:9:46] loaded 62M read pairs
[14:9:57] loaded 63M read pairs
[14:10:8] loaded 64M read pairs
[14:10:19] loaded 65M read pairs
[14:10:29] loaded 66M read pairs
[14:10:40] loaded 67M read pairs
[14:10:51] loaded 68M read pairs
[14:11:2] loaded 69M read pairs
[14:11:13] loaded 70M read pairs
[14:11:23] loaded 71M read pairs
[14:11:34] loaded 72M read pairs
[14:11:45] loaded 73M read pairs
[14:11:56] loaded 74M read pairs
[14:12:6] loaded 75M read pairs
[14:12:17] loaded 76M read pairs
[14:12:28] loaded 77M read pairs
[14:12:39] loaded 78M read pairs
[14:12:50] loaded 79M read pairs
[14:13:0] loaded 80M read pairs
[14:13:11] loaded 81M read pairs
[14:13:22] loaded 82M read pairs
[14:13:32] loaded 83M read pairs
[14:13:43] loaded 84M read pairs
[14:13:55] loaded 85M read pairs
[14:14:6] loaded 86M read pairs
[14:14:17] loaded 87M read pairs
[14:14:29] loaded 88M read pairs
[14:14:40] loaded 89M read pairs
[14:14:52] loaded 90M read pairs
[14:15:3] loaded 91M read pairs
[14:15:15] loaded 92M read pairs
Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[14:17:41] start to load data
[14:17:49] loaded 1M read pairs
[14:17:59] loaded 2M read pairs
[14:18:11] loaded 3M read pairs
[14:18:22] loaded 4M read pairs
[14:18:33] loaded 5M read pairs
[14:18:44] loaded 6M read pairs
[14:18:56] loaded 7M read pairs
[14:19:7] loaded 8M read pairs
[14:19:18] loaded 9M read pairs
[14:19:29] loaded 10M read pairs
[14:19:40] loaded 11M read pairs
[14:19:51] loaded 12M read pairs
[14:20:2] loaded 13M read pairs
[14:20:13] loaded 14M read pairs
[14:20:24] loaded 15M read pairs
[14:20:35] loaded 16M read pairs
[14:20:46] loaded 17M read pairs
[14:20:57] loaded 18M read pairs
[14:21:8] loaded 19M read pairs
[14:21:19] loaded 20M read pairs
[14:21:30] loaded 21M read pairs
[14:21:42] loaded 22M read pairs
[14:21:53] loaded 23M read pairs
[14:22:4] loaded 24M read pairs
[14:22:17] loaded 25M read pairs
[14:22:29] loaded 26M read pairs
[14:22:41] loaded 27M read pairs
[14:22:52] loaded 28M read pairs
[14:23:3] loaded 29M read pairs
[14:23:15] loaded 30M read pairs
[14:23:28] loaded 31M read pairs
[14:23:41] loaded 32M read pairs
[14:23:54] loaded 33M read pairs
[14:24:7] loaded 34M read pairs
[14:24:19] loaded 35M read pairs
[14:24:31] loaded 36M read pairs
[14:24:44] loaded 37M read pairs
[14:24:57] loaded 38M read pairs
[14:25:9] loaded 39M read pairs
[14:25:20] loaded 40M read pairs
[14:25:31] loaded 41M read pairs
[14:25:43] loaded 42M read pairs
[14:25:55] loaded 43M read pairs
[14:26:6] loaded 44M read pairs
[14:26:18] loaded 45M read pairs
[14:26:31] loaded 46M read pairs
[14:26:43] loaded 47M read pairs
[14:26:57] loaded 48M read pairs
[14:27:10] loaded 49M read pairs
[14:27:22] loaded 50M read pairs
[14:27:33] loaded 51M read pairs
[14:27:44] loaded 52M read pairs
[14:27:55] loaded 53M read pairs
[14:28:6] loaded 54M read pairs
[14:28:18] loaded 55M read pairs
[14:28:29] loaded 56M read pairs
[14:28:42] loaded 57M read pairs
[14:28:53] loaded 58M read pairs
[14:29:5] loaded 59M read pairs
[14:29:17] loaded 60M read pairs
[14:29:30] loaded 61M read pairs
[14:29:42] loaded 62M read pairs
[14:29:54] loaded 63M read pairs
[14:30:5] loaded 64M read pairs
[14:30:16] loaded 65M read pairs
[14:30:27] loaded 66M read pairs
[14:30:38] loaded 67M read pairs
[14:30:49] loaded 68M read pairs
[14:31:0] loaded 69M read pairs
[14:31:11] loaded 70M read pairs
[14:31:22] loaded 71M read pairs
[14:31:34] loaded 72M read pairs
[14:31:45] loaded 73M read pairs
[14:31:56] loaded 74M read pairs
[14:32:7] loaded 75M read pairs
[14:32:19] loaded 76M read pairs
[14:32:30] loaded 77M read pairs
[14:32:41] loaded 78M read pairs
[14:32:53] loaded 79M read pairs
[14:33:5] loaded 80M read pairs
[14:33:17] loaded 81M read pairs
[14:33:28] loaded 82M read pairs
[14:33:40] loaded 83M read pairs
[14:33:51] loaded 84M read pairs
[14:34:3] loaded 85M read pairs
[14:34:14] loaded 86M read pairs
[14:34:26] loaded 87M read pairs
[14:34:37] loaded 88M read pairs
[14:34:49] loaded 89M read pairs
[14:35:0] loaded 90M read pairs
[14:35:12] loaded 91M read pairs
[14:35:23] loaded 92M read pairs
[14:35:34] loaded 93M read pairs
[14:35:46] loaded 94M read pairs
[14:35:58] loaded 95M read pairs
[14:36:9] loaded 96M read pairs
[14:36:21] loaded 97M read pairs
[14:36:32] loaded 98M read pairs
[14:36:43] loaded 99M read pairs
[14:36:56] loaded 100M read pairs
[14:37:7] loaded 101M read pairs
[14:37:19] loaded 102M read pairs
[14:37:30] loaded 103M read pairs
[14:37:42] loaded 104M read pairs
[14:37:53] loaded 105M read pairs
[14:38:5] loaded 106M read pairs
[14:38:17] loaded 107M read pairs
[14:38:28] loaded 108M read pairs
[14:38:39] loaded 109M read pairs
[14:38:51] loaded 110M read pairs
[14:39:2] loaded 111M read pairs
[14:39:14] loaded 112M read pairs
[14:39:26] loaded 113M read pairs
[14:39:38] loaded 114M read pairs
[14:39:49] loaded 115M read pairs
[14:40:1] loaded 116M read pairs
[14:40:12] loaded 117M read pairs
[14:40:24] loaded 118M read pairs
[14:40:36] loaded 119M read pairs
[14:40:48] loaded 120M read pairs
[14:41:0] loaded 121M read pairs
[14:41:13] loaded 122M read pairs
[14:41:24] loaded 123M read pairs
[14:41:36] loaded 124M read pairs
[14:41:48] loaded 125M read pairs
[14:42:0] loaded 126M read pairs
[14:42:11] loaded 127M read pairs
[14:42:23] loaded 128M read pairs
[14:42:34] loaded 129M read pairs
[14:42:46] loaded 130M read pairs
[14:42:58] loaded 131M read pairs
[14:43:10] loaded 132M read pairs
[14:43:21] loaded 133M read pairs
[14:43:33] loaded 134M read pairs
[14:43:44] loaded 135M read pairs
[14:43:57] loaded 136M read pairs
[14:44:8] loaded 137M read pairs
[14:44:20] loaded 138M read pairs
[14:44:32] loaded 139M read pairs
[14:44:44] loaded 140M read pairs
[14:44:56] loaded 141M read pairs
[14:45:7] loaded 142M read pairs
[14:45:19] loaded 143M read pairs
[14:45:31] loaded 144M read pairs
[14:45:43] loaded 145M read pairs
[14:45:54] loaded 146M read pairs
[14:46:6] loaded 147M read pairs
[14:46:18] loaded 148M read pairs
[14:46:32] loaded 149M read pairs
[14:46:44] loaded 150M read pairs
[14:46:58] loaded 151M read pairs
[14:47:12] loaded 152M read pairs
[14:47:27] loaded 153M read pairs
[14:47:41] loaded 154M read pairs
[14:47:55] loaded 155M read pairs
[14:48:9] loaded 156M read pairs
[14:48:23] loaded 157M read pairs
[14:48:36] loaded 158M read pairs
[14:48:50] loaded 159M read pairs
[14:49:3] loaded 160M read pairs
[14:49:15] loaded 161M read pairs
[14:49:28] loaded 162M read pairs
[14:49:40] all reads loaded, start to monitor thread status
[14:49:40] thread 1 is processing the 162405 / 162906 pack
[14:49:40] thread 2 is processing the 162406 / 162906 pack
[14:49:40] thread 1 is processing the 162407 / 162906 pack
[14:49:40] thread 2 is processing the 162408 / 162906 pack
[14:49:40] thread 1 is processing the 162409 / 162906 pack
[14:49:40] thread 2 is processing the 162410 / 162906 pack
[14:49:40] thread 1 is processing the 162411 / 162906 pack
[14:49:40] thread 2 is processing the 162412 / 162906 pack
[14:49:40] thread 1 is processing the 162413 / 162906 pack
[14:49:40] thread 2 is processing the 162414 / 162906 pack
[14:49:40] thread 1 is processing the 162415 / 162906 pack
[14:49:40] thread 2 is processing the 162416 / 162906 pack
[14:49:40] thread 1 is processing the 162417 / 162906 pack
[14:49:40] thread 2 is processing the 162418 / 162906 pack
[14:49:40] thread 1 is processing the 162419 / 162906 pack
[14:49:40] thread 2 is processing the 162420 / 162906 pack
[14:49:40] thread 1 is processing the 162421 / 162906 pack
[14:49:40] thread 2 is processing the 162422 / 162906 pack
[14:49:40] thread 1 is processing the 162423 / 162906 pack
[14:49:40] thread 2 is processing the 162424 / 162906 pack
[14:49:40] thread 1 is processing the 162425 / 162906 pack
[14:49:40] thread 2 is processing the 162426 / 162906 pack
[14:49:40] thread 1 is processing the 162427 / 162906 pack
[14:49:40] thread 2 is processing the 162428 / 162906 pack
[14:49:40] thread 1 is processing the 162429 / 162906 pack
[14:49:40] thread 2 is processing the 162430 / 162906 pack
[14:49:40] thread 1 is processing the 162431 / 162906 pack
[14:49:40] thread 2 is processing the 162432 / 162906 pack
[14:49:40] thread 1 is processing the 162433 / 162906 pack
[14:49:40] thread 2 is processing the 162434 / 162906 pack
[14:49:40] thread 1 is processing the 162435 / 162906 pack
[14:49:40] thread 2 is processing the 162436 / 162906 pack
[14:49:40] thread 1 is processing the 162437 / 162906 pack
[14:49:40] thread 2 is processing the 162438 / 162906 pack
[14:49:40] thread 1 is processing the 162439 / 162906 pack
[14:49:40] thread 2 is processing the 162440 / 162906 pack
[14:49:40] thread 1 is processing the 162441 / 162906 pack
[14:49:40] thread 2 is processing the 162442 / 162906 pack
[14:49:40] thread 1 is processing the 162443 / 162906 pack
[14:49:40] thread 2 is processing the 162444 / 162906 pack
[14:49:40] thread 1 is processing the 162445 / 162906 pack
[14:49:40] thread 2 is processing the 162446 / 162906 pack
[14:49:40] thread 1 is processing the 162447 / 162906 pack
[14:49:40] thread 2 is processing the 162448 / 162906 pack
[14:49:40] thread 1 is processing the 162449 / 162906 pack
[14:49:40] thread 2 is processing the 162450 / 162906 pack
[14:49:40] thread 1 is processing the 162451 / 162906 pack
[14:49:40] thread 2 is processing the 162452 / 162906 pack
[14:49:40] thread 1 is processing the 162453 / 162906 pack
[14:49:40] thread 2 is processing the 162454 / 162906 pack
[14:49:40] thread 1 is processing the 162455 / 162906 pack
[14:49:40] thread 2 is processing the 162456 / 162906 pack
[14:49:40] thread 1 is processing the 162457 / 162906 pack
[14:49:40] thread 2 is processing the 162458 / 162906 pack
[14:49:40] thread 1 is processing the 162459 / 162906 pack
[14:49:40] thread 2 is processing the 162460 / 162906 pack
[14:49:40] thread 1 is processing the 162461 / 162906 pack
[14:49:40] thread 2 is processing the 162462 / 162906 pack
[14:49:40] thread 1 is processing the 162463 / 162906 pack
[14:49:40] thread 2 is processing the 162464 / 162906 pack
[14:49:40] thread 1 is processing the 162465 / 162906 pack
[14:49:40] thread 2 is processing the 162466 / 162906 pack
[14:49:40] thread 1 is processing the 162467 / 162906 pack
[14:49:40] thread 2 is processing the 162468 / 162906 pack
[14:49:40] thread 1 is processing the 162469 / 162906 pack
[14:49:40] thread 2 is processing the 162470 / 162906 pack
[14:49:40] thread 1 is processing the 162471 / 162906 pack
[14:49:40] thread 2 is processing the 162472 / 162906 pack
[14:49:40] thread 1 is processing the 162473 / 162906 pack
[14:49:40] thread 2 is processing the 162474 / 162906 pack
[14:49:40] thread 1 is processing the 162475 / 162906 pack
[14:49:40] thread 2 is processing the 162476 / 162906 pack
[14:49:40] thread 1 is processing the 162477 / 162906 pack
[14:49:40] thread 2 is processing the 162478 / 162906 pack
[14:49:40] thread 1 is processing the 162479 / 162906 pack
[14:49:40] thread 2 is processing the 162480 / 162906 pack
[14:49:40] thread 1 is processing the 162481 / 162906 pack
[14:49:40] thread 2 is processing the 162482 / 162906 pack
[14:49:40] thread 1 is processing the 162483 / 162906 pack
[14:49:40] thread 2 is processing the 162484 / 162906 pack
[14:49:40] thread 1 is processing the 162485 / 162906 pack
[14:49:40] thread 2 is processing the 162486 / 162906 pack
[14:49:40] thread 1 is processing the 162487 / 162906 pack
[14:49:40] thread 2 is processing the 162488 / 162906 pack
[14:49:41] thread 1 is processing the 162489 / 162906 pack
[14:49:41] thread 2 is processing the 162490 / 162906 pack
[14:49:41] thread 1 is processing the 162491 / 162906 pack
[14:49:41] thread 2 is processing the 162492 / 162906 pack
[14:49:41] thread 1 is processing the 162493 / 162906 pack
[14:49:41] thread 2 is processing the 162494 / 162906 pack
[14:49:41] thread 1 is processing the 162495 / 162906 pack
[14:49:41] thread 2 is processing the 162496 / 162906 pack
[14:49:41] thread 1 is processing the 162497 / 162906 pack
[14:49:41] thread 2 is processing the 162498 / 162906 pack
[14:49:41] thread 1 is processing the 162499 / 162906 pack
[14:49:41] thread 2 is processing the 162500 / 162906 pack
[14:49:41] thread 1 is processing the 162501 / 162906 pack
[14:49:41] thread 2 is processing the 162502 / 162906 pack
[14:49:41] thread 1 is processing the 162503 / 162906 pack
[14:49:41] thread 2 is processing the 162504 / 162906 pack
[14:49:41] thread 1 is processing the 162505 / 162906 pack
[14:49:41] thread 2 is processing the 162506 / 162906 pack
[14:49:41] thread 1 is processing the 162507 / 162906 pack
[14:49:41] thread 2 is processing the 162508 / 162906 pack
[14:49:41] thread 2 is processing the 162509 / 162906 pack
[14:49:41] thread 1 is processing the 162510 / 162906 pack
[14:49:41] thread 2 is processing the 162511 / 162906 pack
[14:49:41] thread 1 is processing the 162512 / 162906 pack
[14:49:41] thread 2 is processing the 162513 / 162906 pack
[14:49:41] thread 1 is processing the 162514 / 162906 pack
[14:49:41] thread 2 is processing the 162515 / 162906 pack
[14:49:41] thread 1 is processing the 162516 / 162906 pack
[14:49:41] thread 2 is processing the 162517 / 162906 pack
[14:49:41] thread 1 is processing the 162518 / 162906 pack
[14:49:41] thread 2 is processing the 162519 / 162906 pack
[14:49:41] thread 1 is processing the 162520 / 162906 pack
[14:49:41] thread 2 is processing the 162521 / 162906 pack
[14:49:41] thread 1 is processing the 162522 / 162906 pack
[14:49:41] thread 2 is processing the 162523 / 162906 pack
[14:49:41] thread 1 is processing the 162524 / 162906 pack
[14:49:41] thread 2 is processing the 162525 / 162906 pack
[14:49:41] thread 1 is processing the 162526 / 162906 pack
[14:49:41] thread 2 is processing the 162527 / 162906 pack
[14:49:41] thread 1 is processing the 162528 / 162906 pack
[14:49:41] thread 2 is processing the 162529 / 162906 pack
[14:49:41] thread 1 is processing the 162530 / 162906 pack
[14:49:41] thread 2 is processing the 162531 / 162906 pack
[14:49:41] thread 1 is processing the 162532 / 162906 pack
[14:49:41] thread 2 is processing the 162533 / 162906 pack
[14:49:41] thread 1 is processing the 162534 / 162906 pack
[14:49:41] thread 2 is processing the 162535 / 162906 pack
[14:49:41] thread 1 is processing the 162536 / 162906 pack
[14:49:41] thread 2 is processing the 162537 / 162906 pack
[14:49:41] thread 1 is processing the 162538 / 162906 pack
[14:49:41] thread 2 is processing the 162539 / 162906 pack
[14:49:41] thread 1 is processing the 162540 / 162906 pack
[14:49:41] thread 2 is processing the 162541 / 162906 pack
[14:49:41] thread 1 is processing the 162542 / 162906 pack
[14:49:41] thread 2 is processing the 162543 / 162906 pack
[14:49:41] thread 1 is processing the 162544 / 162906 pack
[14:49:41] thread 2 is processing the 162545 / 162906 pack
[14:49:41] thread 1 is processing the 162546 / 162906 pack
[14:49:41] thread 2 is processing the 162547 / 162906 pack
[14:49:41] thread 1 is processing the 162548 / 162906 pack
[14:49:41] thread 2 is processing the 162549 / 162906 pack
[14:49:41] thread 1 is processing the 162550 / 162906 pack
[14:49:41] thread 2 is processing the 162551 / 162906 pack
[14:49:41] thread 1 is processing the 162552 / 162906 pack
[14:49:41] thread 2 is processing the 162553 / 162906 pack
[14:49:41] thread 1 is processing the 162554 / 162906 pack
[14:49:41] thread 2 is processing the 162555 / 162906 pack
[14:49:41] thread 1 is processing the 162556 / 162906 pack
[14:49:41] thread 2 is processing the 162557 / 162906 pack
[14:49:41] thread 1 is processing the 162558 / 162906 pack
[14:49:41] thread 2 is processing the 162559 / 162906 pack
[14:49:41] thread 1 is processing the 162560 / 162906 pack
[14:49:41] thread 2 is processing the 162561 / 162906 pack
[14:49:41] thread 1 is processing the 162562 / 162906 pack
[14:49:41] thread 2 is processing the 162563 / 162906 pack
[14:49:41] thread 1 is processing the 162564 / 162906 pack
[14:49:41] thread 2 is processing the 162565 / 162906 pack
[14:49:41] thread 1 is processing the 162566 / 162906 pack
[14:49:41] thread 2 is processing the 162567 / 162906 pack
[14:49:41] thread 1 is processing the 162568 / 162906 pack
[14:49:41] thread 2 is processing the 162569 / 162906 pack
[14:49:41] thread 1 is processing the 162570 / 162906 pack
[14:49:41] thread 2 is processing the 162571 / 162906 pack
[14:49:41] thread 1 is processing the 162572 / 162906 pack
[14:49:41] thread 2 is processing the 162573 / 162906 pack
[14:49:41] thread 1 is processing the 162574 / 162906 pack
[14:49:41] thread 2 is processing the 162575 / 162906 pack
[14:49:41] thread 1 is processing the 162576 / 162906 pack
[14:49:42] thread 2 is processing the 162577 / 162906 pack
[14:49:42] thread 1 is processing the 162578 / 162906 pack
[14:49:42] thread 2 is processing the 162579 / 162906 pack
[14:49:42] thread 1 is processing the 162580 / 162906 pack
[14:49:42] thread 2 is processing the 162581 / 162906 pack
[14:49:42] thread 1 is processing the 162582 / 162906 pack
[14:49:42] thread 2 is processing the 162583 / 162906 pack
[14:49:42] thread 1 is processing the 162584 / 162906 pack
[14:49:42] thread 2 is processing the 162585 / 162906 pack
[14:49:42] thread 1 is processing the 162586 / 162906 pack
[14:49:42] thread 2 is processing the 162587 / 162906 pack
[14:49:42] thread 1 is processing the 162588 / 162906 pack
[14:49:42] thread 2 is processing the 162589 / 162906 pack
[14:49:42] thread 1 is processing the 162590 / 162906 pack
[14:49:42] thread 2 is processing the 162591 / 162906 pack
[14:49:42] thread 1 is processing the 162592 / 162906 pack
[14:49:42] thread 2 is processing the 162593 / 162906 pack
[14:49:42] thread 1 is processing the 162594 / 162906 pack
[14:49:42] thread 2 is processing the 162595 / 162906 pack
[14:49:42] thread 1 is processing the 162596 / 162906 pack
[14:49:42] thread 2 is processing the 162597 / 162906 pack
[14:49:42] thread 1 is processing the 162598 / 162906 pack
[14:49:42] thread 2 is processing the 162599 / 162906 pack
[14:49:42] thread 1 is processing the 162600 / 162906 pack
[14:49:42] thread 2 is processing the 162601 / 162906 pack
[14:49:42] thread 1 is processing the 162602 / 162906 pack
[14:49:42] thread 2 is processing the 162603 / 162906 pack
[14:49:42] thread 1 is processing the 162604 / 162906 pack
[14:49:42] thread 2 is processing the 162605 / 162906 pack
[14:49:42] thread 1 is processing the 162606 / 162906 pack
[14:49:42] thread 2 is processing the 162607 / 162906 pack
[14:49:42] thread 1 is processing the 162608 / 162906 pack
[14:49:42] thread 2 is processing the 162609 / 162906 pack
[14:49:42] thread 1 is processing the 162610 / 162906 pack
[14:49:42] thread 2 is processing the 162611 / 162906 pack
[14:49:42] thread 1 is processing the 162612 / 162906 pack
[14:49:42] thread 2 is processing the 162613 / 162906 pack
[14:49:42] thread 1 is processing the 162614 / 162906 pack
[14:49:42] thread 2 is processing the 162615 / 162906 pack
[14:49:42] thread 1 is processing the 162616 / 162906 pack
[14:49:42] thread 2 is processing the 162617 / 162906 pack
[14:49:42] thread 1 is processing the 162618 / 162906 pack
[14:49:42] thread 2 is processing the 162619 / 162906 pack
[14:49:42] thread 1 is processing the 162620 / 162906 pack
[14:49:42] thread 2 is processing the 162621 / 162906 pack
[14:49:42] thread 1 is processing the 162622 / 162906 pack
[14:49:42] thread 2 is processing the 162623 / 162906 pack
[14:49:42] thread 1 is processing the 162624 / 162906 pack
[14:49:42] thread 2 is processing the 162625 / 162906 pack
[14:49:42] thread 1 is processing the 162626 / 162906 pack
[14:49:42] thread 2 is processing the 162627 / 162906 pack
[14:49:42] thread 1 is processing the 162628 / 162906 pack
[14:49:42] thread 2 is processing the 162629 / 162906 pack
[14:49:42] thread 1 is processing the 162630 / 162906 pack
[14:49:42] thread 2 is processing the 162631 / 162906 pack
[14:49:42] thread 1 is processing the 162632 / 162906 pack
[14:49:42] thread 2 is processing the 162633 / 162906 pack
[14:49:42] thread 1 is processing the 162634 / 162906 pack
[14:49:42] thread 2 is processing the 162635 / 162906 pack
[14:49:42] thread 1 is processing the 162636 / 162906 pack
[14:49:42] thread 2 is processing the 162637 / 162906 pack
[14:49:42] thread 1 is processing the 162638 / 162906 pack
[14:49:42] thread 2 is processing the 162639 / 162906 pack
[14:49:42] thread 1 is processing the 162640 / 162906 pack
[14:49:42] thread 2 is processing the 162641 / 162906 pack
[14:49:42] thread 1 is processing the 162642 / 162906 pack
[14:49:42] thread 2 is processing the 162643 / 162906 pack
[14:49:42] thread 1 is processing the 162644 / 162906 pack
[14:49:42] thread 2 is processing the 162645 / 162906 pack
[14:49:42] thread 1 is processing the 162646 / 162906 pack
[14:49:42] thread 2 is processing the 162647 / 162906 pack
[14:49:42] thread 1 is processing the 162648 / 162906 pack
[14:49:42] thread 2 is processing the 162649 / 162906 pack
[14:49:42] thread 1 is processing the 162650 / 162906 pack
[14:49:42] thread 2 is processing the 162651 / 162906 pack
[14:49:42] thread 1 is processing the 162652 / 162906 pack
[14:49:42] thread 2 is processing the 162653 / 162906 pack
[14:49:42] thread 1 is processing the 162654 / 162906 pack
[14:49:42] thread 2 is processing the 162655 / 162906 pack
[14:49:42] thread 1 is processing the 162656 / 162906 pack
[14:49:42] thread 2 is processing the 162657 / 162906 pack
[14:49:42] thread 1 is processing the 162658 / 162906 pack
[14:49:42] thread 2 is processing the 162659 / 162906 pack
[14:49:42] thread 1 is processing the 162660 / 162906 pack
[14:49:42] thread 2 is processing the 162661 / 162906 pack
[14:49:43] thread 1 is processing the 162662 / 162906 pack
[14:49:43] thread 2 is processing the 162662 / 162906 pack
[14:49:43] thread 2 is processing the 162664 / 162906 pack
[14:49:43] thread 1 is processing the 162665 / 162906 pack
[14:49:43] thread 2 is processing the 162666 / 162906 pack
[14:49:43] thread 1 is processing the 162667 / 162906 pack
[14:49:43] thread 2 is processing the 162668 / 162906 pack
[14:49:43] thread 1 is processing the 162669 / 162906 pack
[14:49:43] thread 2 is processing the 162670 / 162906 pack
[14:49:43] thread 1 is processing the 162671 / 162906 pack
[14:49:43] thread 2 is processing the 162672 / 162906 pack
[14:49:43] thread 1 is processing the 162673 / 162906 pack
[14:49:43] thread 2 is processing the 162674 / 162906 pack
[14:49:43] thread 1 is processing the 162675 / 162906 pack
[14:49:43] thread 2 is processing the 162676 / 162906 pack
[14:49:43] thread 1 is processing the 162677 / 162906 pack
[14:49:43] thread 2 is processing the 162678 / 162906 pack
[14:49:43] thread 1 is processing the 162679 / 162906 pack
[14:49:43] thread 2 is processing the 162680 / 162906 pack
[14:49:43] thread 1 is processing the 162681 / 162906 pack
[14:49:43] thread 2 is processing the 162682 / 162906 pack
[14:49:43] thread 1 is processing the 162683 / 162906 pack
[14:49:43] thread 2 is processing the 162684 / 162906 pack
[14:49:43] thread 1 is processing the 162685 / 162906 pack
[14:49:43] thread 2 is processing the 162686 / 162906 pack
[14:49:43] thread 1 is processing the 162687 / 162906 pack
[14:49:43] thread 2 is processing the 162688 / 162906 pack
[14:49:43] thread 1 is processing the 162689 / 162906 pack
[14:49:43] thread 2 is processing the 162690 / 162906 pack
[14:49:43] thread 1 is processing the 162691 / 162906 pack
[14:49:43] thread 2 is processing the 162692 / 162906 pack
[14:49:43] thread 1 is processing the 162693 / 162906 pack
[14:49:43] thread 2 is processing the 162694 / 162906 pack
[14:49:43] thread 1 is processing the 162695 / 162906 pack
[14:49:43] thread 2 is processing the 162696 / 162906 pack
[14:49:43] thread 1 is processing the 162697 / 162906 pack
[14:49:43] thread 2 is processing the 162698 / 162906 pack
[14:49:43] thread 1 is processing the 162699 / 162906 pack
[14:49:43] thread 2 is processing the 162700 / 162906 pack
[14:49:43] thread 1 is processing the 162701 / 162906 pack
[14:49:43] thread 2 is processing the 162702 / 162906 pack
[14:49:43] thread 1 is processing the 162703 / 162906 pack
[14:49:43] thread 2 is processing the 162704 / 162906 pack
[14:49:43] thread 1 is processing the 162705 / 162906 pack
[14:49:43] thread 2 is processing the 162706 / 162906 pack
[14:49:43] thread 1 is processing the 162707 / 162906 pack
[14:49:43] thread 2 is processing the 162708 / 162906 pack
[14:49:43] thread 1 is processing the 162709 / 162906 pack
[14:49:43] thread 2 is processing the 162710 / 162906 pack
[14:49:43] thread 1 is processing the 162711 / 162906 pack
[14:49:43] thread 2 is processing the 162712 / 162906 pack
[14:49:43] thread 1 is processing the 162713 / 162906 pack
[14:49:43] thread 2 is processing the 162714 / 162906 pack
[14:49:43] thread 1 is processing the 162715 / 162906 pack
[14:49:43] thread 2 is processing the 162716 / 162906 pack
[14:49:43] thread 1 is processing the 162717 / 162906 pack
[14:49:43] thread 2 is processing the 162718 / 162906 pack
[14:49:43] thread 1 is processing the 162719 / 162906 pack
[14:49:43] thread 2 is processing the 162720 / 162906 pack
[14:49:43] thread 1 is processing the 162721 / 162906 pack
[14:49:43] thread 2 is processing the 162722 / 162906 pack
[14:49:43] thread 1 is processing the 162723 / 162906 pack
[14:49:43] thread 2 is processing the 162724 / 162906 pack
[14:49:43] thread 1 is processing the 162725 / 162906 pack
[14:49:43] thread 2 is processing the 162726 / 162906 pack
[14:49:43] thread 1 is processing the 162727 / 162906 pack
[14:49:43] thread 2 is processing the 162728 / 162906 pack
[14:49:43] thread 1 is processing the 162729 / 162906 pack
[14:49:43] thread 2 is processing the 162730 / 162906 pack
[14:49:43] thread 1 is processing the 162731 / 162906 pack
[14:49:43] thread 2 is processing the 162732 / 162906 pack
[14:49:43] thread 1 is processing the 162733 / 162906 pack
[14:49:43] thread 2 is processing the 162734 / 162906 pack
[14:49:43] thread 1 is processing the 162735 / 162906 pack
[14:49:43] thread 2 is processing the 162736 / 162906 pack
[14:49:43] thread 1 is processing the 162737 / 162906 pack
[14:49:43] thread 2 is processing the 162738 / 162906 pack
[14:49:43] thread 1 is processing the 162739 / 162906 pack
[14:49:43] thread 2 is processing the 162740 / 162906 pack
[14:49:43] thread 1 is processing the 162741 / 162906 pack
[14:49:43] thread 2 is processing the 162742 / 162906 pack
[14:49:43] thread 1 is processing the 162743 / 162906 pack
[14:49:43] thread 2 is processing the 162744 / 162906 pack
[14:49:43] thread 1 is processing the 162745 / 162906 pack
[14:49:43] thread 2 is processing the 162746 / 162906 pack
[14:49:43] thread 1 is processing the 162747 / 162906 pack
[14:49:43] thread 2 is processing the 162748 / 162906 pack
[14:49:44] thread 1 is processing the 162749 / 162906 pack
[14:49:44] thread 2 is processing the 162750 / 162906 pack
[14:49:44] thread 1 is processing the 162751 / 162906 pack
[14:49:44] thread 2 is processing the 162752 / 162906 pack
[14:49:44] thread 1 is processing the 162753 / 162906 pack
[14:49:44] thread 2 is processing the 162754 / 162906 pack
[14:49:44] thread 1 is processing the 162755 / 162906 pack
[14:49:44] thread 2 is processing the 162756 / 162906 pack
[14:49:44] thread 1 is processing the 162757 / 162906 pack
[14:49:44] thread 2 is processing the 162758 / 162906 pack
[14:49:44] thread 2 is processing the 162759 / 162906 pack
[14:49:44] thread 1 is processing the 162760 / 162906 pack
[14:49:44] thread 2 is processing the 162761 / 162906 pack
[14:49:44] thread 1 is processing the 162762 / 162906 pack
[14:49:44] thread 2 is processing the 162763 / 162906 pack
[14:49:44] thread 1 is processing the 162764 / 162906 pack
[14:49:44] thread 2 is processing the 162765 / 162906 pack
[14:49:44] thread 1 is processing the 162766 / 162906 pack
[14:49:44] thread 2 is processing the 162767 / 162906 pack
[14:49:44] thread 1 is processing the 162768 / 162906 pack
[14:49:44] thread 2 is processing the 162769 / 162906 pack
[14:49:44] thread 1 is processing the 162770 / 162906 pack
[14:49:44] thread 2 is processing the 162771 / 162906 pack
[14:49:44] thread 1 is processing the 162772 / 162906 pack
[14:49:44] thread 2 is processing the 162773 / 162906 pack
[14:49:44] thread 1 is processing the 162774 / 162906 pack
[14:49:44] thread 2 is processing the 162775 / 162906 pack
[14:49:44] thread 1 is processing the 162776 / 162906 pack
[14:49:44] thread 2 is processing the 162777 / 162906 pack
[14:49:44] thread 1 is processing the 162778 / 162906 pack
[14:49:44] thread 2 is processing the 162779 / 162906 pack
[14:49:44] thread 1 is processing the 162780 / 162906 pack
[14:49:44] thread 2 is processing the 162781 / 162906 pack
[14:49:44] thread 1 is processing the 162782 / 162906 pack
[14:49:44] thread 2 is processing the 162783 / 162906 pack
[14:49:44] thread 1 is processing the 162784 / 162906 pack
[14:49:44] thread 2 is processing the 162785 / 162906 pack
[14:49:44] thread 1 is processing the 162786 / 162906 pack
[14:49:44] thread 2 is processing the 162787 / 162906 pack
[14:49:44] thread 1 is processing the 162788 / 162906 pack
[14:49:44] thread 2 is processing the 162789 / 162906 pack
[14:49:44] thread 1 is processing the 162790 / 162906 pack
[14:49:44] thread 2 is processing the 162791 / 162906 pack
[14:49:44] thread 1 is processing the 162792 / 162906 pack
[14:49:44] thread 2 is processing the 162793 / 162906 pack
[14:49:44] thread 1 is processing the 162794 / 162906 pack
[14:49:44] thread 2 is processing the 162795 / 162906 pack
[14:49:44] thread 1 is processing the 162796 / 162906 pack
[14:49:44] thread 2 is processing the 162797 / 162906 pack
[14:49:44] thread 1 is processing the 162798 / 162906 pack
[14:49:44] thread 2 is processing the 162799 / 162906 pack
[14:49:44] thread 1 is processing the 162800 / 162906 pack
[14:49:44] thread 2 is processing the 162801 / 162906 pack
[14:49:44] thread 1 is processing the 162802 / 162906 pack
[14:49:44] thread 2 is processing the 162803 / 162906 pack
[14:49:44] thread 1 is processing the 162804 / 162906 pack
[14:49:44] thread 2 is processing the 162805 / 162906 pack
[14:49:44] thread 1 is processing the 162806 / 162906 pack
[14:49:44] thread 2 is processing the 162807 / 162906 pack
[14:49:44] thread 1 is processing the 162808 / 162906 pack
[14:49:44] thread 2 is processing the 162809 / 162906 pack
[14:49:44] thread 1 is processing the 162810 / 162906 pack
[14:49:44] thread 2 is processing the 162811 / 162906 pack
[14:49:44] thread 1 is processing the 162812 / 162906 pack
[14:49:44] thread 2 is processing the 162813 / 162906 pack
[14:49:44] thread 1 is processing the 162814 / 162906 pack
[14:49:44] thread 2 is processing the 162815 / 162906 pack
[14:49:44] thread 1 is processing the 162816 / 162906 pack
[14:49:44] thread 2 is processing the 162817 / 162906 pack
[14:49:44] thread 1 is processing the 162818 / 162906 pack
[14:49:44] thread 2 is processing the 162819 / 162906 pack
[14:49:44] thread 1 is processing the 162820 / 162906 pack
[14:49:44] thread 2 is processing the 162821 / 162906 pack
[14:49:44] thread 1 is processing the 162822 / 162906 pack
[14:49:44] thread 2 is processing the 162823 / 162906 pack
[14:49:44] thread 1 is processing the 162824 / 162906 pack
[14:49:44] thread 2 is processing the 162825 / 162906 pack
[14:49:44] thread 1 is processing the 162826 / 162906 pack
[14:49:44] thread 2 is processing the 162827 / 162906 pack
[14:49:44] thread 1 is processing the 162828 / 162906 pack
[14:49:44] thread 2 is processing the 162829 / 162906 pack
[14:49:44] thread 1 is processing the 162830 / 162906 pack
[14:49:44] thread 2 is processing the 162831 / 162906 pack
[14:49:44] thread 1 is processing the 162832 / 162906 pack
[14:49:44] thread 2 is processing the 162833 / 162906 pack
[14:49:44] thread 1 is processing the 162834 / 162906 pack
[14:49:44] thread 2 is processing the 162835 / 162906 pack
[14:49:45] thread 1 is processing the 162836 / 162906 pack
[14:49:45] thread 2 is processing the 162837 / 162906 pack
[14:49:45] thread 1 is processing the 162838 / 162906 pack
[14:49:45] thread 2 is processing the 162839 / 162906 pack
[14:49:45] thread 1 is processing the 162840 / 162906 pack
[14:49:45] thread 2 is processing the 162841 / 162906 pack
[14:49:45] thread 1 is processing the 162842 / 162906 pack
[14:49:45] thread 2 is processing the 162843 / 162906 pack
[14:49:45] thread 1 is processing the 162844 / 162906 pack
[14:49:45] thread 2 is processing the 162845 / 162906 pack
[14:49:45] thread 1 is processing the 162846 / 162906 pack
[14:49:45] thread 2 is processing the 162847 / 162906 pack
[14:49:45] thread 1 is processing the 162848 / 162906 pack
[14:49:45] thread 2 is processing the 162849 / 162906 pack
[14:49:45] thread 1 is processing the 162850 / 162906 pack
[14:49:45] thread 2 is processing the 162851 / 162906 pack
[14:49:45] thread 1 is processing the 162852 / 162906 pack
[14:49:45] thread 2 is processing the 162853 / 162906 pack
[14:49:45] thread 1 is processing the 162854 / 162906 pack
[14:49:45] thread 2 is processing the 162855 / 162906 pack
[14:49:45] thread 1 is processing the 162856 / 162906 pack
[14:49:45] thread 2 is processing the 162857 / 162906 pack
[14:49:45] thread 1 is processing the 162858 / 162906 pack
[14:49:45] thread 2 is processing the 162859 / 162906 pack
[14:49:45] thread 1 is processing the 162860 / 162906 pack
[14:49:45] thread 2 is processing the 162861 / 162906 pack
[14:49:45] thread 1 is processing the 162862 / 162906 pack
[14:49:45] thread 2 is processing the 162863 / 162906 pack
[14:49:45] thread 1 is processing the 162864 / 162906 pack
[14:49:45] thread 2 is processing the 162865 / 162906 pack
[14:49:45] thread 1 is processing the 162866 / 162906 pack
[14:49:45] thread 2 is processing the 162867 / 162906 pack
[14:49:45] thread 1 is processing the 162868 / 162906 pack
[14:49:45] thread 2 is processing the 162869 / 162906 pack
[14:49:45] thread 1 is processing the 162870 / 162906 pack
[14:49:45] thread 2 is processing the 162871 / 162906 pack
[14:49:45] thread 1 is processing the 162872 / 162906 pack
[14:49:45] thread 2 is processing the 162873 / 162906 pack
[14:49:45] thread 1 is processing the 162874 / 162906 pack
[14:49:45] thread 2 is processing the 162875 / 162906 pack
[14:49:45] thread 1 is processing the 162876 / 162906 pack
[14:49:45] thread 2 is processing the 162877 / 162906 pack
[14:49:45] thread 1 is processing the 162878 / 162906 pack
[14:49:45] thread 2 is processing the 162879 / 162906 pack
[14:49:45] thread 1 is processing the 162880 / 162906 pack
[14:49:45] thread 2 is processing the 162881 / 162906 pack
[14:49:45] thread 1 is processing the 162882 / 162906 pack
[14:49:45] thread 2 is processing the 162883 / 162906 pack
[14:49:45] thread 1 is processing the 162884 / 162906 pack
[14:49:45] thread 2 is processing the 162885 / 162906 pack
[14:49:45] thread 1 is processing the 162886 / 162906 pack
[14:49:45] thread 2 is processing the 162887 / 162906 pack
[14:49:45] thread 1 is processing the 162888 / 162906 pack
[14:49:45] thread 2 is processing the 162889 / 162906 pack
[14:49:45] thread 1 is processing the 162890 / 162906 pack
[14:49:45] thread 2 is processing the 162891 / 162906 pack
[14:49:45] thread 1 is processing the 162892 / 162906 pack
[14:49:45] thread 2 is processing the 162893 / 162906 pack
[14:49:45] thread 1 is processing the 162894 / 162906 pack
[14:49:45] thread 2 is processing the 162895 / 162906 pack
[14:49:45] thread 1 is processing the 162896 / 162906 pack
[14:49:45] thread 2 is processing the 162897 / 162906 pack
[14:49:45] thread 1 is processing the 162898 / 162906 pack
[14:49:45] thread 2 is processing the 162899 / 162906 pack
[14:49:45] thread 1 is processing the 162900 / 162906 pack
[14:49:45] thread 2 is processing the 162901 / 162906 pack
[14:49:45] thread 1 is processing the 162902 / 162906 pack
[14:49:45] thread 2 is processing the 162903 / 162906 pack
[14:49:45] thread 1 is processing the 162904 / 162906 pack
[14:49:45] thread 2 is processing the 162905 / 162906 pack
[14:49:45] thread 2 data processing completed
[14:49:45] thread 2 finished
[14:49:45] thread 1 data processing completed
[14:49:45] thread 1 finished
[14:49:45] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/Eggs_QC_R1.fastq.gz writer finished
[14:49:45] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/Eggs_QC_R2.fastq.gz writer finished
[14:49:45] start to generate reports

Read1 before filtering:
total reads: 162905530
total bases: 24598735030
Q20 bases: 24117262223(98.0427%)
Q30 bases: 23297539445(94.7103%)

Read2 before filtering:
total reads: 162905530
total bases: 24598735030
Q20 bases: 23834884186(96.8948%)
Q30 bases: 22694729390(92.2597%)

Read1 after filtering:
total reads: 160628824
total bases: 23925292330
Q20 bases: 23508926707(98.2597%)
Q30 bases: 22738265195(95.0386%)

Read2 after filtering:
total reads: 160628824
total bases: 23923485245
Q20 bases: 23344799534(97.5811%)
Q30 bases: 22276660789(93.1163%)

Filtering result:
reads passed filter: 321257648
reads failed due to low quality: 4452040
reads failed due to too many N: 14612
reads failed due to too short: 86760
reads with adapter trimmed: 33180067
bases trimmed due to adapters: 658849630

Duplication rate: 5.39017%

Insert size peak (evaluated by paired-end reads): 213

Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[13:58:1] start to load data
[13:58:10] loaded 1M read pairs
[13:58:21] loaded 2M read pairs
[13:58:34] loaded 3M read pairs
[13:58:45] loaded 4M read pairs
[13:58:57] loaded 5M read pairs
[13:59:8] loaded 6M read pairs
[13:59:20] loaded 7M read pairs
[13:59:31] loaded 8M read pairs
[13:59:43] loaded 9M read pairs
[13:59:54] loaded 10M read pairs
[14:0:6] loaded 11M read pairs
[14:0:17] loaded 12M read pairs
[14:0:29] loaded 13M read pairs
[14:0:40] loaded 14M read pairs
[14:0:51] loaded 15M read pairs
[14:1:4] loaded 16M read pairs
[14:1:15] loaded 17M read pairs
[14:1:26] loaded 18M read pairs
[14:1:36] loaded 19M read pairs
[14:1:47] loaded 20M read pairs
[14:1:58] loaded 21M read pairs
[14:2:8] loaded 22M read pairs
[14:2:19] loaded 23M read pairs
[14:2:30] loaded 24M read pairs
[14:2:41] loaded 25M read pairs
[14:2:51] loaded 26M read pairs
[14:3:2] loaded 27M read pairs
[14:3:13] loaded 28M read pairs
[14:3:24] loaded 29M read pairs
[14:3:35] loaded 30M read pairs
[14:3:46] loaded 31M read pairs
[14:3:57] loaded 32M read pairs
[14:4:8] loaded 33M read pairs
[14:4:19] loaded 34M read pairs
[14:4:29] loaded 35M read pairs
[14:4:40] loaded 36M read pairs
[14:4:51] loaded 37M read pairs
[14:5:2] loaded 38M read pairs
[14:5:13] loaded 39M read pairs
[14:5:24] loaded 40M read pairs
[14:5:35] loaded 41M read pairs
[14:5:47] loaded 42M read pairs
[14:5:59] loaded 43M read pairs
[14:6:10] loaded 44M read pairs
[14:6:22] loaded 45M read pairs
[14:6:34] loaded 46M read pairs
[14:6:45] loaded 47M read pairs
[14:6:56] loaded 48M read pairs
[14:7:6] loaded 49M read pairs
[14:7:17] loaded 50M read pairs
[14:7:28] loaded 51M read pairs
[14:7:39] loaded 52M read pairs
[14:7:50] loaded 53M read pairs
[14:8:1] loaded 54M read pairs
[14:8:12] loaded 55M read pairs
[14:8:23] loaded 56M read pairs
[14:8:35] loaded 57M read pairs
[14:8:46] loaded 58M read pairs
[14:8:57] loaded 59M read pairs
[14:9:8] loaded 60M read pairs
[14:9:19] loaded 61M read pairs
[14:9:29] loaded 62M read pairs
[14:9:40] loaded 63M read pairs
[14:9:51] loaded 64M read pairs
[14:10:2] loaded 65M read pairs
[14:10:13] loaded 66M read pairs
[14:10:24] loaded 67M read pairs
[14:10:35] loaded 68M read pairs
[14:10:46] loaded 69M read pairs
[14:10:57] loaded 70M read pairs
[14:11:8] loaded 71M read pairs
[14:11:19] loaded 72M read pairs
[14:11:30] loaded 73M read pairs
[14:11:41] loaded 74M read pairs
[14:11:52] loaded 75M read pairs
[14:12:4] loaded 76M read pairs
[14:12:15] loaded 77M read pairs
[14:12:26] loaded 78M read pairs
[14:12:37] loaded 79M read pairs
[14:12:48] loaded 80M read pairs
[14:12:59] loaded 81M read pairs
[14:13:10] loaded 82M read pairs
[14:13:21] loaded 83M read pairs
[14:13:32] loaded 84M read pairs
[14:13:43] loaded 85M read pairs
[14:13:54] loaded 86M read pairs
[14:14:5] loaded 87M read pairs
[14:14:16] loaded 88M read pairs
[14:14:27] loaded 89M read pairs
[14:14:38] loaded 90M read pairs
[14:14:49] loaded 91M read pairs
[14:15:0] loaded 92M read pairs
[14:15:11] loaded 93M read pairs
[14:15:22] loaded 94M read pairs
Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[14:17:37] start to load data
[14:17:45] loaded 1M read pairs
[14:17:55] loaded 2M read pairs
[14:18:6] loaded 3M read pairs
[14:18:17] loaded 4M read pairs
[14:18:29] loaded 5M read pairs
[14:18:41] loaded 6M read pairs
[14:18:53] loaded 7M read pairs
[14:19:4] loaded 8M read pairs
[14:19:16] loaded 9M read pairs
[14:19:27] loaded 10M read pairs
[14:19:39] loaded 11M read pairs
[14:19:50] loaded 12M read pairs
[14:20:1] loaded 13M read pairs
[14:20:12] loaded 14M read pairs
[14:20:23] loaded 15M read pairs
[14:20:35] loaded 16M read pairs
[14:20:47] loaded 17M read pairs
[14:20:58] loaded 18M read pairs
[14:21:9] loaded 19M read pairs
[14:21:21] loaded 20M read pairs
[14:21:33] loaded 21M read pairs
[14:21:45] loaded 22M read pairs
[14:21:57] loaded 23M read pairs
[14:22:8] loaded 24M read pairs
[14:22:20] loaded 25M read pairs
[14:22:32] loaded 26M read pairs
[14:22:43] loaded 27M read pairs
[14:22:55] loaded 28M read pairs
[14:23:6] loaded 29M read pairs
[14:23:18] loaded 30M read pairs
[14:23:29] loaded 31M read pairs
[14:23:40] loaded 32M read pairs
[14:23:51] loaded 33M read pairs
[14:24:1] loaded 34M read pairs
[14:24:12] loaded 35M read pairs
[14:24:23] loaded 36M read pairs
[14:24:34] loaded 37M read pairs
[14:24:45] loaded 38M read pairs
[14:24:56] loaded 39M read pairs
[14:25:8] loaded 40M read pairs
[14:25:20] loaded 41M read pairs
[14:25:31] loaded 42M read pairs
[14:25:43] loaded 43M read pairs
[14:25:54] loaded 44M read pairs
[14:26:6] loaded 45M read pairs
[14:26:17] loaded 46M read pairs
[14:26:28] loaded 47M read pairs
[14:26:39] loaded 48M read pairs
[14:26:50] loaded 49M read pairs
[14:27:1] loaded 50M read pairs
[14:27:12] loaded 51M read pairs
[14:27:23] loaded 52M read pairs
[14:27:35] loaded 53M read pairs
[14:27:46] loaded 54M read pairs
[14:27:57] loaded 55M read pairs
[14:28:9] loaded 56M read pairs
[14:28:20] loaded 57M read pairs
[14:28:33] loaded 58M read pairs
[14:28:45] loaded 59M read pairs
[14:28:56] loaded 60M read pairs
[14:29:8] loaded 61M read pairs
[14:29:20] loaded 62M read pairs
[14:29:32] loaded 63M read pairs
[14:29:43] loaded 64M read pairs
[14:29:55] loaded 65M read pairs
[14:30:6] loaded 66M read pairs
[14:30:18] loaded 67M read pairs
[14:30:29] loaded 68M read pairs
[14:30:41] loaded 69M read pairs
[14:30:52] loaded 70M read pairs
[14:31:4] loaded 71M read pairs
[14:31:15] loaded 72M read pairs
[14:31:27] loaded 73M read pairs
[14:31:38] loaded 74M read pairs
[14:31:50] loaded 75M read pairs
[14:32:1] loaded 76M read pairs
[14:32:12] loaded 77M read pairs
[14:32:23] loaded 78M read pairs
[14:32:34] loaded 79M read pairs
[14:32:46] loaded 80M read pairs
[14:32:58] loaded 81M read pairs
[14:33:9] loaded 82M read pairs
[14:33:21] loaded 83M read pairs
[14:33:33] loaded 84M read pairs
[14:33:45] loaded 85M read pairs
[14:33:57] loaded 86M read pairs
[14:34:8] loaded 87M read pairs
[14:34:20] loaded 88M read pairs
[14:34:31] loaded 89M read pairs
[14:34:43] loaded 90M read pairs
[14:34:55] loaded 91M read pairs
[14:35:6] loaded 92M read pairs
[14:35:18] loaded 93M read pairs
[14:35:30] loaded 94M read pairs
[14:35:41] loaded 95M read pairs
[14:35:53] loaded 96M read pairs
[14:36:5] loaded 97M read pairs
[14:36:16] loaded 98M read pairs
[14:36:28] loaded 99M read pairs
[14:36:40] loaded 100M read pairs
[14:36:52] loaded 101M read pairs
[14:37:4] loaded 102M read pairs
[14:37:15] loaded 103M read pairs
[14:37:26] loaded 104M read pairs
[14:37:39] loaded 105M read pairs
[14:37:50] loaded 106M read pairs
[14:38:2] loaded 107M read pairs
[14:38:14] loaded 108M read pairs
[14:38:26] loaded 109M read pairs
[14:38:38] loaded 110M read pairs
[14:38:49] loaded 111M read pairs
[14:39:1] loaded 112M read pairs
[14:39:12] loaded 113M read pairs
[14:39:25] loaded 114M read pairs
[14:39:36] loaded 115M read pairs
[14:39:47] loaded 116M read pairs
[14:39:59] loaded 117M read pairs
[14:40:11] loaded 118M read pairs
[14:40:23] loaded 119M read pairs
[14:40:34] loaded 120M read pairs
[14:40:46] loaded 121M read pairs
[14:40:57] loaded 122M read pairs
[14:41:9] loaded 123M read pairs
[14:41:20] loaded 124M read pairs
[14:41:32] loaded 125M read pairs
[14:41:44] loaded 126M read pairs
[14:41:56] loaded 127M read pairs
[14:42:7] loaded 128M read pairs
[14:42:19] loaded 129M read pairs
[14:42:31] loaded 130M read pairs
[14:42:42] loaded 131M read pairs
[14:42:54] loaded 132M read pairs
[14:43:6] loaded 133M read pairs
[14:43:17] loaded 134M read pairs
[14:43:29] loaded 135M read pairs
[14:43:41] loaded 136M read pairs
[14:43:53] loaded 137M read pairs
[14:44:4] loaded 138M read pairs
[14:44:15] loaded 139M read pairs
[14:44:27] loaded 140M read pairs
[14:44:39] loaded 141M read pairs
[14:44:50] loaded 142M read pairs
[14:45:1] loaded 143M read pairs
[14:45:13] loaded 144M read pairs
[14:45:24] loaded 145M read pairs
[14:45:35] loaded 146M read pairs
[14:45:47] loaded 147M read pairs
[14:45:58] loaded 148M read pairs
[14:46:10] loaded 149M read pairs
[14:46:21] loaded 150M read pairs
[14:46:32] loaded 151M read pairs
[14:46:43] loaded 152M read pairs
[14:46:54] loaded 153M read pairs
[14:47:6] loaded 154M read pairs
[14:47:17] loaded 155M read pairs
[14:47:28] loaded 156M read pairs
[14:47:39] loaded 157M read pairs
[14:47:44] all reads loaded, start to monitor thread status
[14:47:44] thread 1 is processing the 156921 / 157421 pack
[14:47:44] thread 2 is processing the 156922 / 157421 pack
[14:47:44] thread 1 is processing the 156923 / 157421 pack
[14:47:44] thread 2 is processing the 156924 / 157421 pack
[14:47:44] thread 1 is processing the 156925 / 157421 pack
[14:47:44] thread 2 is processing the 156926 / 157421 pack
[14:47:44] thread 1 is processing the 156927 / 157421 pack
[14:47:44] thread 2 is processing the 156928 / 157421 pack
[14:47:44] thread 1 is processing the 156929 / 157421 pack
[14:47:44] thread 2 is processing the 156930 / 157421 pack
[14:47:44] thread 1 is processing the 156931 / 157421 pack
[14:47:44] thread 2 is processing the 156931 / 157421 pack
[14:47:44] thread 2 is processing the 156933 / 157421 pack
[14:47:44] thread 1 is processing the 156934 / 157421 pack
[14:47:44] thread 1 is processing the 156935 / 157421 pack
[14:47:44] thread 2 is processing the 156936 / 157421 pack
[14:47:44] thread 2 is processing the 156937 / 157421 pack
[14:47:44] thread 1 is processing the 156937 / 157421 pack
[14:47:44] thread 2 is processing the 156939 / 157421 pack
[14:47:44] thread 1 is processing the 156940 / 157421 pack
[14:47:44] thread 2 is processing the 156941 / 157421 pack
[14:47:44] thread 1 is processing the 156942 / 157421 pack
[14:47:44] thread 2 is processing the 156943 / 157421 pack
[14:47:44] thread 1 is processing the 156944 / 157421 pack
[14:47:44] thread 1 is processing the 156945 / 157421 pack
[14:47:44] thread 2 is processing the 156946 / 157421 pack
[14:47:44] thread 1 is processing the 156947 / 157421 pack
[14:47:44] thread 2 is processing the 156947 / 157421 pack
[14:47:44] thread 1 is processing the 156949 / 157421 pack
[14:47:44] thread 2 is processing the 156950 / 157421 pack
[14:47:44] thread 1 is processing the 156951 / 157421 pack
[14:47:44] thread 2 is processing the 156952 / 157421 pack
[14:47:44] thread 1 is processing the 156953 / 157421 pack
[14:47:44] thread 2 is processing the 156954 / 157421 pack
[14:47:44] thread 1 is processing the 156955 / 157421 pack
[14:47:44] thread 2 is processing the 156956 / 157421 pack
[14:47:44] thread 1 is processing the 156957 / 157421 pack
[14:47:44] thread 2 is processing the 156957 / 157421 pack
[14:47:44] thread 2 is processing the 156959 / 157421 pack
[14:47:44] thread 1 is processing the 156959 / 157421 pack
[14:47:44] thread 1 is processing the 156961 / 157421 pack
[14:47:44] thread 2 is processing the 156962 / 157421 pack
[14:47:44] thread 1 is processing the 156963 / 157421 pack
[14:47:44] thread 2 is processing the 156964 / 157421 pack
[14:47:44] thread 2 is processing the 156965 / 157421 pack
[14:47:44] thread 1 is processing the 156965 / 157421 pack
[14:47:44] thread 1 is processing the 156967 / 157421 pack
[14:47:44] thread 2 is processing the 156968 / 157421 pack
[14:47:44] thread 1 is processing the 156969 / 157421 pack
[14:47:44] thread 2 is processing the 156970 / 157421 pack
[14:47:44] thread 1 is processing the 156971 / 157421 pack
[14:47:44] thread 2 is processing the 156972 / 157421 pack
[14:47:44] thread 1 is processing the 156973 / 157421 pack
[14:47:44] thread 2 is processing the 156974 / 157421 pack
[14:47:44] thread 1 is processing the 156975 / 157421 pack
[14:47:44] thread 2 is processing the 156976 / 157421 pack
[14:47:44] thread 1 is processing the 156977 / 157421 pack
[14:47:44] thread 2 is processing the 156978 / 157421 pack
[14:47:44] thread 1 is processing the 156979 / 157421 pack
[14:47:44] thread 2 is processing the 156980 / 157421 pack
[14:47:44] thread 1 is processing the 156981 / 157421 pack
[14:47:44] thread 2 is processing the 156982 / 157421 pack
[14:47:44] thread 1 is processing the 156983 / 157421 pack
[14:47:44] thread 2 is processing the 156984 / 157421 pack
[14:47:44] thread 1 is processing the 156985 / 157421 pack
[14:47:44] thread 2 is processing the 156985 / 157421 pack
[14:47:44] thread 1 is processing the 156987 / 157421 pack
[14:47:44] thread 2 is processing the 156987 / 157421 pack
[14:47:44] thread 1 is processing the 156989 / 157421 pack
[14:47:44] thread 2 is processing the 156990 / 157421 pack
[14:47:44] thread 1 is processing the 156991 / 157421 pack
[14:47:44] thread 2 is processing the 156992 / 157421 pack
[14:47:44] thread 2 is processing the 156993 / 157421 pack
[14:47:44] thread 1 is processing the 156994 / 157421 pack
[14:47:44] thread 1 is processing the 156995 / 157421 pack
[14:47:44] thread 2 is processing the 156996 / 157421 pack
[14:47:44] thread 1 is processing the 156997 / 157421 pack
[14:47:44] thread 2 is processing the 156998 / 157421 pack
[14:47:44] thread 2 is processing the 156999 / 157421 pack
[14:47:44] thread 1 is processing the 156999 / 157421 pack
[14:47:44] thread 1 is processing the 157001 / 157421 pack
[14:47:44] thread 2 is processing the 157001 / 157421 pack
[14:47:45] thread 1 is processing the 157003 / 157421 pack
[14:47:45] thread 2 is processing the 157004 / 157421 pack
[14:47:45] thread 1 is processing the 157005 / 157421 pack
[14:47:45] thread 2 is processing the 157006 / 157421 pack
[14:47:45] thread 2 is processing the 157007 / 157421 pack
[14:47:45] thread 1 is processing the 157008 / 157421 pack
[14:47:45] thread 1 is processing the 157009 / 157421 pack
[14:47:45] thread 2 is processing the 157010 / 157421 pack
[14:47:45] thread 1 is processing the 157011 / 157421 pack
[14:47:45] thread 2 is processing the 157011 / 157421 pack
[14:47:45] thread 2 is processing the 157013 / 157421 pack
[14:47:45] thread 1 is processing the 157014 / 157421 pack
[14:47:45] thread 1 is processing the 157015 / 157421 pack
[14:47:45] thread 2 is processing the 157016 / 157421 pack
[14:47:45] thread 1 is processing the 157017 / 157421 pack
[14:47:45] thread 2 is processing the 157018 / 157421 pack
[14:47:45] thread 1 is processing the 157019 / 157421 pack
[14:47:45] thread 2 is processing the 157020 / 157421 pack
[14:47:45] thread 1 is processing the 157021 / 157421 pack
[14:47:45] thread 2 is processing the 157022 / 157421 pack
[14:47:45] thread 1 is processing the 157023 / 157421 pack
[14:47:45] thread 2 is processing the 157024 / 157421 pack
[14:47:45] thread 1 is processing the 157025 / 157421 pack
[14:47:45] thread 2 is processing the 157026 / 157421 pack
[14:47:45] thread 1 is processing the 157027 / 157421 pack
[14:47:45] thread 2 is processing the 157028 / 157421 pack
[14:47:45] thread 1 is processing the 157029 / 157421 pack
[14:47:45] thread 2 is processing the 157030 / 157421 pack
[14:47:45] thread 1 is processing the 157031 / 157421 pack
[14:47:45] thread 2 is processing the 157032 / 157421 pack
[14:47:45] thread 1 is processing the 157033 / 157421 pack
[14:47:45] thread 2 is processing the 157034 / 157421 pack
[14:47:45] thread 1 is processing the 157035 / 157421 pack
[14:47:45] thread 2 is processing the 157036 / 157421 pack
[14:47:45] thread 1 is processing the 157037 / 157421 pack
[14:47:45] thread 2 is processing the 157038 / 157421 pack
[14:47:45] thread 1 is processing the 157039 / 157421 pack
[14:47:45] thread 2 is processing the 157040 / 157421 pack
[14:47:45] thread 1 is processing the 157041 / 157421 pack
[14:47:45] thread 2 is processing the 157042 / 157421 pack
[14:47:45] thread 1 is processing the 157043 / 157421 pack
[14:47:45] thread 2 is processing the 157044 / 157421 pack
[14:47:45] thread 1 is processing the 157045 / 157421 pack
[14:47:45] thread 2 is processing the 157046 / 157421 pack
[14:47:45] thread 1 is processing the 157047 / 157421 pack
[14:47:45] thread 2 is processing the 157048 / 157421 pack
[14:47:45] thread 1 is processing the 157049 / 157421 pack
[14:47:45] thread 2 is processing the 157050 / 157421 pack
[14:47:45] thread 1 is processing the 157051 / 157421 pack
[14:47:45] thread 2 is processing the 157052 / 157421 pack
[14:47:45] thread 1 is processing the 157053 / 157421 pack
[14:47:45] thread 2 is processing the 157054 / 157421 pack
[14:47:45] thread 1 is processing the 157055 / 157421 pack
[14:47:45] thread 2 is processing the 157056 / 157421 pack
[14:47:45] thread 1 is processing the 157057 / 157421 pack
[14:47:45] thread 2 is processing the 157058 / 157421 pack
[14:47:45] thread 1 is processing the 157059 / 157421 pack
[14:47:45] thread 2 is processing the 157060 / 157421 pack
[14:47:45] thread 1 is processing the 157061 / 157421 pack
[14:47:45] thread 2 is processing the 157062 / 157421 pack
[14:47:45] thread 1 is processing the 157063 / 157421 pack
[14:47:45] thread 2 is processing the 157064 / 157421 pack
[14:47:45] thread 1 is processing the 157065 / 157421 pack
[14:47:45] thread 2 is processing the 157066 / 157421 pack
[14:47:45] thread 1 is processing the 157067 / 157421 pack
[14:47:45] thread 2 is processing the 157068 / 157421 pack
[14:47:45] thread 1 is processing the 157069 / 157421 pack
[14:47:45] thread 2 is processing the 157070 / 157421 pack
[14:47:45] thread 1 is processing the 157071 / 157421 pack
[14:47:45] thread 2 is processing the 157072 / 157421 pack
[14:47:45] thread 1 is processing the 157073 / 157421 pack
[14:47:45] thread 2 is processing the 157074 / 157421 pack
[14:47:45] thread 1 is processing the 157075 / 157421 pack
[14:47:45] thread 2 is processing the 157076 / 157421 pack
[14:47:45] thread 1 is processing the 157077 / 157421 pack
[14:47:45] thread 2 is processing the 157078 / 157421 pack
[14:47:45] thread 1 is processing the 157079 / 157421 pack
[14:47:45] thread 2 is processing the 157080 / 157421 pack
[14:47:45] thread 1 is processing the 157081 / 157421 pack
[14:47:45] thread 2 is processing the 157082 / 157421 pack
[14:47:45] thread 1 is processing the 157083 / 157421 pack
[14:47:45] thread 2 is processing the 157084 / 157421 pack
[14:47:45] thread 1 is processing the 157085 / 157421 pack
[14:47:45] thread 2 is processing the 157086 / 157421 pack
[14:47:46] thread 1 is processing the 157087 / 157421 pack
[14:47:46] thread 2 is processing the 157088 / 157421 pack
[14:47:46] thread 1 is processing the 157089 / 157421 pack
[14:47:46] thread 2 is processing the 157090 / 157421 pack
[14:47:46] thread 2 is processing the 157091 / 157421 pack
[14:47:46] thread 1 is processing the 157092 / 157421 pack
[14:47:46] thread 2 is processing the 157093 / 157421 pack
[14:47:46] thread 1 is processing the 157094 / 157421 pack
[14:47:46] thread 2 is processing the 157095 / 157421 pack
[14:47:46] thread 1 is processing the 157096 / 157421 pack
[14:47:46] thread 2 is processing the 157097 / 157421 pack
[14:47:46] thread 1 is processing the 157098 / 157421 pack
[14:47:46] thread 2 is processing the 157099 / 157421 pack
[14:47:46] thread 1 is processing the 157100 / 157421 pack
[14:47:46] thread 2 is processing the 157101 / 157421 pack
[14:47:46] thread 1 is processing the 157102 / 157421 pack
[14:47:46] thread 2 is processing the 157103 / 157421 pack
[14:47:46] thread 1 is processing the 157104 / 157421 pack
[14:47:46] thread 2 is processing the 157105 / 157421 pack
[14:47:46] thread 1 is processing the 157106 / 157421 pack
[14:47:46] thread 2 is processing the 157107 / 157421 pack
[14:47:46] thread 1 is processing the 157108 / 157421 pack
[14:47:46] thread 1 is processing the 157109 / 157421 pack
[14:47:46] thread 2 is processing the 157110 / 157421 pack
[14:47:46] thread 2 is processing the 157111 / 157421 pack
[14:47:46] thread 1 is processing the 157111 / 157421 pack
[14:47:46] thread 2 is processing the 157113 / 157421 pack
[14:47:46] thread 1 is processing the 157114 / 157421 pack
[14:47:46] thread 1 is processing the 157115 / 157421 pack
[14:47:46] thread 2 is processing the 157116 / 157421 pack
[14:47:46] thread 1 is processing the 157117 / 157421 pack
[14:47:46] thread 2 is processing the 157118 / 157421 pack
[14:47:46] thread 1 is processing the 157119 / 157421 pack
[14:47:46] thread 2 is processing the 157120 / 157421 pack
[14:47:46] thread 1 is processing the 157121 / 157421 pack
[14:47:46] thread 2 is processing the 157122 / 157421 pack
[14:47:46] thread 2 is processing the 157123 / 157421 pack
[14:47:46] thread 1 is processing the 157124 / 157421 pack
[14:47:46] thread 1 is processing the 157125 / 157421 pack
[14:47:46] thread 2 is processing the 157126 / 157421 pack
[14:47:46] thread 1 is processing the 157127 / 157421 pack
[14:47:46] thread 2 is processing the 157127 / 157421 pack
[14:47:46] thread 2 is processing the 157129 / 157421 pack
[14:47:46] thread 1 is processing the 157130 / 157421 pack
[14:47:46] thread 1 is processing the 157131 / 157421 pack
[14:47:46] thread 2 is processing the 157131 / 157421 pack
[14:47:46] thread 1 is processing the 157133 / 157421 pack
[14:47:46] thread 2 is processing the 157133 / 157421 pack
[14:47:46] thread 1 is processing the 157135 / 157421 pack
[14:47:46] thread 2 is processing the 157136 / 157421 pack
[14:47:46] thread 1 is processing the 157137 / 157421 pack
[14:47:46] thread 2 is processing the 157138 / 157421 pack
[14:47:46] thread 1 is processing the 157139 / 157421 pack
[14:47:46] thread 2 is processing the 157140 / 157421 pack
[14:47:46] thread 1 is processing the 157141 / 157421 pack
[14:47:46] thread 2 is processing the 157142 / 157421 pack
[14:47:46] thread 2 is processing the 157143 / 157421 pack
[14:47:46] thread 1 is processing the 157144 / 157421 pack
[14:47:46] thread 2 is processing the 157145 / 157421 pack
[14:47:46] thread 1 is processing the 157145 / 157421 pack
[14:47:46] thread 2 is processing the 157147 / 157421 pack
[14:47:46] thread 1 is processing the 157148 / 157421 pack
[14:47:46] thread 2 is processing the 157149 / 157421 pack
[14:47:46] thread 1 is processing the 157150 / 157421 pack
[14:47:46] thread 2 is processing the 157151 / 157421 pack
[14:47:46] thread 1 is processing the 157152 / 157421 pack
[14:47:46] thread 2 is processing the 157153 / 157421 pack
[14:47:46] thread 1 is processing the 157154 / 157421 pack
[14:47:46] thread 2 is processing the 157155 / 157421 pack
[14:47:46] thread 1 is processing the 157156 / 157421 pack
[14:47:46] thread 2 is processing the 157157 / 157421 pack
[14:47:46] thread 1 is processing the 157158 / 157421 pack
[14:47:46] thread 2 is processing the 157159 / 157421 pack
[14:47:46] thread 1 is processing the 157160 / 157421 pack
[14:47:46] thread 2 is processing the 157161 / 157421 pack
[14:47:46] thread 1 is processing the 157162 / 157421 pack
[14:47:46] thread 2 is processing the 157163 / 157421 pack
[14:47:46] thread 1 is processing the 157164 / 157421 pack
[14:47:46] thread 2 is processing the 157165 / 157421 pack
[14:47:46] thread 1 is processing the 157166 / 157421 pack
[14:47:46] thread 2 is processing the 157167 / 157421 pack
[14:47:46] thread 1 is processing the 157168 / 157421 pack
[14:47:46] thread 2 is processing the 157169 / 157421 pack
[14:47:46] thread 1 is processing the 157170 / 157421 pack
[14:47:46] thread 2 is processing the 157171 / 157421 pack
[14:47:46] thread 1 is processing the 157172 / 157421 pack
[14:47:47] thread 2 is processing the 157173 / 157421 pack
[14:47:47] thread 1 is processing the 157174 / 157421 pack
[14:47:47] thread 2 is processing the 157175 / 157421 pack
[14:47:47] thread 1 is processing the 157176 / 157421 pack
[14:47:47] thread 2 is processing the 157177 / 157421 pack
[14:47:47] thread 1 is processing the 157178 / 157421 pack
[14:47:47] thread 2 is processing the 157179 / 157421 pack
[14:47:47] thread 1 is processing the 157180 / 157421 pack
[14:47:47] thread 2 is processing the 157181 / 157421 pack
[14:47:47] thread 1 is processing the 157182 / 157421 pack
[14:47:47] thread 2 is processing the 157183 / 157421 pack
[14:47:47] thread 1 is processing the 157184 / 157421 pack
[14:47:47] thread 2 is processing the 157185 / 157421 pack
[14:47:47] thread 1 is processing the 157186 / 157421 pack
[14:47:47] thread 2 is processing the 157187 / 157421 pack
[14:47:47] thread 1 is processing the 157188 / 157421 pack
[14:47:47] thread 2 is processing the 157189 / 157421 pack
[14:47:47] thread 1 is processing the 157190 / 157421 pack
[14:47:47] thread 2 is processing the 157191 / 157421 pack
[14:47:47] thread 1 is processing the 157192 / 157421 pack
[14:47:47] thread 2 is processing the 157193 / 157421 pack
[14:47:47] thread 1 is processing the 157194 / 157421 pack
[14:47:47] thread 2 is processing the 157195 / 157421 pack
[14:47:47] thread 1 is processing the 157196 / 157421 pack
[14:47:47] thread 2 is processing the 157197 / 157421 pack
[14:47:47] thread 1 is processing the 157198 / 157421 pack
[14:47:47] thread 2 is processing the 157199 / 157421 pack
[14:47:47] thread 1 is processing the 157200 / 157421 pack
[14:47:47] thread 2 is processing the 157201 / 157421 pack
[14:47:47] thread 1 is processing the 157202 / 157421 pack
[14:47:47] thread 2 is processing the 157203 / 157421 pack
[14:47:47] thread 1 is processing the 157204 / 157421 pack
[14:47:47] thread 2 is processing the 157205 / 157421 pack
[14:47:47] thread 1 is processing the 157206 / 157421 pack
[14:47:47] thread 2 is processing the 157207 / 157421 pack
[14:47:47] thread 1 is processing the 157208 / 157421 pack
[14:47:47] thread 2 is processing the 157209 / 157421 pack
[14:47:47] thread 1 is processing the 157210 / 157421 pack
[14:47:47] thread 2 is processing the 157211 / 157421 pack
[14:47:47] thread 1 is processing the 157212 / 157421 pack
[14:47:47] thread 2 is processing the 157213 / 157421 pack
[14:47:47] thread 1 is processing the 157214 / 157421 pack
[14:47:47] thread 2 is processing the 157215 / 157421 pack
[14:47:47] thread 1 is processing the 157216 / 157421 pack
[14:47:47] thread 2 is processing the 157217 / 157421 pack
[14:47:47] thread 1 is processing the 157218 / 157421 pack
[14:47:47] thread 2 is processing the 157219 / 157421 pack
[14:47:47] thread 1 is processing the 157220 / 157421 pack
[14:47:47] thread 2 is processing the 157221 / 157421 pack
[14:47:47] thread 1 is processing the 157222 / 157421 pack
[14:47:47] thread 2 is processing the 157223 / 157421 pack
[14:47:47] thread 1 is processing the 157224 / 157421 pack
[14:47:47] thread 2 is processing the 157225 / 157421 pack
[14:47:47] thread 1 is processing the 157226 / 157421 pack
[14:47:47] thread 2 is processing the 157227 / 157421 pack
[14:47:47] thread 1 is processing the 157228 / 157421 pack
[14:47:47] thread 2 is processing the 157229 / 157421 pack
[14:47:47] thread 1 is processing the 157230 / 157421 pack
[14:47:47] thread 2 is processing the 157231 / 157421 pack
[14:47:47] thread 1 is processing the 157232 / 157421 pack
[14:47:47] thread 2 is processing the 157233 / 157421 pack
[14:47:47] thread 1 is processing the 157234 / 157421 pack
[14:47:47] thread 2 is processing the 157235 / 157421 pack
[14:47:47] thread 1 is processing the 157236 / 157421 pack
[14:47:47] thread 2 is processing the 157237 / 157421 pack
[14:47:47] thread 1 is processing the 157238 / 157421 pack
[14:47:47] thread 2 is processing the 157239 / 157421 pack
[14:47:47] thread 1 is processing the 157240 / 157421 pack
[14:47:47] thread 2 is processing the 157241 / 157421 pack
[14:47:47] thread 1 is processing the 157242 / 157421 pack
[14:47:47] thread 2 is processing the 157243 / 157421 pack
[14:47:47] thread 1 is processing the 157244 / 157421 pack
[14:47:47] thread 2 is processing the 157245 / 157421 pack
[14:47:47] thread 1 is processing the 157246 / 157421 pack
[14:47:47] thread 2 is processing the 157247 / 157421 pack
[14:47:47] thread 1 is processing the 157248 / 157421 pack
[14:47:47] thread 2 is processing the 157249 / 157421 pack
[14:47:47] thread 1 is processing the 157250 / 157421 pack
[14:47:47] thread 2 is processing the 157251 / 157421 pack
[14:47:47] thread 1 is processing the 157252 / 157421 pack
[14:47:48] thread 2 is processing the 157253 / 157421 pack
[14:47:48] thread 1 is processing the 157254 / 157421 pack
[14:47:48] thread 2 is processing the 157255 / 157421 pack
[14:47:48] thread 1 is processing the 157256 / 157421 pack
[14:47:48] thread 2 is processing the 157257 / 157421 pack
[14:47:48] thread 1 is processing the 157258 / 157421 pack
[14:47:48] thread 2 is processing the 157259 / 157421 pack
[14:47:48] thread 1 is processing the 157260 / 157421 pack
[14:47:48] thread 2 is processing the 157261 / 157421 pack
[14:47:48] thread 1 is processing the 157262 / 157421 pack
[14:47:48] thread 2 is processing the 157263 / 157421 pack
[14:47:48] thread 1 is processing the 157264 / 157421 pack
[14:47:48] thread 2 is processing the 157265 / 157421 pack
[14:47:48] thread 1 is processing the 157266 / 157421 pack
[14:47:48] thread 2 is processing the 157267 / 157421 pack
[14:47:48] thread 1 is processing the 157268 / 157421 pack
[14:47:48] thread 2 is processing the 157269 / 157421 pack
[14:47:48] thread 1 is processing the 157270 / 157421 pack
[14:47:48] thread 2 is processing the 157271 / 157421 pack
[14:47:48] thread 1 is processing the 157272 / 157421 pack
[14:47:48] thread 2 is processing the 157273 / 157421 pack
[14:47:48] thread 1 is processing the 157274 / 157421 pack
[14:47:48] thread 2 is processing the 157275 / 157421 pack
[14:47:48] thread 1 is processing the 157276 / 157421 pack
[14:47:48] thread 2 is processing the 157277 / 157421 pack
[14:47:48] thread 1 is processing the 157278 / 157421 pack
[14:47:48] thread 2 is processing the 157279 / 157421 pack
[14:47:48] thread 1 is processing the 157280 / 157421 pack
[14:47:48] thread 2 is processing the 157281 / 157421 pack
[14:47:48] thread 1 is processing the 157282 / 157421 pack
[14:47:48] thread 2 is processing the 157283 / 157421 pack
[14:47:48] thread 1 is processing the 157284 / 157421 pack
[14:47:48] thread 2 is processing the 157285 / 157421 pack
[14:47:48] thread 1 is processing the 157286 / 157421 pack
[14:47:48] thread 2 is processing the 157287 / 157421 pack
[14:47:48] thread 1 is processing the 157288 / 157421 pack
[14:47:48] thread 2 is processing the 157289 / 157421 pack
[14:47:48] thread 1 is processing the 157290 / 157421 pack
[14:47:48] thread 2 is processing the 157291 / 157421 pack
[14:47:48] thread 1 is processing the 157292 / 157421 pack
[14:47:48] thread 2 is processing the 157293 / 157421 pack
[14:47:48] thread 1 is processing the 157294 / 157421 pack
[14:47:48] thread 2 is processing the 157295 / 157421 pack
[14:47:48] thread 1 is processing the 157296 / 157421 pack
[14:47:48] thread 2 is processing the 157297 / 157421 pack
[14:47:48] thread 1 is processing the 157298 / 157421 pack
[14:47:48] thread 2 is processing the 157299 / 157421 pack
[14:47:48] thread 1 is processing the 157300 / 157421 pack
[14:47:48] thread 2 is processing the 157301 / 157421 pack
[14:47:48] thread 1 is processing the 157302 / 157421 pack
[14:47:48] thread 2 is processing the 157303 / 157421 pack
[14:47:48] thread 1 is processing the 157304 / 157421 pack
[14:47:48] thread 2 is processing the 157305 / 157421 pack
[14:47:48] thread 1 is processing the 157306 / 157421 pack
[14:47:48] thread 2 is processing the 157307 / 157421 pack
[14:47:48] thread 1 is processing the 157308 / 157421 pack
[14:47:48] thread 2 is processing the 157309 / 157421 pack
[14:47:48] thread 1 is processing the 157310 / 157421 pack
[14:47:48] thread 2 is processing the 157311 / 157421 pack
[14:47:48] thread 1 is processing the 157312 / 157421 pack
[14:47:48] thread 2 is processing the 157313 / 157421 pack
[14:47:48] thread 1 is processing the 157314 / 157421 pack
[14:47:48] thread 2 is processing the 157315 / 157421 pack
[14:47:48] thread 1 is processing the 157316 / 157421 pack
[14:47:48] thread 2 is processing the 157317 / 157421 pack
[14:47:48] thread 1 is processing the 157318 / 157421 pack
[14:47:48] thread 2 is processing the 157319 / 157421 pack
[14:47:48] thread 1 is processing the 157320 / 157421 pack
[14:47:48] thread 2 is processing the 157321 / 157421 pack
[14:47:48] thread 1 is processing the 157322 / 157421 pack
[14:47:48] thread 2 is processing the 157323 / 157421 pack
[14:47:48] thread 1 is processing the 157324 / 157421 pack
[14:47:48] thread 2 is processing the 157325 / 157421 pack
[14:47:48] thread 1 is processing the 157326 / 157421 pack
[14:47:48] thread 2 is processing the 157327 / 157421 pack
[14:47:48] thread 1 is processing the 157328 / 157421 pack
[14:47:48] thread 2 is processing the 157329 / 157421 pack
[14:47:48] thread 1 is processing the 157330 / 157421 pack
[14:47:49] thread 2 is processing the 157331 / 157421 pack
[14:47:49] thread 1 is processing the 157332 / 157421 pack
[14:47:49] thread 2 is processing the 157333 / 157421 pack
[14:47:49] thread 1 is processing the 157334 / 157421 pack
[14:47:49] thread 2 is processing the 157335 / 157421 pack
[14:47:49] thread 1 is processing the 157336 / 157421 pack
[14:47:49] thread 2 is processing the 157337 / 157421 pack
[14:47:49] thread 1 is processing the 157338 / 157421 pack
[14:47:49] thread 2 is processing the 157339 / 157421 pack
[14:47:49] thread 1 is processing the 157340 / 157421 pack
[14:47:49] thread 2 is processing the 157341 / 157421 pack
[14:47:49] thread 1 is processing the 157342 / 157421 pack
[14:47:49] thread 2 is processing the 157343 / 157421 pack
[14:47:49] thread 1 is processing the 157344 / 157421 pack
[14:47:49] thread 2 is processing the 157345 / 157421 pack
[14:47:49] thread 1 is processing the 157346 / 157421 pack
[14:47:49] thread 2 is processing the 157347 / 157421 pack
[14:47:49] thread 1 is processing the 157348 / 157421 pack
[14:47:49] thread 2 is processing the 157349 / 157421 pack
[14:47:49] thread 1 is processing the 157350 / 157421 pack
[14:47:49] thread 2 is processing the 157351 / 157421 pack
[14:47:49] thread 1 is processing the 157352 / 157421 pack
[14:47:49] thread 2 is processing the 157353 / 157421 pack
[14:47:49] thread 1 is processing the 157354 / 157421 pack
[14:47:49] thread 2 is processing the 157355 / 157421 pack
[14:47:49] thread 1 is processing the 157356 / 157421 pack
[14:47:49] thread 2 is processing the 157357 / 157421 pack
[14:47:49] thread 1 is processing the 157358 / 157421 pack
[14:47:49] thread 2 is processing the 157359 / 157421 pack
[14:47:49] thread 1 is processing the 157360 / 157421 pack
[14:47:49] thread 2 is processing the 157361 / 157421 pack
[14:47:49] thread 1 is processing the 157362 / 157421 pack
[14:47:49] thread 2 is processing the 157363 / 157421 pack
[14:47:49] thread 1 is processing the 157364 / 157421 pack
[14:47:49] thread 2 is processing the 157365 / 157421 pack
[14:47:49] thread 1 is processing the 157366 / 157421 pack
[14:47:49] thread 2 is processing the 157367 / 157421 pack
[14:47:49] thread 1 is processing the 157368 / 157421 pack
[14:47:49] thread 2 is processing the 157369 / 157421 pack
[14:47:49] thread 1 is processing the 157370 / 157421 pack
[14:47:49] thread 2 is processing the 157371 / 157421 pack
[14:47:49] thread 1 is processing the 157372 / 157421 pack
[14:47:49] thread 2 is processing the 157373 / 157421 pack
[14:47:49] thread 1 is processing the 157374 / 157421 pack
[14:47:49] thread 2 is processing the 157375 / 157421 pack
[14:47:49] thread 1 is processing the 157376 / 157421 pack
[14:47:49] thread 2 is processing the 157377 / 157421 pack
[14:47:49] thread 1 is processing the 157378 / 157421 pack
[14:47:49] thread 2 is processing the 157379 / 157421 pack
[14:47:49] thread 1 is processing the 157380 / 157421 pack
[14:47:49] thread 2 is processing the 157381 / 157421 pack
[14:47:49] thread 1 is processing the 157382 / 157421 pack
[14:47:49] thread 2 is processing the 157383 / 157421 pack
[14:47:49] thread 1 is processing the 157384 / 157421 pack
[14:47:49] thread 2 is processing the 157385 / 157421 pack
[14:47:49] thread 1 is processing the 157386 / 157421 pack
[14:47:49] thread 2 is processing the 157387 / 157421 pack
[14:47:49] thread 1 is processing the 157388 / 157421 pack
[14:47:49] thread 2 is processing the 157389 / 157421 pack
[14:47:49] thread 1 is processing the 157390 / 157421 pack
[14:47:49] thread 2 is processing the 157391 / 157421 pack
[14:47:49] thread 1 is processing the 157392 / 157421 pack
[14:47:49] thread 2 is processing the 157393 / 157421 pack
[14:47:49] thread 1 is processing the 157394 / 157421 pack
[14:47:49] thread 2 is processing the 157395 / 157421 pack
[14:47:49] thread 1 is processing the 157396 / 157421 pack
[14:47:49] thread 2 is processing the 157397 / 157421 pack
[14:47:49] thread 1 is processing the 157398 / 157421 pack
[14:47:49] thread 2 is processing the 157399 / 157421 pack
[14:47:49] thread 1 is processing the 157400 / 157421 pack
[14:47:49] thread 2 is processing the 157401 / 157421 pack
[14:47:49] thread 1 is processing the 157402 / 157421 pack
[14:47:49] thread 2 is processing the 157403 / 157421 pack
[14:47:49] thread 1 is processing the 157404 / 157421 pack
[14:47:49] thread 2 is processing the 157405 / 157421 pack
[14:47:49] thread 1 is processing the 157406 / 157421 pack
[14:47:49] thread 2 is processing the 157407 / 157421 pack
[14:47:49] thread 1 is processing the 157408 / 157421 pack
[14:47:49] thread 2 is processing the 157409 / 157421 pack
[14:47:49] thread 1 is processing the 157410 / 157421 pack
[14:47:49] thread 2 is processing the 157411 / 157421 pack
[14:47:49] thread 1 is processing the 157412 / 157421 pack
[14:47:50] thread 2 is processing the 157413 / 157421 pack
[14:47:50] thread 1 is processing the 157414 / 157421 pack
[14:47:50] thread 2 is processing the 157415 / 157421 pack
[14:47:50] thread 1 is processing the 157416 / 157421 pack
[14:47:50] thread 2 is processing the 157417 / 157421 pack
[14:47:50] thread 1 is processing the 157418 / 157421 pack
[14:47:50] thread 2 is processing the 157419 / 157421 pack
[14:47:50] thread 1 is processing the 157420 / 157421 pack
[14:47:50] thread 1 data processing completed
[14:47:50] thread 1 finished
[14:47:50] thread 2 data processing completed
[14:47:50] thread 2 finished
[14:47:50] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/Adults_QC_R1.fastq.gz writer finished
[14:47:50] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/Adults_QC_R2.fastq.gz writer finished
[14:47:50] start to generate reports

Read1 before filtering:
total reads: 157420508
total bases: 23770496708
Q20 bases: 23286376588(97.9634%)
Q30 bases: 22484150003(94.5885%)

Read2 before filtering:
total reads: 157420508
total bases: 23770496708
Q20 bases: 22991414051(96.7225%)
Q30 bases: 21884737777(92.0668%)

Read1 after filtering:
total reads: 154852151
total bases: 23174337326
Q20 bases: 22761473207(98.2184%)
Q30 bases: 22007138294(94.9634%)

Read2 after filtering:
total reads: 154852151
total bases: 23172450744
Q20 bases: 22598288230(97.5222%)
Q30 bases: 21560078397(93.0419%)

Filtering result:
reads passed filter: 309704302
reads failed due to low quality: 5017680
reads failed due to too many N: 14224
reads failed due to too short: 104810
reads with adapter trimmed: 22005747
bases trimmed due to adapters: 417626588

Duplication rate: 7.86516%

Insert size peak (evaluated by paired-end reads): 213

Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[13:57:28] start to load data
[13:57:37] loaded 1M read pairs
[13:57:49] loaded 2M read pairs
[13:58:3] loaded 3M read pairs
[13:58:16] loaded 4M read pairs
[13:58:30] loaded 5M read pairs
[13:58:43] loaded 6M read pairs
[13:58:57] loaded 7M read pairs
[13:59:10] loaded 8M read pairs
[13:59:23] loaded 9M read pairs
[13:59:36] loaded 10M read pairs
[13:59:49] loaded 11M read pairs
[14:0:2] loaded 12M read pairs
[14:0:16] loaded 13M read pairs
[14:0:29] loaded 14M read pairs
[14:0:42] loaded 15M read pairs
[14:0:55] loaded 16M read pairs
[14:1:8] loaded 17M read pairs
[14:1:21] loaded 18M read pairs
[14:1:33] loaded 19M read pairs
[14:1:45] loaded 20M read pairs
[14:1:57] loaded 21M read pairs
[14:2:10] loaded 22M read pairs
[14:2:22] loaded 23M read pairs
[14:2:34] loaded 24M read pairs
[14:2:46] loaded 25M read pairs
[14:2:59] loaded 26M read pairs
[14:3:10] loaded 27M read pairs
[14:3:23] loaded 28M read pairs
[14:3:34] loaded 29M read pairs
[14:3:46] loaded 30M read pairs
[14:3:58] loaded 31M read pairs
[14:4:10] loaded 32M read pairs
[14:4:21] loaded 33M read pairs
[14:4:33] loaded 34M read pairs
[14:4:45] loaded 35M read pairs
[14:4:57] loaded 36M read pairs
[14:5:9] loaded 37M read pairs
[14:5:21] loaded 38M read pairs
[14:5:34] loaded 39M read pairs
[14:5:47] loaded 40M read pairs
[14:5:59] loaded 41M read pairs
[14:6:11] loaded 42M read pairs
[14:6:23] loaded 43M read pairs
[14:6:35] loaded 44M read pairs
[14:6:47] loaded 45M read pairs
[14:6:59] loaded 46M read pairs
[14:7:12] loaded 47M read pairs
[14:7:24] loaded 48M read pairs
[14:7:36] loaded 49M read pairs
[14:7:47] loaded 50M read pairs
[14:7:59] loaded 51M read pairs
[14:8:11] loaded 52M read pairs
[14:8:23] loaded 53M read pairs
[14:8:35] loaded 54M read pairs
[14:8:47] loaded 55M read pairs
[14:8:59] loaded 56M read pairs
[14:9:11] loaded 57M read pairs
[14:9:23] loaded 58M read pairs
[14:9:35] loaded 59M read pairs
[14:9:47] loaded 60M read pairs
[14:9:58] loaded 61M read pairs
[14:10:10] loaded 62M read pairs
[14:10:22] loaded 63M read pairs
[14:10:34] loaded 64M read pairs
[14:10:46] loaded 65M read pairs
[14:10:58] loaded 66M read pairs
[14:11:9] loaded 67M read pairs
[14:11:22] loaded 68M read pairs
[14:11:34] loaded 69M read pairs
[14:11:47] loaded 70M read pairs
[14:11:59] loaded 71M read pairs
[14:12:12] loaded 72M read pairs
[14:12:25] loaded 73M read pairs
[14:12:37] loaded 74M read pairs
[14:12:50] loaded 75M read pairs
[14:13:2] loaded 76M read pairs
[14:13:14] loaded 77M read pairs
[14:13:26] loaded 78M read pairs
[14:13:38] loaded 79M read pairs
[14:13:49] loaded 80M read pairs
[14:14:1] loaded 81M read pairs
[14:14:13] loaded 82M read pairs
[14:14:26] loaded 83M read pairs
[14:14:38] loaded 84M read pairs
[14:14:51] loaded 85M read pairs
[14:15:3] loaded 86M read pairs
[14:15:15] loaded 87M read pairs
Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[14:17:39] start to load data
[14:17:46] loaded 1M read pairs
[14:17:56] loaded 2M read pairs
[14:18:7] loaded 3M read pairs
[14:18:18] loaded 4M read pairs
[14:18:29] loaded 5M read pairs
[14:18:41] loaded 6M read pairs
[14:18:52] loaded 7M read pairs
[14:19:4] loaded 8M read pairs
[14:19:17] loaded 9M read pairs
[14:19:29] loaded 10M read pairs
[14:19:40] loaded 11M read pairs
[14:19:53] loaded 12M read pairs
[14:20:7] loaded 13M read pairs
[14:20:20] loaded 14M read pairs
[14:20:32] loaded 15M read pairs
[14:20:44] loaded 16M read pairs
[14:20:57] loaded 17M read pairs
[14:21:11] loaded 18M read pairs
[14:21:23] loaded 19M read pairs
[14:21:34] loaded 20M read pairs
[14:21:46] loaded 21M read pairs
[14:21:57] loaded 22M read pairs
[14:22:9] loaded 23M read pairs
[14:22:20] loaded 24M read pairs
[14:22:31] loaded 25M read pairs
[14:22:43] loaded 26M read pairs
[14:22:54] loaded 27M read pairs
[14:23:5] loaded 28M read pairs
[14:23:16] loaded 29M read pairs
[14:23:27] loaded 30M read pairs
[14:23:38] loaded 31M read pairs
[14:23:50] loaded 32M read pairs
[14:24:0] loaded 33M read pairs
[14:24:11] loaded 34M read pairs
[14:24:22] loaded 35M read pairs
[14:24:34] loaded 36M read pairs
[14:24:45] loaded 37M read pairs
[14:24:56] loaded 38M read pairs
[14:25:8] loaded 39M read pairs
[14:25:19] loaded 40M read pairs
[14:25:31] loaded 41M read pairs
[14:25:43] loaded 42M read pairs
[14:25:54] loaded 43M read pairs
[14:26:6] loaded 44M read pairs
[14:26:18] loaded 45M read pairs
[14:26:29] loaded 46M read pairs
[14:26:40] loaded 47M read pairs
[14:26:51] loaded 48M read pairs
[14:27:3] loaded 49M read pairs
[14:27:14] loaded 50M read pairs
[14:27:27] loaded 51M read pairs
[14:27:39] loaded 52M read pairs
[14:27:51] loaded 53M read pairs
[14:28:3] loaded 54M read pairs
[14:28:16] loaded 55M read pairs
[14:28:28] loaded 56M read pairs
[14:28:40] loaded 57M read pairs
[14:28:51] loaded 58M read pairs
[14:29:3] loaded 59M read pairs
[14:29:15] loaded 60M read pairs
[14:29:26] loaded 61M read pairs
[14:29:38] loaded 62M read pairs
[14:29:50] loaded 63M read pairs
[14:30:1] loaded 64M read pairs
[14:30:13] loaded 65M read pairs
[14:30:24] loaded 66M read pairs
[14:30:36] loaded 67M read pairs
[14:30:48] loaded 68M read pairs
[14:31:1] loaded 69M read pairs
[14:31:14] loaded 70M read pairs
[14:31:27] loaded 71M read pairs
[14:31:40] loaded 72M read pairs
[14:31:53] loaded 73M read pairs
[14:32:6] loaded 74M read pairs
[14:32:18] loaded 75M read pairs
[14:32:31] loaded 76M read pairs
[14:32:43] loaded 77M read pairs
[14:32:55] loaded 78M read pairs
[14:33:7] loaded 79M read pairs
[14:33:18] loaded 80M read pairs
[14:33:30] loaded 81M read pairs
[14:33:42] loaded 82M read pairs
[14:33:54] loaded 83M read pairs
[14:34:5] loaded 84M read pairs
[14:34:17] loaded 85M read pairs
[14:34:29] loaded 86M read pairs
[14:34:41] loaded 87M read pairs
[14:34:52] loaded 88M read pairs
[14:35:4] loaded 89M read pairs
[14:35:16] loaded 90M read pairs
[14:35:29] loaded 91M read pairs
[14:35:41] loaded 92M read pairs
[14:35:53] loaded 93M read pairs
[14:36:5] loaded 94M read pairs
[14:36:17] loaded 95M read pairs
[14:36:28] loaded 96M read pairs
[14:36:40] loaded 97M read pairs
[14:36:52] loaded 98M read pairs
[14:37:4] loaded 99M read pairs
[14:37:17] loaded 100M read pairs
[14:37:28] loaded 101M read pairs
[14:37:40] loaded 102M read pairs
[14:37:52] loaded 103M read pairs
[14:38:4] loaded 104M read pairs
[14:38:15] loaded 105M read pairs
[14:38:27] loaded 106M read pairs
[14:38:39] loaded 107M read pairs
[14:38:51] loaded 108M read pairs
[14:39:3] loaded 109M read pairs
[14:39:15] loaded 110M read pairs
[14:39:26] loaded 111M read pairs
[14:39:38] loaded 112M read pairs
[14:39:49] loaded 113M read pairs
[14:40:2] loaded 114M read pairs
[14:40:15] loaded 115M read pairs
[14:40:27] loaded 116M read pairs
[14:40:39] loaded 117M read pairs
[14:40:51] loaded 118M read pairs
[14:41:3] loaded 119M read pairs
[14:41:15] loaded 120M read pairs
[14:41:27] loaded 121M read pairs
[14:41:40] loaded 122M read pairs
[14:41:51] loaded 123M read pairs
[14:42:3] loaded 124M read pairs
[14:42:15] loaded 125M read pairs
[14:42:27] loaded 126M read pairs
[14:42:39] loaded 127M read pairs
[14:42:50] loaded 128M read pairs
[14:43:2] loaded 129M read pairs
[14:43:14] loaded 130M read pairs
[14:43:25] loaded 131M read pairs
[14:43:37] loaded 132M read pairs
[14:43:49] loaded 133M read pairs
[14:44:1] loaded 134M read pairs
[14:44:13] loaded 135M read pairs
[14:44:25] loaded 136M read pairs
[14:44:37] loaded 137M read pairs
[14:44:49] loaded 138M read pairs
[14:45:0] loaded 139M read pairs
[14:45:12] loaded 140M read pairs
[14:45:24] loaded 141M read pairs
[14:45:36] loaded 142M read pairs
[14:45:47] loaded 143M read pairs
[14:45:59] loaded 144M read pairs
[14:46:11] loaded 145M read pairs
[14:46:22] loaded 146M read pairs
[14:46:33] loaded 147M read pairs
[14:46:45] loaded 148M read pairs
[14:46:56] loaded 149M read pairs
[14:47:7] loaded 150M read pairs
[14:47:18] loaded 151M read pairs
[14:47:29] loaded 152M read pairs
[14:47:41] loaded 153M read pairs
[14:47:52] loaded 154M read pairs
[14:48:3] loaded 155M read pairs
[14:48:14] loaded 156M read pairs
[14:48:25] loaded 157M read pairs
[14:48:36] loaded 158M read pairs
[14:48:47] loaded 159M read pairs
[14:48:58] loaded 160M read pairs
[14:49:9] loaded 161M read pairs
[14:49:19] loaded 162M read pairs
[14:49:30] loaded 163M read pairs
[14:49:41] loaded 164M read pairs
[14:49:51] loaded 165M read pairs
[14:49:56] all reads loaded, start to monitor thread status
[14:49:56] thread 2 is processing the 164901 / 165401 pack
[14:49:56] thread 1 is processing the 164902 / 165401 pack
[14:49:56] thread 2 is processing the 164903 / 165401 pack
[14:49:56] thread 1 is processing the 164904 / 165401 pack
[14:49:56] thread 2 is processing the 164905 / 165401 pack
[14:49:56] thread 1 is processing the 164906 / 165401 pack
[14:49:56] thread 2 is processing the 164907 / 165401 pack
[14:49:56] thread 1 is processing the 164908 / 165401 pack
[14:49:56] thread 2 is processing the 164909 / 165401 pack
[14:49:56] thread 1 is processing the 164910 / 165401 pack
[14:49:56] thread 2 is processing the 164911 / 165401 pack
[14:49:56] thread 1 is processing the 164912 / 165401 pack
[14:49:56] thread 2 is processing the 164913 / 165401 pack
[14:49:56] thread 1 is processing the 164914 / 165401 pack
[14:49:56] thread 2 is processing the 164915 / 165401 pack
[14:49:56] thread 1 is processing the 164916 / 165401 pack
[14:49:56] thread 2 is processing the 164917 / 165401 pack
[14:49:56] thread 1 is processing the 164918 / 165401 pack
[14:49:56] thread 2 is processing the 164919 / 165401 pack
[14:49:56] thread 1 is processing the 164920 / 165401 pack
[14:49:56] thread 2 is processing the 164921 / 165401 pack
[14:49:56] thread 1 is processing the 164922 / 165401 pack
[14:49:56] thread 2 is processing the 164923 / 165401 pack
[14:49:56] thread 1 is processing the 164924 / 165401 pack
[14:49:56] thread 2 is processing the 164925 / 165401 pack
[14:49:56] thread 1 is processing the 164926 / 165401 pack
[14:49:56] thread 2 is processing the 164927 / 165401 pack
[14:49:56] thread 1 is processing the 164928 / 165401 pack
[14:49:56] thread 2 is processing the 164929 / 165401 pack
[14:49:56] thread 1 is processing the 164930 / 165401 pack
[14:49:56] thread 2 is processing the 164931 / 165401 pack
[14:49:56] thread 1 is processing the 164932 / 165401 pack
[14:49:56] thread 2 is processing the 164933 / 165401 pack
[14:49:56] thread 1 is processing the 164934 / 165401 pack
[14:49:56] thread 2 is processing the 164935 / 165401 pack
[14:49:56] thread 1 is processing the 164936 / 165401 pack
[14:49:56] thread 2 is processing the 164937 / 165401 pack
[14:49:56] thread 1 is processing the 164938 / 165401 pack
[14:49:56] thread 2 is processing the 164939 / 165401 pack
[14:49:56] thread 1 is processing the 164940 / 165401 pack
[14:49:56] thread 2 is processing the 164941 / 165401 pack
[14:49:56] thread 1 is processing the 164942 / 165401 pack
[14:49:56] thread 2 is processing the 164943 / 165401 pack
[14:49:56] thread 1 is processing the 164944 / 165401 pack
[14:49:56] thread 2 is processing the 164945 / 165401 pack
[14:49:56] thread 1 is processing the 164946 / 165401 pack
[14:49:56] thread 2 is processing the 164947 / 165401 pack
[14:49:56] thread 1 is processing the 164948 / 165401 pack
[14:49:56] thread 2 is processing the 164949 / 165401 pack
[14:49:56] thread 1 is processing the 164950 / 165401 pack
[14:49:56] thread 2 is processing the 164951 / 165401 pack
[14:49:56] thread 1 is processing the 164952 / 165401 pack
[14:49:56] thread 2 is processing the 164953 / 165401 pack
[14:49:56] thread 1 is processing the 164954 / 165401 pack
[14:49:56] thread 2 is processing the 164955 / 165401 pack
[14:49:56] thread 1 is processing the 164956 / 165401 pack
[14:49:56] thread 2 is processing the 164957 / 165401 pack
[14:49:56] thread 1 is processing the 164958 / 165401 pack
[14:49:56] thread 2 is processing the 164959 / 165401 pack
[14:49:56] thread 1 is processing the 164960 / 165401 pack
[14:49:56] thread 2 is processing the 164961 / 165401 pack
[14:49:56] thread 1 is processing the 164962 / 165401 pack
[14:49:56] thread 2 is processing the 164963 / 165401 pack
[14:49:56] thread 1 is processing the 164964 / 165401 pack
[14:49:56] thread 2 is processing the 164965 / 165401 pack
[14:49:56] thread 1 is processing the 164966 / 165401 pack
[14:49:56] thread 2 is processing the 164967 / 165401 pack
[14:49:56] thread 1 is processing the 164968 / 165401 pack
[14:49:56] thread 2 is processing the 164969 / 165401 pack
[14:49:56] thread 1 is processing the 164970 / 165401 pack
[14:49:56] thread 2 is processing the 164971 / 165401 pack
[14:49:56] thread 1 is processing the 164972 / 165401 pack
[14:49:56] thread 2 is processing the 164973 / 165401 pack
[14:49:56] thread 1 is processing the 164974 / 165401 pack
[14:49:56] thread 2 is processing the 164975 / 165401 pack
[14:49:56] thread 1 is processing the 164976 / 165401 pack
[14:49:56] thread 2 is processing the 164977 / 165401 pack
[14:49:56] thread 1 is processing the 164978 / 165401 pack
[14:49:56] thread 2 is processing the 164979 / 165401 pack
[14:49:56] thread 1 is processing the 164980 / 165401 pack
[14:49:56] thread 2 is processing the 164981 / 165401 pack
[14:49:57] thread 1 is processing the 164982 / 165401 pack
[14:49:57] thread 2 is processing the 164983 / 165401 pack
[14:49:57] thread 1 is processing the 164984 / 165401 pack
[14:49:57] thread 2 is processing the 164985 / 165401 pack
[14:49:57] thread 1 is processing the 164986 / 165401 pack
[14:49:57] thread 2 is processing the 164987 / 165401 pack
[14:49:57] thread 1 is processing the 164988 / 165401 pack
[14:49:57] thread 2 is processing the 164989 / 165401 pack
[14:49:57] thread 1 is processing the 164990 / 165401 pack
[14:49:57] thread 2 is processing the 164991 / 165401 pack
[14:49:57] thread 1 is processing the 164992 / 165401 pack
[14:49:57] thread 2 is processing the 164993 / 165401 pack
[14:49:57] thread 1 is processing the 164994 / 165401 pack
[14:49:57] thread 2 is processing the 164995 / 165401 pack
[14:49:57] thread 1 is processing the 164996 / 165401 pack
[14:49:57] thread 2 is processing the 164997 / 165401 pack
[14:49:57] thread 1 is processing the 164998 / 165401 pack
[14:49:57] thread 2 is processing the 164999 / 165401 pack
[14:49:57] thread 1 is processing the 165000 / 165401 pack
[14:49:57] thread 2 is processing the 165001 / 165401 pack
[14:49:57] thread 1 is processing the 165002 / 165401 pack
[14:49:57] thread 2 is processing the 165003 / 165401 pack
[14:49:57] thread 1 is processing the 165004 / 165401 pack
[14:49:57] thread 2 is processing the 165005 / 165401 pack
[14:49:57] thread 1 is processing the 165006 / 165401 pack
[14:49:57] thread 2 is processing the 165007 / 165401 pack
[14:49:57] thread 1 is processing the 165008 / 165401 pack
[14:49:57] thread 2 is processing the 165009 / 165401 pack
[14:49:57] thread 1 is processing the 165010 / 165401 pack
[14:49:57] thread 2 is processing the 165011 / 165401 pack
[14:49:57] thread 1 is processing the 165012 / 165401 pack
[14:49:57] thread 2 is processing the 165013 / 165401 pack
[14:49:57] thread 1 is processing the 165014 / 165401 pack
[14:49:57] thread 2 is processing the 165015 / 165401 pack
[14:49:57] thread 1 is processing the 165016 / 165401 pack
[14:49:57] thread 2 is processing the 165017 / 165401 pack
[14:49:57] thread 1 is processing the 165018 / 165401 pack
[14:49:57] thread 2 is processing the 165019 / 165401 pack
[14:49:57] thread 1 is processing the 165020 / 165401 pack
[14:49:57] thread 2 is processing the 165021 / 165401 pack
[14:49:57] thread 1 is processing the 165022 / 165401 pack
[14:49:57] thread 2 is processing the 165023 / 165401 pack
[14:49:57] thread 1 is processing the 165024 / 165401 pack
[14:49:57] thread 2 is processing the 165025 / 165401 pack
[14:49:57] thread 1 is processing the 165026 / 165401 pack
[14:49:57] thread 2 is processing the 165027 / 165401 pack
[14:49:57] thread 1 is processing the 165028 / 165401 pack
[14:49:57] thread 2 is processing the 165029 / 165401 pack
[14:49:57] thread 1 is processing the 165030 / 165401 pack
[14:49:57] thread 2 is processing the 165031 / 165401 pack
[14:49:57] thread 1 is processing the 165032 / 165401 pack
[14:49:57] thread 2 is processing the 165033 / 165401 pack
[14:49:57] thread 1 is processing the 165034 / 165401 pack
[14:49:57] thread 2 is processing the 165035 / 165401 pack
[14:49:57] thread 1 is processing the 165036 / 165401 pack
[14:49:57] thread 2 is processing the 165037 / 165401 pack
[14:49:57] thread 1 is processing the 165038 / 165401 pack
[14:49:57] thread 2 is processing the 165039 / 165401 pack
[14:49:57] thread 1 is processing the 165040 / 165401 pack
[14:49:57] thread 2 is processing the 165041 / 165401 pack
[14:49:57] thread 1 is processing the 165042 / 165401 pack
[14:49:57] thread 2 is processing the 165043 / 165401 pack
[14:49:57] thread 1 is processing the 165044 / 165401 pack
[14:49:57] thread 2 is processing the 165045 / 165401 pack
[14:49:57] thread 1 is processing the 165046 / 165401 pack
[14:49:57] thread 2 is processing the 165047 / 165401 pack
[14:49:57] thread 1 is processing the 165048 / 165401 pack
[14:49:57] thread 2 is processing the 165049 / 165401 pack
[14:49:57] thread 1 is processing the 165050 / 165401 pack
[14:49:57] thread 2 is processing the 165051 / 165401 pack
[14:49:57] thread 1 is processing the 165052 / 165401 pack
[14:49:57] thread 2 is processing the 165053 / 165401 pack
[14:49:57] thread 1 is processing the 165054 / 165401 pack
[14:49:57] thread 2 is processing the 165055 / 165401 pack
[14:49:57] thread 1 is processing the 165056 / 165401 pack
[14:49:57] thread 2 is processing the 165057 / 165401 pack
[14:49:57] thread 1 is processing the 165058 / 165401 pack
[14:49:57] thread 2 is processing the 165059 / 165401 pack
[14:49:57] thread 1 is processing the 165060 / 165401 pack
[14:49:57] thread 2 is processing the 165061 / 165401 pack
[14:49:57] thread 1 is processing the 165062 / 165401 pack
[14:49:57] thread 2 is processing the 165063 / 165401 pack
[14:49:57] thread 1 is processing the 165064 / 165401 pack
[14:49:57] thread 2 is processing the 165065 / 165401 pack
[14:49:57] thread 1 is processing the 165066 / 165401 pack
[14:49:57] thread 2 is processing the 165067 / 165401 pack
[14:49:57] thread 1 is processing the 165068 / 165401 pack
[14:49:57] thread 2 is processing the 165069 / 165401 pack
[14:49:57] thread 1 is processing the 165070 / 165401 pack
[14:49:57] thread 2 is processing the 165071 / 165401 pack
[14:49:57] thread 1 is processing the 165072 / 165401 pack
[14:49:57] thread 2 is processing the 165073 / 165401 pack
[14:49:58] thread 2 is processing the 165074 / 165401 pack
[14:49:58] thread 1 is processing the 165074 / 165401 pack
[14:49:58] thread 2 is processing the 165076 / 165401 pack
[14:49:58] thread 1 is processing the 165077 / 165401 pack
[14:49:58] thread 2 is processing the 165078 / 165401 pack
[14:49:58] thread 1 is processing the 165079 / 165401 pack
[14:49:58] thread 2 is processing the 165080 / 165401 pack
[14:49:58] thread 1 is processing the 165081 / 165401 pack
[14:49:58] thread 2 is processing the 165082 / 165401 pack
[14:49:58] thread 1 is processing the 165083 / 165401 pack
[14:49:58] thread 2 is processing the 165084 / 165401 pack
[14:49:58] thread 1 is processing the 165085 / 165401 pack
[14:49:58] thread 2 is processing the 165086 / 165401 pack
[14:49:58] thread 1 is processing the 165087 / 165401 pack
[14:49:58] thread 2 is processing the 165088 / 165401 pack
[14:49:58] thread 1 is processing the 165089 / 165401 pack
[14:49:58] thread 2 is processing the 165090 / 165401 pack
[14:49:58] thread 1 is processing the 165091 / 165401 pack
[14:49:58] thread 2 is processing the 165092 / 165401 pack
[14:49:58] thread 1 is processing the 165093 / 165401 pack
[14:49:58] thread 2 is processing the 165094 / 165401 pack
[14:49:58] thread 1 is processing the 165095 / 165401 pack
[14:49:58] thread 2 is processing the 165096 / 165401 pack
[14:49:58] thread 1 is processing the 165097 / 165401 pack
[14:49:58] thread 2 is processing the 165098 / 165401 pack
[14:49:58] thread 1 is processing the 165099 / 165401 pack
[14:49:58] thread 2 is processing the 165100 / 165401 pack
[14:49:58] thread 1 is processing the 165101 / 165401 pack
[14:49:58] thread 2 is processing the 165102 / 165401 pack
[14:49:58] thread 1 is processing the 165103 / 165401 pack
[14:49:58] thread 2 is processing the 165104 / 165401 pack
[14:49:58] thread 1 is processing the 165105 / 165401 pack
[14:49:58] thread 2 is processing the 165106 / 165401 pack
[14:49:58] thread 1 is processing the 165107 / 165401 pack
[14:49:58] thread 2 is processing the 165108 / 165401 pack
[14:49:58] thread 1 is processing the 165109 / 165401 pack
[14:49:58] thread 2 is processing the 165110 / 165401 pack
[14:49:58] thread 1 is processing the 165111 / 165401 pack
[14:49:58] thread 2 is processing the 165112 / 165401 pack
[14:49:58] thread 1 is processing the 165113 / 165401 pack
[14:49:58] thread 2 is processing the 165114 / 165401 pack
[14:49:58] thread 1 is processing the 165115 / 165401 pack
[14:49:58] thread 2 is processing the 165116 / 165401 pack
[14:49:58] thread 1 is processing the 165117 / 165401 pack
[14:49:58] thread 2 is processing the 165118 / 165401 pack
[14:49:58] thread 1 is processing the 165119 / 165401 pack
[14:49:58] thread 2 is processing the 165120 / 165401 pack
[14:49:58] thread 1 is processing the 165121 / 165401 pack
[14:49:58] thread 2 is processing the 165122 / 165401 pack
[14:49:58] thread 1 is processing the 165123 / 165401 pack
[14:49:58] thread 2 is processing the 165124 / 165401 pack
[14:49:58] thread 1 is processing the 165125 / 165401 pack
[14:49:58] thread 2 is processing the 165126 / 165401 pack
[14:49:58] thread 1 is processing the 165127 / 165401 pack
[14:49:58] thread 2 is processing the 165128 / 165401 pack
[14:49:58] thread 1 is processing the 165129 / 165401 pack
[14:49:58] thread 2 is processing the 165130 / 165401 pack
[14:49:58] thread 1 is processing the 165131 / 165401 pack
[14:49:58] thread 2 is processing the 165132 / 165401 pack
[14:49:58] thread 1 is processing the 165133 / 165401 pack
[14:49:58] thread 2 is processing the 165134 / 165401 pack
[14:49:58] thread 1 is processing the 165135 / 165401 pack
[14:49:58] thread 2 is processing the 165136 / 165401 pack
[14:49:58] thread 1 is processing the 165137 / 165401 pack
[14:49:58] thread 2 is processing the 165138 / 165401 pack
[14:49:58] thread 1 is processing the 165139 / 165401 pack
[14:49:58] thread 2 is processing the 165140 / 165401 pack
[14:49:58] thread 1 is processing the 165141 / 165401 pack
[14:49:58] thread 2 is processing the 165142 / 165401 pack
[14:49:58] thread 1 is processing the 165143 / 165401 pack
[14:49:58] thread 2 is processing the 165144 / 165401 pack
[14:49:58] thread 1 is processing the 165145 / 165401 pack
[14:49:58] thread 2 is processing the 165146 / 165401 pack
[14:49:58] thread 1 is processing the 165147 / 165401 pack
[14:49:58] thread 2 is processing the 165148 / 165401 pack
[14:49:58] thread 1 is processing the 165149 / 165401 pack
[14:49:58] thread 2 is processing the 165150 / 165401 pack
[14:49:58] thread 1 is processing the 165151 / 165401 pack
[14:49:58] thread 2 is processing the 165152 / 165401 pack
[14:49:58] thread 1 is processing the 165153 / 165401 pack
[14:49:58] thread 2 is processing the 165154 / 165401 pack
[14:49:58] thread 1 is processing the 165155 / 165401 pack
[14:49:58] thread 2 is processing the 165156 / 165401 pack
[14:49:58] thread 1 is processing the 165157 / 165401 pack
[14:49:58] thread 2 is processing the 165158 / 165401 pack
[14:49:58] thread 1 is processing the 165159 / 165401 pack
[14:49:58] thread 2 is processing the 165160 / 165401 pack
[14:49:58] thread 1 is processing the 165161 / 165401 pack
[14:49:58] thread 2 is processing the 165162 / 165401 pack
[14:49:58] thread 1 is processing the 165163 / 165401 pack
[14:49:59] thread 2 is processing the 165164 / 165401 pack
[14:49:59] thread 1 is processing the 165165 / 165401 pack
[14:49:59] thread 2 is processing the 165166 / 165401 pack
[14:49:59] thread 1 is processing the 165167 / 165401 pack
[14:49:59] thread 2 is processing the 165168 / 165401 pack
[14:49:59] thread 1 is processing the 165169 / 165401 pack
[14:49:59] thread 2 is processing the 165170 / 165401 pack
[14:49:59] thread 1 is processing the 165171 / 165401 pack
[14:49:59] thread 2 is processing the 165172 / 165401 pack
[14:49:59] thread 1 is processing the 165173 / 165401 pack
[14:49:59] thread 2 is processing the 165174 / 165401 pack
[14:49:59] thread 1 is processing the 165175 / 165401 pack
[14:49:59] thread 2 is processing the 165176 / 165401 pack
[14:49:59] thread 1 is processing the 165177 / 165401 pack
[14:49:59] thread 2 is processing the 165178 / 165401 pack
[14:49:59] thread 1 is processing the 165179 / 165401 pack
[14:49:59] thread 2 is processing the 165180 / 165401 pack
[14:49:59] thread 1 is processing the 165181 / 165401 pack
[14:49:59] thread 2 is processing the 165182 / 165401 pack
[14:49:59] thread 1 is processing the 165183 / 165401 pack
[14:49:59] thread 2 is processing the 165184 / 165401 pack
[14:49:59] thread 1 is processing the 165185 / 165401 pack
[14:49:59] thread 2 is processing the 165186 / 165401 pack
[14:49:59] thread 1 is processing the 165187 / 165401 pack
[14:49:59] thread 2 is processing the 165188 / 165401 pack
[14:49:59] thread 1 is processing the 165189 / 165401 pack
[14:49:59] thread 2 is processing the 165190 / 165401 pack
[14:49:59] thread 1 is processing the 165191 / 165401 pack
[14:49:59] thread 2 is processing the 165192 / 165401 pack
[14:49:59] thread 1 is processing the 165193 / 165401 pack
[14:49:59] thread 2 is processing the 165194 / 165401 pack
[14:49:59] thread 1 is processing the 165195 / 165401 pack
[14:49:59] thread 2 is processing the 165196 / 165401 pack
[14:49:59] thread 1 is processing the 165197 / 165401 pack
[14:49:59] thread 2 is processing the 165198 / 165401 pack
[14:49:59] thread 1 is processing the 165199 / 165401 pack
[14:49:59] thread 2 is processing the 165200 / 165401 pack
[14:49:59] thread 1 is processing the 165201 / 165401 pack
[14:49:59] thread 2 is processing the 165202 / 165401 pack
[14:49:59] thread 1 is processing the 165203 / 165401 pack
[14:49:59] thread 2 is processing the 165204 / 165401 pack
[14:49:59] thread 1 is processing the 165205 / 165401 pack
[14:49:59] thread 2 is processing the 165206 / 165401 pack
[14:49:59] thread 1 is processing the 165207 / 165401 pack
[14:49:59] thread 2 is processing the 165208 / 165401 pack
[14:49:59] thread 1 is processing the 165209 / 165401 pack
[14:49:59] thread 2 is processing the 165210 / 165401 pack
[14:49:59] thread 1 is processing the 165211 / 165401 pack
[14:49:59] thread 2 is processing the 165212 / 165401 pack
[14:49:59] thread 1 is processing the 165213 / 165401 pack
[14:49:59] thread 2 is processing the 165214 / 165401 pack
[14:49:59] thread 1 is processing the 165215 / 165401 pack
[14:49:59] thread 2 is processing the 165216 / 165401 pack
[14:49:59] thread 1 is processing the 165217 / 165401 pack
[14:49:59] thread 2 is processing the 165218 / 165401 pack
[14:49:59] thread 1 is processing the 165219 / 165401 pack
[14:49:59] thread 2 is processing the 165220 / 165401 pack
[14:49:59] thread 1 is processing the 165221 / 165401 pack
[14:49:59] thread 2 is processing the 165222 / 165401 pack
[14:49:59] thread 1 is processing the 165223 / 165401 pack
[14:49:59] thread 2 is processing the 165224 / 165401 pack
[14:49:59] thread 1 is processing the 165225 / 165401 pack
[14:49:59] thread 2 is processing the 165226 / 165401 pack
[14:49:59] thread 1 is processing the 165227 / 165401 pack
[14:49:59] thread 2 is processing the 165228 / 165401 pack
[14:49:59] thread 1 is processing the 165229 / 165401 pack
[14:49:59] thread 2 is processing the 165230 / 165401 pack
[14:49:59] thread 1 is processing the 165231 / 165401 pack
[14:49:59] thread 2 is processing the 165232 / 165401 pack
[14:49:59] thread 1 is processing the 165233 / 165401 pack
[14:49:59] thread 2 is processing the 165234 / 165401 pack
[14:49:59] thread 1 is processing the 165235 / 165401 pack
[14:49:59] thread 2 is processing the 165236 / 165401 pack
[14:49:59] thread 1 is processing the 165237 / 165401 pack
[14:49:59] thread 2 is processing the 165238 / 165401 pack
[14:49:59] thread 1 is processing the 165239 / 165401 pack
[14:49:59] thread 2 is processing the 165240 / 165401 pack
[14:49:59] thread 1 is processing the 165241 / 165401 pack
[14:49:59] thread 2 is processing the 165242 / 165401 pack
[14:49:59] thread 1 is processing the 165243 / 165401 pack
[14:49:59] thread 2 is processing the 165244 / 165401 pack
[14:49:59] thread 1 is processing the 165245 / 165401 pack
[14:49:59] thread 2 is processing the 165246 / 165401 pack
[14:49:59] thread 1 is processing the 165247 / 165401 pack
[14:49:59] thread 2 is processing the 165248 / 165401 pack
[14:49:59] thread 1 is processing the 165249 / 165401 pack
[14:49:59] thread 2 is processing the 165250 / 165401 pack
[14:50:0] thread 1 is processing the 165251 / 165401 pack
[14:50:0] thread 2 is processing the 165252 / 165401 pack
[14:50:0] thread 1 is processing the 165253 / 165401 pack
[14:50:0] thread 2 is processing the 165254 / 165401 pack
[14:50:0] thread 1 is processing the 165255 / 165401 pack
[14:50:0] thread 2 is processing the 165256 / 165401 pack
[14:50:0] thread 1 is processing the 165257 / 165401 pack
[14:50:0] thread 2 is processing the 165258 / 165401 pack
[14:50:0] thread 1 is processing the 165259 / 165401 pack
[14:50:0] thread 2 is processing the 165260 / 165401 pack
[14:50:0] thread 1 is processing the 165261 / 165401 pack
[14:50:0] thread 2 is processing the 165262 / 165401 pack
[14:50:0] thread 1 is processing the 165263 / 165401 pack
[14:50:0] thread 2 is processing the 165264 / 165401 pack
[14:50:0] thread 1 is processing the 165265 / 165401 pack
[14:50:0] thread 2 is processing the 165266 / 165401 pack
[14:50:0] thread 1 is processing the 165267 / 165401 pack
[14:50:0] thread 2 is processing the 165268 / 165401 pack
[14:50:0] thread 1 is processing the 165269 / 165401 pack
[14:50:0] thread 2 is processing the 165270 / 165401 pack
[14:50:0] thread 1 is processing the 165271 / 165401 pack
[14:50:0] thread 2 is processing the 165272 / 165401 pack
[14:50:0] thread 1 is processing the 165273 / 165401 pack
[14:50:0] thread 2 is processing the 165274 / 165401 pack
[14:50:0] thread 1 is processing the 165275 / 165401 pack
[14:50:0] thread 2 is processing the 165276 / 165401 pack
[14:50:0] thread 1 is processing the 165277 / 165401 pack
[14:50:0] thread 2 is processing the 165277 / 165401 pack
[14:50:0] thread 2 is processing the 165279 / 165401 pack
[14:50:0] thread 1 is processing the 165280 / 165401 pack
[14:50:0] thread 1 is processing the 165281 / 165401 pack
[14:50:0] thread 2 is processing the 165282 / 165401 pack
[14:50:0] thread 1 is processing the 165283 / 165401 pack
[14:50:0] thread 2 is processing the 165284 / 165401 pack
[14:50:0] thread 2 is processing the 165285 / 165401 pack
[14:50:0] thread 1 is processing the 165286 / 165401 pack
[14:50:0] thread 2 is processing the 165287 / 165401 pack
[14:50:0] thread 1 is processing the 165288 / 165401 pack
[14:50:0] thread 1 is processing the 165289 / 165401 pack
[14:50:0] thread 2 is processing the 165290 / 165401 pack
[14:50:0] thread 1 is processing the 165291 / 165401 pack
[14:50:0] thread 2 is processing the 165292 / 165401 pack
[14:50:0] thread 1 is processing the 165293 / 165401 pack
[14:50:0] thread 2 is processing the 165294 / 165401 pack
[14:50:0] thread 2 is processing the 165295 / 165401 pack
[14:50:0] thread 1 is processing the 165296 / 165401 pack
[14:50:0] thread 2 is processing the 165297 / 165401 pack
[14:50:0] thread 1 is processing the 165298 / 165401 pack
[14:50:0] thread 2 is processing the 165299 / 165401 pack
[14:50:0] thread 1 is processing the 165300 / 165401 pack
[14:50:0] thread 1 is processing the 165301 / 165401 pack
[14:50:0] thread 2 is processing the 165302 / 165401 pack
[14:50:0] thread 2 is processing the 165303 / 165401 pack
[14:50:0] thread 1 is processing the 165304 / 165401 pack
[14:50:0] thread 2 is processing the 165305 / 165401 pack
[14:50:0] thread 1 is processing the 165306 / 165401 pack
[14:50:0] thread 2 is processing the 165307 / 165401 pack
[14:50:0] thread 1 is processing the 165308 / 165401 pack
[14:50:0] thread 2 is processing the 165309 / 165401 pack
[14:50:0] thread 1 is processing the 165310 / 165401 pack
[14:50:0] thread 2 is processing the 165311 / 165401 pack
[14:50:0] thread 1 is processing the 165312 / 165401 pack
[14:50:0] thread 2 is processing the 165313 / 165401 pack
[14:50:0] thread 1 is processing the 165314 / 165401 pack
[14:50:0] thread 2 is processing the 165315 / 165401 pack
[14:50:0] thread 1 is processing the 165316 / 165401 pack
[14:50:0] thread 2 is processing the 165317 / 165401 pack
[14:50:0] thread 1 is processing the 165318 / 165401 pack
[14:50:0] thread 2 is processing the 165319 / 165401 pack
[14:50:0] thread 1 is processing the 165320 / 165401 pack
[14:50:0] thread 1 is processing the 165321 / 165401 pack
[14:50:0] thread 2 is processing the 165322 / 165401 pack
[14:50:0] thread 2 is processing the 165323 / 165401 pack
[14:50:0] thread 1 is processing the 165323 / 165401 pack
[14:50:0] thread 1 is processing the 165325 / 165401 pack
[14:50:0] thread 2 is processing the 165326 / 165401 pack
[14:50:0] thread 1 is processing the 165327 / 165401 pack
[14:50:0] thread 2 is processing the 165328 / 165401 pack
[14:50:0] thread 1 is processing the 165329 / 165401 pack
[14:50:0] thread 2 is processing the 165330 / 165401 pack
[14:50:0] thread 1 is processing the 165331 / 165401 pack
[14:50:0] thread 2 is processing the 165332 / 165401 pack
[14:50:1] thread 1 is processing the 165333 / 165401 pack
[14:50:1] thread 2 is processing the 165334 / 165401 pack
[14:50:1] thread 1 is processing the 165335 / 165401 pack
[14:50:1] thread 2 is processing the 165336 / 165401 pack
[14:50:1] thread 1 is processing the 165337 / 165401 pack
[14:50:1] thread 2 is processing the 165338 / 165401 pack
[14:50:1] thread 1 is processing the 165339 / 165401 pack
[14:50:1] thread 2 is processing the 165340 / 165401 pack
[14:50:1] thread 1 is processing the 165341 / 165401 pack
[14:50:1] thread 2 is processing the 165342 / 165401 pack
[14:50:1] thread 1 is processing the 165343 / 165401 pack
[14:50:1] thread 2 is processing the 165344 / 165401 pack
[14:50:1] thread 1 is processing the 165345 / 165401 pack
[14:50:1] thread 2 is processing the 165346 / 165401 pack
[14:50:1] thread 1 is processing the 165347 / 165401 pack
[14:50:1] thread 2 is processing the 165348 / 165401 pack
[14:50:1] thread 1 is processing the 165349 / 165401 pack
[14:50:1] thread 2 is processing the 165350 / 165401 pack
[14:50:1] thread 1 is processing the 165351 / 165401 pack
[14:50:1] thread 2 is processing the 165352 / 165401 pack
[14:50:1] thread 1 is processing the 165353 / 165401 pack
[14:50:1] thread 2 is processing the 165354 / 165401 pack
[14:50:1] thread 1 is processing the 165355 / 165401 pack
[14:50:1] thread 2 is processing the 165356 / 165401 pack
[14:50:1] thread 1 is processing the 165357 / 165401 pack
[14:50:1] thread 2 is processing the 165358 / 165401 pack
[14:50:1] thread 1 is processing the 165359 / 165401 pack
[14:50:1] thread 2 is processing the 165360 / 165401 pack
[14:50:1] thread 1 is processing the 165361 / 165401 pack
[14:50:1] thread 2 is processing the 165362 / 165401 pack
[14:50:1] thread 1 is processing the 165363 / 165401 pack
[14:50:1] thread 2 is processing the 165364 / 165401 pack
[14:50:1] thread 1 is processing the 165365 / 165401 pack
[14:50:1] thread 2 is processing the 165366 / 165401 pack
[14:50:1] thread 1 is processing the 165367 / 165401 pack
[14:50:1] thread 2 is processing the 165368 / 165401 pack
[14:50:1] thread 1 is processing the 165369 / 165401 pack
[14:50:1] thread 2 is processing the 165370 / 165401 pack
[14:50:1] thread 1 is processing the 165371 / 165401 pack
[14:50:1] thread 2 is processing the 165372 / 165401 pack
[14:50:1] thread 1 is processing the 165373 / 165401 pack
[14:50:1] thread 2 is processing the 165374 / 165401 pack
[14:50:1] thread 1 is processing the 165375 / 165401 pack
[14:50:1] thread 2 is processing the 165376 / 165401 pack
[14:50:1] thread 1 is processing the 165377 / 165401 pack
[14:50:1] thread 2 is processing the 165378 / 165401 pack
[14:50:1] thread 1 is processing the 165379 / 165401 pack
[14:50:1] thread 2 is processing the 165380 / 165401 pack
[14:50:1] thread 1 is processing the 165381 / 165401 pack
[14:50:1] thread 2 is processing the 165382 / 165401 pack
[14:50:1] thread 1 is processing the 165383 / 165401 pack
[14:50:1] thread 2 is processing the 165384 / 165401 pack
[14:50:1] thread 1 is processing the 165385 / 165401 pack
[14:50:1] thread 2 is processing the 165386 / 165401 pack
[14:50:1] thread 1 is processing the 165387 / 165401 pack
[14:50:1] thread 2 is processing the 165388 / 165401 pack
[14:50:1] thread 1 is processing the 165389 / 165401 pack
[14:50:1] thread 2 is processing the 165390 / 165401 pack
[14:50:1] thread 1 is processing the 165391 / 165401 pack
[14:50:1] thread 2 is processing the 165392 / 165401 pack
[14:50:1] thread 1 is processing the 165393 / 165401 pack
[14:50:1] thread 2 is processing the 165394 / 165401 pack
[14:50:1] thread 1 is processing the 165395 / 165401 pack
[14:50:1] thread 2 is processing the 165396 / 165401 pack
[14:50:1] thread 1 is processing the 165397 / 165401 pack
[14:50:1] thread 2 is processing the 165398 / 165401 pack
[14:50:1] thread 2 is processing the 165399 / 165401 pack
[14:50:1] thread 1 is processing the 165400 / 165401 pack
[14:50:1] thread 1 data processing completed
[14:50:1] thread 1 finished
[14:50:1] thread 2 data processing completed
[14:50:1] thread 2 finished
[14:50:1] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/Prepupae_QC_R2.fastq.gz writer finished
[14:50:1] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/Prepupae_QC_R1.fastq.gz writer finished
[14:50:1] start to generate reports

Read1 before filtering:
total reads: 165400633
total bases: 24975495583
Q20 bases: 24450739973(97.8989%)
Q30 bases: 23584077115(94.4289%)

Read2 before filtering:
total reads: 165400633
total bases: 24975495583
Q20 bases: 24205198671(96.9158%)
Q30 bases: 23116421835(92.5564%)

Read1 after filtering:
total reads: 162713828
total bases: 24313009925
Q20 bases: 23860471531(98.1387%)
Q30 bases: 23045547771(94.7869%)

Read2 after filtering:
total reads: 162713828
total bases: 24311383251
Q20 bases: 23758153521(97.7244%)
Q30 bases: 22743295099(93.55%)

Filtering result:
reads passed filter: 325427656
reads failed due to low quality: 5223666
reads failed due to too many N: 14784
reads failed due to too short: 135160
reads with adapter trimmed: 27430922
bases trimmed due to adapters: 513497699

Duplication rate: 6.4997%

Insert size peak (evaluated by paired-end reads): 198

Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[13:58:21] start to load data
[13:58:30] loaded 1M read pairs
[13:58:42] loaded 2M read pairs
[13:58:55] loaded 3M read pairs
[13:59:8] loaded 4M read pairs
[13:59:21] loaded 5M read pairs
[13:59:34] loaded 6M read pairs
[13:59:47] loaded 7M read pairs
[14:0:0] loaded 8M read pairs
[14:0:13] loaded 9M read pairs
[14:0:26] loaded 10M read pairs
[14:0:40] loaded 11M read pairs
[14:0:53] loaded 12M read pairs
[14:1:6] loaded 13M read pairs
[14:1:18] loaded 14M read pairs
[14:1:30] loaded 15M read pairs
[14:1:41] loaded 16M read pairs
[14:1:53] loaded 17M read pairs
[14:2:3] loaded 18M read pairs
[14:2:14] loaded 19M read pairs
[14:2:25] loaded 20M read pairs
[14:2:35] loaded 21M read pairs
[14:2:46] loaded 22M read pairs
[14:2:59] loaded 23M read pairs
[14:3:10] loaded 24M read pairs
[14:3:20] loaded 25M read pairs
[14:3:31] loaded 26M read pairs
[14:3:42] loaded 27M read pairs
[14:3:52] loaded 28M read pairs
[14:4:3] loaded 29M read pairs
[14:4:14] loaded 30M read pairs
[14:4:25] loaded 31M read pairs
[14:4:35] loaded 32M read pairs
[14:4:46] loaded 33M read pairs
[14:4:57] loaded 34M read pairs
[14:5:8] loaded 35M read pairs
[14:5:19] loaded 36M read pairs
[14:5:29] loaded 37M read pairs
[14:5:40] loaded 38M read pairs
[14:5:50] loaded 39M read pairs
[14:6:1] loaded 40M read pairs
[14:6:12] loaded 41M read pairs
[14:6:23] loaded 42M read pairs
[14:6:34] loaded 43M read pairs
[14:6:44] loaded 44M read pairs
[14:6:55] loaded 45M read pairs
[14:7:6] loaded 46M read pairs
[14:7:17] loaded 47M read pairs
[14:7:28] loaded 48M read pairs
[14:7:38] loaded 49M read pairs
[14:7:49] loaded 50M read pairs
[14:8:0] loaded 51M read pairs
[14:8:10] loaded 52M read pairs
[14:8:21] loaded 53M read pairs
[14:8:32] loaded 54M read pairs
[14:8:42] loaded 55M read pairs
[14:8:53] loaded 56M read pairs
[14:9:4] loaded 57M read pairs
[14:9:14] loaded 58M read pairs
[14:9:25] loaded 59M read pairs
[14:9:36] loaded 60M read pairs
[14:9:47] loaded 61M read pairs
[14:9:57] loaded 62M read pairs
[14:10:8] loaded 63M read pairs
[14:10:19] loaded 64M read pairs
[14:10:29] loaded 65M read pairs
[14:10:40] loaded 66M read pairs
[14:10:51] loaded 67M read pairs
[14:11:2] loaded 68M read pairs
[14:11:12] loaded 69M read pairs
[14:11:23] loaded 70M read pairs
[14:11:34] loaded 71M read pairs
[14:11:44] loaded 72M read pairs
[14:11:55] loaded 73M read pairs
[14:12:6] loaded 74M read pairs
[14:12:17] loaded 75M read pairs
[14:12:27] loaded 76M read pairs
[14:12:38] loaded 77M read pairs
[14:12:49] loaded 78M read pairs
[14:13:0] loaded 79M read pairs
[14:13:11] loaded 80M read pairs
[14:13:21] loaded 81M read pairs
[14:13:32] loaded 82M read pairs
[14:13:43] loaded 83M read pairs
[14:13:54] loaded 84M read pairs
[14:14:6] loaded 85M read pairs
[14:14:17] loaded 86M read pairs
[14:14:29] loaded 87M read pairs
[14:14:40] loaded 88M read pairs
[14:14:51] loaded 89M read pairs
[14:15:3] loaded 90M read pairs
[14:15:14] loaded 91M read pairs
Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[14:17:43] start to load data
[14:17:51] loaded 1M read pairs
[14:18:1] loaded 2M read pairs
[14:18:12] loaded 3M read pairs
[14:18:23] loaded 4M read pairs
[14:18:34] loaded 5M read pairs
[14:18:45] loaded 6M read pairs
[14:18:56] loaded 7M read pairs
[14:19:7] loaded 8M read pairs
[14:19:19] loaded 9M read pairs
[14:19:30] loaded 10M read pairs
[14:19:41] loaded 11M read pairs
[14:19:52] loaded 12M read pairs
[14:20:3] loaded 13M read pairs
[14:20:14] loaded 14M read pairs
[14:20:25] loaded 15M read pairs
[14:20:36] loaded 16M read pairs
[14:20:47] loaded 17M read pairs
[14:20:58] loaded 18M read pairs
[14:21:9] loaded 19M read pairs
[14:21:20] loaded 20M read pairs
[14:21:31] loaded 21M read pairs
[14:21:42] loaded 22M read pairs
[14:21:53] loaded 23M read pairs
[14:22:5] loaded 24M read pairs
[14:22:16] loaded 25M read pairs
[14:22:28] loaded 26M read pairs
[14:22:39] loaded 27M read pairs
[14:22:50] loaded 28M read pairs
[14:23:1] loaded 29M read pairs
[14:23:12] loaded 30M read pairs
[14:23:23] loaded 31M read pairs
[14:23:34] loaded 32M read pairs
[14:23:45] loaded 33M read pairs
[14:23:56] loaded 34M read pairs
[14:24:7] loaded 35M read pairs
[14:24:18] loaded 36M read pairs
[14:24:28] loaded 37M read pairs
[14:24:39] loaded 38M read pairs
[14:24:50] loaded 39M read pairs
[14:25:2] loaded 40M read pairs
[14:25:14] loaded 41M read pairs
[14:25:26] loaded 42M read pairs
[14:25:38] loaded 43M read pairs
[14:25:50] loaded 44M read pairs
[14:26:2] loaded 45M read pairs
[14:26:14] loaded 46M read pairs
[14:26:26] loaded 47M read pairs
[14:26:37] loaded 48M read pairs
[14:26:49] loaded 49M read pairs
[14:27:0] loaded 50M read pairs
[14:27:11] loaded 51M read pairs
[14:27:22] loaded 52M read pairs
[14:27:33] loaded 53M read pairs
[14:27:44] loaded 54M read pairs
[14:27:55] loaded 55M read pairs
[14:28:6] loaded 56M read pairs
[14:28:17] loaded 57M read pairs
[14:28:29] loaded 58M read pairs
[14:28:40] loaded 59M read pairs
[14:28:52] loaded 60M read pairs
[14:29:3] loaded 61M read pairs
[14:29:15] loaded 62M read pairs
[14:29:26] loaded 63M read pairs
[14:29:38] loaded 64M read pairs
[14:29:49] loaded 65M read pairs
[14:30:1] loaded 66M read pairs
[14:30:12] loaded 67M read pairs
[14:30:23] loaded 68M read pairs
[14:30:34] loaded 69M read pairs
[14:30:45] loaded 70M read pairs
[14:30:57] loaded 71M read pairs
[14:31:8] loaded 72M read pairs
[14:31:19] loaded 73M read pairs
[14:31:31] loaded 74M read pairs
[14:31:42] loaded 75M read pairs
[14:31:53] loaded 76M read pairs
[14:32:4] loaded 77M read pairs
[14:32:15] loaded 78M read pairs
[14:32:26] loaded 79M read pairs
[14:32:38] loaded 80M read pairs
[14:32:49] loaded 81M read pairs
[14:33:1] loaded 82M read pairs
[14:33:13] loaded 83M read pairs
[14:33:25] loaded 84M read pairs
[14:33:36] loaded 85M read pairs
[14:33:47] loaded 86M read pairs
[14:33:59] loaded 87M read pairs
[14:34:10] loaded 88M read pairs
[14:34:22] loaded 89M read pairs
[14:34:33] loaded 90M read pairs
[14:34:44] loaded 91M read pairs
[14:34:57] loaded 92M read pairs
[14:35:8] loaded 93M read pairs
[14:35:21] loaded 94M read pairs
[14:35:32] loaded 95M read pairs
[14:35:44] loaded 96M read pairs
[14:35:56] loaded 97M read pairs
[14:36:8] loaded 98M read pairs
[14:36:19] loaded 99M read pairs
[14:36:31] loaded 100M read pairs
[14:36:43] loaded 101M read pairs
[14:36:55] loaded 102M read pairs
[14:37:7] loaded 103M read pairs
[14:37:19] loaded 104M read pairs
[14:37:30] loaded 105M read pairs
[14:37:42] loaded 106M read pairs
[14:37:53] loaded 107M read pairs
[14:38:5] loaded 108M read pairs
[14:38:17] loaded 109M read pairs
[14:38:29] loaded 110M read pairs
[14:38:41] loaded 111M read pairs
[14:38:53] loaded 112M read pairs
[14:39:4] loaded 113M read pairs
[14:39:16] loaded 114M read pairs
[14:39:28] loaded 115M read pairs
[14:39:39] loaded 116M read pairs
[14:39:51] loaded 117M read pairs
[14:40:2] loaded 118M read pairs
[14:40:14] loaded 119M read pairs
[14:40:26] loaded 120M read pairs
[14:40:38] loaded 121M read pairs
[14:40:50] loaded 122M read pairs
[14:41:2] loaded 123M read pairs
[14:41:14] loaded 124M read pairs
[14:41:25] loaded 125M read pairs
[14:41:36] loaded 126M read pairs
[14:41:48] loaded 127M read pairs
[14:41:59] loaded 128M read pairs
[14:42:10] loaded 129M read pairs
[14:42:22] loaded 130M read pairs
[14:42:33] loaded 131M read pairs
[14:42:44] loaded 132M read pairs
[14:42:55] loaded 133M read pairs
[14:43:7] loaded 134M read pairs
[14:43:19] loaded 135M read pairs
[14:43:30] loaded 136M read pairs
[14:43:42] loaded 137M read pairs
[14:43:54] loaded 138M read pairs
[14:44:5] loaded 139M read pairs
[14:44:17] loaded 140M read pairs
[14:44:28] loaded 141M read pairs
[14:44:40] loaded 142M read pairs
[14:44:52] loaded 143M read pairs
[14:45:4] loaded 144M read pairs
[14:45:16] loaded 145M read pairs
[14:45:28] loaded 146M read pairs
[14:45:39] loaded 147M read pairs
[14:45:50] loaded 148M read pairs
[14:46:2] loaded 149M read pairs
[14:46:13] loaded 150M read pairs
[14:46:24] loaded 151M read pairs
[14:46:35] loaded 152M read pairs
[14:46:46] loaded 153M read pairs
[14:46:58] loaded 154M read pairs
[14:47:9] loaded 155M read pairs
[14:47:20] loaded 156M read pairs
[14:47:31] loaded 157M read pairs
[14:47:43] loaded 158M read pairs
[14:47:55] loaded 159M read pairs
[14:48:6] loaded 160M read pairs
[14:48:17] loaded 161M read pairs
[14:48:28] loaded 162M read pairs
[14:48:40] loaded 163M read pairs
[14:48:51] loaded 164M read pairs
[14:49:2] loaded 165M read pairs
[14:49:13] loaded 166M read pairs
[14:49:24] loaded 167M read pairs
[14:49:36] loaded 168M read pairs
[14:49:47] loaded 169M read pairs
[14:49:58] loaded 170M read pairs
[14:50:10] loaded 171M read pairs
[14:50:23] loaded 172M read pairs
[14:50:36] loaded 173M read pairs
[14:50:49] loaded 174M read pairs
[14:51:2] loaded 175M read pairs
[14:51:15] loaded 176M read pairs
[14:51:27] loaded 177M read pairs
[14:51:39] loaded 178M read pairs
[14:51:43] all reads loaded, start to monitor thread status
[14:51:43] thread 1 is processing the 177836 / 178337 pack
[14:51:43] thread 2 is processing the 177837 / 178337 pack
[14:51:43] thread 1 is processing the 177838 / 178337 pack
[14:51:43] thread 2 is processing the 177839 / 178337 pack
[14:51:43] thread 1 is processing the 177840 / 178337 pack
[14:51:43] thread 2 is processing the 177841 / 178337 pack
[14:51:43] thread 2 is processing the 177842 / 178337 pack
[14:51:43] thread 1 is processing the 177843 / 178337 pack
[14:51:44] thread 2 is processing the 177844 / 178337 pack
[14:51:44] thread 1 is processing the 177845 / 178337 pack
[14:51:44] thread 2 is processing the 177846 / 178337 pack
[14:51:44] thread 1 is processing the 177847 / 178337 pack
[14:51:44] thread 2 is processing the 177848 / 178337 pack
[14:51:44] thread 1 is processing the 177849 / 178337 pack
[14:51:44] thread 2 is processing the 177850 / 178337 pack
[14:51:44] thread 1 is processing the 177851 / 178337 pack
[14:51:44] thread 2 is processing the 177852 / 178337 pack
[14:51:44] thread 1 is processing the 177853 / 178337 pack
[14:51:44] thread 2 is processing the 177854 / 178337 pack
[14:51:44] thread 1 is processing the 177855 / 178337 pack
[14:51:44] thread 2 is processing the 177856 / 178337 pack
[14:51:44] thread 1 is processing the 177857 / 178337 pack
[14:51:44] thread 2 is processing the 177858 / 178337 pack
[14:51:44] thread 1 is processing the 177859 / 178337 pack
[14:51:44] thread 2 is processing the 177860 / 178337 pack
[14:51:44] thread 1 is processing the 177861 / 178337 pack
[14:51:44] thread 2 is processing the 177862 / 178337 pack
[14:51:44] thread 1 is processing the 177863 / 178337 pack
[14:51:44] thread 2 is processing the 177864 / 178337 pack
[14:51:44] thread 1 is processing the 177865 / 178337 pack
[14:51:44] thread 2 is processing the 177866 / 178337 pack
[14:51:44] thread 1 is processing the 177867 / 178337 pack
[14:51:44] thread 2 is processing the 177868 / 178337 pack
[14:51:44] thread 1 is processing the 177869 / 178337 pack
[14:51:44] thread 2 is processing the 177870 / 178337 pack
[14:51:44] thread 1 is processing the 177871 / 178337 pack
[14:51:44] thread 2 is processing the 177872 / 178337 pack
[14:51:44] thread 1 is processing the 177873 / 178337 pack
[14:51:44] thread 2 is processing the 177874 / 178337 pack
[14:51:44] thread 1 is processing the 177875 / 178337 pack
[14:51:44] thread 2 is processing the 177876 / 178337 pack
[14:51:44] thread 1 is processing the 177877 / 178337 pack
[14:51:44] thread 2 is processing the 177878 / 178337 pack
[14:51:44] thread 1 is processing the 177879 / 178337 pack
[14:51:44] thread 2 is processing the 177880 / 178337 pack
[14:51:44] thread 1 is processing the 177881 / 178337 pack
[14:51:44] thread 2 is processing the 177882 / 178337 pack
[14:51:44] thread 1 is processing the 177883 / 178337 pack
[14:51:44] thread 2 is processing the 177884 / 178337 pack
[14:51:44] thread 1 is processing the 177885 / 178337 pack
[14:51:44] thread 2 is processing the 177886 / 178337 pack
[14:51:44] thread 1 is processing the 177887 / 178337 pack
[14:51:44] thread 2 is processing the 177888 / 178337 pack
[14:51:44] thread 1 is processing the 177889 / 178337 pack
[14:51:44] thread 2 is processing the 177890 / 178337 pack
[14:51:44] thread 1 is processing the 177891 / 178337 pack
[14:51:44] thread 2 is processing the 177892 / 178337 pack
[14:51:44] thread 1 is processing the 177893 / 178337 pack
[14:51:44] thread 2 is processing the 177894 / 178337 pack
[14:51:44] thread 1 is processing the 177895 / 178337 pack
[14:51:44] thread 2 is processing the 177896 / 178337 pack
[14:51:44] thread 1 is processing the 177897 / 178337 pack
[14:51:44] thread 2 is processing the 177898 / 178337 pack
[14:51:44] thread 1 is processing the 177899 / 178337 pack
[14:51:44] thread 2 is processing the 177900 / 178337 pack
[14:51:44] thread 1 is processing the 177901 / 178337 pack
[14:51:44] thread 2 is processing the 177902 / 178337 pack
[14:51:44] thread 1 is processing the 177903 / 178337 pack
[14:51:44] thread 2 is processing the 177904 / 178337 pack
[14:51:44] thread 1 is processing the 177905 / 178337 pack
[14:51:44] thread 2 is processing the 177906 / 178337 pack
[14:51:44] thread 1 is processing the 177907 / 178337 pack
[14:51:44] thread 2 is processing the 177908 / 178337 pack
[14:51:44] thread 1 is processing the 177909 / 178337 pack
[14:51:44] thread 2 is processing the 177910 / 178337 pack
[14:51:44] thread 1 is processing the 177911 / 178337 pack
[14:51:44] thread 2 is processing the 177912 / 178337 pack
[14:51:44] thread 1 is processing the 177913 / 178337 pack
[14:51:44] thread 2 is processing the 177914 / 178337 pack
[14:51:44] thread 1 is processing the 177915 / 178337 pack
[14:51:44] thread 1 is processing the 177916 / 178337 pack
[14:51:44] thread 2 is processing the 177917 / 178337 pack
[14:51:44] thread 1 is processing the 177918 / 178337 pack
[14:51:44] thread 2 is processing the 177919 / 178337 pack
[14:51:44] thread 1 is processing the 177920 / 178337 pack
[14:51:44] thread 2 is processing the 177921 / 178337 pack
[14:51:44] thread 1 is processing the 177922 / 178337 pack
[14:51:44] thread 2 is processing the 177923 / 178337 pack
[14:51:44] thread 1 is processing the 177924 / 178337 pack
[14:51:45] thread 2 is processing the 177925 / 178337 pack
[14:51:45] thread 1 is processing the 177926 / 178337 pack
[14:51:45] thread 2 is processing the 177927 / 178337 pack
[14:51:45] thread 1 is processing the 177928 / 178337 pack
[14:51:45] thread 2 is processing the 177929 / 178337 pack
[14:51:45] thread 1 is processing the 177930 / 178337 pack
[14:51:45] thread 2 is processing the 177931 / 178337 pack
[14:51:45] thread 1 is processing the 177932 / 178337 pack
[14:51:45] thread 2 is processing the 177933 / 178337 pack
[14:51:45] thread 1 is processing the 177934 / 178337 pack
[14:51:45] thread 2 is processing the 177935 / 178337 pack
[14:51:45] thread 1 is processing the 177936 / 178337 pack
[14:51:45] thread 2 is processing the 177937 / 178337 pack
[14:51:45] thread 1 is processing the 177938 / 178337 pack
[14:51:45] thread 2 is processing the 177939 / 178337 pack
[14:51:45] thread 1 is processing the 177940 / 178337 pack
[14:51:45] thread 2 is processing the 177941 / 178337 pack
[14:51:45] thread 1 is processing the 177942 / 178337 pack
[14:51:45] thread 2 is processing the 177943 / 178337 pack
[14:51:45] thread 1 is processing the 177944 / 178337 pack
[14:51:45] thread 2 is processing the 177945 / 178337 pack
[14:51:45] thread 1 is processing the 177946 / 178337 pack
[14:51:45] thread 2 is processing the 177947 / 178337 pack
[14:51:45] thread 1 is processing the 177948 / 178337 pack
[14:51:45] thread 2 is processing the 177949 / 178337 pack
[14:51:45] thread 1 is processing the 177950 / 178337 pack
[14:51:45] thread 2 is processing the 177951 / 178337 pack
[14:51:45] thread 1 is processing the 177952 / 178337 pack
[14:51:45] thread 2 is processing the 177953 / 178337 pack
[14:51:45] thread 1 is processing the 177954 / 178337 pack
[14:51:45] thread 1 is processing the 177955 / 178337 pack
[14:51:45] thread 2 is processing the 177956 / 178337 pack
[14:51:45] thread 1 is processing the 177957 / 178337 pack
[14:51:45] thread 2 is processing the 177958 / 178337 pack
[14:51:45] thread 1 is processing the 177959 / 178337 pack
[14:51:45] thread 2 is processing the 177960 / 178337 pack
[14:51:45] thread 1 is processing the 177961 / 178337 pack
[14:51:45] thread 2 is processing the 177962 / 178337 pack
[14:51:45] thread 1 is processing the 177963 / 178337 pack
[14:51:45] thread 2 is processing the 177964 / 178337 pack
[14:51:45] thread 1 is processing the 177965 / 178337 pack
[14:51:45] thread 2 is processing the 177966 / 178337 pack
[14:51:45] thread 1 is processing the 177967 / 178337 pack
[14:51:45] thread 2 is processing the 177968 / 178337 pack
[14:51:45] thread 1 is processing the 177969 / 178337 pack
[14:51:45] thread 2 is processing the 177970 / 178337 pack
[14:51:45] thread 1 is processing the 177971 / 178337 pack
[14:51:45] thread 2 is processing the 177972 / 178337 pack
[14:51:45] thread 1 is processing the 177973 / 178337 pack
[14:51:45] thread 2 is processing the 177974 / 178337 pack
[14:51:45] thread 1 is processing the 177975 / 178337 pack
[14:51:45] thread 2 is processing the 177976 / 178337 pack
[14:51:45] thread 1 is processing the 177977 / 178337 pack
[14:51:45] thread 2 is processing the 177978 / 178337 pack
[14:51:45] thread 1 is processing the 177979 / 178337 pack
[14:51:45] thread 1 is processing the 177980 / 178337 pack
[14:51:45] thread 2 is processing the 177981 / 178337 pack
[14:51:45] thread 1 is processing the 177982 / 178337 pack
[14:51:45] thread 2 is processing the 177983 / 178337 pack
[14:51:45] thread 1 is processing the 177984 / 178337 pack
[14:51:45] thread 2 is processing the 177985 / 178337 pack
[14:51:45] thread 1 is processing the 177986 / 178337 pack
[14:51:45] thread 2 is processing the 177987 / 178337 pack
[14:51:45] thread 1 is processing the 177988 / 178337 pack
[14:51:45] thread 2 is processing the 177989 / 178337 pack
[14:51:45] thread 1 is processing the 177990 / 178337 pack
[14:51:45] thread 2 is processing the 177991 / 178337 pack
[14:51:45] thread 1 is processing the 177992 / 178337 pack
[14:51:45] thread 2 is processing the 177993 / 178337 pack
[14:51:45] thread 1 is processing the 177994 / 178337 pack
[14:51:45] thread 2 is processing the 177995 / 178337 pack
[14:51:45] thread 1 is processing the 177996 / 178337 pack
[14:51:45] thread 2 is processing the 177997 / 178337 pack
[14:51:45] thread 1 is processing the 177998 / 178337 pack
[14:51:45] thread 2 is processing the 177999 / 178337 pack
[14:51:45] thread 1 is processing the 178000 / 178337 pack
[14:51:45] thread 1 is processing the 178001 / 178337 pack
[14:51:45] thread 2 is processing the 178002 / 178337 pack
[14:51:45] thread 1 is processing the 178003 / 178337 pack
[14:51:45] thread 2 is processing the 178004 / 178337 pack
[14:51:45] thread 1 is processing the 178005 / 178337 pack
[14:51:45] thread 2 is processing the 178006 / 178337 pack
[14:51:46] thread 1 is processing the 178007 / 178337 pack
[14:51:46] thread 2 is processing the 178008 / 178337 pack
[14:51:46] thread 1 is processing the 178009 / 178337 pack
[14:51:46] thread 2 is processing the 178010 / 178337 pack
[14:51:46] thread 1 is processing the 178011 / 178337 pack
[14:51:46] thread 2 is processing the 178012 / 178337 pack
[14:51:46] thread 1 is processing the 178013 / 178337 pack
[14:51:46] thread 2 is processing the 178014 / 178337 pack
[14:51:46] thread 1 is processing the 178015 / 178337 pack
[14:51:46] thread 2 is processing the 178016 / 178337 pack
[14:51:46] thread 1 is processing the 178017 / 178337 pack
[14:51:46] thread 2 is processing the 178018 / 178337 pack
[14:51:46] thread 1 is processing the 178019 / 178337 pack
[14:51:46] thread 2 is processing the 178020 / 178337 pack
[14:51:46] thread 1 is processing the 178021 / 178337 pack
[14:51:46] thread 2 is processing the 178022 / 178337 pack
[14:51:46] thread 1 is processing the 178023 / 178337 pack
[14:51:46] thread 2 is processing the 178024 / 178337 pack
[14:51:46] thread 1 is processing the 178025 / 178337 pack
[14:51:46] thread 2 is processing the 178026 / 178337 pack
[14:51:46] thread 1 is processing the 178027 / 178337 pack
[14:51:46] thread 1 is processing the 178028 / 178337 pack
[14:51:46] thread 2 is processing the 178029 / 178337 pack
[14:51:46] thread 1 is processing the 178030 / 178337 pack
[14:51:46] thread 2 is processing the 178031 / 178337 pack
[14:51:46] thread 1 is processing the 178032 / 178337 pack
[14:51:46] thread 2 is processing the 178033 / 178337 pack
[14:51:46] thread 1 is processing the 178034 / 178337 pack
[14:51:46] thread 2 is processing the 178035 / 178337 pack
[14:51:46] thread 1 is processing the 178036 / 178337 pack
[14:51:46] thread 2 is processing the 178037 / 178337 pack
[14:51:46] thread 1 is processing the 178038 / 178337 pack
[14:51:46] thread 2 is processing the 178039 / 178337 pack
[14:51:46] thread 1 is processing the 178040 / 178337 pack
[14:51:46] thread 2 is processing the 178041 / 178337 pack
[14:51:46] thread 1 is processing the 178042 / 178337 pack
[14:51:46] thread 2 is processing the 178043 / 178337 pack
[14:51:46] thread 1 is processing the 178044 / 178337 pack
[14:51:46] thread 2 is processing the 178045 / 178337 pack
[14:51:46] thread 1 is processing the 178046 / 178337 pack
[14:51:46] thread 2 is processing the 178047 / 178337 pack
[14:51:46] thread 1 is processing the 178048 / 178337 pack
[14:51:46] thread 1 is processing the 178049 / 178337 pack
[14:51:46] thread 2 is processing the 178050 / 178337 pack
[14:51:46] thread 1 is processing the 178051 / 178337 pack
[14:51:46] thread 2 is processing the 178052 / 178337 pack
[14:51:46] thread 1 is processing the 178053 / 178337 pack
[14:51:46] thread 2 is processing the 178054 / 178337 pack
[14:51:46] thread 1 is processing the 178055 / 178337 pack
[14:51:46] thread 2 is processing the 178056 / 178337 pack
[14:51:46] thread 1 is processing the 178057 / 178337 pack
[14:51:46] thread 2 is processing the 178058 / 178337 pack
[14:51:46] thread 1 is processing the 178059 / 178337 pack
[14:51:46] thread 2 is processing the 178060 / 178337 pack
[14:51:46] thread 1 is processing the 178061 / 178337 pack
[14:51:46] thread 2 is processing the 178062 / 178337 pack
[14:51:46] thread 1 is processing the 178063 / 178337 pack
[14:51:46] thread 2 is processing the 178064 / 178337 pack
[14:51:46] thread 1 is processing the 178065 / 178337 pack
[14:51:46] thread 2 is processing the 178066 / 178337 pack
[14:51:46] thread 1 is processing the 178067 / 178337 pack
[14:51:46] thread 2 is processing the 178068 / 178337 pack
[14:51:46] thread 1 is processing the 178069 / 178337 pack
[14:51:46] thread 1 is processing the 178070 / 178337 pack
[14:51:46] thread 2 is processing the 178071 / 178337 pack
[14:51:46] thread 1 is processing the 178072 / 178337 pack
[14:51:46] thread 2 is processing the 178073 / 178337 pack
[14:51:46] thread 1 is processing the 178074 / 178337 pack
[14:51:46] thread 2 is processing the 178075 / 178337 pack
[14:51:46] thread 1 is processing the 178076 / 178337 pack
[14:51:46] thread 2 is processing the 178077 / 178337 pack
[14:51:46] thread 1 is processing the 178078 / 178337 pack
[14:51:46] thread 2 is processing the 178079 / 178337 pack
[14:51:46] thread 1 is processing the 178080 / 178337 pack
[14:51:46] thread 2 is processing the 178081 / 178337 pack
[14:51:46] thread 1 is processing the 178082 / 178337 pack
[14:51:46] thread 2 is processing the 178083 / 178337 pack
[14:51:46] thread 1 is processing the 178083 / 178337 pack
[14:51:47] thread 1 is processing the 178085 / 178337 pack
[14:51:47] thread 2 is processing the 178086 / 178337 pack
[14:51:47] thread 1 is processing the 178087 / 178337 pack
[14:51:47] thread 2 is processing the 178088 / 178337 pack
[14:51:47] thread 1 is processing the 178089 / 178337 pack
[14:51:47] thread 2 is processing the 178090 / 178337 pack
[14:51:47] thread 1 is processing the 178091 / 178337 pack
[14:51:47] thread 2 is processing the 178092 / 178337 pack
[14:51:47] thread 1 is processing the 178093 / 178337 pack
[14:51:47] thread 2 is processing the 178094 / 178337 pack
[14:51:47] thread 1 is processing the 178095 / 178337 pack
[14:51:47] thread 2 is processing the 178096 / 178337 pack
[14:51:47] thread 1 is processing the 178097 / 178337 pack
[14:51:47] thread 1 is processing the 178098 / 178337 pack
[14:51:47] thread 2 is processing the 178099 / 178337 pack
[14:51:47] thread 1 is processing the 178100 / 178337 pack
[14:51:47] thread 2 is processing the 178101 / 178337 pack
[14:51:47] thread 1 is processing the 178102 / 178337 pack
[14:51:47] thread 2 is processing the 178103 / 178337 pack
[14:51:47] thread 1 is processing the 178104 / 178337 pack
[14:51:47] thread 2 is processing the 178105 / 178337 pack
[14:51:47] thread 1 is processing the 178106 / 178337 pack
[14:51:47] thread 2 is processing the 178107 / 178337 pack
[14:51:47] thread 1 is processing the 178108 / 178337 pack
[14:51:47] thread 2 is processing the 178109 / 178337 pack
[14:51:47] thread 1 is processing the 178110 / 178337 pack
[14:51:47] thread 2 is processing the 178111 / 178337 pack
[14:51:47] thread 1 is processing the 178112 / 178337 pack
[14:51:47] thread 2 is processing the 178113 / 178337 pack
[14:51:47] thread 1 is processing the 178114 / 178337 pack
[14:51:47] thread 2 is processing the 178115 / 178337 pack
[14:51:47] thread 1 is processing the 178116 / 178337 pack
[14:51:47] thread 1 is processing the 178117 / 178337 pack
[14:51:47] thread 2 is processing the 178118 / 178337 pack
[14:51:47] thread 1 is processing the 178119 / 178337 pack
[14:51:47] thread 2 is processing the 178120 / 178337 pack
[14:51:47] thread 1 is processing the 178121 / 178337 pack
[14:51:47] thread 2 is processing the 178122 / 178337 pack
[14:51:47] thread 1 is processing the 178123 / 178337 pack
[14:51:47] thread 2 is processing the 178124 / 178337 pack
[14:51:47] thread 1 is processing the 178125 / 178337 pack
[14:51:47] thread 2 is processing the 178126 / 178337 pack
[14:51:47] thread 1 is processing the 178127 / 178337 pack
[14:51:47] thread 2 is processing the 178128 / 178337 pack
[14:51:47] thread 1 is processing the 178129 / 178337 pack
[14:51:47] thread 2 is processing the 178130 / 178337 pack
[14:51:47] thread 1 is processing the 178131 / 178337 pack
[14:51:47] thread 2 is processing the 178132 / 178337 pack
[14:51:47] thread 1 is processing the 178133 / 178337 pack
[14:51:47] thread 1 is processing the 178134 / 178337 pack
[14:51:47] thread 2 is processing the 178135 / 178337 pack
[14:51:47] thread 1 is processing the 178136 / 178337 pack
[14:51:47] thread 2 is processing the 178137 / 178337 pack
[14:51:47] thread 1 is processing the 178138 / 178337 pack
[14:51:47] thread 2 is processing the 178139 / 178337 pack
[14:51:47] thread 1 is processing the 178140 / 178337 pack
[14:51:47] thread 2 is processing the 178141 / 178337 pack
[14:51:47] thread 1 is processing the 178142 / 178337 pack
[14:51:47] thread 2 is processing the 178143 / 178337 pack
[14:51:47] thread 1 is processing the 178144 / 178337 pack
[14:51:47] thread 2 is processing the 178145 / 178337 pack
[14:51:47] thread 1 is processing the 178146 / 178337 pack
[14:51:47] thread 2 is processing the 178147 / 178337 pack
[14:51:47] thread 1 is processing the 178148 / 178337 pack
[14:51:47] thread 2 is processing the 178149 / 178337 pack
[14:51:47] thread 1 is processing the 178150 / 178337 pack
[14:51:48] thread 2 is processing the 178151 / 178337 pack
[14:51:48] thread 1 is processing the 178152 / 178337 pack
[14:51:48] thread 1 is processing the 178153 / 178337 pack
[14:51:48] thread 2 is processing the 178154 / 178337 pack
[14:51:48] thread 1 is processing the 178155 / 178337 pack
[14:51:48] thread 2 is processing the 178156 / 178337 pack
[14:51:48] thread 1 is processing the 178157 / 178337 pack
[14:51:48] thread 2 is processing the 178158 / 178337 pack
[14:51:48] thread 1 is processing the 178159 / 178337 pack
[14:51:48] thread 2 is processing the 178160 / 178337 pack
[14:51:48] thread 1 is processing the 178161 / 178337 pack
[14:51:48] thread 2 is processing the 178162 / 178337 pack
[14:51:48] thread 1 is processing the 178163 / 178337 pack
[14:51:48] thread 2 is processing the 178164 / 178337 pack
[14:51:48] thread 1 is processing the 178165 / 178337 pack
[14:51:48] thread 2 is processing the 178166 / 178337 pack
[14:51:48] thread 1 is processing the 178167 / 178337 pack
[14:51:48] thread 2 is processing the 178168 / 178337 pack
[14:51:48] thread 1 is processing the 178169 / 178337 pack
[14:51:48] thread 2 is processing the 178170 / 178337 pack
[14:51:48] thread 1 is processing the 178171 / 178337 pack
[14:51:48] thread 2 is processing the 178172 / 178337 pack
[14:51:48] thread 1 is processing the 178173 / 178337 pack
[14:51:48] thread 2 is processing the 178174 / 178337 pack
[14:51:48] thread 1 is processing the 178175 / 178337 pack
[14:51:48] thread 2 is processing the 178176 / 178337 pack
[14:51:48] thread 1 is processing the 178177 / 178337 pack
[14:51:48] thread 2 is processing the 178178 / 178337 pack
[14:51:48] thread 1 is processing the 178179 / 178337 pack
[14:51:48] thread 2 is processing the 178180 / 178337 pack
[14:51:48] thread 1 is processing the 178181 / 178337 pack
[14:51:48] thread 2 is processing the 178182 / 178337 pack
[14:51:48] thread 1 is processing the 178183 / 178337 pack
[14:51:48] thread 2 is processing the 178184 / 178337 pack
[14:51:48] thread 1 is processing the 178185 / 178337 pack
[14:51:48] thread 2 is processing the 178186 / 178337 pack
[14:51:48] thread 1 is processing the 178187 / 178337 pack
[14:51:48] thread 2 is processing the 178188 / 178337 pack
[14:51:48] thread 1 is processing the 178189 / 178337 pack
[14:51:48] thread 2 is processing the 178190 / 178337 pack
[14:51:48] thread 1 is processing the 178191 / 178337 pack
[14:51:48] thread 2 is processing the 178192 / 178337 pack
[14:51:48] thread 1 is processing the 178193 / 178337 pack
[14:51:48] thread 2 is processing the 178194 / 178337 pack
[14:51:48] thread 1 is processing the 178195 / 178337 pack
[14:51:48] thread 2 is processing the 178196 / 178337 pack
[14:51:48] thread 1 is processing the 178197 / 178337 pack
[14:51:48] thread 2 is processing the 178198 / 178337 pack
[14:51:48] thread 1 is processing the 178199 / 178337 pack
[14:51:48] thread 2 is processing the 178200 / 178337 pack
[14:51:48] thread 1 is processing the 178201 / 178337 pack
[14:51:48] thread 2 is processing the 178202 / 178337 pack
[14:51:48] thread 1 is processing the 178203 / 178337 pack
[14:51:48] thread 2 is processing the 178204 / 178337 pack
[14:51:48] thread 1 is processing the 178205 / 178337 pack
[14:51:48] thread 2 is processing the 178206 / 178337 pack
[14:51:48] thread 1 is processing the 178207 / 178337 pack
[14:51:48] thread 2 is processing the 178208 / 178337 pack
[14:51:48] thread 1 is processing the 178209 / 178337 pack
[14:51:48] thread 2 is processing the 178210 / 178337 pack
[14:51:48] thread 1 is processing the 178211 / 178337 pack
[14:51:48] thread 2 is processing the 178212 / 178337 pack
[14:51:48] thread 1 is processing the 178213 / 178337 pack
[14:51:48] thread 2 is processing the 178214 / 178337 pack
[14:51:48] thread 1 is processing the 178215 / 178337 pack
[14:51:48] thread 2 is processing the 178216 / 178337 pack
[14:51:48] thread 1 is processing the 178217 / 178337 pack
[14:51:48] thread 2 is processing the 178218 / 178337 pack
[14:51:48] thread 1 is processing the 178219 / 178337 pack
[14:51:48] thread 2 is processing the 178220 / 178337 pack
[14:51:48] thread 1 is processing the 178221 / 178337 pack
[14:51:48] thread 2 is processing the 178222 / 178337 pack
[14:51:48] thread 1 is processing the 178223 / 178337 pack
[14:51:48] thread 2 is processing the 178224 / 178337 pack
[14:51:48] thread 1 is processing the 178225 / 178337 pack
[14:51:48] thread 2 is processing the 178226 / 178337 pack
[14:51:48] thread 1 is processing the 178227 / 178337 pack
[14:51:48] thread 2 is processing the 178228 / 178337 pack
[14:51:48] thread 1 is processing the 178229 / 178337 pack
[14:51:48] thread 2 is processing the 178230 / 178337 pack
[14:51:48] thread 1 is processing the 178231 / 178337 pack
[14:51:48] thread 2 is processing the 178232 / 178337 pack
[14:51:49] thread 1 is processing the 178233 / 178337 pack
[14:51:49] thread 2 is processing the 178234 / 178337 pack
[14:51:49] thread 1 is processing the 178235 / 178337 pack
[14:51:49] thread 2 is processing the 178236 / 178337 pack
[14:51:49] thread 1 is processing the 178237 / 178337 pack
[14:51:49] thread 2 is processing the 178238 / 178337 pack
[14:51:49] thread 1 is processing the 178239 / 178337 pack
[14:51:49] thread 2 is processing the 178240 / 178337 pack
[14:51:49] thread 1 is processing the 178241 / 178337 pack
[14:51:49] thread 2 is processing the 178242 / 178337 pack
[14:51:49] thread 1 is processing the 178243 / 178337 pack
[14:51:49] thread 2 is processing the 178244 / 178337 pack
[14:51:49] thread 1 is processing the 178245 / 178337 pack
[14:51:49] thread 2 is processing the 178246 / 178337 pack
[14:51:49] thread 1 is processing the 178247 / 178337 pack
[14:51:49] thread 2 is processing the 178248 / 178337 pack
[14:51:49] thread 1 is processing the 178249 / 178337 pack
[14:51:49] thread 2 is processing the 178250 / 178337 pack
[14:51:49] thread 1 is processing the 178251 / 178337 pack
[14:51:49] thread 2 is processing the 178252 / 178337 pack
[14:51:49] thread 1 is processing the 178253 / 178337 pack
[14:51:49] thread 2 is processing the 178254 / 178337 pack
[14:51:49] thread 1 is processing the 178255 / 178337 pack
[14:51:49] thread 2 is processing the 178256 / 178337 pack
[14:51:49] thread 1 is processing the 178257 / 178337 pack
[14:51:49] thread 2 is processing the 178258 / 178337 pack
[14:51:49] thread 1 is processing the 178259 / 178337 pack
[14:51:49] thread 2 is processing the 178260 / 178337 pack
[14:51:49] thread 1 is processing the 178261 / 178337 pack
[14:51:49] thread 2 is processing the 178262 / 178337 pack
[14:51:49] thread 1 is processing the 178263 / 178337 pack
[14:51:49] thread 2 is processing the 178264 / 178337 pack
[14:51:49] thread 1 is processing the 178265 / 178337 pack
[14:51:49] thread 2 is processing the 178266 / 178337 pack
[14:51:49] thread 1 is processing the 178267 / 178337 pack
[14:51:49] thread 2 is processing the 178268 / 178337 pack
[14:51:49] thread 1 is processing the 178269 / 178337 pack
[14:51:49] thread 2 is processing the 178270 / 178337 pack
[14:51:49] thread 1 is processing the 178271 / 178337 pack
[14:51:49] thread 2 is processing the 178272 / 178337 pack
[14:51:49] thread 1 is processing the 178273 / 178337 pack
[14:51:49] thread 2 is processing the 178274 / 178337 pack
[14:51:49] thread 1 is processing the 178275 / 178337 pack
[14:51:49] thread 2 is processing the 178276 / 178337 pack
[14:51:49] thread 1 is processing the 178277 / 178337 pack
[14:51:49] thread 2 is processing the 178278 / 178337 pack
[14:51:49] thread 1 is processing the 178279 / 178337 pack
[14:51:49] thread 2 is processing the 178280 / 178337 pack
[14:51:49] thread 1 is processing the 178281 / 178337 pack
[14:51:49] thread 2 is processing the 178282 / 178337 pack
[14:51:49] thread 1 is processing the 178283 / 178337 pack
[14:51:49] thread 2 is processing the 178284 / 178337 pack
[14:51:49] thread 1 is processing the 178285 / 178337 pack
[14:51:49] thread 2 is processing the 178286 / 178337 pack
[14:51:49] thread 1 is processing the 178287 / 178337 pack
[14:51:49] thread 2 is processing the 178288 / 178337 pack
[14:51:49] thread 1 is processing the 178289 / 178337 pack
[14:51:49] thread 2 is processing the 178290 / 178337 pack
[14:51:49] thread 1 is processing the 178291 / 178337 pack
[14:51:49] thread 2 is processing the 178292 / 178337 pack
[14:51:49] thread 1 is processing the 178293 / 178337 pack
[14:51:49] thread 2 is processing the 178294 / 178337 pack
[14:51:49] thread 1 is processing the 178295 / 178337 pack
[14:51:49] thread 2 is processing the 178296 / 178337 pack
[14:51:49] thread 1 is processing the 178297 / 178337 pack
[14:51:49] thread 2 is processing the 178298 / 178337 pack
[14:51:49] thread 1 is processing the 178299 / 178337 pack
[14:51:49] thread 2 is processing the 178300 / 178337 pack
[14:51:49] thread 1 is processing the 178301 / 178337 pack
[14:51:49] thread 2 is processing the 178302 / 178337 pack
[14:51:49] thread 1 is processing the 178303 / 178337 pack
[14:51:49] thread 2 is processing the 178304 / 178337 pack
[14:51:49] thread 1 is processing the 178305 / 178337 pack
[14:51:49] thread 2 is processing the 178306 / 178337 pack
[14:51:49] thread 1 is processing the 178307 / 178337 pack
[14:51:49] thread 2 is processing the 178308 / 178337 pack
[14:51:49] thread 1 is processing the 178309 / 178337 pack
[14:51:49] thread 2 is processing the 178310 / 178337 pack
[14:51:49] thread 1 is processing the 178311 / 178337 pack
[14:51:50] thread 2 is processing the 178312 / 178337 pack
[14:51:50] thread 1 is processing the 178313 / 178337 pack
[14:51:50] thread 2 is processing the 178314 / 178337 pack
[14:51:50] thread 1 is processing the 178315 / 178337 pack
[14:51:50] thread 2 is processing the 178316 / 178337 pack
[14:51:50] thread 1 is processing the 178317 / 178337 pack
[14:51:50] thread 2 is processing the 178318 / 178337 pack
[14:51:50] thread 1 is processing the 178319 / 178337 pack
[14:51:50] thread 2 is processing the 178320 / 178337 pack
[14:51:50] thread 1 is processing the 178321 / 178337 pack
[14:51:50] thread 2 is processing the 178322 / 178337 pack
[14:51:50] thread 1 is processing the 178323 / 178337 pack
[14:51:50] thread 2 is processing the 178324 / 178337 pack
[14:51:50] thread 1 is processing the 178325 / 178337 pack
[14:51:50] thread 2 is processing the 178326 / 178337 pack
[14:51:50] thread 1 is processing the 178327 / 178337 pack
[14:51:50] thread 2 is processing the 178328 / 178337 pack
[14:51:50] thread 1 is processing the 178329 / 178337 pack
[14:51:50] thread 2 is processing the 178330 / 178337 pack
[14:51:50] thread 1 is processing the 178331 / 178337 pack
[14:51:50] thread 2 is processing the 178332 / 178337 pack
[14:51:50] thread 1 is processing the 178333 / 178337 pack
[14:51:50] thread 2 is processing the 178334 / 178337 pack
[14:51:50] thread 1 is processing the 178335 / 178337 pack
[14:51:50] thread 2 is processing the 178336 / 178337 pack
[14:51:50] thread 1 data processing completed
[14:51:50] thread 1 finished
[14:51:50] thread 2 data processing completed
[14:51:50] thread 2 finished
[14:51:50] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/Pupae_QC_R1.fastq.gz writer finished
[14:51:50] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/Pupae_QC_R2.fastq.gz writer finished
[14:51:50] start to generate reports

Read1 before filtering:
total reads: 178336838
total bases: 26928862538
Q20 bases: 26392299376(98.0075%)
Q30 bases: 25494886391(94.6749%)

Read2 before filtering:
total reads: 178336838
total bases: 26928862538
Q20 bases: 25991256620(96.5182%)
Q30 bases: 24670073511(91.612%)

Read1 after filtering:
total reads: 175385089
total bases: 26183180307
Q20 bases: 25724348183(98.2476%)
Q30 bases: 24885117577(95.0424%)

Read2 after filtering:
total reads: 175385089
total bases: 26181832281
Q20 bases: 25484379628(97.3361%)
Q30 bases: 24248485366(92.6157%)

Filtering result:
reads passed filter: 350770178
reads failed due to low quality: 5778268
reads failed due to too many N: 15850
reads failed due to too short: 109380
reads with adapter trimmed: 31490787
bases trimmed due to adapters: 599781053

Duplication rate: 7.02538%

Insert size peak (evaluated by paired-end reads): 198

Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[13:57:38] start to load data
[13:57:47] loaded 1M read pairs
[13:57:56] loaded 2M read pairs
[13:58:7] loaded 3M read pairs
[13:58:18] loaded 4M read pairs
[13:58:29] loaded 5M read pairs
[13:58:40] loaded 6M read pairs
[13:58:51] loaded 7M read pairs
[13:59:3] loaded 8M read pairs
[13:59:14] loaded 9M read pairs
[13:59:25] loaded 10M read pairs
[13:59:37] loaded 11M read pairs
[13:59:48] loaded 12M read pairs
[14:0:0] loaded 13M read pairs
[14:0:12] loaded 14M read pairs
[14:0:23] loaded 15M read pairs
[14:0:35] loaded 16M read pairs
[14:0:46] loaded 17M read pairs
[14:0:59] loaded 18M read pairs
[14:1:10] loaded 19M read pairs
[14:1:21] loaded 20M read pairs
[14:1:32] loaded 21M read pairs
[14:1:43] loaded 22M read pairs
[14:1:53] loaded 23M read pairs
[14:2:4] loaded 24M read pairs
[14:2:15] loaded 25M read pairs
[14:2:25] loaded 26M read pairs
[14:2:36] loaded 27M read pairs
[14:2:46] loaded 28M read pairs
[14:2:57] loaded 29M read pairs
[14:3:8] loaded 30M read pairs
[14:3:19] loaded 31M read pairs
[14:3:30] loaded 32M read pairs
[14:3:41] loaded 33M read pairs
[14:3:52] loaded 34M read pairs
[14:4:2] loaded 35M read pairs
[14:4:14] loaded 36M read pairs
[14:4:24] loaded 37M read pairs
[14:4:35] loaded 38M read pairs
[14:4:47] loaded 39M read pairs
[14:4:57] loaded 40M read pairs
[14:5:9] loaded 41M read pairs
[14:5:20] loaded 42M read pairs
[14:5:31] loaded 43M read pairs
[14:5:41] loaded 44M read pairs
[14:5:52] loaded 45M read pairs
[14:6:2] loaded 46M read pairs
[14:6:14] loaded 47M read pairs
[14:6:24] loaded 48M read pairs
[14:6:35] loaded 49M read pairs
[14:6:46] loaded 50M read pairs
[14:6:57] loaded 51M read pairs
[14:7:7] loaded 52M read pairs
[14:7:18] loaded 53M read pairs
[14:7:28] loaded 54M read pairs
[14:7:39] loaded 55M read pairs
[14:7:49] loaded 56M read pairs
[14:8:0] loaded 57M read pairs
[14:8:11] loaded 58M read pairs
[14:8:21] loaded 59M read pairs
[14:8:32] loaded 60M read pairs
[14:8:42] loaded 61M read pairs
[14:8:53] loaded 62M read pairs
[14:9:3] loaded 63M read pairs
[14:9:14] loaded 64M read pairs
[14:9:24] loaded 65M read pairs
[14:9:35] loaded 66M read pairs
[14:9:46] loaded 67M read pairs
[14:9:57] loaded 68M read pairs
[14:10:7] loaded 69M read pairs
[14:10:18] loaded 70M read pairs
[14:10:29] loaded 71M read pairs
[14:10:40] loaded 72M read pairs
[14:10:51] loaded 73M read pairs
[14:11:1] loaded 74M read pairs
[14:11:12] loaded 75M read pairs
[14:11:22] loaded 76M read pairs
[14:11:33] loaded 77M read pairs
[14:11:44] loaded 78M read pairs
[14:11:55] loaded 79M read pairs
[14:12:6] loaded 80M read pairs
[14:12:16] loaded 81M read pairs
[14:12:27] loaded 82M read pairs
[14:12:38] loaded 83M read pairs
[14:12:49] loaded 84M read pairs
[14:12:59] loaded 85M read pairs
[14:13:10] loaded 86M read pairs
[14:13:21] loaded 87M read pairs
[14:13:32] loaded 88M read pairs
[14:13:42] loaded 89M read pairs
[14:13:53] loaded 90M read pairs
[14:14:4] loaded 91M read pairs
[14:14:15] loaded 92M read pairs
[14:14:25] loaded 93M read pairs
[14:14:36] loaded 94M read pairs
[14:14:46] loaded 95M read pairs
[14:14:57] loaded 96M read pairs
[14:15:8] loaded 97M read pairs
[14:15:18] loaded 98M read pairs
Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[14:17:41] start to load data
[14:17:48] loaded 1M read pairs
[14:17:58] loaded 2M read pairs
[14:18:9] loaded 3M read pairs
[14:18:20] loaded 4M read pairs
[14:18:32] loaded 5M read pairs
[14:18:44] loaded 6M read pairs
[14:18:55] loaded 7M read pairs
[14:19:6] loaded 8M read pairs
[14:19:17] loaded 9M read pairs
[14:19:29] loaded 10M read pairs
[14:19:40] loaded 11M read pairs
[14:19:51] loaded 12M read pairs
[14:20:2] loaded 13M read pairs
[14:20:12] loaded 14M read pairs
[14:20:23] loaded 15M read pairs
[14:20:34] loaded 16M read pairs
[14:20:45] loaded 17M read pairs
[14:20:57] loaded 18M read pairs
[14:21:8] loaded 19M read pairs
[14:21:19] loaded 20M read pairs
[14:21:30] loaded 21M read pairs
[14:21:42] loaded 22M read pairs
[14:21:53] loaded 23M read pairs
[14:22:4] loaded 24M read pairs
[14:22:16] loaded 25M read pairs
[14:22:28] loaded 26M read pairs
[14:22:39] loaded 27M read pairs
[14:22:50] loaded 28M read pairs
[14:23:2] loaded 29M read pairs
[14:23:13] loaded 30M read pairs
[14:23:24] loaded 31M read pairs
[14:23:35] loaded 32M read pairs
[14:23:46] loaded 33M read pairs
[14:23:57] loaded 34M read pairs
[14:24:8] loaded 35M read pairs
[14:24:19] loaded 36M read pairs
[14:24:30] loaded 37M read pairs
[14:24:41] loaded 38M read pairs
[14:24:52] loaded 39M read pairs
[14:25:3] loaded 40M read pairs
[14:25:14] loaded 41M read pairs
[14:25:26] loaded 42M read pairs
[14:25:38] loaded 43M read pairs
[14:25:50] loaded 44M read pairs
[14:26:1] loaded 45M read pairs
[14:26:13] loaded 46M read pairs
[14:26:25] loaded 47M read pairs
[14:26:36] loaded 48M read pairs
[14:26:48] loaded 49M read pairs
[14:26:59] loaded 50M read pairs
[14:27:9] loaded 51M read pairs
[14:27:21] loaded 52M read pairs
[14:27:32] loaded 53M read pairs
[14:27:43] loaded 54M read pairs
[14:27:54] loaded 55M read pairs
[14:28:5] loaded 56M read pairs
[14:28:17] loaded 57M read pairs
[14:28:29] loaded 58M read pairs
[14:28:40] loaded 59M read pairs
[14:28:52] loaded 60M read pairs
[14:29:4] loaded 61M read pairs
[14:29:15] loaded 62M read pairs
[14:29:27] loaded 63M read pairs
[14:29:39] loaded 64M read pairs
[14:29:50] loaded 65M read pairs
[14:30:1] loaded 66M read pairs
[14:30:12] loaded 67M read pairs
[14:30:23] loaded 68M read pairs
[14:30:34] loaded 69M read pairs
[14:30:45] loaded 70M read pairs
[14:30:56] loaded 71M read pairs
[14:31:8] loaded 72M read pairs
[14:31:19] loaded 73M read pairs
[14:31:30] loaded 74M read pairs
[14:31:42] loaded 75M read pairs
[14:31:53] loaded 76M read pairs
[14:32:3] loaded 77M read pairs
[14:32:14] loaded 78M read pairs
[14:32:25] loaded 79M read pairs
[14:32:36] loaded 80M read pairs
[14:32:48] loaded 81M read pairs
[14:32:59] loaded 82M read pairs
[14:33:12] loaded 83M read pairs
[14:33:24] loaded 84M read pairs
[14:33:36] loaded 85M read pairs
[14:33:47] loaded 86M read pairs
[14:33:59] loaded 87M read pairs
[14:34:10] loaded 88M read pairs
[14:34:21] loaded 89M read pairs
[14:34:33] loaded 90M read pairs
[14:34:44] loaded 91M read pairs
[14:34:55] loaded 92M read pairs
[14:35:7] loaded 93M read pairs
[14:35:18] loaded 94M read pairs
[14:35:29] loaded 95M read pairs
[14:35:40] loaded 96M read pairs
[14:35:52] loaded 97M read pairs
[14:36:3] loaded 98M read pairs
[14:36:15] loaded 99M read pairs
[14:36:26] loaded 100M read pairs
[14:36:37] loaded 101M read pairs
[14:36:48] loaded 102M read pairs
[14:37:0] loaded 103M read pairs
[14:37:11] loaded 104M read pairs
[14:37:23] loaded 105M read pairs
[14:37:34] loaded 106M read pairs
[14:37:45] loaded 107M read pairs
[14:37:56] loaded 108M read pairs
[14:38:7] loaded 109M read pairs
[14:38:19] loaded 110M read pairs
[14:38:30] loaded 111M read pairs
[14:38:41] loaded 112M read pairs
[14:38:52] loaded 113M read pairs
[14:39:3] loaded 114M read pairs
[14:39:14] loaded 115M read pairs
[14:39:26] loaded 116M read pairs
[14:39:38] loaded 117M read pairs
[14:39:49] loaded 118M read pairs
[14:40:0] loaded 119M read pairs
[14:40:11] loaded 120M read pairs
[14:40:24] loaded 121M read pairs
[14:40:35] loaded 122M read pairs
[14:40:47] loaded 123M read pairs
[14:40:58] loaded 124M read pairs
[14:41:9] loaded 125M read pairs
[14:41:20] loaded 126M read pairs
[14:41:31] loaded 127M read pairs
[14:41:43] loaded 128M read pairs
[14:41:54] loaded 129M read pairs
[14:42:6] loaded 130M read pairs
[14:42:17] loaded 131M read pairs
[14:42:28] loaded 132M read pairs
[14:42:40] loaded 133M read pairs
[14:42:51] loaded 134M read pairs
[14:43:2] loaded 135M read pairs
[14:43:14] loaded 136M read pairs
[14:43:25] loaded 137M read pairs
[14:43:37] loaded 138M read pairs
[14:43:48] loaded 139M read pairs
[14:43:59] loaded 140M read pairs
[14:44:10] loaded 141M read pairs
[14:44:21] loaded 142M read pairs
[14:44:32] loaded 143M read pairs
[14:44:44] loaded 144M read pairs
[14:44:55] loaded 145M read pairs
[14:45:6] loaded 146M read pairs
[14:45:17] loaded 147M read pairs
[14:45:29] loaded 148M read pairs
[14:45:40] loaded 149M read pairs
[14:45:51] loaded 150M read pairs
[14:46:3] loaded 151M read pairs
[14:46:14] loaded 152M read pairs
[14:46:25] loaded 153M read pairs
[14:46:37] loaded 154M read pairs
[14:46:48] loaded 155M read pairs
[14:46:59] loaded 156M read pairs
[14:47:10] loaded 157M read pairs
[14:47:21] loaded 158M read pairs
[14:47:32] loaded 159M read pairs
[14:47:43] loaded 160M read pairs
[14:47:54] loaded 161M read pairs
[14:47:56] all reads loaded, start to monitor thread status
[14:47:56] thread 2 is processing the 160689 / 161190 pack
[14:47:56] thread 1 is processing the 160690 / 161190 pack
[14:47:56] thread 2 is processing the 160691 / 161190 pack
[14:47:56] thread 1 is processing the 160692 / 161190 pack
[14:47:56] thread 2 is processing the 160693 / 161190 pack
[14:47:56] thread 1 is processing the 160694 / 161190 pack
[14:47:56] thread 2 is processing the 160695 / 161190 pack
[14:47:56] thread 1 is processing the 160696 / 161190 pack
[14:47:56] thread 2 is processing the 160697 / 161190 pack
[14:47:56] thread 1 is processing the 160698 / 161190 pack
[14:47:56] thread 2 is processing the 160699 / 161190 pack
[14:47:56] thread 1 is processing the 160700 / 161190 pack
[14:47:56] thread 2 is processing the 160701 / 161190 pack
[14:47:56] thread 1 is processing the 160702 / 161190 pack
[14:47:56] thread 2 is processing the 160703 / 161190 pack
[14:47:56] thread 1 is processing the 160704 / 161190 pack
[14:47:56] thread 2 is processing the 160705 / 161190 pack
[14:47:56] thread 1 is processing the 160706 / 161190 pack
[14:47:56] thread 2 is processing the 160707 / 161190 pack
[14:47:56] thread 1 is processing the 160708 / 161190 pack
[14:47:56] thread 2 is processing the 160709 / 161190 pack
[14:47:56] thread 1 is processing the 160710 / 161190 pack
[14:47:56] thread 2 is processing the 160711 / 161190 pack
[14:47:56] thread 1 is processing the 160712 / 161190 pack
[14:47:56] thread 2 is processing the 160713 / 161190 pack
[14:47:56] thread 1 is processing the 160714 / 161190 pack
[14:47:56] thread 2 is processing the 160715 / 161190 pack
[14:47:56] thread 1 is processing the 160716 / 161190 pack
[14:47:56] thread 2 is processing the 160717 / 161190 pack
[14:47:56] thread 1 is processing the 160718 / 161190 pack
[14:47:56] thread 2 is processing the 160719 / 161190 pack
[14:47:56] thread 1 is processing the 160720 / 161190 pack
[14:47:56] thread 2 is processing the 160721 / 161190 pack
[14:47:56] thread 1 is processing the 160722 / 161190 pack
[14:47:56] thread 2 is processing the 160723 / 161190 pack
[14:47:56] thread 1 is processing the 160724 / 161190 pack
[14:47:56] thread 2 is processing the 160725 / 161190 pack
[14:47:56] thread 1 is processing the 160726 / 161190 pack
[14:47:56] thread 2 is processing the 160727 / 161190 pack
[14:47:56] thread 1 is processing the 160728 / 161190 pack
[14:47:56] thread 2 is processing the 160729 / 161190 pack
[14:47:56] thread 1 is processing the 160730 / 161190 pack
[14:47:56] thread 2 is processing the 160731 / 161190 pack
[14:47:56] thread 1 is processing the 160732 / 161190 pack
[14:47:56] thread 2 is processing the 160733 / 161190 pack
[14:47:56] thread 1 is processing the 160734 / 161190 pack
[14:47:56] thread 2 is processing the 160735 / 161190 pack
[14:47:56] thread 1 is processing the 160736 / 161190 pack
[14:47:56] thread 2 is processing the 160737 / 161190 pack
[14:47:56] thread 1 is processing the 160738 / 161190 pack
[14:47:56] thread 2 is processing the 160739 / 161190 pack
[14:47:56] thread 1 is processing the 160740 / 161190 pack
[14:47:56] thread 2 is processing the 160741 / 161190 pack
[14:47:56] thread 1 is processing the 160742 / 161190 pack
[14:47:56] thread 2 is processing the 160743 / 161190 pack
[14:47:56] thread 1 is processing the 160744 / 161190 pack
[14:47:56] thread 2 is processing the 160745 / 161190 pack
[14:47:56] thread 1 is processing the 160746 / 161190 pack
[14:47:56] thread 2 is processing the 160747 / 161190 pack
[14:47:56] thread 1 is processing the 160748 / 161190 pack
[14:47:56] thread 2 is processing the 160749 / 161190 pack
[14:47:56] thread 1 is processing the 160750 / 161190 pack
[14:47:56] thread 2 is processing the 160751 / 161190 pack
[14:47:56] thread 1 is processing the 160752 / 161190 pack
[14:47:56] thread 2 is processing the 160753 / 161190 pack
[14:47:56] thread 1 is processing the 160754 / 161190 pack
[14:47:56] thread 2 is processing the 160755 / 161190 pack
[14:47:56] thread 1 is processing the 160756 / 161190 pack
[14:47:56] thread 2 is processing the 160757 / 161190 pack
[14:47:56] thread 1 is processing the 160758 / 161190 pack
[14:47:56] thread 2 is processing the 160759 / 161190 pack
[14:47:56] thread 1 is processing the 160760 / 161190 pack
[14:47:56] thread 2 is processing the 160761 / 161190 pack
[14:47:56] thread 1 is processing the 160762 / 161190 pack
[14:47:56] thread 2 is processing the 160763 / 161190 pack
[14:47:56] thread 1 is processing the 160764 / 161190 pack
[14:47:57] thread 2 is processing the 160765 / 161190 pack
[14:47:57] thread 1 is processing the 160766 / 161190 pack
[14:47:57] thread 2 is processing the 160767 / 161190 pack
[14:47:57] thread 1 is processing the 160768 / 161190 pack
[14:47:57] thread 2 is processing the 160769 / 161190 pack
[14:47:57] thread 1 is processing the 160770 / 161190 pack
[14:47:57] thread 2 is processing the 160771 / 161190 pack
[14:47:57] thread 1 is processing the 160772 / 161190 pack
[14:47:57] thread 2 is processing the 160773 / 161190 pack
[14:47:57] thread 1 is processing the 160774 / 161190 pack
[14:47:57] thread 2 is processing the 160775 / 161190 pack
[14:47:57] thread 1 is processing the 160776 / 161190 pack
[14:47:57] thread 2 is processing the 160777 / 161190 pack
[14:47:57] thread 1 is processing the 160778 / 161190 pack
[14:47:57] thread 2 is processing the 160779 / 161190 pack
[14:47:57] thread 1 is processing the 160780 / 161190 pack
[14:47:57] thread 2 is processing the 160781 / 161190 pack
[14:47:57] thread 1 is processing the 160782 / 161190 pack
[14:47:57] thread 2 is processing the 160783 / 161190 pack
[14:47:57] thread 1 is processing the 160784 / 161190 pack
[14:47:57] thread 2 is processing the 160785 / 161190 pack
[14:47:57] thread 1 is processing the 160786 / 161190 pack
[14:47:57] thread 2 is processing the 160787 / 161190 pack
[14:47:57] thread 1 is processing the 160788 / 161190 pack
[14:47:57] thread 2 is processing the 160789 / 161190 pack
[14:47:57] thread 1 is processing the 160790 / 161190 pack
[14:47:57] thread 2 is processing the 160791 / 161190 pack
[14:47:57] thread 1 is processing the 160792 / 161190 pack
[14:47:57] thread 2 is processing the 160793 / 161190 pack
[14:47:57] thread 1 is processing the 160794 / 161190 pack
[14:47:57] thread 2 is processing the 160795 / 161190 pack
[14:47:57] thread 1 is processing the 160796 / 161190 pack
[14:47:57] thread 2 is processing the 160797 / 161190 pack
[14:47:57] thread 1 is processing the 160798 / 161190 pack
[14:47:57] thread 2 is processing the 160799 / 161190 pack
[14:47:57] thread 1 is processing the 160800 / 161190 pack
[14:47:57] thread 2 is processing the 160801 / 161190 pack
[14:47:57] thread 1 is processing the 160802 / 161190 pack
[14:47:57] thread 2 is processing the 160803 / 161190 pack
[14:47:57] thread 1 is processing the 160804 / 161190 pack
[14:47:57] thread 2 is processing the 160805 / 161190 pack
[14:47:57] thread 1 is processing the 160806 / 161190 pack
[14:47:57] thread 2 is processing the 160807 / 161190 pack
[14:47:57] thread 1 is processing the 160808 / 161190 pack
[14:47:57] thread 2 is processing the 160809 / 161190 pack
[14:47:57] thread 1 is processing the 160810 / 161190 pack
[14:47:57] thread 2 is processing the 160811 / 161190 pack
[14:47:57] thread 1 is processing the 160812 / 161190 pack
[14:47:57] thread 2 is processing the 160813 / 161190 pack
[14:47:57] thread 1 is processing the 160814 / 161190 pack
[14:47:57] thread 2 is processing the 160815 / 161190 pack
[14:47:57] thread 1 is processing the 160816 / 161190 pack
[14:47:57] thread 2 is processing the 160817 / 161190 pack
[14:47:57] thread 1 is processing the 160818 / 161190 pack
[14:47:57] thread 2 is processing the 160819 / 161190 pack
[14:47:57] thread 1 is processing the 160820 / 161190 pack
[14:47:57] thread 2 is processing the 160821 / 161190 pack
[14:47:57] thread 1 is processing the 160822 / 161190 pack
[14:47:57] thread 2 is processing the 160823 / 161190 pack
[14:47:57] thread 1 is processing the 160824 / 161190 pack
[14:47:57] thread 2 is processing the 160825 / 161190 pack
[14:47:57] thread 1 is processing the 160826 / 161190 pack
[14:47:57] thread 2 is processing the 160827 / 161190 pack
[14:47:57] thread 1 is processing the 160828 / 161190 pack
[14:47:57] thread 2 is processing the 160829 / 161190 pack
[14:47:57] thread 1 is processing the 160830 / 161190 pack
[14:47:57] thread 2 is processing the 160831 / 161190 pack
[14:47:57] thread 1 is processing the 160832 / 161190 pack
[14:47:57] thread 2 is processing the 160833 / 161190 pack
[14:47:57] thread 1 is processing the 160834 / 161190 pack
[14:47:57] thread 2 is processing the 160835 / 161190 pack
[14:47:57] thread 1 is processing the 160836 / 161190 pack
[14:47:57] thread 2 is processing the 160837 / 161190 pack
[14:47:57] thread 1 is processing the 160838 / 161190 pack
[14:47:57] thread 2 is processing the 160839 / 161190 pack
[14:47:57] thread 1 is processing the 160840 / 161190 pack
[14:47:57] thread 2 is processing the 160841 / 161190 pack
[14:47:57] thread 1 is processing the 160842 / 161190 pack
[14:47:57] thread 2 is processing the 160843 / 161190 pack
[14:47:57] thread 1 is processing the 160844 / 161190 pack
[14:47:57] thread 2 is processing the 160845 / 161190 pack
[14:47:57] thread 1 is processing the 160846 / 161190 pack
[14:47:57] thread 2 is processing the 160847 / 161190 pack
[14:47:57] thread 1 is processing the 160848 / 161190 pack
[14:47:57] thread 2 is processing the 160849 / 161190 pack
[14:47:57] thread 1 is processing the 160850 / 161190 pack
[14:47:57] thread 2 is processing the 160851 / 161190 pack
[14:47:57] thread 1 is processing the 160852 / 161190 pack
[14:47:58] thread 2 is processing the 160853 / 161190 pack
[14:47:58] thread 1 is processing the 160854 / 161190 pack
[14:47:58] thread 2 is processing the 160855 / 161190 pack
[14:47:58] thread 1 is processing the 160856 / 161190 pack
[14:47:58] thread 2 is processing the 160857 / 161190 pack
[14:47:58] thread 1 is processing the 160858 / 161190 pack
[14:47:58] thread 2 is processing the 160859 / 161190 pack
[14:47:58] thread 1 is processing the 160860 / 161190 pack
[14:47:58] thread 2 is processing the 160861 / 161190 pack
[14:47:58] thread 1 is processing the 160862 / 161190 pack
[14:47:58] thread 2 is processing the 160863 / 161190 pack
[14:47:58] thread 1 is processing the 160864 / 161190 pack
[14:47:58] thread 2 is processing the 160865 / 161190 pack
[14:47:58] thread 1 is processing the 160866 / 161190 pack
[14:47:58] thread 2 is processing the 160867 / 161190 pack
[14:47:58] thread 1 is processing the 160868 / 161190 pack
[14:47:58] thread 2 is processing the 160869 / 161190 pack
[14:47:58] thread 1 is processing the 160870 / 161190 pack
[14:47:58] thread 2 is processing the 160871 / 161190 pack
[14:47:58] thread 1 is processing the 160872 / 161190 pack
[14:47:58] thread 2 is processing the 160873 / 161190 pack
[14:47:58] thread 1 is processing the 160874 / 161190 pack
[14:47:58] thread 2 is processing the 160875 / 161190 pack
[14:47:58] thread 1 is processing the 160876 / 161190 pack
[14:47:58] thread 2 is processing the 160877 / 161190 pack
[14:47:58] thread 1 is processing the 160878 / 161190 pack
[14:47:58] thread 2 is processing the 160879 / 161190 pack
[14:47:58] thread 1 is processing the 160880 / 161190 pack
[14:47:58] thread 2 is processing the 160881 / 161190 pack
[14:47:58] thread 1 is processing the 160882 / 161190 pack
[14:47:58] thread 2 is processing the 160883 / 161190 pack
[14:47:58] thread 1 is processing the 160884 / 161190 pack
[14:47:58] thread 2 is processing the 160885 / 161190 pack
[14:47:58] thread 1 is processing the 160886 / 161190 pack
[14:47:58] thread 2 is processing the 160887 / 161190 pack
[14:47:58] thread 1 is processing the 160888 / 161190 pack
[14:47:58] thread 2 is processing the 160889 / 161190 pack
[14:47:58] thread 1 is processing the 160890 / 161190 pack
[14:47:58] thread 2 is processing the 160891 / 161190 pack
[14:47:58] thread 1 is processing the 160892 / 161190 pack
[14:47:58] thread 2 is processing the 160893 / 161190 pack
[14:47:58] thread 1 is processing the 160894 / 161190 pack
[14:47:58] thread 2 is processing the 160895 / 161190 pack
[14:47:58] thread 1 is processing the 160896 / 161190 pack
[14:47:58] thread 2 is processing the 160897 / 161190 pack
[14:47:58] thread 1 is processing the 160898 / 161190 pack
[14:47:58] thread 2 is processing the 160899 / 161190 pack
[14:47:58] thread 1 is processing the 160900 / 161190 pack
[14:47:58] thread 2 is processing the 160901 / 161190 pack
[14:47:58] thread 1 is processing the 160902 / 161190 pack
[14:47:58] thread 2 is processing the 160903 / 161190 pack
[14:47:58] thread 1 is processing the 160904 / 161190 pack
[14:47:58] thread 2 is processing the 160905 / 161190 pack
[14:47:58] thread 1 is processing the 160906 / 161190 pack
[14:47:58] thread 2 is processing the 160907 / 161190 pack
[14:47:58] thread 1 is processing the 160908 / 161190 pack
[14:47:58] thread 2 is processing the 160909 / 161190 pack
[14:47:58] thread 1 is processing the 160910 / 161190 pack
[14:47:58] thread 2 is processing the 160911 / 161190 pack
[14:47:58] thread 1 is processing the 160912 / 161190 pack
[14:47:58] thread 2 is processing the 160913 / 161190 pack
[14:47:58] thread 1 is processing the 160914 / 161190 pack
[14:47:58] thread 2 is processing the 160915 / 161190 pack
[14:47:58] thread 1 is processing the 160916 / 161190 pack
[14:47:58] thread 2 is processing the 160917 / 161190 pack
[14:47:58] thread 1 is processing the 160918 / 161190 pack
[14:47:58] thread 2 is processing the 160919 / 161190 pack
[14:47:58] thread 1 is processing the 160920 / 161190 pack
[14:47:58] thread 2 is processing the 160921 / 161190 pack
[14:47:58] thread 1 is processing the 160922 / 161190 pack
[14:47:58] thread 2 is processing the 160923 / 161190 pack
[14:47:58] thread 1 is processing the 160924 / 161190 pack
[14:47:58] thread 2 is processing the 160925 / 161190 pack
[14:47:58] thread 1 is processing the 160926 / 161190 pack
[14:47:58] thread 2 is processing the 160927 / 161190 pack
[14:47:58] thread 1 is processing the 160928 / 161190 pack
[14:47:58] thread 2 is processing the 160929 / 161190 pack
[14:47:58] thread 1 is processing the 160930 / 161190 pack
[14:47:58] thread 2 is processing the 160931 / 161190 pack
[14:47:58] thread 1 is processing the 160932 / 161190 pack
[14:47:58] thread 2 is processing the 160933 / 161190 pack
[14:47:58] thread 1 is processing the 160934 / 161190 pack
[14:47:58] thread 2 is processing the 160935 / 161190 pack
[14:47:58] thread 1 is processing the 160936 / 161190 pack
[14:47:58] thread 2 is processing the 160937 / 161190 pack
[14:47:58] thread 1 is processing the 160938 / 161190 pack
[14:47:59] thread 2 is processing the 160939 / 161190 pack
[14:47:59] thread 1 is processing the 160940 / 161190 pack
[14:47:59] thread 2 is processing the 160941 / 161190 pack
[14:47:59] thread 1 is processing the 160942 / 161190 pack
[14:47:59] thread 2 is processing the 160943 / 161190 pack
[14:47:59] thread 1 is processing the 160944 / 161190 pack
[14:47:59] thread 2 is processing the 160945 / 161190 pack
[14:47:59] thread 1 is processing the 160946 / 161190 pack
[14:47:59] thread 2 is processing the 160947 / 161190 pack
[14:47:59] thread 1 is processing the 160948 / 161190 pack
[14:47:59] thread 2 is processing the 160949 / 161190 pack
[14:47:59] thread 1 is processing the 160950 / 161190 pack
[14:47:59] thread 2 is processing the 160951 / 161190 pack
[14:47:59] thread 1 is processing the 160952 / 161190 pack
[14:47:59] thread 2 is processing the 160953 / 161190 pack
[14:47:59] thread 1 is processing the 160954 / 161190 pack
[14:47:59] thread 2 is processing the 160955 / 161190 pack
[14:47:59] thread 1 is processing the 160956 / 161190 pack
[14:47:59] thread 2 is processing the 160957 / 161190 pack
[14:47:59] thread 1 is processing the 160958 / 161190 pack
[14:47:59] thread 2 is processing the 160959 / 161190 pack
[14:47:59] thread 1 is processing the 160960 / 161190 pack
[14:47:59] thread 2 is processing the 160961 / 161190 pack
[14:47:59] thread 1 is processing the 160962 / 161190 pack
[14:47:59] thread 2 is processing the 160963 / 161190 pack
[14:47:59] thread 1 is processing the 160964 / 161190 pack
[14:47:59] thread 2 is processing the 160965 / 161190 pack
[14:47:59] thread 1 is processing the 160966 / 161190 pack
[14:47:59] thread 2 is processing the 160967 / 161190 pack
[14:47:59] thread 1 is processing the 160968 / 161190 pack
[14:47:59] thread 2 is processing the 160969 / 161190 pack
[14:47:59] thread 1 is processing the 160970 / 161190 pack
[14:47:59] thread 2 is processing the 160971 / 161190 pack
[14:47:59] thread 1 is processing the 160972 / 161190 pack
[14:47:59] thread 2 is processing the 160973 / 161190 pack
[14:47:59] thread 1 is processing the 160974 / 161190 pack
[14:47:59] thread 2 is processing the 160975 / 161190 pack
[14:47:59] thread 1 is processing the 160976 / 161190 pack
[14:47:59] thread 2 is processing the 160977 / 161190 pack
[14:47:59] thread 1 is processing the 160978 / 161190 pack
[14:47:59] thread 2 is processing the 160979 / 161190 pack
[14:47:59] thread 1 is processing the 160980 / 161190 pack
[14:47:59] thread 2 is processing the 160981 / 161190 pack
[14:47:59] thread 1 is processing the 160982 / 161190 pack
[14:47:59] thread 2 is processing the 160983 / 161190 pack
[14:47:59] thread 1 is processing the 160984 / 161190 pack
[14:47:59] thread 2 is processing the 160985 / 161190 pack
[14:47:59] thread 1 is processing the 160986 / 161190 pack
[14:47:59] thread 2 is processing the 160987 / 161190 pack
[14:47:59] thread 1 is processing the 160988 / 161190 pack
[14:47:59] thread 2 is processing the 160989 / 161190 pack
[14:47:59] thread 1 is processing the 160990 / 161190 pack
[14:47:59] thread 2 is processing the 160991 / 161190 pack
[14:47:59] thread 1 is processing the 160992 / 161190 pack
[14:47:59] thread 2 is processing the 160993 / 161190 pack
[14:47:59] thread 1 is processing the 160994 / 161190 pack
[14:47:59] thread 2 is processing the 160995 / 161190 pack
[14:47:59] thread 1 is processing the 160996 / 161190 pack
[14:47:59] thread 2 is processing the 160997 / 161190 pack
[14:47:59] thread 1 is processing the 160998 / 161190 pack
[14:47:59] thread 2 is processing the 160999 / 161190 pack
[14:47:59] thread 1 is processing the 161000 / 161190 pack
[14:47:59] thread 2 is processing the 161001 / 161190 pack
[14:47:59] thread 1 is processing the 161002 / 161190 pack
[14:47:59] thread 2 is processing the 161003 / 161190 pack
[14:47:59] thread 1 is processing the 161004 / 161190 pack
[14:47:59] thread 2 is processing the 161005 / 161190 pack
[14:47:59] thread 1 is processing the 161006 / 161190 pack
[14:47:59] thread 2 is processing the 161007 / 161190 pack
[14:47:59] thread 1 is processing the 161008 / 161190 pack
[14:47:59] thread 2 is processing the 161009 / 161190 pack
[14:47:59] thread 1 is processing the 161010 / 161190 pack
[14:47:59] thread 2 is processing the 161011 / 161190 pack
[14:47:59] thread 1 is processing the 161012 / 161190 pack
[14:47:59] thread 2 is processing the 161013 / 161190 pack
[14:47:59] thread 1 is processing the 161014 / 161190 pack
[14:47:59] thread 2 is processing the 161015 / 161190 pack
[14:47:59] thread 1 is processing the 161016 / 161190 pack
[14:47:59] thread 2 is processing the 161017 / 161190 pack
[14:47:59] thread 1 is processing the 161018 / 161190 pack
[14:47:59] thread 2 is processing the 161019 / 161190 pack
[14:47:59] thread 1 is processing the 161020 / 161190 pack
[14:47:59] thread 2 is processing the 161021 / 161190 pack
[14:47:59] thread 1 is processing the 161022 / 161190 pack
[14:47:59] thread 2 is processing the 161023 / 161190 pack
[14:47:59] thread 1 is processing the 161024 / 161190 pack
[14:47:59] thread 2 is processing the 161025 / 161190 pack
[14:47:59] thread 1 is processing the 161026 / 161190 pack
[14:48:0] thread 2 is processing the 161027 / 161190 pack
[14:48:0] thread 1 is processing the 161028 / 161190 pack
[14:48:0] thread 2 is processing the 161029 / 161190 pack
[14:48:0] thread 1 is processing the 161030 / 161190 pack
[14:48:0] thread 2 is processing the 161031 / 161190 pack
[14:48:0] thread 1 is processing the 161032 / 161190 pack
[14:48:0] thread 2 is processing the 161033 / 161190 pack
[14:48:0] thread 1 is processing the 161034 / 161190 pack
[14:48:0] thread 2 is processing the 161035 / 161190 pack
[14:48:0] thread 1 is processing the 161036 / 161190 pack
[14:48:0] thread 2 is processing the 161037 / 161190 pack
[14:48:0] thread 1 is processing the 161038 / 161190 pack
[14:48:0] thread 2 is processing the 161039 / 161190 pack
[14:48:0] thread 1 is processing the 161040 / 161190 pack
[14:48:0] thread 2 is processing the 161041 / 161190 pack
[14:48:0] thread 1 is processing the 161042 / 161190 pack
[14:48:0] thread 2 is processing the 161043 / 161190 pack
[14:48:0] thread 1 is processing the 161044 / 161190 pack
[14:48:0] thread 2 is processing the 161045 / 161190 pack
[14:48:0] thread 1 is processing the 161046 / 161190 pack
[14:48:0] thread 2 is processing the 161047 / 161190 pack
[14:48:0] thread 1 is processing the 161048 / 161190 pack
[14:48:0] thread 2 is processing the 161049 / 161190 pack
[14:48:0] thread 1 is processing the 161050 / 161190 pack
[14:48:0] thread 2 is processing the 161051 / 161190 pack
[14:48:0] thread 1 is processing the 161052 / 161190 pack
[14:48:0] thread 2 is processing the 161053 / 161190 pack
[14:48:0] thread 1 is processing the 161054 / 161190 pack
[14:48:0] thread 2 is processing the 161055 / 161190 pack
[14:48:0] thread 1 is processing the 161056 / 161190 pack
[14:48:0] thread 2 is processing the 161057 / 161190 pack
[14:48:0] thread 1 is processing the 161058 / 161190 pack
[14:48:0] thread 2 is processing the 161059 / 161190 pack
[14:48:0] thread 1 is processing the 161060 / 161190 pack
[14:48:0] thread 2 is processing the 161061 / 161190 pack
[14:48:0] thread 1 is processing the 161062 / 161190 pack
[14:48:0] thread 2 is processing the 161063 / 161190 pack
[14:48:0] thread 1 is processing the 161064 / 161190 pack
[14:48:0] thread 2 is processing the 161065 / 161190 pack
[14:48:0] thread 1 is processing the 161066 / 161190 pack
[14:48:0] thread 2 is processing the 161067 / 161190 pack
[14:48:0] thread 1 is processing the 161068 / 161190 pack
[14:48:0] thread 2 is processing the 161069 / 161190 pack
[14:48:0] thread 1 is processing the 161070 / 161190 pack
[14:48:0] thread 2 is processing the 161071 / 161190 pack
[14:48:0] thread 1 is processing the 161071 / 161190 pack
[14:48:0] thread 2 is processing the 161073 / 161190 pack
[14:48:0] thread 1 is processing the 161073 / 161190 pack
[14:48:0] thread 1 is processing the 161075 / 161190 pack
[14:48:0] thread 2 is processing the 161075 / 161190 pack
[14:48:0] thread 1 is processing the 161077 / 161190 pack
[14:48:0] thread 2 is processing the 161078 / 161190 pack
[14:48:0] thread 1 is processing the 161079 / 161190 pack
[14:48:0] thread 2 is processing the 161080 / 161190 pack
[14:48:0] thread 2 is processing the 161081 / 161190 pack
[14:48:0] thread 1 is processing the 161082 / 161190 pack
[14:48:0] thread 2 is processing the 161083 / 161190 pack
[14:48:0] thread 1 is processing the 161084 / 161190 pack
[14:48:0] thread 2 is processing the 161085 / 161190 pack
[14:48:0] thread 1 is processing the 161086 / 161190 pack
[14:48:0] thread 2 is processing the 161087 / 161190 pack
[14:48:0] thread 1 is processing the 161088 / 161190 pack
[14:48:0] thread 2 is processing the 161089 / 161190 pack
[14:48:0] thread 1 is processing the 161090 / 161190 pack
[14:48:0] thread 1 is processing the 161091 / 161190 pack
[14:48:0] thread 2 is processing the 161092 / 161190 pack
[14:48:0] thread 1 is processing the 161093 / 161190 pack
[14:48:0] thread 2 is processing the 161093 / 161190 pack
[14:48:0] thread 1 is processing the 161095 / 161190 pack
[14:48:0] thread 2 is processing the 161096 / 161190 pack
[14:48:0] thread 1 is processing the 161097 / 161190 pack
[14:48:0] thread 2 is processing the 161098 / 161190 pack
[14:48:0] thread 1 is processing the 161099 / 161190 pack
[14:48:0] thread 2 is processing the 161099 / 161190 pack
[14:48:0] thread 1 is processing the 161101 / 161190 pack
[14:48:0] thread 2 is processing the 161102 / 161190 pack
[14:48:0] thread 1 is processing the 161103 / 161190 pack
[14:48:0] thread 2 is processing the 161104 / 161190 pack
[14:48:0] thread 1 is processing the 161105 / 161190 pack
[14:48:0] thread 2 is processing the 161106 / 161190 pack
[14:48:0] thread 1 is processing the 161107 / 161190 pack
[14:48:0] thread 2 is processing the 161108 / 161190 pack
[14:48:0] thread 1 is processing the 161109 / 161190 pack
[14:48:0] thread 2 is processing the 161110 / 161190 pack
[14:48:0] thread 1 is processing the 161111 / 161190 pack
[14:48:0] thread 2 is processing the 161112 / 161190 pack
[14:48:1] thread 1 is processing the 161113 / 161190 pack
[14:48:1] thread 2 is processing the 161114 / 161190 pack
[14:48:1] thread 1 is processing the 161115 / 161190 pack
[14:48:1] thread 2 is processing the 161116 / 161190 pack
[14:48:1] thread 1 is processing the 161117 / 161190 pack
[14:48:1] thread 2 is processing the 161118 / 161190 pack
[14:48:1] thread 1 is processing the 161119 / 161190 pack
[14:48:1] thread 2 is processing the 161120 / 161190 pack
[14:48:1] thread 1 is processing the 161121 / 161190 pack
[14:48:1] thread 2 is processing the 161122 / 161190 pack
[14:48:1] thread 1 is processing the 161123 / 161190 pack
[14:48:1] thread 2 is processing the 161124 / 161190 pack
[14:48:1] thread 1 is processing the 161125 / 161190 pack
[14:48:1] thread 2 is processing the 161126 / 161190 pack
[14:48:1] thread 1 is processing the 161127 / 161190 pack
[14:48:1] thread 2 is processing the 161128 / 161190 pack
[14:48:1] thread 1 is processing the 161129 / 161190 pack
[14:48:1] thread 2 is processing the 161130 / 161190 pack
[14:48:1] thread 1 is processing the 161131 / 161190 pack
[14:48:1] thread 2 is processing the 161132 / 161190 pack
[14:48:1] thread 1 is processing the 161133 / 161190 pack
[14:48:1] thread 2 is processing the 161134 / 161190 pack
[14:48:1] thread 1 is processing the 161135 / 161190 pack
[14:48:1] thread 2 is processing the 161136 / 161190 pack
[14:48:1] thread 1 is processing the 161137 / 161190 pack
[14:48:1] thread 2 is processing the 161138 / 161190 pack
[14:48:1] thread 1 is processing the 161139 / 161190 pack
[14:48:1] thread 2 is processing the 161140 / 161190 pack
[14:48:1] thread 1 is processing the 161141 / 161190 pack
[14:48:1] thread 2 is processing the 161142 / 161190 pack
[14:48:1] thread 1 is processing the 161143 / 161190 pack
[14:48:1] thread 2 is processing the 161144 / 161190 pack
[14:48:1] thread 1 is processing the 161145 / 161190 pack
[14:48:1] thread 2 is processing the 161146 / 161190 pack
[14:48:1] thread 1 is processing the 161147 / 161190 pack
[14:48:1] thread 2 is processing the 161148 / 161190 pack
[14:48:1] thread 1 is processing the 161149 / 161190 pack
[14:48:1] thread 2 is processing the 161150 / 161190 pack
[14:48:1] thread 1 is processing the 161151 / 161190 pack
[14:48:1] thread 2 is processing the 161152 / 161190 pack
[14:48:1] thread 1 is processing the 161153 / 161190 pack
[14:48:1] thread 2 is processing the 161154 / 161190 pack
[14:48:1] thread 1 is processing the 161155 / 161190 pack
[14:48:1] thread 2 is processing the 161156 / 161190 pack
[14:48:1] thread 1 is processing the 161157 / 161190 pack
[14:48:1] thread 2 is processing the 161158 / 161190 pack
[14:48:1] thread 1 is processing the 161159 / 161190 pack
[14:48:1] thread 2 is processing the 161160 / 161190 pack
[14:48:1] thread 1 is processing the 161161 / 161190 pack
[14:48:1] thread 2 is processing the 161162 / 161190 pack
[14:48:1] thread 1 is processing the 161163 / 161190 pack
[14:48:1] thread 2 is processing the 161164 / 161190 pack
[14:48:1] thread 1 is processing the 161165 / 161190 pack
[14:48:1] thread 2 is processing the 161166 / 161190 pack
[14:48:1] thread 1 is processing the 161167 / 161190 pack
[14:48:1] thread 2 is processing the 161168 / 161190 pack
[14:48:1] thread 1 is processing the 161169 / 161190 pack
[14:48:1] thread 2 is processing the 161170 / 161190 pack
[14:48:1] thread 1 is processing the 161171 / 161190 pack
[14:48:1] thread 2 is processing the 161172 / 161190 pack
[14:48:1] thread 1 is processing the 161173 / 161190 pack
[14:48:1] thread 2 is processing the 161174 / 161190 pack
[14:48:1] thread 1 is processing the 161175 / 161190 pack
[14:48:1] thread 2 is processing the 161176 / 161190 pack
[14:48:1] thread 1 is processing the 161177 / 161190 pack
[14:48:1] thread 2 is processing the 161178 / 161190 pack
[14:48:1] thread 1 is processing the 161179 / 161190 pack
[14:48:1] thread 2 is processing the 161180 / 161190 pack
[14:48:1] thread 1 is processing the 161181 / 161190 pack
[14:48:1] thread 2 is processing the 161182 / 161190 pack
[14:48:1] thread 1 is processing the 161183 / 161190 pack
[14:48:1] thread 2 is processing the 161183 / 161190 pack
[14:48:1] thread 1 is processing the 161185 / 161190 pack
[14:48:1] thread 2 is processing the 161186 / 161190 pack
[14:48:1] thread 1 is processing the 161187 / 161190 pack
[14:48:1] thread 2 is processing the 161188 / 161190 pack
[14:48:1] thread 1 is processing the 161189 / 161190 pack
[14:48:1] thread 2 data processing completed
[14:48:1] thread 2 finished
[14:48:1] thread 1 data processing completed
[14:48:1] thread 1 finished
[14:48:1] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/YoungLarvae_QC_R1.fastq.gz writer finished
[14:48:1] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/YoungLarvae_QC_R2.fastq.gz writer finished
[14:48:1] start to generate reports

Read1 before filtering:
total reads: 161189227
total bases: 24339573277
Q20 bases: 23849414790(97.9862%)
Q30 bases: 23013418903(94.5514%)

Read2 before filtering:
total reads: 161189227
total bases: 24339573277
Q20 bases: 23533332752(96.6875%)
Q30 bases: 22406215989(92.0567%)

Read1 after filtering:
total reads: 158277328
total bases: 23635001823
Q20 bases: 23211291256(98.2073%)
Q30 bases: 22425992029(94.8847%)

Read2 after filtering:
total reads: 158277328
total bases: 23632793700
Q20 bases: 23066047454(97.6019%)
Q30 bases: 22019370495(93.1729%)

Filtering result:
reads passed filter: 316554656
reads failed due to low quality: 5660392
reads failed due to too many N: 14372
reads failed due to too short: 149034
reads with adapter trimmed: 28107492
bases trimmed due to adapters: 531370219

Duplication rate: 6.90022%

Insert size peak (evaluated by paired-end reads): 205

Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[13:58:26] start to load data
[13:58:35] loaded 1M read pairs
[13:58:46] loaded 2M read pairs
[13:58:59] loaded 3M read pairs
[13:59:12] loaded 4M read pairs
[13:59:24] loaded 5M read pairs
[13:59:37] loaded 6M read pairs
[13:59:49] loaded 7M read pairs
[14:0:1] loaded 8M read pairs
[14:0:13] loaded 9M read pairs
[14:0:26] loaded 10M read pairs
[14:0:38] loaded 11M read pairs
[14:0:52] loaded 12M read pairs
[14:1:8] loaded 13M read pairs
[14:1:22] loaded 14M read pairs
[14:1:36] loaded 15M read pairs
[14:1:49] loaded 16M read pairs
[14:2:1] loaded 17M read pairs
[14:2:14] loaded 18M read pairs
[14:2:26] loaded 19M read pairs
[14:2:38] loaded 20M read pairs
[14:2:51] loaded 21M read pairs
[14:3:4] loaded 22M read pairs
[14:3:17] loaded 23M read pairs
[14:3:30] loaded 24M read pairs
[14:3:43] loaded 25M read pairs
[14:3:56] loaded 26M read pairs
[14:4:9] loaded 27M read pairs
[14:4:23] loaded 28M read pairs
[14:4:37] loaded 29M read pairs
[14:4:50] loaded 30M read pairs
[14:5:3] loaded 31M read pairs
[14:5:16] loaded 32M read pairs
[14:5:30] loaded 33M read pairs
[14:5:42] loaded 34M read pairs
[14:5:53] loaded 35M read pairs
[14:6:4] loaded 36M read pairs
[14:6:15] loaded 37M read pairs
[14:6:26] loaded 38M read pairs
[14:6:37] loaded 39M read pairs
[14:6:48] loaded 40M read pairs
[14:6:58] loaded 41M read pairs
[14:7:9] loaded 42M read pairs
[14:7:20] loaded 43M read pairs
[14:7:30] loaded 44M read pairs
[14:7:41] loaded 45M read pairs
[14:7:52] loaded 46M read pairs
[14:8:2] loaded 47M read pairs
[14:8:13] loaded 48M read pairs
[14:8:24] loaded 49M read pairs
[14:8:35] loaded 50M read pairs
[14:8:46] loaded 51M read pairs
[14:8:57] loaded 52M read pairs
[14:9:8] loaded 53M read pairs
[14:9:19] loaded 54M read pairs
[14:9:30] loaded 55M read pairs
[14:9:40] loaded 56M read pairs
[14:9:51] loaded 57M read pairs
[14:10:2] loaded 58M read pairs
[14:10:13] loaded 59M read pairs
[14:10:24] loaded 60M read pairs
[14:10:35] loaded 61M read pairs
[14:10:45] loaded 62M read pairs
[14:10:56] loaded 63M read pairs
[14:11:7] loaded 64M read pairs
[14:11:18] loaded 65M read pairs
[14:11:29] loaded 66M read pairs
[14:11:40] loaded 67M read pairs
[14:11:51] loaded 68M read pairs
[14:12:2] loaded 69M read pairs
[14:12:13] loaded 70M read pairs
[14:12:24] loaded 71M read pairs
[14:12:35] loaded 72M read pairs
[14:12:46] loaded 73M read pairs
[14:12:57] loaded 74M read pairs
[14:13:8] loaded 75M read pairs
[14:13:18] loaded 76M read pairs
[14:13:29] loaded 77M read pairs
[14:13:40] loaded 78M read pairs
[14:13:51] loaded 79M read pairs
[14:14:2] loaded 80M read pairs
[14:14:13] loaded 81M read pairs
[14:14:24] loaded 82M read pairs
[14:14:35] loaded 83M read pairs
[14:14:47] loaded 84M read pairs
[14:14:58] loaded 85M read pairs
[14:15:9] loaded 86M read pairs
[14:15:20] loaded 87M read pairs
Detecting adapter sequence for read1...
>Illumina TruSeq Adapter Read 1
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA

Detecting adapter sequence for read2...
>Illumina TruSeq Adapter Read 2
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT

[14:17:49] start to load data
[14:17:56] loaded 1M read pairs
[14:18:8] loaded 2M read pairs
[14:18:20] loaded 3M read pairs
[14:18:33] loaded 4M read pairs
[14:18:45] loaded 5M read pairs
[14:18:58] loaded 6M read pairs
[14:19:11] loaded 7M read pairs
[14:19:23] loaded 8M read pairs
[14:19:35] loaded 9M read pairs
[14:19:46] loaded 10M read pairs
[14:19:57] loaded 11M read pairs
[14:20:10] loaded 12M read pairs
[14:20:22] loaded 13M read pairs
[14:20:35] loaded 14M read pairs
[14:20:47] loaded 15M read pairs
[14:20:59] loaded 16M read pairs
[14:21:12] loaded 17M read pairs
[14:21:25] loaded 18M read pairs
[14:21:37] loaded 19M read pairs
[14:21:50] loaded 20M read pairs
[14:22:3] loaded 21M read pairs
[14:22:15] loaded 22M read pairs
[14:22:28] loaded 23M read pairs
[14:22:41] loaded 24M read pairs
[14:22:53] loaded 25M read pairs
[14:23:4] loaded 26M read pairs
[14:23:15] loaded 27M read pairs
[14:23:27] loaded 28M read pairs
[14:23:40] loaded 29M read pairs
[14:23:52] loaded 30M read pairs
[14:24:5] loaded 31M read pairs
[14:24:17] loaded 32M read pairs
[14:24:30] loaded 33M read pairs
[14:24:43] loaded 34M read pairs
[14:24:55] loaded 35M read pairs
[14:25:8] loaded 36M read pairs
[14:25:21] loaded 37M read pairs
[14:25:34] loaded 38M read pairs
[14:25:47] loaded 39M read pairs
[14:25:58] loaded 40M read pairs
[14:26:8] loaded 41M read pairs
[14:26:21] loaded 42M read pairs
[14:26:33] loaded 43M read pairs
[14:26:46] loaded 44M read pairs
[14:26:58] loaded 45M read pairs
[14:27:10] loaded 46M read pairs
[14:27:23] loaded 47M read pairs
[14:27:36] loaded 48M read pairs
[14:27:49] loaded 49M read pairs
[14:28:2] loaded 50M read pairs
[14:28:15] loaded 51M read pairs
[14:28:27] loaded 52M read pairs
[14:28:40] loaded 53M read pairs
[14:28:53] loaded 54M read pairs
[14:29:6] loaded 55M read pairs
[14:29:17] loaded 56M read pairs
[14:29:28] loaded 57M read pairs
[14:29:39] loaded 58M read pairs
[14:29:51] loaded 59M read pairs
[14:30:4] loaded 60M read pairs
[14:30:16] loaded 61M read pairs
[14:30:29] loaded 62M read pairs
[14:30:42] loaded 63M read pairs
[14:30:54] loaded 64M read pairs
[14:31:7] loaded 65M read pairs
[14:31:20] loaded 66M read pairs
[14:31:33] loaded 67M read pairs
[14:31:46] loaded 68M read pairs
[14:31:59] loaded 69M read pairs
[14:32:12] loaded 70M read pairs
[14:32:25] loaded 71M read pairs
[14:32:37] loaded 72M read pairs
[14:32:49] loaded 73M read pairs
[14:33:1] loaded 74M read pairs
[14:33:13] loaded 75M read pairs
[14:33:26] loaded 76M read pairs
[14:33:38] loaded 77M read pairs
[14:33:49] loaded 78M read pairs
[14:34:0] loaded 79M read pairs
[14:34:10] loaded 80M read pairs
[14:34:21] loaded 81M read pairs
[14:34:32] loaded 82M read pairs
[14:34:43] loaded 83M read pairs
[14:34:54] loaded 84M read pairs
[14:35:6] loaded 85M read pairs
[14:35:17] loaded 86M read pairs
[14:35:28] loaded 87M read pairs
[14:35:40] loaded 88M read pairs
[14:35:52] loaded 89M read pairs
[14:36:4] loaded 90M read pairs
[14:36:16] loaded 91M read pairs
[14:36:29] loaded 92M read pairs
[14:36:42] loaded 93M read pairs
[14:36:54] loaded 94M read pairs
[14:37:7] loaded 95M read pairs
[14:37:19] loaded 96M read pairs
[14:37:32] loaded 97M read pairs
[14:37:45] loaded 98M read pairs
[14:37:57] loaded 99M read pairs
[14:38:10] loaded 100M read pairs
[14:38:23] loaded 101M read pairs
[14:38:36] loaded 102M read pairs
[14:38:49] loaded 103M read pairs
[14:39:2] loaded 104M read pairs
[14:39:15] loaded 105M read pairs
[14:39:28] loaded 106M read pairs
[14:39:41] loaded 107M read pairs
[14:39:53] loaded 108M read pairs
[14:40:6] loaded 109M read pairs
[14:40:19] loaded 110M read pairs
[14:40:32] loaded 111M read pairs
[14:40:45] loaded 112M read pairs
[14:40:57] loaded 113M read pairs
[14:41:10] loaded 114M read pairs
[14:41:23] loaded 115M read pairs
[14:41:35] loaded 116M read pairs
[14:41:48] loaded 117M read pairs
[14:42:0] loaded 118M read pairs
[14:42:13] loaded 119M read pairs
[14:42:26] loaded 120M read pairs
[14:42:38] loaded 121M read pairs
[14:42:51] loaded 122M read pairs
[14:43:4] loaded 123M read pairs
[14:43:16] loaded 124M read pairs
[14:43:29] loaded 125M read pairs
[14:43:42] loaded 126M read pairs
[14:43:55] loaded 127M read pairs
[14:44:8] loaded 128M read pairs
[14:44:21] loaded 129M read pairs
[14:44:34] loaded 130M read pairs
[14:44:47] loaded 131M read pairs
[14:45:0] loaded 132M read pairs
[14:45:13] loaded 133M read pairs
[14:45:26] loaded 134M read pairs
[14:45:39] loaded 135M read pairs
[14:45:52] loaded 136M read pairs
[14:46:5] loaded 137M read pairs
[14:46:18] loaded 138M read pairs
[14:46:30] loaded 139M read pairs
[14:46:43] loaded 140M read pairs
[14:46:56] loaded 141M read pairs
[14:47:8] loaded 142M read pairs
[14:47:21] loaded 143M read pairs
[14:47:34] loaded 144M read pairs
[14:47:46] loaded 145M read pairs
[14:47:59] loaded 146M read pairs
[14:48:12] loaded 147M read pairs
[14:48:24] loaded 148M read pairs
[14:48:37] loaded 149M read pairs
[14:48:50] loaded 150M read pairs
[14:49:3] loaded 151M read pairs
[14:49:15] loaded 152M read pairs
[14:49:28] loaded 153M read pairs
[14:49:41] loaded 154M read pairs
[14:49:53] loaded 155M read pairs
[14:50:5] loaded 156M read pairs
[14:50:16] loaded 157M read pairs
[14:50:28] loaded 158M read pairs
[14:50:40] loaded 159M read pairs
[14:50:51] loaded 160M read pairs
[14:51:3] loaded 161M read pairs
[14:51:15] loaded 162M read pairs
[14:51:27] loaded 163M read pairs
[14:51:38] loaded 164M read pairs
[14:51:50] loaded 165M read pairs
[14:52:1] loaded 166M read pairs
[14:52:13] loaded 167M read pairs
[14:52:24] loaded 168M read pairs
[14:52:36] loaded 169M read pairs
[14:52:48] loaded 170M read pairs
[14:52:59] loaded 171M read pairs
[14:53:11] loaded 172M read pairs
[14:53:22] loaded 173M read pairs
[14:53:33] loaded 174M read pairs
[14:53:45] loaded 175M read pairs
[14:53:56] loaded 176M read pairs
[14:54:8] loaded 177M read pairs
[14:54:19] loaded 178M read pairs
[14:54:32] loaded 179M read pairs
[14:54:44] loaded 180M read pairs
[14:54:56] loaded 181M read pairs
[14:55:9] loaded 182M read pairs
[14:55:21] loaded 183M read pairs
[14:55:34] loaded 184M read pairs
[14:55:35] all reads loaded, start to monitor thread status
[14:55:35] thread 2 is processing the 183589 / 184090 pack
[14:55:35] thread 1 is processing the 183590 / 184090 pack
[14:55:35] thread 2 is processing the 183591 / 184090 pack
[14:55:35] thread 1 is processing the 183592 / 184090 pack
[14:55:35] thread 2 is processing the 183593 / 184090 pack
[14:55:35] thread 1 is processing the 183594 / 184090 pack
[14:55:35] thread 2 is processing the 183595 / 184090 pack
[14:55:35] thread 1 is processing the 183596 / 184090 pack
[14:55:35] thread 2 is processing the 183597 / 184090 pack
[14:55:35] thread 1 is processing the 183598 / 184090 pack
[14:55:35] thread 2 is processing the 183599 / 184090 pack
[14:55:35] thread 1 is processing the 183600 / 184090 pack
[14:55:35] thread 2 is processing the 183601 / 184090 pack
[14:55:35] thread 1 is processing the 183602 / 184090 pack
[14:55:35] thread 2 is processing the 183603 / 184090 pack
[14:55:35] thread 1 is processing the 183604 / 184090 pack
[14:55:35] thread 2 is processing the 183605 / 184090 pack
[14:55:35] thread 1 is processing the 183606 / 184090 pack
[14:55:35] thread 2 is processing the 183607 / 184090 pack
[14:55:35] thread 1 is processing the 183608 / 184090 pack
[14:55:35] thread 2 is processing the 183609 / 184090 pack
[14:55:35] thread 1 is processing the 183610 / 184090 pack
[14:55:35] thread 2 is processing the 183611 / 184090 pack
[14:55:35] thread 1 is processing the 183612 / 184090 pack
[14:55:35] thread 2 is processing the 183613 / 184090 pack
[14:55:35] thread 1 is processing the 183614 / 184090 pack
[14:55:35] thread 2 is processing the 183615 / 184090 pack
[14:55:35] thread 1 is processing the 183616 / 184090 pack
[14:55:35] thread 2 is processing the 183617 / 184090 pack
[14:55:35] thread 1 is processing the 183618 / 184090 pack
[14:55:35] thread 2 is processing the 183619 / 184090 pack
[14:55:35] thread 1 is processing the 183620 / 184090 pack
[14:55:35] thread 2 is processing the 183621 / 184090 pack
[14:55:35] thread 1 is processing the 183622 / 184090 pack
[14:55:35] thread 2 is processing the 183623 / 184090 pack
[14:55:35] thread 1 is processing the 183624 / 184090 pack
[14:55:35] thread 2 is processing the 183625 / 184090 pack
[14:55:35] thread 1 is processing the 183626 / 184090 pack
[14:55:35] thread 2 is processing the 183627 / 184090 pack
[14:55:35] thread 1 is processing the 183628 / 184090 pack
[14:55:35] thread 2 is processing the 183629 / 184090 pack
[14:55:35] thread 1 is processing the 183630 / 184090 pack
[14:55:35] thread 2 is processing the 183631 / 184090 pack
[14:55:35] thread 1 is processing the 183632 / 184090 pack
[14:55:35] thread 2 is processing the 183633 / 184090 pack
[14:55:35] thread 1 is processing the 183634 / 184090 pack
[14:55:35] thread 2 is processing the 183635 / 184090 pack
[14:55:35] thread 1 is processing the 183636 / 184090 pack
[14:55:35] thread 2 is processing the 183637 / 184090 pack
[14:55:35] thread 1 is processing the 183638 / 184090 pack
[14:55:35] thread 2 is processing the 183639 / 184090 pack
[14:55:35] thread 1 is processing the 183640 / 184090 pack
[14:55:35] thread 2 is processing the 183641 / 184090 pack
[14:55:35] thread 1 is processing the 183642 / 184090 pack
[14:55:35] thread 2 is processing the 183643 / 184090 pack
[14:55:35] thread 1 is processing the 183644 / 184090 pack
[14:55:35] thread 2 is processing the 183645 / 184090 pack
[14:55:35] thread 1 is processing the 183646 / 184090 pack
[14:55:35] thread 2 is processing the 183647 / 184090 pack
[14:55:35] thread 1 is processing the 183648 / 184090 pack
[14:55:35] thread 2 is processing the 183649 / 184090 pack
[14:55:35] thread 1 is processing the 183650 / 184090 pack
[14:55:35] thread 2 is processing the 183651 / 184090 pack
[14:55:35] thread 1 is processing the 183652 / 184090 pack
[14:55:35] thread 2 is processing the 183653 / 184090 pack
[14:55:35] thread 1 is processing the 183654 / 184090 pack
[14:55:35] thread 2 is processing the 183655 / 184090 pack
[14:55:35] thread 1 is processing the 183656 / 184090 pack
[14:55:35] thread 2 is processing the 183657 / 184090 pack
[14:55:35] thread 1 is processing the 183658 / 184090 pack
[14:55:35] thread 2 is processing the 183659 / 184090 pack
[14:55:35] thread 1 is processing the 183660 / 184090 pack
[14:55:35] thread 2 is processing the 183661 / 184090 pack
[14:55:35] thread 1 is processing the 183662 / 184090 pack
[14:55:35] thread 2 is processing the 183663 / 184090 pack
[14:55:35] thread 1 is processing the 183664 / 184090 pack
[14:55:35] thread 2 is processing the 183665 / 184090 pack
[14:55:35] thread 1 is processing the 183666 / 184090 pack
[14:55:35] thread 2 is processing the 183667 / 184090 pack
[14:55:35] thread 1 is processing the 183668 / 184090 pack
[14:55:35] thread 2 is processing the 183669 / 184090 pack
[14:55:35] thread 1 is processing the 183670 / 184090 pack
[14:55:35] thread 2 is processing the 183671 / 184090 pack
[14:55:35] thread 1 is processing the 183672 / 184090 pack
[14:55:35] thread 2 is processing the 183673 / 184090 pack
[14:55:35] thread 1 is processing the 183674 / 184090 pack
[14:55:35] thread 2 is processing the 183675 / 184090 pack
[14:55:35] thread 1 is processing the 183676 / 184090 pack
[14:55:35] thread 2 is processing the 183677 / 184090 pack
[14:55:36] thread 1 is processing the 183678 / 184090 pack
[14:55:36] thread 2 is processing the 183679 / 184090 pack
[14:55:36] thread 1 is processing the 183680 / 184090 pack
[14:55:36] thread 2 is processing the 183681 / 184090 pack
[14:55:36] thread 1 is processing the 183682 / 184090 pack
[14:55:36] thread 2 is processing the 183683 / 184090 pack
[14:55:36] thread 1 is processing the 183684 / 184090 pack
[14:55:36] thread 2 is processing the 183685 / 184090 pack
[14:55:36] thread 1 is processing the 183686 / 184090 pack
[14:55:36] thread 2 is processing the 183687 / 184090 pack
[14:55:36] thread 1 is processing the 183688 / 184090 pack
[14:55:36] thread 2 is processing the 183689 / 184090 pack
[14:55:36] thread 1 is processing the 183690 / 184090 pack
[14:55:36] thread 2 is processing the 183691 / 184090 pack
[14:55:36] thread 1 is processing the 183692 / 184090 pack
[14:55:36] thread 2 is processing the 183693 / 184090 pack
[14:55:36] thread 1 is processing the 183694 / 184090 pack
[14:55:36] thread 2 is processing the 183695 / 184090 pack
[14:55:36] thread 1 is processing the 183696 / 184090 pack
[14:55:36] thread 2 is processing the 183697 / 184090 pack
[14:55:36] thread 1 is processing the 183698 / 184090 pack
[14:55:36] thread 2 is processing the 183699 / 184090 pack
[14:55:36] thread 1 is processing the 183700 / 184090 pack
[14:55:36] thread 2 is processing the 183701 / 184090 pack
[14:55:36] thread 1 is processing the 183702 / 184090 pack
[14:55:36] thread 2 is processing the 183703 / 184090 pack
[14:55:36] thread 1 is processing the 183704 / 184090 pack
[14:55:36] thread 2 is processing the 183705 / 184090 pack
[14:55:36] thread 1 is processing the 183706 / 184090 pack
[14:55:36] thread 2 is processing the 183707 / 184090 pack
[14:55:36] thread 1 is processing the 183708 / 184090 pack
[14:55:36] thread 2 is processing the 183709 / 184090 pack
[14:55:36] thread 1 is processing the 183710 / 184090 pack
[14:55:36] thread 2 is processing the 183711 / 184090 pack
[14:55:36] thread 1 is processing the 183712 / 184090 pack
[14:55:36] thread 2 is processing the 183713 / 184090 pack
[14:55:36] thread 1 is processing the 183714 / 184090 pack
[14:55:36] thread 2 is processing the 183715 / 184090 pack
[14:55:36] thread 1 is processing the 183716 / 184090 pack
[14:55:36] thread 2 is processing the 183717 / 184090 pack
[14:55:36] thread 1 is processing the 183718 / 184090 pack
[14:55:36] thread 2 is processing the 183719 / 184090 pack
[14:55:36] thread 1 is processing the 183720 / 184090 pack
[14:55:36] thread 2 is processing the 183721 / 184090 pack
[14:55:36] thread 1 is processing the 183722 / 184090 pack
[14:55:36] thread 2 is processing the 183723 / 184090 pack
[14:55:36] thread 1 is processing the 183724 / 184090 pack
[14:55:36] thread 2 is processing the 183725 / 184090 pack
[14:55:36] thread 1 is processing the 183726 / 184090 pack
[14:55:36] thread 2 is processing the 183727 / 184090 pack
[14:55:36] thread 1 is processing the 183728 / 184090 pack
[14:55:36] thread 2 is processing the 183729 / 184090 pack
[14:55:36] thread 1 is processing the 183730 / 184090 pack
[14:55:36] thread 2 is processing the 183731 / 184090 pack
[14:55:36] thread 1 is processing the 183732 / 184090 pack
[14:55:36] thread 2 is processing the 183733 / 184090 pack
[14:55:36] thread 1 is processing the 183734 / 184090 pack
[14:55:36] thread 2 is processing the 183735 / 184090 pack
[14:55:36] thread 1 is processing the 183736 / 184090 pack
[14:55:36] thread 2 is processing the 183737 / 184090 pack
[14:55:36] thread 1 is processing the 183738 / 184090 pack
[14:55:36] thread 2 is processing the 183739 / 184090 pack
[14:55:36] thread 1 is processing the 183740 / 184090 pack
[14:55:36] thread 2 is processing the 183741 / 184090 pack
[14:55:36] thread 1 is processing the 183742 / 184090 pack
[14:55:36] thread 2 is processing the 183743 / 184090 pack
[14:55:36] thread 1 is processing the 183744 / 184090 pack
[14:55:36] thread 2 is processing the 183745 / 184090 pack
[14:55:36] thread 1 is processing the 183746 / 184090 pack
[14:55:36] thread 2 is processing the 183747 / 184090 pack
[14:55:36] thread 1 is processing the 183748 / 184090 pack
[14:55:36] thread 2 is processing the 183749 / 184090 pack
[14:55:36] thread 1 is processing the 183750 / 184090 pack
[14:55:36] thread 2 is processing the 183751 / 184090 pack
[14:55:36] thread 1 is processing the 183752 / 184090 pack
[14:55:36] thread 2 is processing the 183753 / 184090 pack
[14:55:36] thread 1 is processing the 183754 / 184090 pack
[14:55:36] thread 2 is processing the 183755 / 184090 pack
[14:55:36] thread 1 is processing the 183756 / 184090 pack
[14:55:36] thread 2 is processing the 183757 / 184090 pack
[14:55:36] thread 1 is processing the 183758 / 184090 pack
[14:55:36] thread 2 is processing the 183759 / 184090 pack
[14:55:36] thread 1 is processing the 183760 / 184090 pack
[14:55:36] thread 2 is processing the 183761 / 184090 pack
[14:55:36] thread 1 is processing the 183762 / 184090 pack
[14:55:36] thread 2 is processing the 183763 / 184090 pack
[14:55:36] thread 1 is processing the 183764 / 184090 pack
[14:55:36] thread 2 is processing the 183765 / 184090 pack
[14:55:36] thread 1 is processing the 183766 / 184090 pack
[14:55:36] thread 2 is processing the 183767 / 184090 pack
[14:55:36] thread 1 is processing the 183768 / 184090 pack
[14:55:36] thread 2 is processing the 183769 / 184090 pack
[14:55:36] thread 1 is processing the 183770 / 184090 pack
[14:55:36] thread 2 is processing the 183771 / 184090 pack
[14:55:36] thread 1 is processing the 183772 / 184090 pack
[14:55:37] thread 2 is processing the 183773 / 184090 pack
[14:55:37] thread 1 is processing the 183774 / 184090 pack
[14:55:37] thread 2 is processing the 183775 / 184090 pack
[14:55:37] thread 1 is processing the 183776 / 184090 pack
[14:55:37] thread 2 is processing the 183777 / 184090 pack
[14:55:37] thread 1 is processing the 183778 / 184090 pack
[14:55:37] thread 2 is processing the 183779 / 184090 pack
[14:55:37] thread 1 is processing the 183780 / 184090 pack
[14:55:37] thread 2 is processing the 183781 / 184090 pack
[14:55:37] thread 1 is processing the 183782 / 184090 pack
[14:55:37] thread 2 is processing the 183783 / 184090 pack
[14:55:37] thread 1 is processing the 183784 / 184090 pack
[14:55:37] thread 2 is processing the 183785 / 184090 pack
[14:55:37] thread 1 is processing the 183786 / 184090 pack
[14:55:37] thread 2 is processing the 183787 / 184090 pack
[14:55:37] thread 1 is processing the 183788 / 184090 pack
[14:55:37] thread 2 is processing the 183789 / 184090 pack
[14:55:37] thread 1 is processing the 183790 / 184090 pack
[14:55:37] thread 2 is processing the 183791 / 184090 pack
[14:55:37] thread 1 is processing the 183792 / 184090 pack
[14:55:37] thread 2 is processing the 183793 / 184090 pack
[14:55:37] thread 1 is processing the 183794 / 184090 pack
[14:55:37] thread 2 is processing the 183795 / 184090 pack
[14:55:37] thread 1 is processing the 183796 / 184090 pack
[14:55:37] thread 2 is processing the 183797 / 184090 pack
[14:55:37] thread 1 is processing the 183798 / 184090 pack
[14:55:37] thread 2 is processing the 183799 / 184090 pack
[14:55:37] thread 1 is processing the 183800 / 184090 pack
[14:55:37] thread 2 is processing the 183801 / 184090 pack
[14:55:37] thread 1 is processing the 183802 / 184090 pack
[14:55:37] thread 2 is processing the 183803 / 184090 pack
[14:55:37] thread 1 is processing the 183804 / 184090 pack
[14:55:37] thread 2 is processing the 183805 / 184090 pack
[14:55:37] thread 1 is processing the 183806 / 184090 pack
[14:55:37] thread 2 is processing the 183807 / 184090 pack
[14:55:37] thread 1 is processing the 183808 / 184090 pack
[14:55:37] thread 2 is processing the 183809 / 184090 pack
[14:55:37] thread 1 is processing the 183810 / 184090 pack
[14:55:37] thread 2 is processing the 183811 / 184090 pack
[14:55:37] thread 1 is processing the 183812 / 184090 pack
[14:55:37] thread 2 is processing the 183813 / 184090 pack
[14:55:37] thread 1 is processing the 183814 / 184090 pack
[14:55:37] thread 2 is processing the 183815 / 184090 pack
[14:55:37] thread 1 is processing the 183816 / 184090 pack
[14:55:37] thread 2 is processing the 183817 / 184090 pack
[14:55:37] thread 1 is processing the 183818 / 184090 pack
[14:55:37] thread 2 is processing the 183819 / 184090 pack
[14:55:37] thread 1 is processing the 183820 / 184090 pack
[14:55:37] thread 2 is processing the 183821 / 184090 pack
[14:55:37] thread 1 is processing the 183822 / 184090 pack
[14:55:37] thread 2 is processing the 183823 / 184090 pack
[14:55:37] thread 1 is processing the 183824 / 184090 pack
[14:55:37] thread 2 is processing the 183825 / 184090 pack
[14:55:37] thread 1 is processing the 183826 / 184090 pack
[14:55:37] thread 2 is processing the 183827 / 184090 pack
[14:55:37] thread 1 is processing the 183828 / 184090 pack
[14:55:37] thread 2 is processing the 183829 / 184090 pack
[14:55:37] thread 1 is processing the 183830 / 184090 pack
[14:55:37] thread 2 is processing the 183831 / 184090 pack
[14:55:37] thread 1 is processing the 183832 / 184090 pack
[14:55:37] thread 2 is processing the 183833 / 184090 pack
[14:55:37] thread 1 is processing the 183834 / 184090 pack
[14:55:37] thread 2 is processing the 183835 / 184090 pack
[14:55:37] thread 1 is processing the 183836 / 184090 pack
[14:55:37] thread 2 is processing the 183837 / 184090 pack
[14:55:37] thread 1 is processing the 183838 / 184090 pack
[14:55:37] thread 2 is processing the 183839 / 184090 pack
[14:55:37] thread 1 is processing the 183840 / 184090 pack
[14:55:37] thread 2 is processing the 183841 / 184090 pack
[14:55:37] thread 1 is processing the 183842 / 184090 pack
[14:55:37] thread 2 is processing the 183843 / 184090 pack
[14:55:37] thread 1 is processing the 183844 / 184090 pack
[14:55:37] thread 2 is processing the 183845 / 184090 pack
[14:55:37] thread 1 is processing the 183846 / 184090 pack
[14:55:37] thread 2 is processing the 183847 / 184090 pack
[14:55:37] thread 1 is processing the 183848 / 184090 pack
[14:55:37] thread 2 is processing the 183849 / 184090 pack
[14:55:37] thread 1 is processing the 183850 / 184090 pack
[14:55:37] thread 2 is processing the 183851 / 184090 pack
[14:55:37] thread 1 is processing the 183852 / 184090 pack
[14:55:37] thread 2 is processing the 183853 / 184090 pack
[14:55:37] thread 1 is processing the 183854 / 184090 pack
[14:55:37] thread 2 is processing the 183855 / 184090 pack
[14:55:37] thread 1 is processing the 183856 / 184090 pack
[14:55:37] thread 2 is processing the 183857 / 184090 pack
[14:55:37] thread 1 is processing the 183858 / 184090 pack
[14:55:37] thread 2 is processing the 183859 / 184090 pack
[14:55:37] thread 1 is processing the 183860 / 184090 pack
[14:55:37] thread 2 is processing the 183861 / 184090 pack
[14:55:37] thread 1 is processing the 183862 / 184090 pack
[14:55:37] thread 2 is processing the 183863 / 184090 pack
[14:55:37] thread 1 is processing the 183864 / 184090 pack
[14:55:37] thread 2 is processing the 183865 / 184090 pack
[14:55:37] thread 1 is processing the 183866 / 184090 pack
[14:55:38] thread 2 is processing the 183867 / 184090 pack
[14:55:38] thread 1 is processing the 183868 / 184090 pack
[14:55:38] thread 2 is processing the 183869 / 184090 pack
[14:55:38] thread 1 is processing the 183870 / 184090 pack
[14:55:38] thread 2 is processing the 183871 / 184090 pack
[14:55:38] thread 1 is processing the 183872 / 184090 pack
[14:55:38] thread 2 is processing the 183873 / 184090 pack
[14:55:38] thread 1 is processing the 183874 / 184090 pack
[14:55:38] thread 2 is processing the 183875 / 184090 pack
[14:55:38] thread 1 is processing the 183876 / 184090 pack
[14:55:38] thread 2 is processing the 183877 / 184090 pack
[14:55:38] thread 1 is processing the 183878 / 184090 pack
[14:55:38] thread 2 is processing the 183879 / 184090 pack
[14:55:38] thread 1 is processing the 183880 / 184090 pack
[14:55:38] thread 2 is processing the 183881 / 184090 pack
[14:55:38] thread 1 is processing the 183882 / 184090 pack
[14:55:38] thread 2 is processing the 183883 / 184090 pack
[14:55:38] thread 1 is processing the 183884 / 184090 pack
[14:55:38] thread 2 is processing the 183885 / 184090 pack
[14:55:38] thread 1 is processing the 183886 / 184090 pack
[14:55:38] thread 2 is processing the 183887 / 184090 pack
[14:55:38] thread 1 is processing the 183888 / 184090 pack
[14:55:38] thread 2 is processing the 183889 / 184090 pack
[14:55:38] thread 1 is processing the 183890 / 184090 pack
[14:55:38] thread 2 is processing the 183891 / 184090 pack
[14:55:38] thread 1 is processing the 183892 / 184090 pack
[14:55:38] thread 2 is processing the 183893 / 184090 pack
[14:55:38] thread 1 is processing the 183894 / 184090 pack
[14:55:38] thread 2 is processing the 183895 / 184090 pack
[14:55:38] thread 1 is processing the 183896 / 184090 pack
[14:55:38] thread 2 is processing the 183897 / 184090 pack
[14:55:38] thread 1 is processing the 183898 / 184090 pack
[14:55:38] thread 2 is processing the 183899 / 184090 pack
[14:55:38] thread 1 is processing the 183900 / 184090 pack
[14:55:38] thread 2 is processing the 183901 / 184090 pack
[14:55:38] thread 1 is processing the 183902 / 184090 pack
[14:55:38] thread 2 is processing the 183903 / 184090 pack
[14:55:38] thread 1 is processing the 183904 / 184090 pack
[14:55:38] thread 2 is processing the 183905 / 184090 pack
[14:55:38] thread 1 is processing the 183906 / 184090 pack
[14:55:38] thread 2 is processing the 183907 / 184090 pack
[14:55:38] thread 1 is processing the 183908 / 184090 pack
[14:55:38] thread 2 is processing the 183909 / 184090 pack
[14:55:38] thread 1 is processing the 183910 / 184090 pack
[14:55:38] thread 2 is processing the 183911 / 184090 pack
[14:55:38] thread 1 is processing the 183912 / 184090 pack
[14:55:38] thread 2 is processing the 183913 / 184090 pack
[14:55:38] thread 1 is processing the 183914 / 184090 pack
[14:55:38] thread 2 is processing the 183915 / 184090 pack
[14:55:38] thread 1 is processing the 183916 / 184090 pack
[14:55:38] thread 2 is processing the 183917 / 184090 pack
[14:55:38] thread 1 is processing the 183918 / 184090 pack
[14:55:38] thread 2 is processing the 183919 / 184090 pack
[14:55:38] thread 1 is processing the 183920 / 184090 pack
[14:55:38] thread 2 is processing the 183921 / 184090 pack
[14:55:38] thread 1 is processing the 183922 / 184090 pack
[14:55:38] thread 2 is processing the 183923 / 184090 pack
[14:55:38] thread 1 is processing the 183924 / 184090 pack
[14:55:38] thread 2 is processing the 183925 / 184090 pack
[14:55:38] thread 1 is processing the 183926 / 184090 pack
[14:55:38] thread 2 is processing the 183927 / 184090 pack
[14:55:38] thread 1 is processing the 183928 / 184090 pack
[14:55:38] thread 2 is processing the 183929 / 184090 pack
[14:55:38] thread 1 is processing the 183930 / 184090 pack
[14:55:38] thread 2 is processing the 183931 / 184090 pack
[14:55:38] thread 1 is processing the 183932 / 184090 pack
[14:55:38] thread 2 is processing the 183933 / 184090 pack
[14:55:38] thread 1 is processing the 183934 / 184090 pack
[14:55:38] thread 2 is processing the 183935 / 184090 pack
[14:55:38] thread 1 is processing the 183936 / 184090 pack
[14:55:38] thread 2 is processing the 183937 / 184090 pack
[14:55:38] thread 1 is processing the 183938 / 184090 pack
[14:55:38] thread 2 is processing the 183939 / 184090 pack
[14:55:38] thread 1 is processing the 183940 / 184090 pack
[14:55:38] thread 2 is processing the 183941 / 184090 pack
[14:55:38] thread 1 is processing the 183942 / 184090 pack
[14:55:38] thread 2 is processing the 183943 / 184090 pack
[14:55:38] thread 1 is processing the 183944 / 184090 pack
[14:55:38] thread 2 is processing the 183945 / 184090 pack
[14:55:38] thread 1 is processing the 183946 / 184090 pack
[14:55:38] thread 2 is processing the 183947 / 184090 pack
[14:55:38] thread 1 is processing the 183948 / 184090 pack
[14:55:38] thread 2 is processing the 183949 / 184090 pack
[14:55:38] thread 1 is processing the 183950 / 184090 pack
[14:55:38] thread 2 is processing the 183951 / 184090 pack
[14:55:38] thread 1 is processing the 183952 / 184090 pack
[14:55:38] thread 2 is processing the 183953 / 184090 pack
[14:55:38] thread 1 is processing the 183954 / 184090 pack
[14:55:38] thread 2 is processing the 183955 / 184090 pack
[14:55:38] thread 1 is processing the 183956 / 184090 pack
[14:55:38] thread 2 is processing the 183957 / 184090 pack
[14:55:38] thread 1 is processing the 183958 / 184090 pack
[14:55:38] thread 2 is processing the 183959 / 184090 pack
[14:55:38] thread 1 is processing the 183960 / 184090 pack
[14:55:39] thread 2 is processing the 183961 / 184090 pack
[14:55:39] thread 1 is processing the 183962 / 184090 pack
[14:55:39] thread 2 is processing the 183963 / 184090 pack
[14:55:39] thread 1 is processing the 183964 / 184090 pack
[14:55:39] thread 2 is processing the 183965 / 184090 pack
[14:55:39] thread 1 is processing the 183966 / 184090 pack
[14:55:39] thread 2 is processing the 183967 / 184090 pack
[14:55:39] thread 1 is processing the 183968 / 184090 pack
[14:55:39] thread 2 is processing the 183969 / 184090 pack
[14:55:39] thread 1 is processing the 183970 / 184090 pack
[14:55:39] thread 2 is processing the 183971 / 184090 pack
[14:55:39] thread 1 is processing the 183972 / 184090 pack
[14:55:39] thread 2 is processing the 183973 / 184090 pack
[14:55:39] thread 1 is processing the 183974 / 184090 pack
[14:55:39] thread 2 is processing the 183975 / 184090 pack
[14:55:39] thread 1 is processing the 183976 / 184090 pack
[14:55:39] thread 2 is processing the 183977 / 184090 pack
[14:55:39] thread 1 is processing the 183978 / 184090 pack
[14:55:39] thread 2 is processing the 183979 / 184090 pack
[14:55:39] thread 1 is processing the 183980 / 184090 pack
[14:55:39] thread 2 is processing the 183981 / 184090 pack
[14:55:39] thread 1 is processing the 183982 / 184090 pack
[14:55:39] thread 2 is processing the 183983 / 184090 pack
[14:55:39] thread 1 is processing the 183984 / 184090 pack
[14:55:39] thread 2 is processing the 183985 / 184090 pack
[14:55:39] thread 1 is processing the 183986 / 184090 pack
[14:55:39] thread 2 is processing the 183987 / 184090 pack
[14:55:39] thread 1 is processing the 183988 / 184090 pack
[14:55:39] thread 2 is processing the 183989 / 184090 pack
[14:55:39] thread 1 is processing the 183990 / 184090 pack
[14:55:39] thread 2 is processing the 183991 / 184090 pack
[14:55:39] thread 1 is processing the 183992 / 184090 pack
[14:55:39] thread 2 is processing the 183993 / 184090 pack
[14:55:39] thread 1 is processing the 183994 / 184090 pack
[14:55:39] thread 2 is processing the 183995 / 184090 pack
[14:55:39] thread 1 is processing the 183996 / 184090 pack
[14:55:39] thread 2 is processing the 183997 / 184090 pack
[14:55:39] thread 1 is processing the 183998 / 184090 pack
[14:55:39] thread 2 is processing the 183999 / 184090 pack
[14:55:39] thread 1 is processing the 184000 / 184090 pack
[14:55:39] thread 2 is processing the 184001 / 184090 pack
[14:55:39] thread 1 is processing the 184002 / 184090 pack
[14:55:39] thread 2 is processing the 184003 / 184090 pack
[14:55:39] thread 1 is processing the 184004 / 184090 pack
[14:55:39] thread 2 is processing the 184005 / 184090 pack
[14:55:39] thread 1 is processing the 184006 / 184090 pack
[14:55:39] thread 2 is processing the 184007 / 184090 pack
[14:55:39] thread 1 is processing the 184008 / 184090 pack
[14:55:39] thread 2 is processing the 184009 / 184090 pack
[14:55:39] thread 1 is processing the 184010 / 184090 pack
[14:55:39] thread 2 is processing the 184011 / 184090 pack
[14:55:39] thread 1 is processing the 184012 / 184090 pack
[14:55:39] thread 2 is processing the 184013 / 184090 pack
[14:55:39] thread 1 is processing the 184014 / 184090 pack
[14:55:39] thread 2 is processing the 184015 / 184090 pack
[14:55:39] thread 1 is processing the 184016 / 184090 pack
[14:55:39] thread 2 is processing the 184017 / 184090 pack
[14:55:39] thread 1 is processing the 184018 / 184090 pack
[14:55:39] thread 2 is processing the 184019 / 184090 pack
[14:55:39] thread 1 is processing the 184020 / 184090 pack
[14:55:39] thread 2 is processing the 184021 / 184090 pack
[14:55:39] thread 1 is processing the 184022 / 184090 pack
[14:55:39] thread 2 is processing the 184023 / 184090 pack
[14:55:39] thread 1 is processing the 184024 / 184090 pack
[14:55:39] thread 2 is processing the 184025 / 184090 pack
[14:55:39] thread 1 is processing the 184026 / 184090 pack
[14:55:39] thread 2 is processing the 184027 / 184090 pack
[14:55:39] thread 1 is processing the 184028 / 184090 pack
[14:55:39] thread 2 is processing the 184029 / 184090 pack
[14:55:39] thread 1 is processing the 184030 / 184090 pack
[14:55:39] thread 2 is processing the 184031 / 184090 pack
[14:55:39] thread 1 is processing the 184032 / 184090 pack
[14:55:39] thread 2 is processing the 184033 / 184090 pack
[14:55:39] thread 1 is processing the 184034 / 184090 pack
[14:55:39] thread 1 is processing the 184035 / 184090 pack
[14:55:39] thread 2 is processing the 184036 / 184090 pack
[14:55:39] thread 1 is processing the 184037 / 184090 pack
[14:55:39] thread 2 is processing the 184038 / 184090 pack
[14:55:39] thread 1 is processing the 184039 / 184090 pack
[14:55:39] thread 2 is processing the 184040 / 184090 pack
[14:55:39] thread 1 is processing the 184041 / 184090 pack
[14:55:39] thread 2 is processing the 184042 / 184090 pack
[14:55:39] thread 2 is processing the 184043 / 184090 pack
[14:55:39] thread 1 is processing the 184044 / 184090 pack
[14:55:39] thread 2 is processing the 184045 / 184090 pack
[14:55:39] thread 1 is processing the 184046 / 184090 pack
[14:55:39] thread 1 is processing the 184047 / 184090 pack
[14:55:39] thread 2 is processing the 184048 / 184090 pack
[14:55:39] thread 1 is processing the 184049 / 184090 pack
[14:55:39] thread 2 is processing the 184050 / 184090 pack
[14:55:39] thread 1 is processing the 184051 / 184090 pack
[14:55:39] thread 2 is processing the 184052 / 184090 pack
[14:55:39] thread 1 is processing the 184053 / 184090 pack
[14:55:39] thread 2 is processing the 184054 / 184090 pack
[14:55:40] thread 1 is processing the 184055 / 184090 pack
[14:55:40] thread 2 is processing the 184056 / 184090 pack
[14:55:40] thread 1 is processing the 184057 / 184090 pack
[14:55:40] thread 2 is processing the 184058 / 184090 pack
[14:55:40] thread 1 is processing the 184059 / 184090 pack
[14:55:40] thread 2 is processing the 184060 / 184090 pack
[14:55:40] thread 1 is processing the 184061 / 184090 pack
[14:55:40] thread 2 is processing the 184062 / 184090 pack
[14:55:40] thread 1 is processing the 184063 / 184090 pack
[14:55:40] thread 2 is processing the 184064 / 184090 pack
[14:55:40] thread 1 is processing the 184065 / 184090 pack
[14:55:40] thread 2 is processing the 184066 / 184090 pack
[14:55:40] thread 1 is processing the 184067 / 184090 pack
[14:55:40] thread 2 is processing the 184068 / 184090 pack
[14:55:40] thread 1 is processing the 184069 / 184090 pack
[14:55:40] thread 2 is processing the 184070 / 184090 pack
[14:55:40] thread 1 is processing the 184071 / 184090 pack
[14:55:40] thread 2 is processing the 184072 / 184090 pack
[14:55:40] thread 1 is processing the 184073 / 184090 pack
[14:55:40] thread 2 is processing the 184074 / 184090 pack
[14:55:40] thread 1 is processing the 184075 / 184090 pack
[14:55:40] thread 2 is processing the 184076 / 184090 pack
[14:55:40] thread 1 is processing the 184077 / 184090 pack
[14:55:40] thread 2 is processing the 184078 / 184090 pack
[14:55:40] thread 1 is processing the 184079 / 184090 pack
[14:55:40] thread 2 is processing the 184080 / 184090 pack
[14:55:40] thread 1 is processing the 184081 / 184090 pack
[14:55:40] thread 2 is processing the 184082 / 184090 pack
[14:55:40] thread 1 is processing the 184083 / 184090 pack
[14:55:40] thread 2 is processing the 184084 / 184090 pack
[14:55:40] thread 1 is processing the 184085 / 184090 pack
[14:55:40] thread 2 is processing the 184086 / 184090 pack
[14:55:40] thread 1 is processing the 184087 / 184090 pack
[14:55:40] thread 2 is processing the 184088 / 184090 pack
[14:55:40] thread 1 is processing the 184089 / 184090 pack
[14:55:40] thread 2 data processing completed
[14:55:40] thread 2 finished
[14:55:40] thread 1 data processing completed
[14:55:40] thread 1 finished
[14:55:40] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/FourthInstarLarvae_QC_R1.fastq.gz writer finished
[14:55:40] /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/FQ_QC/FourthInstarLarvae_QC_R2.fastq.gz writer finished
[14:55:40] start to generate reports

Read1 before filtering:
total reads: 184089709
total bases: 27797546059
Q20 bases: 27297236243(98.2002%)
Q30 bases: 26424291421(95.0598%)

Read2 before filtering:
total reads: 184089709
total bases: 27797546059
Q20 bases: 27027934452(97.2314%)
Q30 bases: 25851947614(93.0008%)

Read1 after filtering:
total reads: 181723832
total bases: 27178905460
Q20 bases: 26741041808(98.389%)
Q30 bases: 25913090743(95.3427%)

Read2 after filtering:
total reads: 181723832
total bases: 27176915562
Q20 bases: 26595094925(97.8591%)
Q30 bases: 25491533278(93.7985%)

Filtering result:
reads passed filter: 363447664
reads failed due to low quality: 4639092
reads failed due to too many N: 16652
reads failed due to too short: 76010
reads with adapter trimmed: 28665853
bases trimmed due to adapters: 523523602

Duplication rate: 10.749%

Insert size peak (evaluated by paired-end reads): 196

```
</details>
## Alignment of FastQ to reference genome (Hisat2)

This section descibes how fastQ is aligned to the reference genome using Rsubread.

<details>
```{r , include=TRUE}
FourthInstarLarvae_splitter_1 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("FourthInstarLarvae_splitFQ1_part1.fq.gz", "FourthInstarLarvae_splitFQ1_part2.fq.gz", "FourthInstarLarvae_splitFQ1_part3.fq.gz", "FourthInstarLarvae_splitFQ1_part4.fq.gz", "FourthInstarLarvae_splitFQ1_part5.fq.gz", "FourthInstarLarvae_splitFQ1_part6.fq.gz", "FourthInstarLarvae_splitFQ1_part7.fq.gz", "FourthInstarLarvae_splitFQ1_part8.fq.gz", "FourthInstarLarvae_splitFQ1_part9.fq.gz", "FourthInstarLarvae_splitFQ1_part10.fq.gz", "FourthInstarLarvae_splitFQ1_part11.fq.gz", "FourthInstarLarvae_splitFQ1_part12.fq.gz", "FourthInstarLarvae_splitFQ1_part13.fq.gz", "FourthInstarLarvae_splitFQ1_part14.fq.gz", "FourthInstarLarvae_splitFQ1_part15.fq.gz", "FourthInstarLarvae_splitFQ1_part16.fq.gz", "FourthInstarLarvae_splitFQ1_part17.fq.gz", "FourthInstarLarvae_splitFQ1_part18.fq.gz", "FourthInstarLarvae_splitFQ1_part19.fq.gz", "FourthInstarLarvae_splitFQ1_part20.fq.gz", "FourthInstarLarvae_splitFQ1_part21.fq.gz", "FourthInstarLarvae_splitFQ1_part22.fq.gz", "FourthInstarLarvae_splitFQ1_part23.fq.gz", "FourthInstarLarvae_splitFQ1_part24.fq.gz", "FourthInstarLarvae_splitFQ1_part25.fq.gz", "FourthInstarLarvae_splitFQ1_part26.fq.gz", "FourthInstarLarvae_splitFQ1_part27.fq.gz", "FourthInstarLarvae_splitFQ1_part28.fq.gz", "FourthInstarLarvae_splitFQ1_part29.fq.gz", "FourthInstarLarvae_splitFQ1_part30.fq.gz", "FourthInstarLarvae_splitFQ1_part31.fq.gz", "FourthInstarLarvae_splitFQ1_part32.fq.gz", "FourthInstarLarvae_splitFQ1_part33.fq.gz", "FourthInstarLarvae_splitFQ1_part34.fq.gz", "FourthInstarLarvae_splitFQ1_part35.fq.gz", "FourthInstarLarvae_splitFQ1_part36.fq.gz", "FourthInstarLarvae_splitFQ1_part37.fq.gz", "FourthInstarLarvae_splitFQ1_part38.fq.gz", "FourthInstarLarvae_splitFQ1_part39.fq.gz", "FourthInstarLarvae_splitFQ1_part40.fq.gz")
      args <- c("-i", FourthInstarLarvae_Pair1FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitter_2 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("Pupae_splitFQ2_part1.fq.gz", "Pupae_splitFQ2_part2.fq.gz", "Pupae_splitFQ2_part3.fq.gz", "Pupae_splitFQ2_part4.fq.gz", "Pupae_splitFQ2_part5.fq.gz", "Pupae_splitFQ2_part6.fq.gz", "Pupae_splitFQ2_part7.fq.gz", "Pupae_splitFQ2_part8.fq.gz", "Pupae_splitFQ2_part9.fq.gz", "Pupae_splitFQ2_part10.fq.gz", "Pupae_splitFQ2_part11.fq.gz", "Pupae_splitFQ2_part12.fq.gz", "Pupae_splitFQ2_part13.fq.gz", "Pupae_splitFQ2_part14.fq.gz", "Pupae_splitFQ2_part15.fq.gz", "Pupae_splitFQ2_part16.fq.gz", "Pupae_splitFQ2_part17.fq.gz", "Pupae_splitFQ2_part18.fq.gz", "Pupae_splitFQ2_part19.fq.gz", "Pupae_splitFQ2_part20.fq.gz", "Pupae_splitFQ2_part21.fq.gz", "Pupae_splitFQ2_part22.fq.gz", "Pupae_splitFQ2_part23.fq.gz", "Pupae_splitFQ2_part24.fq.gz", "Pupae_splitFQ2_part25.fq.gz", "Pupae_splitFQ2_part26.fq.gz", "Pupae_splitFQ2_part27.fq.gz", "Pupae_splitFQ2_part28.fq.gz", "Pupae_splitFQ2_part29.fq.gz", "Pupae_splitFQ2_part30.fq.gz", "Pupae_splitFQ2_part31.fq.gz", "Pupae_splitFQ2_part32.fq.gz", "Pupae_splitFQ2_part33.fq.gz", "Pupae_splitFQ2_part34.fq.gz", "Pupae_splitFQ2_part35.fq.gz", "Pupae_splitFQ2_part36.fq.gz", "Pupae_splitFQ2_part37.fq.gz", "Pupae_splitFQ2_part38.fq.gz", "Pupae_splitFQ2_part39.fq.gz", "Pupae_splitFQ2_part40.fq.gz")
      args <- c("-i", Pupae_Pair2FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitter_1 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("Eggs_splitFQ1_part1.fq.gz", "Eggs_splitFQ1_part2.fq.gz", "Eggs_splitFQ1_part3.fq.gz", "Eggs_splitFQ1_part4.fq.gz", "Eggs_splitFQ1_part5.fq.gz", "Eggs_splitFQ1_part6.fq.gz", "Eggs_splitFQ1_part7.fq.gz", "Eggs_splitFQ1_part8.fq.gz", "Eggs_splitFQ1_part9.fq.gz", "Eggs_splitFQ1_part10.fq.gz", "Eggs_splitFQ1_part11.fq.gz", "Eggs_splitFQ1_part12.fq.gz", "Eggs_splitFQ1_part13.fq.gz", "Eggs_splitFQ1_part14.fq.gz", "Eggs_splitFQ1_part15.fq.gz", "Eggs_splitFQ1_part16.fq.gz", "Eggs_splitFQ1_part17.fq.gz", "Eggs_splitFQ1_part18.fq.gz", "Eggs_splitFQ1_part19.fq.gz", "Eggs_splitFQ1_part20.fq.gz", "Eggs_splitFQ1_part21.fq.gz", "Eggs_splitFQ1_part22.fq.gz", "Eggs_splitFQ1_part23.fq.gz", "Eggs_splitFQ1_part24.fq.gz", "Eggs_splitFQ1_part25.fq.gz", "Eggs_splitFQ1_part26.fq.gz", "Eggs_splitFQ1_part27.fq.gz", "Eggs_splitFQ1_part28.fq.gz", "Eggs_splitFQ1_part29.fq.gz", "Eggs_splitFQ1_part30.fq.gz", "Eggs_splitFQ1_part31.fq.gz", "Eggs_splitFQ1_part32.fq.gz", "Eggs_splitFQ1_part33.fq.gz", "Eggs_splitFQ1_part34.fq.gz", "Eggs_splitFQ1_part35.fq.gz", "Eggs_splitFQ1_part36.fq.gz", "Eggs_splitFQ1_part37.fq.gz", "Eggs_splitFQ1_part38.fq.gz", "Eggs_splitFQ1_part39.fq.gz", "Eggs_splitFQ1_part40.fq.gz")
      args <- c("-i", Eggs_Pair1FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitter_2 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("Eggs_splitFQ2_part1.fq.gz", "Eggs_splitFQ2_part2.fq.gz", "Eggs_splitFQ2_part3.fq.gz", "Eggs_splitFQ2_part4.fq.gz", "Eggs_splitFQ2_part5.fq.gz", "Eggs_splitFQ2_part6.fq.gz", "Eggs_splitFQ2_part7.fq.gz", "Eggs_splitFQ2_part8.fq.gz", "Eggs_splitFQ2_part9.fq.gz", "Eggs_splitFQ2_part10.fq.gz", "Eggs_splitFQ2_part11.fq.gz", "Eggs_splitFQ2_part12.fq.gz", "Eggs_splitFQ2_part13.fq.gz", "Eggs_splitFQ2_part14.fq.gz", "Eggs_splitFQ2_part15.fq.gz", "Eggs_splitFQ2_part16.fq.gz", "Eggs_splitFQ2_part17.fq.gz", "Eggs_splitFQ2_part18.fq.gz", "Eggs_splitFQ2_part19.fq.gz", "Eggs_splitFQ2_part20.fq.gz", "Eggs_splitFQ2_part21.fq.gz", "Eggs_splitFQ2_part22.fq.gz", "Eggs_splitFQ2_part23.fq.gz", "Eggs_splitFQ2_part24.fq.gz", "Eggs_splitFQ2_part25.fq.gz", "Eggs_splitFQ2_part26.fq.gz", "Eggs_splitFQ2_part27.fq.gz", "Eggs_splitFQ2_part28.fq.gz", "Eggs_splitFQ2_part29.fq.gz", "Eggs_splitFQ2_part30.fq.gz", "Eggs_splitFQ2_part31.fq.gz", "Eggs_splitFQ2_part32.fq.gz", "Eggs_splitFQ2_part33.fq.gz", "Eggs_splitFQ2_part34.fq.gz", "Eggs_splitFQ2_part35.fq.gz", "Eggs_splitFQ2_part36.fq.gz", "Eggs_splitFQ2_part37.fq.gz", "Eggs_splitFQ2_part38.fq.gz", "Eggs_splitFQ2_part39.fq.gz", "Eggs_splitFQ2_part40.fq.gz")
      args <- c("-i", Eggs_Pair2FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitter_1 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("Adults_splitFQ1_part1.fq.gz", "Adults_splitFQ1_part2.fq.gz", "Adults_splitFQ1_part3.fq.gz", "Adults_splitFQ1_part4.fq.gz", "Adults_splitFQ1_part5.fq.gz", "Adults_splitFQ1_part6.fq.gz", "Adults_splitFQ1_part7.fq.gz", "Adults_splitFQ1_part8.fq.gz", "Adults_splitFQ1_part9.fq.gz", "Adults_splitFQ1_part10.fq.gz", "Adults_splitFQ1_part11.fq.gz", "Adults_splitFQ1_part12.fq.gz", "Adults_splitFQ1_part13.fq.gz", "Adults_splitFQ1_part14.fq.gz", "Adults_splitFQ1_part15.fq.gz", "Adults_splitFQ1_part16.fq.gz", "Adults_splitFQ1_part17.fq.gz", "Adults_splitFQ1_part18.fq.gz", "Adults_splitFQ1_part19.fq.gz", "Adults_splitFQ1_part20.fq.gz", "Adults_splitFQ1_part21.fq.gz", "Adults_splitFQ1_part22.fq.gz", "Adults_splitFQ1_part23.fq.gz", "Adults_splitFQ1_part24.fq.gz", "Adults_splitFQ1_part25.fq.gz", "Adults_splitFQ1_part26.fq.gz", "Adults_splitFQ1_part27.fq.gz", "Adults_splitFQ1_part28.fq.gz", "Adults_splitFQ1_part29.fq.gz", "Adults_splitFQ1_part30.fq.gz", "Adults_splitFQ1_part31.fq.gz", "Adults_splitFQ1_part32.fq.gz", "Adults_splitFQ1_part33.fq.gz", "Adults_splitFQ1_part34.fq.gz", "Adults_splitFQ1_part35.fq.gz", "Adults_splitFQ1_part36.fq.gz", "Adults_splitFQ1_part37.fq.gz", "Adults_splitFQ1_part38.fq.gz", "Adults_splitFQ1_part39.fq.gz", "Adults_splitFQ1_part40.fq.gz")
      args <- c("-i", Adults_Pair1FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitter_2 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("Adults_splitFQ2_part1.fq.gz", "Adults_splitFQ2_part2.fq.gz", "Adults_splitFQ2_part3.fq.gz", "Adults_splitFQ2_part4.fq.gz", "Adults_splitFQ2_part5.fq.gz", "Adults_splitFQ2_part6.fq.gz", "Adults_splitFQ2_part7.fq.gz", "Adults_splitFQ2_part8.fq.gz", "Adults_splitFQ2_part9.fq.gz", "Adults_splitFQ2_part10.fq.gz", "Adults_splitFQ2_part11.fq.gz", "Adults_splitFQ2_part12.fq.gz", "Adults_splitFQ2_part13.fq.gz", "Adults_splitFQ2_part14.fq.gz", "Adults_splitFQ2_part15.fq.gz", "Adults_splitFQ2_part16.fq.gz", "Adults_splitFQ2_part17.fq.gz", "Adults_splitFQ2_part18.fq.gz", "Adults_splitFQ2_part19.fq.gz", "Adults_splitFQ2_part20.fq.gz", "Adults_splitFQ2_part21.fq.gz", "Adults_splitFQ2_part22.fq.gz", "Adults_splitFQ2_part23.fq.gz", "Adults_splitFQ2_part24.fq.gz", "Adults_splitFQ2_part25.fq.gz", "Adults_splitFQ2_part26.fq.gz", "Adults_splitFQ2_part27.fq.gz", "Adults_splitFQ2_part28.fq.gz", "Adults_splitFQ2_part29.fq.gz", "Adults_splitFQ2_part30.fq.gz", "Adults_splitFQ2_part31.fq.gz", "Adults_splitFQ2_part32.fq.gz", "Adults_splitFQ2_part33.fq.gz", "Adults_splitFQ2_part34.fq.gz", "Adults_splitFQ2_part35.fq.gz", "Adults_splitFQ2_part36.fq.gz", "Adults_splitFQ2_part37.fq.gz", "Adults_splitFQ2_part38.fq.gz", "Adults_splitFQ2_part39.fq.gz", "Adults_splitFQ2_part40.fq.gz")
      args <- c("-i", Adults_Pair2FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitter_1 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("Prepupae_splitFQ1_part1.fq.gz", "Prepupae_splitFQ1_part2.fq.gz", "Prepupae_splitFQ1_part3.fq.gz", "Prepupae_splitFQ1_part4.fq.gz", "Prepupae_splitFQ1_part5.fq.gz", "Prepupae_splitFQ1_part6.fq.gz", "Prepupae_splitFQ1_part7.fq.gz", "Prepupae_splitFQ1_part8.fq.gz", "Prepupae_splitFQ1_part9.fq.gz", "Prepupae_splitFQ1_part10.fq.gz", "Prepupae_splitFQ1_part11.fq.gz", "Prepupae_splitFQ1_part12.fq.gz", "Prepupae_splitFQ1_part13.fq.gz", "Prepupae_splitFQ1_part14.fq.gz", "Prepupae_splitFQ1_part15.fq.gz", "Prepupae_splitFQ1_part16.fq.gz", "Prepupae_splitFQ1_part17.fq.gz", "Prepupae_splitFQ1_part18.fq.gz", "Prepupae_splitFQ1_part19.fq.gz", "Prepupae_splitFQ1_part20.fq.gz", "Prepupae_splitFQ1_part21.fq.gz", "Prepupae_splitFQ1_part22.fq.gz", "Prepupae_splitFQ1_part23.fq.gz", "Prepupae_splitFQ1_part24.fq.gz", "Prepupae_splitFQ1_part25.fq.gz", "Prepupae_splitFQ1_part26.fq.gz", "Prepupae_splitFQ1_part27.fq.gz", "Prepupae_splitFQ1_part28.fq.gz", "Prepupae_splitFQ1_part29.fq.gz", "Prepupae_splitFQ1_part30.fq.gz", "Prepupae_splitFQ1_part31.fq.gz", "Prepupae_splitFQ1_part32.fq.gz", "Prepupae_splitFQ1_part33.fq.gz", "Prepupae_splitFQ1_part34.fq.gz", "Prepupae_splitFQ1_part35.fq.gz", "Prepupae_splitFQ1_part36.fq.gz", "Prepupae_splitFQ1_part37.fq.gz", "Prepupae_splitFQ1_part38.fq.gz", "Prepupae_splitFQ1_part39.fq.gz", "Prepupae_splitFQ1_part40.fq.gz")
      args <- c("-i", Prepupae_Pair1FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitter_1 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("YoungLarvae_splitFQ1_part1.fq.gz", "YoungLarvae_splitFQ1_part2.fq.gz", "YoungLarvae_splitFQ1_part3.fq.gz", "YoungLarvae_splitFQ1_part4.fq.gz", "YoungLarvae_splitFQ1_part5.fq.gz", "YoungLarvae_splitFQ1_part6.fq.gz", "YoungLarvae_splitFQ1_part7.fq.gz", "YoungLarvae_splitFQ1_part8.fq.gz", "YoungLarvae_splitFQ1_part9.fq.gz", "YoungLarvae_splitFQ1_part10.fq.gz", "YoungLarvae_splitFQ1_part11.fq.gz", "YoungLarvae_splitFQ1_part12.fq.gz", "YoungLarvae_splitFQ1_part13.fq.gz", "YoungLarvae_splitFQ1_part14.fq.gz", "YoungLarvae_splitFQ1_part15.fq.gz", "YoungLarvae_splitFQ1_part16.fq.gz", "YoungLarvae_splitFQ1_part17.fq.gz", "YoungLarvae_splitFQ1_part18.fq.gz", "YoungLarvae_splitFQ1_part19.fq.gz", "YoungLarvae_splitFQ1_part20.fq.gz", "YoungLarvae_splitFQ1_part21.fq.gz", "YoungLarvae_splitFQ1_part22.fq.gz", "YoungLarvae_splitFQ1_part23.fq.gz", "YoungLarvae_splitFQ1_part24.fq.gz", "YoungLarvae_splitFQ1_part25.fq.gz", "YoungLarvae_splitFQ1_part26.fq.gz", "YoungLarvae_splitFQ1_part27.fq.gz", "YoungLarvae_splitFQ1_part28.fq.gz", "YoungLarvae_splitFQ1_part29.fq.gz", "YoungLarvae_splitFQ1_part30.fq.gz", "YoungLarvae_splitFQ1_part31.fq.gz", "YoungLarvae_splitFQ1_part32.fq.gz", "YoungLarvae_splitFQ1_part33.fq.gz", "YoungLarvae_splitFQ1_part34.fq.gz", "YoungLarvae_splitFQ1_part35.fq.gz", "YoungLarvae_splitFQ1_part36.fq.gz", "YoungLarvae_splitFQ1_part37.fq.gz", "YoungLarvae_splitFQ1_part38.fq.gz", "YoungLarvae_splitFQ1_part39.fq.gz", "YoungLarvae_splitFQ1_part40.fq.gz")
      args <- c("-i", YoungLarvae_Pair1FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitter_2 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("Prepupae_splitFQ2_part1.fq.gz", "Prepupae_splitFQ2_part2.fq.gz", "Prepupae_splitFQ2_part3.fq.gz", "Prepupae_splitFQ2_part4.fq.gz", "Prepupae_splitFQ2_part5.fq.gz", "Prepupae_splitFQ2_part6.fq.gz", "Prepupae_splitFQ2_part7.fq.gz", "Prepupae_splitFQ2_part8.fq.gz", "Prepupae_splitFQ2_part9.fq.gz", "Prepupae_splitFQ2_part10.fq.gz", "Prepupae_splitFQ2_part11.fq.gz", "Prepupae_splitFQ2_part12.fq.gz", "Prepupae_splitFQ2_part13.fq.gz", "Prepupae_splitFQ2_part14.fq.gz", "Prepupae_splitFQ2_part15.fq.gz", "Prepupae_splitFQ2_part16.fq.gz", "Prepupae_splitFQ2_part17.fq.gz", "Prepupae_splitFQ2_part18.fq.gz", "Prepupae_splitFQ2_part19.fq.gz", "Prepupae_splitFQ2_part20.fq.gz", "Prepupae_splitFQ2_part21.fq.gz", "Prepupae_splitFQ2_part22.fq.gz", "Prepupae_splitFQ2_part23.fq.gz", "Prepupae_splitFQ2_part24.fq.gz", "Prepupae_splitFQ2_part25.fq.gz", "Prepupae_splitFQ2_part26.fq.gz", "Prepupae_splitFQ2_part27.fq.gz", "Prepupae_splitFQ2_part28.fq.gz", "Prepupae_splitFQ2_part29.fq.gz", "Prepupae_splitFQ2_part30.fq.gz", "Prepupae_splitFQ2_part31.fq.gz", "Prepupae_splitFQ2_part32.fq.gz", "Prepupae_splitFQ2_part33.fq.gz", "Prepupae_splitFQ2_part34.fq.gz", "Prepupae_splitFQ2_part35.fq.gz", "Prepupae_splitFQ2_part36.fq.gz", "Prepupae_splitFQ2_part37.fq.gz", "Prepupae_splitFQ2_part38.fq.gz", "Prepupae_splitFQ2_part39.fq.gz", "Prepupae_splitFQ2_part40.fq.gz")
      args <- c("-i", Prepupae_Pair2FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitter_1 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("Pupae_splitFQ1_part1.fq.gz", "Pupae_splitFQ1_part2.fq.gz", "Pupae_splitFQ1_part3.fq.gz", "Pupae_splitFQ1_part4.fq.gz", "Pupae_splitFQ1_part5.fq.gz", "Pupae_splitFQ1_part6.fq.gz", "Pupae_splitFQ1_part7.fq.gz", "Pupae_splitFQ1_part8.fq.gz", "Pupae_splitFQ1_part9.fq.gz", "Pupae_splitFQ1_part10.fq.gz", "Pupae_splitFQ1_part11.fq.gz", "Pupae_splitFQ1_part12.fq.gz", "Pupae_splitFQ1_part13.fq.gz", "Pupae_splitFQ1_part14.fq.gz", "Pupae_splitFQ1_part15.fq.gz", "Pupae_splitFQ1_part16.fq.gz", "Pupae_splitFQ1_part17.fq.gz", "Pupae_splitFQ1_part18.fq.gz", "Pupae_splitFQ1_part19.fq.gz", "Pupae_splitFQ1_part20.fq.gz", "Pupae_splitFQ1_part21.fq.gz", "Pupae_splitFQ1_part22.fq.gz", "Pupae_splitFQ1_part23.fq.gz", "Pupae_splitFQ1_part24.fq.gz", "Pupae_splitFQ1_part25.fq.gz", "Pupae_splitFQ1_part26.fq.gz", "Pupae_splitFQ1_part27.fq.gz", "Pupae_splitFQ1_part28.fq.gz", "Pupae_splitFQ1_part29.fq.gz", "Pupae_splitFQ1_part30.fq.gz", "Pupae_splitFQ1_part31.fq.gz", "Pupae_splitFQ1_part32.fq.gz", "Pupae_splitFQ1_part33.fq.gz", "Pupae_splitFQ1_part34.fq.gz", "Pupae_splitFQ1_part35.fq.gz", "Pupae_splitFQ1_part36.fq.gz", "Pupae_splitFQ1_part37.fq.gz", "Pupae_splitFQ1_part38.fq.gz", "Pupae_splitFQ1_part39.fq.gz", "Pupae_splitFQ1_part40.fq.gz")
      args <- c("-i", Pupae_Pair1FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitter_2 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("YoungLarvae_splitFQ2_part1.fq.gz", "YoungLarvae_splitFQ2_part2.fq.gz", "YoungLarvae_splitFQ2_part3.fq.gz", "YoungLarvae_splitFQ2_part4.fq.gz", "YoungLarvae_splitFQ2_part5.fq.gz", "YoungLarvae_splitFQ2_part6.fq.gz", "YoungLarvae_splitFQ2_part7.fq.gz", "YoungLarvae_splitFQ2_part8.fq.gz", "YoungLarvae_splitFQ2_part9.fq.gz", "YoungLarvae_splitFQ2_part10.fq.gz", "YoungLarvae_splitFQ2_part11.fq.gz", "YoungLarvae_splitFQ2_part12.fq.gz", "YoungLarvae_splitFQ2_part13.fq.gz", "YoungLarvae_splitFQ2_part14.fq.gz", "YoungLarvae_splitFQ2_part15.fq.gz", "YoungLarvae_splitFQ2_part16.fq.gz", "YoungLarvae_splitFQ2_part17.fq.gz", "YoungLarvae_splitFQ2_part18.fq.gz", "YoungLarvae_splitFQ2_part19.fq.gz", "YoungLarvae_splitFQ2_part20.fq.gz", "YoungLarvae_splitFQ2_part21.fq.gz", "YoungLarvae_splitFQ2_part22.fq.gz", "YoungLarvae_splitFQ2_part23.fq.gz", "YoungLarvae_splitFQ2_part24.fq.gz", "YoungLarvae_splitFQ2_part25.fq.gz", "YoungLarvae_splitFQ2_part26.fq.gz", "YoungLarvae_splitFQ2_part27.fq.gz", "YoungLarvae_splitFQ2_part28.fq.gz", "YoungLarvae_splitFQ2_part29.fq.gz", "YoungLarvae_splitFQ2_part30.fq.gz", "YoungLarvae_splitFQ2_part31.fq.gz", "YoungLarvae_splitFQ2_part32.fq.gz", "YoungLarvae_splitFQ2_part33.fq.gz", "YoungLarvae_splitFQ2_part34.fq.gz", "YoungLarvae_splitFQ2_part35.fq.gz", "YoungLarvae_splitFQ2_part36.fq.gz", "YoungLarvae_splitFQ2_part37.fq.gz", "YoungLarvae_splitFQ2_part38.fq.gz", "YoungLarvae_splitFQ2_part39.fq.gz", "YoungLarvae_splitFQ2_part40.fq.gz")
      args <- c("-i", YoungLarvae_Pair2FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitter_2 <- {
  Herper::with_CondaEnv(general_env,
    {
      command <- "fastqsplitter"
      files <- c("FourthInstarLarvae_splitFQ2_part1.fq.gz", "FourthInstarLarvae_splitFQ2_part2.fq.gz", "FourthInstarLarvae_splitFQ2_part3.fq.gz", "FourthInstarLarvae_splitFQ2_part4.fq.gz", "FourthInstarLarvae_splitFQ2_part5.fq.gz", "FourthInstarLarvae_splitFQ2_part6.fq.gz", "FourthInstarLarvae_splitFQ2_part7.fq.gz", "FourthInstarLarvae_splitFQ2_part8.fq.gz", "FourthInstarLarvae_splitFQ2_part9.fq.gz", "FourthInstarLarvae_splitFQ2_part10.fq.gz", "FourthInstarLarvae_splitFQ2_part11.fq.gz", "FourthInstarLarvae_splitFQ2_part12.fq.gz", "FourthInstarLarvae_splitFQ2_part13.fq.gz", "FourthInstarLarvae_splitFQ2_part14.fq.gz", "FourthInstarLarvae_splitFQ2_part15.fq.gz", "FourthInstarLarvae_splitFQ2_part16.fq.gz", "FourthInstarLarvae_splitFQ2_part17.fq.gz", "FourthInstarLarvae_splitFQ2_part18.fq.gz", "FourthInstarLarvae_splitFQ2_part19.fq.gz", "FourthInstarLarvae_splitFQ2_part20.fq.gz", "FourthInstarLarvae_splitFQ2_part21.fq.gz", "FourthInstarLarvae_splitFQ2_part22.fq.gz", "FourthInstarLarvae_splitFQ2_part23.fq.gz", "FourthInstarLarvae_splitFQ2_part24.fq.gz", "FourthInstarLarvae_splitFQ2_part25.fq.gz", "FourthInstarLarvae_splitFQ2_part26.fq.gz", "FourthInstarLarvae_splitFQ2_part27.fq.gz", "FourthInstarLarvae_splitFQ2_part28.fq.gz", "FourthInstarLarvae_splitFQ2_part29.fq.gz", "FourthInstarLarvae_splitFQ2_part30.fq.gz", "FourthInstarLarvae_splitFQ2_part31.fq.gz", "FourthInstarLarvae_splitFQ2_part32.fq.gz", "FourthInstarLarvae_splitFQ2_part33.fq.gz", "FourthInstarLarvae_splitFQ2_part34.fq.gz", "FourthInstarLarvae_splitFQ2_part35.fq.gz", "FourthInstarLarvae_splitFQ2_part36.fq.gz", "FourthInstarLarvae_splitFQ2_part37.fq.gz", "FourthInstarLarvae_splitFQ2_part38.fq.gz", "FourthInstarLarvae_splitFQ2_part39.fq.gz", "FourthInstarLarvae_splitFQ2_part40.fq.gz")
      args <- c("-i", FourthInstarLarvae_Pair2FQ, paste("-o", file.path(output_dir, files)))
      system2(command, args = args)
      return(files)
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part32 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[32L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[32L]), "-2", file.path(output_dir, temp2[32L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part33 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[33L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[33L]), "-2", file.path(output_dir, temp2[33L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part34 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[34L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[34L]), "-2", file.path(output_dir, temp2[34L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part35 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[35L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[35L]), "-2", file.path(output_dir, temp2[35L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part36 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[36L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[36L]), "-2", file.path(output_dir, temp2[36L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part37 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[37L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[37L]), "-2", file.path(output_dir, temp2[37L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part38 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[38L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[38L]), "-2", file.path(output_dir, temp2[38L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part39 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[39L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[39L]), "-2", file.path(output_dir, temp2[39L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part40 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[40L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[40L]), "-2", file.path(output_dir, temp2[40L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part10 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[10L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[10L]), "-2", file.path(output_dir, temp2[10L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part11 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[11L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[11L]), "-2", file.path(output_dir, temp2[11L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part12 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[12L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[12L]), "-2", file.path(output_dir, temp2[12L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part13 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[13L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[13L]), "-2", file.path(output_dir, temp2[13L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part14 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[14L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[14L]), "-2", file.path(output_dir, temp2[14L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part15 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[15L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[15L]), "-2", file.path(output_dir, temp2[15L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part16 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[16L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[16L]), "-2", file.path(output_dir, temp2[16L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part17 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[17L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[17L]), "-2", file.path(output_dir, temp2[17L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part18 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[18L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[18L]), "-2", file.path(output_dir, temp2[18L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part19 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[19L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[19L]), "-2", file.path(output_dir, temp2[19L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part20 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[20L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[20L]), "-2", file.path(output_dir, temp2[20L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part21 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[21L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[21L]), "-2", file.path(output_dir, temp2[21L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part22 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[22L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[22L]), "-2", file.path(output_dir, temp2[22L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part23 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[23L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[23L]), "-2", file.path(output_dir, temp2[23L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part24 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[24L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[24L]), "-2", file.path(output_dir, temp2[24L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part25 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[25L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[25L]), "-2", file.path(output_dir, temp2[25L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part26 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[26L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[26L]), "-2", file.path(output_dir, temp2[26L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part27 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[27L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[27L]), "-2", file.path(output_dir, temp2[27L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part28 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[28L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[28L]), "-2", file.path(output_dir, temp2[28L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part29 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[29L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[29L]), "-2", file.path(output_dir, temp2[29L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part1 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[1L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[1L]), "-2", file.path(output_dir, temp2[1L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part2 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[2L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[2L]), "-2", file.path(output_dir, temp2[2L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part3 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[3L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[3L]), "-2", file.path(output_dir, temp2[3L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part4 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[4L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[4L]), "-2", file.path(output_dir, temp2[4L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part5 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[5L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[5L]), "-2", file.path(output_dir, temp2[5L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part6 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[6L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[6L]), "-2", file.path(output_dir, temp2[6L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part7 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[7L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[7L]), "-2", file.path(output_dir, temp2[7L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part8 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[8L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[8L]), "-2", file.path(output_dir, temp2[8L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part9 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[9L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[9L]), "-2", file.path(output_dir, temp2[9L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part30 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[30L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[30L]), "-2", file.path(output_dir, temp2[30L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_splitFQ1_part31 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Eggs_splitter_1
      temp2 <- Eggs_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[31L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[31L]), "-2", file.path(output_dir, temp2[31L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part1 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[1L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[1L]), "-2", file.path(output_dir, temp2[1L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part10 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[10L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[10L]), "-2", file.path(output_dir, temp2[10L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part11 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[11L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[11L]), "-2", file.path(output_dir, temp2[11L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part12 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[12L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[12L]), "-2", file.path(output_dir, temp2[12L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part13 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[13L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[13L]), "-2", file.path(output_dir, temp2[13L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part14 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[14L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[14L]), "-2", file.path(output_dir, temp2[14L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part15 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[15L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[15L]), "-2", file.path(output_dir, temp2[15L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part16 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[16L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[16L]), "-2", file.path(output_dir, temp2[16L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part17 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[17L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[17L]), "-2", file.path(output_dir, temp2[17L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part18 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[18L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[18L]), "-2", file.path(output_dir, temp2[18L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part19 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[19L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[19L]), "-2", file.path(output_dir, temp2[19L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part2 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[2L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[2L]), "-2", file.path(output_dir, temp2[2L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part20 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[20L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[20L]), "-2", file.path(output_dir, temp2[20L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part21 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[21L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[21L]), "-2", file.path(output_dir, temp2[21L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part22 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[22L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[22L]), "-2", file.path(output_dir, temp2[22L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part23 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[23L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[23L]), "-2", file.path(output_dir, temp2[23L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part24 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[24L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[24L]), "-2", file.path(output_dir, temp2[24L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part25 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[25L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[25L]), "-2", file.path(output_dir, temp2[25L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part26 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[26L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[26L]), "-2", file.path(output_dir, temp2[26L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part27 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[27L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[27L]), "-2", file.path(output_dir, temp2[27L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part28 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[28L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[28L]), "-2", file.path(output_dir, temp2[28L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part29 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[29L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[29L]), "-2", file.path(output_dir, temp2[29L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part3 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[3L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[3L]), "-2", file.path(output_dir, temp2[3L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part30 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[30L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[30L]), "-2", file.path(output_dir, temp2[30L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part31 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[31L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[31L]), "-2", file.path(output_dir, temp2[31L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part32 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[32L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[32L]), "-2", file.path(output_dir, temp2[32L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part33 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[33L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[33L]), "-2", file.path(output_dir, temp2[33L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part34 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[34L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[34L]), "-2", file.path(output_dir, temp2[34L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part35 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[35L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[35L]), "-2", file.path(output_dir, temp2[35L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part36 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[36L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[36L]), "-2", file.path(output_dir, temp2[36L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part37 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[37L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[37L]), "-2", file.path(output_dir, temp2[37L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part38 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[38L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[38L]), "-2", file.path(output_dir, temp2[38L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part39 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[39L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[39L]), "-2", file.path(output_dir, temp2[39L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part4 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[4L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[4L]), "-2", file.path(output_dir, temp2[4L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part40 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[40L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[40L]), "-2", file.path(output_dir, temp2[40L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part5 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[5L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[5L]), "-2", file.path(output_dir, temp2[5L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part6 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[6L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[6L]), "-2", file.path(output_dir, temp2[6L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part7 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[7L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[7L]), "-2", file.path(output_dir, temp2[7L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part8 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[8L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[8L]), "-2", file.path(output_dir, temp2[8L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Adults_splitFQ1_part9 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Adults_splitter_1
      temp2 <- Adults_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[9L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[9L]), "-2", file.path(output_dir, temp2[9L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part1 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[1L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[1L]), "-2", file.path(output_dir, temp2[1L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part10 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[10L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[10L]), "-2", file.path(output_dir, temp2[10L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part11 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[11L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[11L]), "-2", file.path(output_dir, temp2[11L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part12 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[12L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[12L]), "-2", file.path(output_dir, temp2[12L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part13 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[13L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[13L]), "-2", file.path(output_dir, temp2[13L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part14 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[14L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[14L]), "-2", file.path(output_dir, temp2[14L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part15 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[15L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[15L]), "-2", file.path(output_dir, temp2[15L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part16 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[16L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[16L]), "-2", file.path(output_dir, temp2[16L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part17 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[17L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[17L]), "-2", file.path(output_dir, temp2[17L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part18 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[18L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[18L]), "-2", file.path(output_dir, temp2[18L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part19 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[19L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[19L]), "-2", file.path(output_dir, temp2[19L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part2 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[2L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[2L]), "-2", file.path(output_dir, temp2[2L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part20 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[20L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[20L]), "-2", file.path(output_dir, temp2[20L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part21 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[21L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[21L]), "-2", file.path(output_dir, temp2[21L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part22 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[22L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[22L]), "-2", file.path(output_dir, temp2[22L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part23 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[23L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[23L]), "-2", file.path(output_dir, temp2[23L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part24 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[24L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[24L]), "-2", file.path(output_dir, temp2[24L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part25 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[25L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[25L]), "-2", file.path(output_dir, temp2[25L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part26 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[26L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[26L]), "-2", file.path(output_dir, temp2[26L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part27 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[27L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[27L]), "-2", file.path(output_dir, temp2[27L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part28 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[28L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[28L]), "-2", file.path(output_dir, temp2[28L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part29 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[29L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[29L]), "-2", file.path(output_dir, temp2[29L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part3 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[3L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[3L]), "-2", file.path(output_dir, temp2[3L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part30 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[30L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[30L]), "-2", file.path(output_dir, temp2[30L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part31 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[31L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[31L]), "-2", file.path(output_dir, temp2[31L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part32 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[32L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[32L]), "-2", file.path(output_dir, temp2[32L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part33 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[33L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[33L]), "-2", file.path(output_dir, temp2[33L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part34 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[34L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[34L]), "-2", file.path(output_dir, temp2[34L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part35 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[35L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[35L]), "-2", file.path(output_dir, temp2[35L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part36 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[36L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[36L]), "-2", file.path(output_dir, temp2[36L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part37 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[37L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[37L]), "-2", file.path(output_dir, temp2[37L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part38 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[38L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[38L]), "-2", file.path(output_dir, temp2[38L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part39 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[39L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[39L]), "-2", file.path(output_dir, temp2[39L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part4 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[4L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[4L]), "-2", file.path(output_dir, temp2[4L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part40 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[40L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[40L]), "-2", file.path(output_dir, temp2[40L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part5 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[5L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[5L]), "-2", file.path(output_dir, temp2[5L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part6 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[6L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[6L]), "-2", file.path(output_dir, temp2[6L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part7 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[7L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[7L]), "-2", file.path(output_dir, temp2[7L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part8 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[8L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[8L]), "-2", file.path(output_dir, temp2[8L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_splitFQ1_part9 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Prepupae_splitter_1
      temp2 <- Prepupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[9L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[9L]), "-2", file.path(output_dir, temp2[9L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part1 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[1L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[1L]), "-2", file.path(output_dir, temp2[1L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part10 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[10L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[10L]), "-2", file.path(output_dir, temp2[10L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part11 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[11L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[11L]), "-2", file.path(output_dir, temp2[11L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part12 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[12L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[12L]), "-2", file.path(output_dir, temp2[12L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part13 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[13L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[13L]), "-2", file.path(output_dir, temp2[13L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part14 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[14L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[14L]), "-2", file.path(output_dir, temp2[14L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part15 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[15L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[15L]), "-2", file.path(output_dir, temp2[15L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part16 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[16L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[16L]), "-2", file.path(output_dir, temp2[16L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part17 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[17L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[17L]), "-2", file.path(output_dir, temp2[17L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part18 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[18L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[18L]), "-2", file.path(output_dir, temp2[18L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part19 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[19L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[19L]), "-2", file.path(output_dir, temp2[19L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part2 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[2L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[2L]), "-2", file.path(output_dir, temp2[2L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part20 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[20L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[20L]), "-2", file.path(output_dir, temp2[20L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part21 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[21L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[21L]), "-2", file.path(output_dir, temp2[21L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part22 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[22L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[22L]), "-2", file.path(output_dir, temp2[22L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part23 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[23L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[23L]), "-2", file.path(output_dir, temp2[23L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part24 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[24L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[24L]), "-2", file.path(output_dir, temp2[24L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part25 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[25L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[25L]), "-2", file.path(output_dir, temp2[25L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part26 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[26L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[26L]), "-2", file.path(output_dir, temp2[26L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part27 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[27L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[27L]), "-2", file.path(output_dir, temp2[27L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part28 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[28L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[28L]), "-2", file.path(output_dir, temp2[28L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part29 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[29L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[29L]), "-2", file.path(output_dir, temp2[29L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part3 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[3L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[3L]), "-2", file.path(output_dir, temp2[3L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part30 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[30L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[30L]), "-2", file.path(output_dir, temp2[30L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part31 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[31L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[31L]), "-2", file.path(output_dir, temp2[31L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part32 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[32L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[32L]), "-2", file.path(output_dir, temp2[32L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part33 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[33L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[33L]), "-2", file.path(output_dir, temp2[33L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part34 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[34L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[34L]), "-2", file.path(output_dir, temp2[34L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part35 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[35L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[35L]), "-2", file.path(output_dir, temp2[35L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part36 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[36L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[36L]), "-2", file.path(output_dir, temp2[36L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part37 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[37L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[37L]), "-2", file.path(output_dir, temp2[37L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part38 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[38L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[38L]), "-2", file.path(output_dir, temp2[38L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part39 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[39L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[39L]), "-2", file.path(output_dir, temp2[39L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part4 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[4L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[4L]), "-2", file.path(output_dir, temp2[4L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part40 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[40L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[40L]), "-2", file.path(output_dir, temp2[40L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part5 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[5L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[5L]), "-2", file.path(output_dir, temp2[5L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part6 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[6L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[6L]), "-2", file.path(output_dir, temp2[6L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part7 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[7L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[7L]), "-2", file.path(output_dir, temp2[7L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part8 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[8L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[8L]), "-2", file.path(output_dir, temp2[8L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Pupae_splitFQ1_part9 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- Pupae_splitter_1
      temp2 <- Pupae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[9L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[9L]), "-2", file.path(output_dir, temp2[9L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part1 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[1L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[1L]), "-2", file.path(output_dir, temp2[1L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part10 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[10L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[10L]), "-2", file.path(output_dir, temp2[10L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part11 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[11L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[11L]), "-2", file.path(output_dir, temp2[11L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part12 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[12L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[12L]), "-2", file.path(output_dir, temp2[12L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part13 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[13L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[13L]), "-2", file.path(output_dir, temp2[13L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part14 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[14L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[14L]), "-2", file.path(output_dir, temp2[14L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part15 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[15L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[15L]), "-2", file.path(output_dir, temp2[15L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part16 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[16L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[16L]), "-2", file.path(output_dir, temp2[16L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part17 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[17L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[17L]), "-2", file.path(output_dir, temp2[17L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part18 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[18L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[18L]), "-2", file.path(output_dir, temp2[18L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part19 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[19L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[19L]), "-2", file.path(output_dir, temp2[19L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part2 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[2L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[2L]), "-2", file.path(output_dir, temp2[2L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part20 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[20L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[20L]), "-2", file.path(output_dir, temp2[20L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part21 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[21L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[21L]), "-2", file.path(output_dir, temp2[21L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part22 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[22L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[22L]), "-2", file.path(output_dir, temp2[22L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part23 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[23L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[23L]), "-2", file.path(output_dir, temp2[23L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part24 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[24L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[24L]), "-2", file.path(output_dir, temp2[24L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part25 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[25L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[25L]), "-2", file.path(output_dir, temp2[25L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part26 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[26L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[26L]), "-2", file.path(output_dir, temp2[26L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part27 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[27L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[27L]), "-2", file.path(output_dir, temp2[27L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part28 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[28L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[28L]), "-2", file.path(output_dir, temp2[28L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part29 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[29L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[29L]), "-2", file.path(output_dir, temp2[29L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part3 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[3L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[3L]), "-2", file.path(output_dir, temp2[3L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part30 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[30L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[30L]), "-2", file.path(output_dir, temp2[30L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part31 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[31L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[31L]), "-2", file.path(output_dir, temp2[31L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part32 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[32L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[32L]), "-2", file.path(output_dir, temp2[32L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part33 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[33L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[33L]), "-2", file.path(output_dir, temp2[33L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part34 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[34L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[34L]), "-2", file.path(output_dir, temp2[34L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part35 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[35L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[35L]), "-2", file.path(output_dir, temp2[35L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part36 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[36L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[36L]), "-2", file.path(output_dir, temp2[36L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part37 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[37L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[37L]), "-2", file.path(output_dir, temp2[37L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part38 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[38L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[38L]), "-2", file.path(output_dir, temp2[38L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part39 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[39L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[39L]), "-2", file.path(output_dir, temp2[39L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part4 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[4L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[4L]), "-2", file.path(output_dir, temp2[4L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part40 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[40L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[40L]), "-2", file.path(output_dir, temp2[40L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part5 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[5L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[5L]), "-2", file.path(output_dir, temp2[5L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part6 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[6L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[6L]), "-2", file.path(output_dir, temp2[6L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part7 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[7L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[7L]), "-2", file.path(output_dir, temp2[7L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part8 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[8L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[8L]), "-2", file.path(output_dir, temp2[8L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_splitFQ1_part9 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- YoungLarvae_splitter_1
      temp2 <- YoungLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[9L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[9L]), "-2", file.path(output_dir, temp2[9L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part20 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[20L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[20L]), "-2", file.path(output_dir, temp2[20L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part21 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[21L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[21L]), "-2", file.path(output_dir, temp2[21L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part22 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[22L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[22L]), "-2", file.path(output_dir, temp2[22L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part23 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[23L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[23L]), "-2", file.path(output_dir, temp2[23L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part1 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[1L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[1L]), "-2", file.path(output_dir, temp2[1L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part24 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[24L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[24L]), "-2", file.path(output_dir, temp2[24L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part2 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[2L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[2L]), "-2", file.path(output_dir, temp2[2L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part25 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[25L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[25L]), "-2", file.path(output_dir, temp2[25L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part3 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[3L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[3L]), "-2", file.path(output_dir, temp2[3L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part26 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[26L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[26L]), "-2", file.path(output_dir, temp2[26L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part4 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[4L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[4L]), "-2", file.path(output_dir, temp2[4L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part27 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[27L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[27L]), "-2", file.path(output_dir, temp2[27L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part5 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[5L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[5L]), "-2", file.path(output_dir, temp2[5L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part28 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[28L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[28L]), "-2", file.path(output_dir, temp2[28L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part6 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[6L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[6L]), "-2", file.path(output_dir, temp2[6L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part29 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[29L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[29L]), "-2", file.path(output_dir, temp2[29L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part7 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[7L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[7L]), "-2", file.path(output_dir, temp2[7L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part8 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[8L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[8L]), "-2", file.path(output_dir, temp2[8L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part9 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[9L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[9L]), "-2", file.path(output_dir, temp2[9L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part10 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[10L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[10L]), "-2", file.path(output_dir, temp2[10L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part11 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[11L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[11L]), "-2", file.path(output_dir, temp2[11L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part12 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[12L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[12L]), "-2", file.path(output_dir, temp2[12L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part13 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[13L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[13L]), "-2", file.path(output_dir, temp2[13L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part14 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[14L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[14L]), "-2", file.path(output_dir, temp2[14L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part15 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[15L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[15L]), "-2", file.path(output_dir, temp2[15L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part16 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[16L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[16L]), "-2", file.path(output_dir, temp2[16L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part17 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[17L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[17L]), "-2", file.path(output_dir, temp2[17L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part18 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[18L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[18L]), "-2", file.path(output_dir, temp2[18L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part19 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[19L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[19L]), "-2", file.path(output_dir, temp2[19L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part30 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[30L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[30L]), "-2", file.path(output_dir, temp2[30L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part31 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[31L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[31L]), "-2", file.path(output_dir, temp2[31L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part32 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[32L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[32L]), "-2", file.path(output_dir, temp2[32L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part33 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[33L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[33L]), "-2", file.path(output_dir, temp2[33L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part34 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[34L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[34L]), "-2", file.path(output_dir, temp2[34L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part35 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[35L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[35L]), "-2", file.path(output_dir, temp2[35L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part36 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[36L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[36L]), "-2", file.path(output_dir, temp2[36L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part37 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[37L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[37L]), "-2", file.path(output_dir, temp2[37L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part38 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[38L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[38L]), "-2", file.path(output_dir, temp2[38L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part39 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[39L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[39L]), "-2", file.path(output_dir, temp2[39L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_splitFQ1_part40 <- {
  Herper::with_CondaEnv(general_env,
    {
      temp1 <- FourthInstarLarvae_splitter_1
      temp2 <- FourthInstarLarvae_splitter_2
      samTemp <- file.path(output_dir, gsub("\\.fq\\.gz", "\\.sam", temp1[40L]))
      hisat_index_path <- dirname(BRCDK54_hisat2index)
      hisat_index_file <- gsub("\\.1\\.ht2", "", basename(BRCDK54_hisat2index))
      hisat_index <- file.path(hisat_index_path, hisat_index_file)
      command <- "hisat2"
      args <- c("-x", hisat_index, "-1", file.path(output_dir, temp1[40L]), "-2", file.path(output_dir, temp2[40L]), "-p", 4, "-S", samTemp)
      system2(command, args = args)
      samcmd <- paste("samtools sort -@4 --output-fmt BAM", samTemp, ">", file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp))))
      system(samcmd)
      unlink(samTemp)
      file.path(output_dir, gsub("\\.sam", "\\.bam", basename(samTemp)))
    },
    pathToMiniConda = general_conda
  )
}
Eggs_hisat2 <- {
  Herper::with_CondaEnv(general_env,
    {
      bamFiles <- list(Eggs_splitFQ1_part1, Eggs_splitFQ1_part2, Eggs_splitFQ1_part3, Eggs_splitFQ1_part4, Eggs_splitFQ1_part5, Eggs_splitFQ1_part6, Eggs_splitFQ1_part7, Eggs_splitFQ1_part8, Eggs_splitFQ1_part9, Eggs_splitFQ1_part10, Eggs_splitFQ1_part11, Eggs_splitFQ1_part12, Eggs_splitFQ1_part13, Eggs_splitFQ1_part14, Eggs_splitFQ1_part15, Eggs_splitFQ1_part16, Eggs_splitFQ1_part17, Eggs_splitFQ1_part18, Eggs_splitFQ1_part19, Eggs_splitFQ1_part20, Eggs_splitFQ1_part21, Eggs_splitFQ1_part22, Eggs_splitFQ1_part23, Eggs_splitFQ1_part24, Eggs_splitFQ1_part25, Eggs_splitFQ1_part26, Eggs_splitFQ1_part27, Eggs_splitFQ1_part28, Eggs_splitFQ1_part29, Eggs_splitFQ1_part30, Eggs_splitFQ1_part31, Eggs_splitFQ1_part32, Eggs_splitFQ1_part33, Eggs_splitFQ1_part34, Eggs_splitFQ1_part35, Eggs_splitFQ1_part36, Eggs_splitFQ1_part37, Eggs_splitFQ1_part38, Eggs_splitFQ1_part39, Eggs_splitFQ1_part40)
      mergecmd <- paste("samtools merge -@4", file.path(output_dir, "Eggs_hisat2.bam"), do.call(paste, bamFiles))
      system(mergecmd)
      file.path(output_dir, "Eggs_hisat2.bam")
    },
    pathToMiniConda = general_conda
  )
}
Adults_hisat2 <- {
  Herper::with_CondaEnv(general_env,
    {
      bamFiles <- list(Adults_splitFQ1_part1, Adults_splitFQ1_part2, Adults_splitFQ1_part3, Adults_splitFQ1_part4, Adults_splitFQ1_part5, Adults_splitFQ1_part6, Adults_splitFQ1_part7, Adults_splitFQ1_part8, Adults_splitFQ1_part9, Adults_splitFQ1_part10, Adults_splitFQ1_part11, Adults_splitFQ1_part12, Adults_splitFQ1_part13, Adults_splitFQ1_part14, Adults_splitFQ1_part15, Adults_splitFQ1_part16, Adults_splitFQ1_part17, Adults_splitFQ1_part18, Adults_splitFQ1_part19, Adults_splitFQ1_part20, Adults_splitFQ1_part21, Adults_splitFQ1_part22, Adults_splitFQ1_part23, Adults_splitFQ1_part24, Adults_splitFQ1_part25, Adults_splitFQ1_part26, Adults_splitFQ1_part27, Adults_splitFQ1_part28, Adults_splitFQ1_part29, Adults_splitFQ1_part30, Adults_splitFQ1_part31, Adults_splitFQ1_part32, Adults_splitFQ1_part33, Adults_splitFQ1_part34, Adults_splitFQ1_part35, Adults_splitFQ1_part36, Adults_splitFQ1_part37, Adults_splitFQ1_part38, Adults_splitFQ1_part39, Adults_splitFQ1_part40)
      mergecmd <- paste("samtools merge -@4", file.path(output_dir, "Adults_hisat2.bam"), do.call(paste, bamFiles))
      system(mergecmd)
      file.path(output_dir, "Adults_hisat2.bam")
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_hisat2 <- {
  Herper::with_CondaEnv(general_env,
    {
      bamFiles <- list(Prepupae_splitFQ1_part1, Prepupae_splitFQ1_part2, Prepupae_splitFQ1_part3, Prepupae_splitFQ1_part4, Prepupae_splitFQ1_part5, Prepupae_splitFQ1_part6, Prepupae_splitFQ1_part7, Prepupae_splitFQ1_part8, Prepupae_splitFQ1_part9, Prepupae_splitFQ1_part10, Prepupae_splitFQ1_part11, Prepupae_splitFQ1_part12, Prepupae_splitFQ1_part13, Prepupae_splitFQ1_part14, Prepupae_splitFQ1_part15, Prepupae_splitFQ1_part16, Prepupae_splitFQ1_part17, Prepupae_splitFQ1_part18, Prepupae_splitFQ1_part19, Prepupae_splitFQ1_part20, Prepupae_splitFQ1_part21, Prepupae_splitFQ1_part22, Prepupae_splitFQ1_part23, Prepupae_splitFQ1_part24, Prepupae_splitFQ1_part25, Prepupae_splitFQ1_part26, Prepupae_splitFQ1_part27, Prepupae_splitFQ1_part28, Prepupae_splitFQ1_part29, Prepupae_splitFQ1_part30, Prepupae_splitFQ1_part31, Prepupae_splitFQ1_part32, Prepupae_splitFQ1_part33, Prepupae_splitFQ1_part34, Prepupae_splitFQ1_part35, Prepupae_splitFQ1_part36, Prepupae_splitFQ1_part37, Prepupae_splitFQ1_part38, Prepupae_splitFQ1_part39, Prepupae_splitFQ1_part40)
      mergecmd <- paste("samtools merge -@4", file.path(output_dir, "Prepupae_hisat2.bam"), do.call(paste, bamFiles))
      system(mergecmd)
      file.path(output_dir, "Prepupae_hisat2.bam")
    },
    pathToMiniConda = general_conda
  )
}
Pupae_hisat2 <- {
  Herper::with_CondaEnv(general_env,
    {
      bamFiles <- list(Pupae_splitFQ1_part1, Pupae_splitFQ1_part2, Pupae_splitFQ1_part3, Pupae_splitFQ1_part4, Pupae_splitFQ1_part5, Pupae_splitFQ1_part6, Pupae_splitFQ1_part7, Pupae_splitFQ1_part8, Pupae_splitFQ1_part9, Pupae_splitFQ1_part10, Pupae_splitFQ1_part11, Pupae_splitFQ1_part12, Pupae_splitFQ1_part13, Pupae_splitFQ1_part14, Pupae_splitFQ1_part15, Pupae_splitFQ1_part16, Pupae_splitFQ1_part17, Pupae_splitFQ1_part18, Pupae_splitFQ1_part19, Pupae_splitFQ1_part20, Pupae_splitFQ1_part21, Pupae_splitFQ1_part22, Pupae_splitFQ1_part23, Pupae_splitFQ1_part24, Pupae_splitFQ1_part25, Pupae_splitFQ1_part26, Pupae_splitFQ1_part27, Pupae_splitFQ1_part28, Pupae_splitFQ1_part29, Pupae_splitFQ1_part30, Pupae_splitFQ1_part31, Pupae_splitFQ1_part32, Pupae_splitFQ1_part33, Pupae_splitFQ1_part34, Pupae_splitFQ1_part35, Pupae_splitFQ1_part36, Pupae_splitFQ1_part37, Pupae_splitFQ1_part38, Pupae_splitFQ1_part39, Pupae_splitFQ1_part40)
      mergecmd <- paste("samtools merge -@4", file.path(output_dir, "Pupae_hisat2.bam"), do.call(paste, bamFiles))
      system(mergecmd)
      file.path(output_dir, "Pupae_hisat2.bam")
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_hisat2 <- {
  Herper::with_CondaEnv(general_env,
    {
      bamFiles <- list(YoungLarvae_splitFQ1_part1, YoungLarvae_splitFQ1_part2, YoungLarvae_splitFQ1_part3, YoungLarvae_splitFQ1_part4, YoungLarvae_splitFQ1_part5, YoungLarvae_splitFQ1_part6, YoungLarvae_splitFQ1_part7, YoungLarvae_splitFQ1_part8, YoungLarvae_splitFQ1_part9, YoungLarvae_splitFQ1_part10, YoungLarvae_splitFQ1_part11, YoungLarvae_splitFQ1_part12, YoungLarvae_splitFQ1_part13, YoungLarvae_splitFQ1_part14, YoungLarvae_splitFQ1_part15, YoungLarvae_splitFQ1_part16, YoungLarvae_splitFQ1_part17, YoungLarvae_splitFQ1_part18, YoungLarvae_splitFQ1_part19, YoungLarvae_splitFQ1_part20, YoungLarvae_splitFQ1_part21, YoungLarvae_splitFQ1_part22, YoungLarvae_splitFQ1_part23, YoungLarvae_splitFQ1_part24, YoungLarvae_splitFQ1_part25, YoungLarvae_splitFQ1_part26, YoungLarvae_splitFQ1_part27, YoungLarvae_splitFQ1_part28, YoungLarvae_splitFQ1_part29, YoungLarvae_splitFQ1_part30, YoungLarvae_splitFQ1_part31, YoungLarvae_splitFQ1_part32, YoungLarvae_splitFQ1_part33, YoungLarvae_splitFQ1_part34, YoungLarvae_splitFQ1_part35, YoungLarvae_splitFQ1_part36, YoungLarvae_splitFQ1_part37, YoungLarvae_splitFQ1_part38, YoungLarvae_splitFQ1_part39, YoungLarvae_splitFQ1_part40)
      mergecmd <- paste("samtools merge -@4", file.path(output_dir, "YoungLarvae_hisat2.bam"), do.call(paste, bamFiles))
      system(mergecmd)
      file.path(output_dir, "YoungLarvae_hisat2.bam")
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_hisat2 <- {
  Herper::with_CondaEnv(general_env,
    {
      bamFiles <- list(FourthInstarLarvae_splitFQ1_part1, FourthInstarLarvae_splitFQ1_part2, FourthInstarLarvae_splitFQ1_part3, FourthInstarLarvae_splitFQ1_part4, FourthInstarLarvae_splitFQ1_part5, FourthInstarLarvae_splitFQ1_part6, FourthInstarLarvae_splitFQ1_part7, FourthInstarLarvae_splitFQ1_part8, FourthInstarLarvae_splitFQ1_part9, FourthInstarLarvae_splitFQ1_part10, FourthInstarLarvae_splitFQ1_part11, FourthInstarLarvae_splitFQ1_part12, FourthInstarLarvae_splitFQ1_part13, FourthInstarLarvae_splitFQ1_part14, FourthInstarLarvae_splitFQ1_part15, FourthInstarLarvae_splitFQ1_part16, FourthInstarLarvae_splitFQ1_part17, FourthInstarLarvae_splitFQ1_part18, FourthInstarLarvae_splitFQ1_part19, FourthInstarLarvae_splitFQ1_part20, FourthInstarLarvae_splitFQ1_part21, FourthInstarLarvae_splitFQ1_part22, FourthInstarLarvae_splitFQ1_part23, FourthInstarLarvae_splitFQ1_part24, FourthInstarLarvae_splitFQ1_part25, FourthInstarLarvae_splitFQ1_part26, FourthInstarLarvae_splitFQ1_part27, FourthInstarLarvae_splitFQ1_part28, FourthInstarLarvae_splitFQ1_part29, FourthInstarLarvae_splitFQ1_part30, FourthInstarLarvae_splitFQ1_part31, FourthInstarLarvae_splitFQ1_part32, FourthInstarLarvae_splitFQ1_part33, FourthInstarLarvae_splitFQ1_part34, FourthInstarLarvae_splitFQ1_part35, FourthInstarLarvae_splitFQ1_part36, FourthInstarLarvae_splitFQ1_part37, FourthInstarLarvae_splitFQ1_part38, FourthInstarLarvae_splitFQ1_part39, FourthInstarLarvae_splitFQ1_part40)
      mergecmd <- paste("samtools merge -@4", file.path(output_dir, "FourthInstarLarvae_hisat2.bam"), do.call(paste, bamFiles))
      system(mergecmd)
      file.path(output_dir, "FourthInstarLarvae_hisat2.bam")
    },
    pathToMiniConda = general_conda
  )
}
```
```



```
</details>
## Create gene models using stringtie

In this section we create our per sample gene models.

<details>
```{r , include=TRUE}
Eggs_stringtie <- {
  Herper::with_CondaEnv(general_env,
    {
      stringtie_directory <- file.path(output_dir, "Eggs_stringtie.gtf")
      stringtie_gtf <- file.path(stringtie_directory, "Eggs_stringtie.gtf")
      dir.create(stringtie_directory, showWarnings = FALSE, recursive = TRUE)
      args <- c(Eggs_hisat2, "-o", stringtie_gtf)
      command <- "stringtie"
      processx::run(command, args = args, echo = TRUE)
      stringtie_gtf
    },
    pathToMiniConda = general_conda
  )
}
Adults_stringtie <- {
  Herper::with_CondaEnv(general_env,
    {
      stringtie_directory <- file.path(output_dir, "Adults_stringtie.gtf")
      stringtie_gtf <- file.path(stringtie_directory, "Adults_stringtie.gtf")
      dir.create(stringtie_directory, showWarnings = FALSE, recursive = TRUE)
      args <- c(Adults_hisat2, "-o", stringtie_gtf)
      command <- "stringtie"
      processx::run(command, args = args, echo = TRUE)
      stringtie_gtf
    },
    pathToMiniConda = general_conda
  )
}
Prepupae_stringtie <- {
  Herper::with_CondaEnv(general_env,
    {
      stringtie_directory <- file.path(output_dir, "Prepupae_stringtie.gtf")
      stringtie_gtf <- file.path(stringtie_directory, "Prepupae_stringtie.gtf")
      dir.create(stringtie_directory, showWarnings = FALSE, recursive = TRUE)
      args <- c(Prepupae_hisat2, "-o", stringtie_gtf)
      command <- "stringtie"
      processx::run(command, args = args, echo = TRUE)
      stringtie_gtf
    },
    pathToMiniConda = general_conda
  )
}
Pupae_stringtie <- {
  Herper::with_CondaEnv(general_env,
    {
      stringtie_directory <- file.path(output_dir, "Pupae_stringtie.gtf")
      stringtie_gtf <- file.path(stringtie_directory, "Pupae_stringtie.gtf")
      dir.create(stringtie_directory, showWarnings = FALSE, recursive = TRUE)
      args <- c(Pupae_hisat2, "-o", stringtie_gtf)
      command <- "stringtie"
      processx::run(command, args = args, echo = TRUE)
      stringtie_gtf
    },
    pathToMiniConda = general_conda
  )
}
YoungLarvae_stringtie <- {
  Herper::with_CondaEnv(general_env,
    {
      stringtie_directory <- file.path(output_dir, "YoungLarvae_stringtie.gtf")
      stringtie_gtf <- file.path(stringtie_directory, "YoungLarvae_stringtie.gtf")
      dir.create(stringtie_directory, showWarnings = FALSE, recursive = TRUE)
      args <- c(YoungLarvae_hisat2, "-o", stringtie_gtf)
      command <- "stringtie"
      processx::run(command, args = args, echo = TRUE)
      stringtie_gtf
    },
    pathToMiniConda = general_conda
  )
}
FourthInstarLarvae_stringtie <- {
  Herper::with_CondaEnv(general_env,
    {
      stringtie_directory <- file.path(output_dir, "FourthInstarLarvae_stringtie.gtf")
      stringtie_gtf <- file.path(stringtie_directory, "FourthInstarLarvae_stringtie.gtf")
      dir.create(stringtie_directory, showWarnings = FALSE, recursive = TRUE)
      args <- c(FourthInstarLarvae_hisat2, "-o", stringtie_gtf)
      command <- "stringtie"
      processx::run(command, args = args, echo = TRUE)
      stringtie_gtf
    },
    pathToMiniConda = general_conda
  )
}
```
```






```
</details>
## Merge gtfs 

In this section we merge gtfs across samples

<details>
```{r , include=TRUE}
gtfList <- {
  Herper::with_CondaEnv(general_env,
    {
      gtf_list_file <- file.path(output_dir, "stringtie", "gtfList.txt")
      dir.create(dirname(gtf_list_file))
      gtf_to_merge <- data.frame(x = c(Adults_stringtie, Eggs_stringtie, FourthInstarLarvae_stringtie, Prepupae_stringtie, Pupae_stringtie, YoungLarvae_stringtie))
      write.table(gtf_to_merge, file = gtf_list_file, row.names = FALSE, col.names = FALSE, quote = FALSE, sep = "\t")
      gtf_list_file
    },
    pathToMiniConda = general_conda
  )
}
merged_stringtie <- {
  Herper::with_CondaEnv(general_env,
    {
      minFPKM <- 0.5
      minTPM <- 2
      minIso <- 0.1
      merged_gtf <- file.path(output_dir, "stringtie", "merged_stringtie.gtf")
      ARGS <- c(paste0("--merge"), paste0("-o", " ", merged_gtf), "-i", paste0("-F", " ", minFPKM), paste0("-T", " ", minTPM), paste0("-f", " ", minIso), gtfList)
      system2("stringtie", args = ARGS, stdout = file.path(dirname(merged_gtf), "merged_stringtie.out"), stderr = file.path(dirname(merged_gtf), "merged_stringtie.err"))
      merged_gtf
    },
    pathToMiniConda = general_conda
  )
}
```
```



```
</details>

## Identify coding regions in GFF.

In this section we use transdecoder to identify coding regions in our gene models.


<details>
```{r , include=TRUE}
transdecoder_env <- {
  "transdecoder"
}
merged_stringtie_transdecode <- {
  Herper::with_CondaEnv(transdecoder_env,
    {
      transcode_output <- file.path(output_dir, "transdecode")
      unzippedFasta <- file.path(tempdir(), gsub("\\.bgz", "", basename(BRCDK54_Bzipped)))
      transfasta <- file.path(tempdir(), "transcripts.fasta")
      gffTemp <- file.path(tempdir(), gsub("\\.gtf", ".gff", basename(merged_stringtie)))
      R.utils::gunzip(BRCDK54_Bzipped, unzippedFasta, remove = FALSE)
      cmd <- "gtf_genome_to_cdna_fasta.pl"
      args <- paste(merged_stringtie, unzippedFasta, ">", transfasta)
      system(paste(cmd, args))
      cmd <- "gtf_to_alignment_gff3.pl"
      args <- paste(merged_stringtie, ">", gffTemp)
      system(paste(cmd, args))
      system(paste("TransDecoder.LongOrfs --output_dir", transcode_output, "-t", transfasta))
      system(paste("TransDecoder.Predict --output_dir", transcode_output, "-t", transfasta))
      system(paste("cdna_alignment_orf_to_genome_orf.pl", paste0(file.path(transcode_output, basename(transfasta)), ".transdecoder.gff3"), gffTemp, transfasta, ">", paste0(file.path(transcode_output, basename(transfasta)), ".transdecoder.genome.gff3")))
      system(paste("gffread -E -T", file.path(transcode_output, paste0(basename(transfasta), ".transdecoder.genome.gff3")), ">", file.path(output_dir, "merged_stringtie_transdecode.gtf")), )
      R.utils::gzip(file.path(output_dir, "merged_stringtie_transdecode.gtf"), paste0(file.path(output_dir, "merged_stringtie_transdecode.gtf"), ".gz"), remove = FALSE)
      paste0(file.path(output_dir, "merged_stringtie_transdecode.gtf"), ".gz")
    },
    pathToMiniConda = general_conda
  )
}
```
```



CMD: mkdir -p /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/transdecode/transcripts.fasta.transdecoder_dir
CMD: touch /rugpfs/fs0/brc/scratch/brc_pipeline/analysis/kip_redo/BRCDK_noGTF/workflow_data/transdecode/transcripts.fasta.transdecoder_dir/__checkpoints_longorfs/TD.longorfs.ok


```
</details>

<table id="example" class="display" width="100%"></table>

  <script src="https://code.jquery.com/jquery-3.5.0.js"></script>
  <script src="https://cdn.datatables.net/1.11.3/js/jquery.dataTables.min.js"></script>
  <link rel="stylesheet" href="https://cdn.datatables.net/1.11.3/css/jquery.dataTables.min.css">


  <script>

  var tablearray = [];

(function() {
  var testjson = "https://raw.githubusercontent.com/RockefellerUniversity/GlossyBRC/master/inst/json/glossary2.json";
  $.getJSON( testjson, function(jsonToParse){
    console.log(jsonToParse)
  })
  .done(function( data ) {
    $.each( data, function(i,data) {
      //console.log(data);
      var wordcount = 0;
      console.log(data.Tags)
      $('p').contents().filter(function() {
        return this.nodeType === 3;
      }).replaceWith(function() {
        console.log("<a href='"+"#example"+"'>"+data.Tags+"</a>");
        // return this.nodeValue.replace(data.Tags,data.URLS);
        count = (this.nodeValue.toLowerCase().match(data.Tags.toLowerCase()) || []).length;
        console.log(count);
        wordcount = wordcount+count;
        return this.nodeValue.toLowerCase().replace(data.Tags.toLowerCase(),"<a href='"+"#example"+"'>"+data.Tags+"</a>");
      });
      console.log(wordcount);
      if(wordcount > 0){
        console.log("ooh");
        x = new Array();
        x.push(data.Tags,"<a href='"+data.URLS+"'>"+data.Tags+"</a>",data.Description)
        console.log(x)
        tablearray.push(x)
        console.log(tablearray)
      }
    });
  });
})();



$(document).ready(function() {
  $('#example').DataTable( {
    data: tablearray,    columns: [
      { title: "Tag" },      { title: "URL" },      { title: "Description" }
    ]
  } );
} );
</script>
